<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-11-16" />

<title>Power analysis</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis_new</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis_new">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Power analysis</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-11-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis_new/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree449bd0104148e8e21c4e4d42c2833f17e6764f39targetblank449bd01a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis_new/tree/449bd0104148e8e21c4e4d42c2833f17e6764f39" target="_blank">449bd01</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree449bd0104148e8e21c4e4d42c2833f17e6764f39targetblank449bd01a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis_new/tree/449bd0104148e8e21c4e4d42c2833f17e6764f39" target="_blank">449bd01</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/revision_power_simulate.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/revision_power_simulate.Rmd</code>) and HTML
(<code>docs/revision_power_simulate.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/449bd0104148e8e21c4e4d42c2833f17e6764f39/analysis/revision_power_simulate.Rmd" target="_blank">449bd01</a>
</td>
<td>
XSun
</td>
<td>
2025-11-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c978c3b5a80c5825b30d3c3970fd6e18e8f2e478/analysis/revision_power_simulate.Rmd" target="_blank">c978c3b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/c978c3b5a80c5825b30d3c3970fd6e18e8f2e478/docs/revision_power_simulate.html" target="_blank">c978c3b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e4a28fbdf4e9c0a85510fe97012ed983845089c1/analysis/revision_power_simulate.Rmd" target="_blank">e4a28fb</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/e4a28fbdf4e9c0a85510fe97012ed983845089c1/docs/revision_power_simulate.html" target="_blank">e4a28fb</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fbe076e3aa90193f18f8c528addc262634c1bb8/analysis/revision_power_simulate.Rmd" target="_blank">4fbe076</a>
</td>
<td>
XSun
</td>
<td>
2025-11-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/4fbe076e3aa90193f18f8c528addc262634c1bb8/docs/revision_power_simulate.html" target="_blank">4fbe076</a>
</td>
<td>
XSun
</td>
<td>
2025-11-20
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/7a6312af336e57c5b5a3002a481bb7df3952a145/analysis/revision_power_simulate.Rmd" target="_blank">7a6312a</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/7a6312af336e57c5b5a3002a481bb7df3952a145/docs/revision_power_simulate.html" target="_blank">7a6312a</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/a52423948461c96ac48e24fbb9e444663a039133/analysis/revision_power_simulate.Rmd" target="_blank">a524239</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/a52423948461c96ac48e24fbb9e444663a039133/docs/revision_power_simulate.html" target="_blank">a524239</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e7faa4bc2947e2db5e6c4e3180d3c63498575053/analysis/revision_power_simulate.Rmd" target="_blank">e7faa4b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/e7faa4bc2947e2db5e6c4e3180d3c63498575053/docs/revision_power_simulate.html" target="_blank">e7faa4b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0fb1c37548b1b850b3036a1870bc5bed17db3e20/analysis/revision_power_simulate.Rmd" target="_blank">0fb1c37</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/0fb1c37548b1b850b3036a1870bc5bed17db3e20/docs/revision_power_simulate.html" target="_blank">0fb1c37</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/bb3beb06fe4f5af31ddb1995851a83492ca15718/analysis/revision_power_simulate.Rmd" target="_blank">bb3beb0</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/bb3beb06fe4f5af31ddb1995851a83492ca15718/docs/revision_power_simulate.html" target="_blank">bb3beb0</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/d42343d6b42ab1e52cb3aa4c75d6170cded5cc3c/analysis/revision_power_simulate.Rmd" target="_blank">d42343d</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/977037c765dc7f77a1fec555ab6788033e846f25/analysis/revision_power_simulate.Rmd" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/977037c765dc7f77a1fec555ab6788033e846f25/docs/revision_power_simulate.html" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/analysis/revision_power_simulate.Rmd" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/6920c6bf856ac9c848365efe0a92384b867c3177/docs/revision_power_simulate.html" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/analysis/revision_power_simulate.Rmd" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/revision_power_simulate.html" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/analysis/revision_power_simulate.Rmd" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/revision_power_simulate.html" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/23a11ad02272616669af2010c84af8abe09401d9/analysis/revision_power_simulate.Rmd" target="_blank">23a11ad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>To evaluate the statistical power of the ECT under reduced GWAS
sample sizes, we performed a simulation procedure for some factor-trait
pairs.</p>
<ul>
<li>Estimating the true effect size of factor on trait</li>
</ul>
<p>ECT models the relationship between SNP effects on the factor
(exposure) and on the trait (outcome) using a no-intercept
regression:</p>
<p><span class="math display">\[
\beta^{\text{GWAS}} = \gamma \cdot \beta^{\text{factor}} + \varepsilon ,
\]</span></p>
<p>where <span class="math inline">\(\gamma\)</span> represents the
causal effect of the factor on the trait.</p>
<p>For each factor–trait pair, we extracted supporting SNPs with <span
class="math inline">\(\mathrm{FDR}_{\text{exposure}} &lt; 0.2\)</span>
and estimated <span class="math inline">\(\gamma\)</span> by
fitting:</p>
<p><span class="math display">\[
\beta^{\text{GWAS}}_j = \gamma \, \beta^{\text{factor}}_j +
\varepsilon_j .
\]</span> The fitted slope <span
class="math inline">\(\hat{\gamma}\)</span> was taken as the estimated
true effect.</p>
<p>For each SNP <span class="math inline">\(j\)</span>, the expected
(true) GWAS effect is therefore:</p>
<p><span class="math display">\[
\beta^{\text{true}}_j = \hat{\gamma} \cdot \beta^{\text{factor}}_j .
\]</span></p>
<ul>
<li>Simulating down-sampled GWAS effect estimates</li>
</ul>
<p>Assuming the standard error of a GWAS effect estimate scales with
sample size as <span class="math inline">\(SE \propto N^{-1/2}\)</span>,
a down-sampled GWAS using a proportion <span
class="math inline">\(d\)</span> of the original sample has:</p>
<p><span class="math display">\[
SE^{\text{new}}_j = \frac{SE_j}{\sqrt{d}} .
\]</span></p>
<p>For each supporting SNP, we simulated GWAS estimates under the
down-sampled scenario according to:</p>
<p><span class="math display">\[
\hat{\beta}_j \sim N\!\left(
      \beta^{\text{true}}_j,\
      SE^{\text{new}}_j
\right).
\]</span></p>
<p>We generated 100 independent simulations per SNP, yielding 100
complete GWAS-like datasets.</p>
<ul>
<li>Re-running ECT on simulated datasets</li>
</ul>
<p>For each simulated dataset, we replaced the observed GWAS effect
sizes with the simulated values <span
class="math inline">\(\hat{\beta}_j\)</span>, retained the original
exposure effect estimates and <span
class="math inline">\(p\)</span>-values, and re-ran ECT for the
corresponding factor–trait pair.</p>
<p>This procedure produced 100 ECT results, each representing the
inference obtained under a GWAS with reduced sample size.</p>
<ul>
<li>Power estimation</li>
</ul>
<p>Power was estimated as the proportion of simulation rounds in which
the ECT result remained statistically significant under the specified
down-sampling proportion.</p>
<pre class="r"><code>suppressMessages(library(ggplot2))
suppressMessages(library(gridExtra))
suppressMessages(library(data.table))
suppressMessages(library(DT))
suppressMessages(library(knitr))
suppressMessages(library(htmltools))
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:gridExtra&#39;:

    combine</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)
library(tidyr)

celltypes &lt;- c(&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;CD15_positive_leukocyte&quot;,&quot;platelet&quot;,&quot;T_cell&quot;,&quot;thymocyte&quot;)
types &lt;- c(&quot;B cell (CD19+)&quot;,&quot;Monocyte (CD14+)&quot;,&quot;Granulocyte (CD15+)&quot;,&quot;Platelet&quot;,&quot;T cell (CD4+)&quot;,&quot;T cell (CD8+)&quot;)
names(celltypes) &lt;- types
names(types) &lt;- celltypes


DT::datatable(matrix())</code></pre>
<div id="htmlwidget-0c923cdc0df2ab45b6f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c923cdc0df2ab45b6f5">{"x":{"filter":"none","vertical":false,"data":[[null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="summary-for-strongweak-pairs" class="section level1">
<h1>Summary for strong/weak pairs</h1>
<div id="strong-ect-fdr-0.1-8-pairs" class="section level2">
<h2>Strong – ECT FDR &lt;0.1, 8 pairs</h2>
<pre class="r"><code>df_all_fdr01 &lt;- c()
df_all_fdr02 &lt;- c()
for(celltype in celltypes){

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  df &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]

  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2 &amp; df$ECT_FDR_5suppSNP &gt; 0.1,]
  df_all_fdr01 &lt;- rbind(df_all_fdr01,df_fdr01)
  df_all_fdr02 &lt;- rbind(df_all_fdr02,df_fdr02)

}

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = F),]
df_all_fdr01 &lt;- df_all_fdr01[order(df_all_fdr01$ECT_FDR_5suppSNP,decreasing = F),]</code></pre>
<pre class="r"><code>#for (i in 2:2) {

power_all_pairs &lt;- c()
for (i in 1:nrow(df_all_fdr01)) {  
  trait=df_all_fdr01$trait[i]
  factor=df_all_fdr01$factor[i]
  celltype=df_all_fdr01$celltype[i]

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

  df$celltype &lt;- types[df$celltype]
  
  dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
  p_obs &lt;- dt_pair$p_ECT
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)
  
  p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)
  
  power_all &lt;- data.frame(
    celltype = character(),
    factor = character(),
    trait = character(),
    downsample_size = numeric(),
    power_fdr01     = numeric(),
    power_fdr02     = numeric(),
    stringsAsFactors = FALSE
  )
  for (downsample_size in c(0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 1)){
    
    file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
    ECT_res_allround_df&lt;- readRDS(file_power)
    
    power_all &lt;- rbind(
      power_all,
      data.frame(
        celltype = celltype,
        factor = factor,
        trait = trait,
        downsample_size = downsample_size,
        power_fdr01     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01),
        power_fdr02     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
      )
    )
    

    
  }
  
 
  # colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
  # rownames(power_all) &lt;- NULL
  # power_all &lt;- as.data.frame(power_all)
  power_all$celltype &lt;- types[celltype]

  
  power_all_pairs &lt;- rbind(power_all_pairs, power_all)
  
}


power_summary &lt;- power_all_pairs %&gt;%
  group_by(downsample_size) %&gt;%
  summarise(
    mean_fdr01 = mean(power_fdr01, na.rm = TRUE),
    se_fdr01   = sd(power_fdr01, na.rm = TRUE) / sqrt(n()),
    mean_fdr02 = mean(power_fdr02, na.rm = TRUE),
    se_fdr02   = sd(power_fdr02, na.rm = TRUE) / sqrt(n())
  )

# Reshape to long format for ggplot
power_long &lt;- power_summary %&gt;%
  pivot_longer(
    cols = c(mean_fdr01, mean_fdr02, se_fdr01, se_fdr02),
    names_to = c(&quot;metric&quot;, &quot;fdr&quot;),
    names_pattern = &quot;(mean|se)_(fdr\\d+)&quot;,
    values_to = &quot;value&quot;
  ) %&gt;%
  pivot_wider(
    names_from = metric,
    values_from = value
  )

ggplot(power_long, aes(x = downsample_size, y = mean, color = fdr, group = fdr)) +
  geom_line(size = 1.1) +
  geom_point(size = 2) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.02) +
  theme_bw(base_size = 14) +
  labs(
    x = &quot;Downsample proportion&quot;,
    y = &quot;Mean power&quot;,
    color = &quot;FDR cutoff&quot;,
    title = &quot;Power vs Downsample Size Across All Strong Pairs&quot;
  )+
  ylim(0, 100)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c978c3b5a80c5825b30d3c3970fd6e18e8f2e478/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">c978c3b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e4a28fbdf4e9c0a85510fe97012ed983845089c1/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">e4a28fb</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fbe076e3aa90193f18f8c528addc262634c1bb8/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">4fbe076</a>
</td>
<td>
XSun
</td>
<td>
2025-11-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/a52423948461c96ac48e24fbb9e444663a039133/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">a524239</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e7faa4bc2947e2db5e6c4e3180d3c63498575053/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">e7faa4b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="weak-bottom-8-from-0.1-ect-fdr-0.2" class="section level2">
<h2>Weak – Bottom 8 from 0.1 &lt; ECT FDR &lt; 0.2</h2>
<pre class="r"><code>#for (i in 2:2) {

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = T),]

power_all_pairs &lt;- c()
for (i in 1:8) {  
  trait=df_all_fdr02$trait[i]
  factor=df_all_fdr02$factor[i]
  celltype=df_all_fdr02$celltype[i]

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

  df$celltype &lt;- types[df$celltype]
  
  dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
  p_obs &lt;- dt_pair$p_ECT
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)
  
  p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)
  
  power_all &lt;- data.frame(
    celltype = character(),
    factor = character(),
    trait = character(),
    downsample_size = numeric(),
    power_fdr01     = numeric(),
    power_fdr02     = numeric(),
    stringsAsFactors = FALSE
  )
  for (downsample_size in c(0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 1)){
    
    file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
    ECT_res_allround_df&lt;- readRDS(file_power)
    
    power_all &lt;- rbind(
      power_all,
      data.frame(
        celltype = celltype,
        factor = factor,
        trait = trait,
        downsample_size = downsample_size,
        power_fdr01     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01),
        power_fdr02     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
      )
    )
    

    
  }
  
 
  # colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
  # rownames(power_all) &lt;- NULL
  # power_all &lt;- as.data.frame(power_all)
  power_all$celltype &lt;- types[celltype]

  
  power_all_pairs &lt;- rbind(power_all_pairs, power_all)
  
}


power_summary &lt;- power_all_pairs %&gt;%
  group_by(downsample_size) %&gt;%
  summarise(
    mean_fdr01 = mean(power_fdr01, na.rm = TRUE),
    se_fdr01   = sd(power_fdr01, na.rm = TRUE) / sqrt(n()),
    mean_fdr02 = mean(power_fdr02, na.rm = TRUE),
    se_fdr02   = sd(power_fdr02, na.rm = TRUE) / sqrt(n())
  )

# Reshape to long format for ggplot
power_long &lt;- power_summary %&gt;%
  pivot_longer(
    cols = c(mean_fdr01, mean_fdr02, se_fdr01, se_fdr02),
    names_to = c(&quot;metric&quot;, &quot;fdr&quot;),
    names_pattern = &quot;(mean|se)_(fdr\\d+)&quot;,
    values_to = &quot;value&quot;
  ) %&gt;%
  pivot_wider(
    names_from = metric,
    values_from = value
  )

ggplot(power_long, aes(x = downsample_size, y = mean, color = fdr, group = fdr)) +
  geom_line(size = 1.1) +
  geom_point(size = 2) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.02) +
  theme_bw(base_size = 14) +
  labs(
    x = &quot;Downsample proportion&quot;,
    y = &quot;Mean power&quot;,
    color = &quot;FDR cutoff&quot;,
    title = &quot;Power vs Downsample Size Across All Weak Pairs&quot;
  )+
  ylim(0, 100)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c978c3b5a80c5825b30d3c3970fd6e18e8f2e478/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-4-1.png" target="_blank">c978c3b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e4a28fbdf4e9c0a85510fe97012ed983845089c1/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-4-1.png" target="_blank">e4a28fb</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-4-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-4-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-4-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="weak-bottom-15-from-0.1-ect-fdr-0.2" class="section level2">
<h2>Weak – Bottom 15 from 0.1 &lt; ECT FDR &lt; 0.2</h2>
<pre class="r"><code>#for (i in 2:2) {

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = T),]

power_all_pairs &lt;- c()
for (i in 1:15) {  
  trait=df_all_fdr02$trait[i]
  factor=df_all_fdr02$factor[i]
  celltype=df_all_fdr02$celltype[i]

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

  df$celltype &lt;- types[df$celltype]
  
  dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
  p_obs &lt;- dt_pair$p_ECT
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)
  
  p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)
  
  power_all &lt;- data.frame(
    celltype = character(),
    factor = character(),
    trait = character(),
    downsample_size = numeric(),
    power_fdr01     = numeric(),
    power_fdr02     = numeric(),
    stringsAsFactors = FALSE
  )
  for (downsample_size in c(0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 1)){
    
    file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
    ECT_res_allround_df&lt;- readRDS(file_power)
    
    power_all &lt;- rbind(
      power_all,
      data.frame(
        celltype = celltype,
        factor = factor,
        trait = trait,
        downsample_size = downsample_size,
        power_fdr01     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01),
        power_fdr02     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
      )
    )
    

    
  }
  
 
  # colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
  # rownames(power_all) &lt;- NULL
  # power_all &lt;- as.data.frame(power_all)
  power_all$celltype &lt;- types[celltype]

  
  power_all_pairs &lt;- rbind(power_all_pairs, power_all)
  
}


power_summary &lt;- power_all_pairs %&gt;%
  group_by(downsample_size) %&gt;%
  summarise(
    mean_fdr01 = mean(power_fdr01, na.rm = TRUE),
    se_fdr01   = sd(power_fdr01, na.rm = TRUE) / sqrt(n()),
    mean_fdr02 = mean(power_fdr02, na.rm = TRUE),
    se_fdr02   = sd(power_fdr02, na.rm = TRUE) / sqrt(n())
  )

# Reshape to long format for ggplot
power_long &lt;- power_summary %&gt;%
  pivot_longer(
    cols = c(mean_fdr01, mean_fdr02, se_fdr01, se_fdr02),
    names_to = c(&quot;metric&quot;, &quot;fdr&quot;),
    names_pattern = &quot;(mean|se)_(fdr\\d+)&quot;,
    values_to = &quot;value&quot;
  ) %&gt;%
  pivot_wider(
    names_from = metric,
    values_from = value
  )

ggplot(power_long, aes(x = downsample_size, y = mean, color = fdr, group = fdr)) +
  geom_line(size = 1.1) +
  geom_point(size = 2) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.02) +
  theme_bw(base_size = 14) +
  labs(
    x = &quot;Downsample proportion&quot;,
    y = &quot;Mean power&quot;,
    color = &quot;FDR cutoff&quot;,
    title = &quot;Power vs Downsample Size Across All Weak Pairs&quot;
  )+
  ylim(0, 100)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e4a28fbdf4e9c0a85510fe97012ed983845089c1/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">e4a28fb</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fbe076e3aa90193f18f8c528addc262634c1bb8/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">4fbe076</a>
</td>
<td>
XSun
</td>
<td>
2025-11-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/a52423948461c96ac48e24fbb9e444663a039133/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">a524239</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e7faa4bc2947e2db5e6c4e3180d3c63498575053/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">e7faa4b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="weak-top22-from-0.1-ect-fdr-0.2" class="section level2">
<h2>Weak – TOP22 from 0.1 &lt; ECT FDR &lt; 0.2</h2>
<pre class="r"><code>#for (i in 2:2) {

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = F),]

power_all_pairs &lt;- c()
for (i in 1:22) {  
  trait=df_all_fdr02$trait[i]
  factor=df_all_fdr02$factor[i]
  celltype=df_all_fdr02$celltype[i]

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

  df$celltype &lt;- types[df$celltype]
  
  dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
  p_obs &lt;- dt_pair$p_ECT
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)
  
  p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)
  
  power_all &lt;- data.frame(
    celltype = character(),
    factor = character(),
    trait = character(),
    downsample_size = numeric(),
    power_fdr01     = numeric(),
    power_fdr02     = numeric(),
    stringsAsFactors = FALSE
  )
  for (downsample_size in c(0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 1)){
    
    file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
    ECT_res_allround_df&lt;- readRDS(file_power)
    
    power_all &lt;- rbind(
      power_all,
      data.frame(
        celltype = celltype,
        factor = factor,
        trait = trait,
        downsample_size = downsample_size,
        power_fdr01     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01),
        power_fdr02     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
      )
    )
    

    
  }
  
 
  # colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
  # rownames(power_all) &lt;- NULL
  # power_all &lt;- as.data.frame(power_all)
  power_all$celltype &lt;- types[celltype]

  
  power_all_pairs &lt;- rbind(power_all_pairs, power_all)
  
}


power_summary &lt;- power_all_pairs %&gt;%
  group_by(downsample_size) %&gt;%
  summarise(
    mean_fdr01 = mean(power_fdr01, na.rm = TRUE),
    se_fdr01   = sd(power_fdr01, na.rm = TRUE) / sqrt(n()),
    mean_fdr02 = mean(power_fdr02, na.rm = TRUE),
    se_fdr02   = sd(power_fdr02, na.rm = TRUE) / sqrt(n())
  )

# Reshape to long format for ggplot
power_long &lt;- power_summary %&gt;%
  pivot_longer(
    cols = c(mean_fdr01, mean_fdr02, se_fdr01, se_fdr02),
    names_to = c(&quot;metric&quot;, &quot;fdr&quot;),
    names_pattern = &quot;(mean|se)_(fdr\\d+)&quot;,
    values_to = &quot;value&quot;
  ) %&gt;%
  pivot_wider(
    names_from = metric,
    values_from = value
  )

ggplot(power_long, aes(x = downsample_size, y = mean, color = fdr, group = fdr)) +
  geom_line(size = 1.1) +
  geom_point(size = 2) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.02) +
  theme_bw(base_size = 14) +
  labs(
    x = &quot;Downsample proportion&quot;,
    y = &quot;Mean power&quot;,
    color = &quot;FDR cutoff&quot;,
    title = &quot;Power vs Downsample Size Across All Weak Pairs&quot;
  )+
  ylim(0, 100)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e4a28fbdf4e9c0a85510fe97012ed983845089c1/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-6-1.png" target="_blank">e4a28fb</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-6-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-6-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-6-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="weak-all-pairs-with-results-22-15-37-from-0.1-ect-fdr-0.2"
class="section level2">
<h2>Weak – all pairs with results (22 + 15 = 37) from 0.1 &lt; ECT FDR
&lt; 0.2</h2>
<pre class="r"><code>#for (i in 2:2) {

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = T),]
df_all_fdr02_bottom &lt;- df_all_fdr02[1:15,]

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = F),]
df_all_fdr02_top &lt;- df_all_fdr02[1:22,]

df_all_fdr02_topbottom &lt;- rbind(df_all_fdr02_top, df_all_fdr02_bottom)

power_all_pairs &lt;- c()
for (i in 1:nrow(df_all_fdr02_topbottom)) {  
  trait=df_all_fdr02_topbottom$trait[i]
  factor=df_all_fdr02_topbottom$factor[i]
  celltype=df_all_fdr02_topbottom$celltype[i]
  
  

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

  df$celltype &lt;- types[df$celltype]
  
  dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
  p_obs &lt;- dt_pair$p_ECT
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)
  
  p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)
  
  power_all &lt;- data.frame(
    celltype = character(),
    factor = character(),
    trait = character(),
    downsample_size = numeric(),
    power_fdr01     = numeric(),
    power_fdr02     = numeric(),
    stringsAsFactors = FALSE
  )
  for (downsample_size in c(0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 1)){
    
    file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
    ECT_res_allround_df&lt;- readRDS(file_power)
    
    power_all &lt;- rbind(
      power_all,
      data.frame(
        celltype = celltype,
        factor = factor,
        trait = trait,
        downsample_size = downsample_size,
        power_fdr01     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01),
        power_fdr02     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
      )
    )
    

    
  }
  
 
  # colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
  # rownames(power_all) &lt;- NULL
  # power_all &lt;- as.data.frame(power_all)
  power_all$celltype &lt;- types[celltype]

  
  power_all_pairs &lt;- rbind(power_all_pairs, power_all)
  
}


power_summary &lt;- power_all_pairs %&gt;%
  group_by(downsample_size) %&gt;%
  summarise(
    mean_fdr01 = mean(power_fdr01, na.rm = TRUE),
    se_fdr01   = sd(power_fdr01, na.rm = TRUE) / sqrt(n()),
    mean_fdr02 = mean(power_fdr02, na.rm = TRUE),
    se_fdr02   = sd(power_fdr02, na.rm = TRUE) / sqrt(n())
  )

# Reshape to long format for ggplot
power_long &lt;- power_summary %&gt;%
  pivot_longer(
    cols = c(mean_fdr01, mean_fdr02, se_fdr01, se_fdr02),
    names_to = c(&quot;metric&quot;, &quot;fdr&quot;),
    names_pattern = &quot;(mean|se)_(fdr\\d+)&quot;,
    values_to = &quot;value&quot;
  ) %&gt;%
  pivot_wider(
    names_from = metric,
    values_from = value
  )

ggplot(power_long, aes(x = downsample_size, y = mean, color = fdr, group = fdr)) +
  geom_line(size = 1.1) +
  geom_point(size = 2) +
  geom_errorbar(aes(ymin = mean - se, ymax = mean + se), width = 0.02) +
  theme_bw(base_size = 14) +
  labs(
    x = &quot;Downsample proportion&quot;,
    y = &quot;Mean power&quot;,
    color = &quot;FDR cutoff&quot;,
    title = &quot;Power vs Downsample Size Across All Weak Pairs&quot;
  )+
  ylim(0, 100)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e4a28fbdf4e9c0a85510fe97012ed983845089c1/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" target="_blank">e4a28fb</a>
</td>
<td>
XSun
</td>
<td>
2025-11-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fbe076e3aa90193f18f8c528addc262634c1bb8/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" target="_blank">4fbe076</a>
</td>
<td>
XSun
</td>
<td>
2025-11-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/7a6312af336e57c5b5a3002a481bb7df3952a145/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" target="_blank">7a6312a</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/a52423948461c96ac48e24fbb9e444663a039133/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" target="_blank">a524239</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e7faa4bc2947e2db5e6c4e3180d3c63498575053/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" target="_blank">e7faa4b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/977037c765dc7f77a1fec555ab6788033e846f25/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="strong-examples-ect-fdr-0.1" class="section level1">
<h1>Strong examples ECT FDR &lt; 0.1</h1>
<pre class="r"><code>df_all_fdr01 &lt;- c()
df_all_fdr02 &lt;- c()
for(celltype in celltypes){

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  df &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]

  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2 &amp; df$ECT_FDR_5suppSNP &gt; 0.1,]
  df_all_fdr01 &lt;- rbind(df_all_fdr01,df_fdr01)
  df_all_fdr02 &lt;- rbind(df_all_fdr02,df_fdr02)

}

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = F),]
df_all_fdr01 &lt;- df_all_fdr01[order(df_all_fdr01$ECT_FDR_5suppSNP,decreasing = F),]</code></pre>
<pre class="r"><code>#for (i in 2:2) {

power_all_pairs &lt;- c()
for (i in 1:nrow(df_all_fdr01)) {
  trait=df_all_fdr01$trait[i]
  factor=df_all_fdr01$factor[i]
  celltype=df_all_fdr01$celltype[i]

  cat(sprintf(&quot;&lt;h3 style=&#39;margin-top:1px&#39;&gt;Top %d - %s - %s - %s &lt;/h3&gt;&quot;, i, celltype, factor, trait))

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

  df$celltype &lt;- types[df$celltype]

  dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
  p_obs &lt;- dt_pair$p_ECT


  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]

  htmltools::div(
  style = &quot;margin-top: 30px; margin-bottom: 30px;&quot;,
  DT::datatable(df_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;pairs with FDR &lt; 0.2&#39;),options = list(pageLength = 5) )
)

  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)

  p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)


  p &lt;- list()
  power_all &lt;- data.frame(
  downsample_size = numeric(),
  power_fdr01     = numeric(),
  power_fdr02     = numeric(),
  stringsAsFactors = FALSE
)
  for (downsample_size in c(0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 1)){

    file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
    ECT_res_allround_df&lt;- readRDS(file_power)

    # tmp &lt;- c(downsample_size, sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01), sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02))
    # power_all &lt;- rbind(power_all, tmp)

    power_all &lt;- rbind(
      power_all,
      data.frame(
        celltype = celltype,
        factor = factor,
        trait = trait,
        downsample_size = downsample_size,
        power_fdr01     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01),
        power_fdr02     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
      )
    )

    num_leq_fdr01 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01)
    num_leq_fdr02 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)

    p[[as.character(downsample_size)]] &lt;- ggplot(ECT_res_allround_df, aes(x = -log10(p_ECT))) +
      geom_histogram(bins = 30, color = &quot;white&quot;, fill = &quot;steelblue&quot;) +

      # cutoff lines
      geom_vline(xintercept = -log10(p_cutoff_fdr01), color = &quot;red&quot;, size = 1) +
      geom_vline(xintercept = -log10(p_cutoff_fdr02), color = &quot;red&quot;, linetype = &quot;dashed&quot;, size = 1) +

      # stacked annotation block
      annotate(
        &quot;label&quot;,
        x = max(-log10(ECT_res_allround_df$p_ECT)),
        y = Inf,
        hjust = 1, vjust = 1.2,
        label = paste0(
          &quot;Count p &lt;= 0.1 cutoff: &quot;, num_leq_fdr01, &quot;\n&quot;,
          &quot;Count p &lt;= 0.2 cutoff: &quot;, num_leq_fdr02
        ),
        size = 4,
        fill = &quot;white&quot;,
        label.size = 0.3
      ) +

      theme_bw(base_size = 13) +
      labs(
        x = &quot;-log10(ECT p-value)&quot;,
        y = &quot;Count&quot;,
        title = paste0(&quot;downsample size &quot;, downsample_size)
      )


  }

  all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]],p[[7]], nrow = 2)


  # colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
  # rownames(power_all) &lt;- NULL
  # power_all &lt;- as.data.frame(power_all)
  power_all$celltype &lt;- types[celltype]

  table &lt;- DT::datatable(
    power_all,
    caption = htmltools::tags$caption(
      style = &#39;caption-side:left; text-align:left; color:black; font-size:140%;&#39;,
      paste0(&quot;Power for &quot;, celltype, &quot; - &quot;, factor, &quot; - &quot;, trait)
    ),
    options = list(pageLength = 10)
  )

  cat(knit_print(table))

  power_all_pairs &lt;- rbind(power_all_pairs, power_all)

}</code></pre>
<h3 style="margin-top:1px">
Top 1 - CD15_positive_leukocyte - 04910_PC4 - MCH-ebi-a-GCST90002390
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fbe076e3aa90193f18f8c528addc262634c1bb8/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">4fbe076</a>
</td>
<td>
XSun
</td>
<td>
2025-11-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/7a6312af336e57c5b5a3002a481bb7df3952a145/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">7a6312a</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/a52423948461c96ac48e24fbb9e444663a039133/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">a524239</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e7faa4bc2947e2db5e6c4e3180d3c63498575053/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">e7faa4b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/977037c765dc7f77a1fec555ab6788033e846f25/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="htmlwidget-ad41175326dc68f35292" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ad41175326dc68f35292">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for CD15_positive_leukocyte - 04910_PC4 - MCH-ebi-a-GCST90002390<\/caption>","data":[["1","2","3","4","5","6","7"],["Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)"],["04910_PC4","04910_PC4","04910_PC4","04910_PC4","04910_PC4","04910_PC4","04910_PC4"],["MCH-ebi-a-GCST90002390","MCH-ebi-a-GCST90002390","MCH-ebi-a-GCST90002390","MCH-ebi-a-GCST90002390","MCH-ebi-a-GCST90002390","MCH-ebi-a-GCST90002390","MCH-ebi-a-GCST90002390"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[1,9,14,22,31,38,48],[1,9,14,22,31,38,48]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>trait<\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 2 - platelet - 04064_PC1 - HB-ebi-a-GCST90002384
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-2.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-66aaeb19fe992dfb51c2" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-66aaeb19fe992dfb51c2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04064_PC1 - HB-ebi-a-GCST90002384<\/caption>","data":[["1","2","3","4","5","6","7"],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"],["04064_PC1","04064_PC1","04064_PC1","04064_PC1","04064_PC1","04064_PC1","04064_PC1"],["HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[5,14,24,44,57,67,72],[5,14,24,44,57,67,72]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>trait<\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 3 - platelet - 04390_PC5 - RET-ebi-a-GCST90002405
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-3.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-1814506231684700bf53" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1814506231684700bf53">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04390_PC5 - RET-ebi-a-GCST90002405<\/caption>","data":[["1","2","3","4","5","6","7"],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"],["04390_PC5","04390_PC5","04390_PC5","04390_PC5","04390_PC5","04390_PC5","04390_PC5"],["RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[2,7,10,19,22,27,29],[15,38,40,53,56,60,66]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>trait<\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 4 - platelet - 04714_PC4 - RET-ebi-a-GCST90002405
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-4.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-43c5071fe8a890b96ecd" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-43c5071fe8a890b96ecd">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04714_PC4 - RET-ebi-a-GCST90002405<\/caption>","data":[["1","2","3","4","5","6","7"],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"],["04714_PC4","04714_PC4","04714_PC4","04714_PC4","04714_PC4","04714_PC4","04714_PC4"],["RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[2,46,78,99,100,100,100],[43,96,100,100,100,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>trait<\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 5 - platelet - 00230_PC1 - RET-ebi-a-GCST90002405
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-5.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-3f22443f93eeeda03f7b" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3f22443f93eeeda03f7b">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 00230_PC1 - RET-ebi-a-GCST90002405<\/caption>","data":[["1","2","3","4","5","6","7"],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"],["00230_PC1","00230_PC1","00230_PC1","00230_PC1","00230_PC1","00230_PC1","00230_PC1"],["RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[8,34,58,91,99,99,100],[35,87,100,100,100,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>trait<\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 6 - B_cell - 04062_PC2 - Psoriasis-ebi-a-GCST90038681
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-6.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-198b30f56771bc600b2c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-198b30f56771bc600b2c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04062_PC2 - Psoriasis-ebi-a-GCST90038681<\/caption>","data":[["1","2","3","4","5","6","7"],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"],["04062_PC2","04062_PC2","04062_PC2","04062_PC2","04062_PC2","04062_PC2","04062_PC2"],["Psoriasis-ebi-a-GCST90038681","Psoriasis-ebi-a-GCST90038681","Psoriasis-ebi-a-GCST90038681","Psoriasis-ebi-a-GCST90038681","Psoriasis-ebi-a-GCST90038681","Psoriasis-ebi-a-GCST90038681","Psoriasis-ebi-a-GCST90038681"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[2,8,17,30,40,42,45],[9,26,40,57,73,79,79]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>trait<\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 7 - thymocyte - 04971_PC2 - IBD-ebi-a-GCST003043
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-7.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-b544a055166914a18edc" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b544a055166914a18edc">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04971_PC2 - IBD-ebi-a-GCST003043<\/caption>","data":[["1","2","3","4","5","6","7"],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"],["04971_PC2","04971_PC2","04971_PC2","04971_PC2","04971_PC2","04971_PC2","04971_PC2"],["IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[7,25,39,47,49,50,50],[7,25,39,47,49,50,50]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>trait<\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 8 - thymocyte - 04670_PC1 - IBD-ebi-a-GCST003043
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-8.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-0d0e9bf024e70c3c54e0" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0d0e9bf024e70c3c54e0">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04670_PC1 - IBD-ebi-a-GCST003043<\/caption>","data":[["1","2","3","4","5","6","7"],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"],["04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1"],["IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[4,32,73,96,98,100,100],[27,83,94,100,100,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>trait<\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="weak-examples-0.1-ect-fdr-0.2" class="section level1">
<h1>Weak examples 0.1 &lt; ECT FDR &lt; 0.2</h1>
<pre class="r"><code>#for (i in 2:2) {

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = T),]
df_all_fdr02_bottom &lt;- df_all_fdr02[1:15,]

df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP,decreasing = F),]
df_all_fdr02_top &lt;- df_all_fdr02[1:22,]

df_all_fdr02_topbottom &lt;- rbind(df_all_fdr02_top, df_all_fdr02_bottom)

for (i in 1:nrow(df_all_fdr02_topbottom) ) {

  trait=df_all_fdr02_topbottom$trait[i]
  factor=df_all_fdr02_topbottom$factor[i]
  celltype=df_all_fdr02_topbottom$celltype[i]

  cat(sprintf(&quot;&lt;h3 style=&#39;margin-top:1px&#39;&gt;Top %d - %s - %s - %s &lt;/h3&gt;&quot;, i+8, celltype, factor, trait))

  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

  df$celltype &lt;- types[df$celltype]

  dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
  p_obs &lt;- dt_pair$p_ECT


  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]

  htmltools::div(
  style = &quot;margin-top: 30px; margin-bottom: 30px;&quot;,
  DT::datatable(df_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;pairs with FDR &lt; 0.2&#39;),options = list(pageLength = 5) )
)

  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)

  p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)


  p &lt;- list()
  power_all &lt;- data.frame(
  downsample_size = numeric(),
  power_fdr01     = numeric(),
  power_fdr02     = numeric(),
  stringsAsFactors = FALSE
)
  for (downsample_size in c(0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 1)){

    file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
    ECT_res_allround_df&lt;- readRDS(file_power)

    tmp &lt;- c(downsample_size, sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01), sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02))
    # power_all &lt;- rbind(power_all, tmp)

    power_all &lt;- rbind(
      power_all,
      data.frame(
        downsample_size = downsample_size,
        power_fdr01     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01),
        power_fdr02     = sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
      )
    )

    num_leq_fdr01 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01)
    num_leq_fdr02 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)

    p[[as.character(downsample_size)]] &lt;- ggplot(ECT_res_allround_df, aes(x = -log10(p_ECT))) +
      geom_histogram(bins = 30, color = &quot;white&quot;, fill = &quot;steelblue&quot;) +

      # cutoff lines
      geom_vline(xintercept = -log10(p_cutoff_fdr01), color = &quot;red&quot;, size = 1) +
      geom_vline(xintercept = -log10(p_cutoff_fdr02), color = &quot;red&quot;, linetype = &quot;dashed&quot;, size = 1) +

      # stacked annotation block
      annotate(
        &quot;label&quot;,
        x = max(-log10(ECT_res_allround_df$p_ECT)),
        y = Inf,
        hjust = 1, vjust = 1.2,
        label = paste0(
          &quot;Count p &lt;= 0.1 cutoff: &quot;, num_leq_fdr01, &quot;\n&quot;,
          &quot;Count p &lt;= 0.2 cutoff: &quot;, num_leq_fdr02
        ),
        size = 4,
        fill = &quot;white&quot;,
        label.size = 0.3
      ) +

      theme_bw(base_size = 13) +
      labs(
        x = &quot;-log10(ECT p-value)&quot;,
        y = &quot;Count&quot;,
        title = paste0(&quot;downsample size &quot;, downsample_size)
      )


  }

  all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]],p[[7]], nrow = 2)


  # colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
  # rownames(power_all) &lt;- NULL
  # power_all &lt;- as.data.frame(power_all)
  power_all$celltype &lt;- types[celltype]

  table &lt;- DT::datatable(
    power_all,
    caption = htmltools::tags$caption(
      style = &#39;caption-side:left; text-align:left; color:black; font-size:140%;&#39;,
      paste0(&quot;Power for &quot;, celltype, &quot; - &quot;, factor, &quot; - &quot;, trait)
    ),
    options = list(pageLength = 10)
  )

  cat(knit_print(table))


}</code></pre>
<h3 style="margin-top:1px">
Top 9 - thymocyte - 04971_PC1 - IBD-ebi-a-GCST003043
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/977037c765dc7f77a1fec555ab6788033e846f25/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-10-1.png" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-10-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-10-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-10-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="htmlwidget-854fff1bc9f845d6a98c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-854fff1bc9f845d6a98c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04971_PC1 - IBD-ebi-a-GCST003043<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[3,11,16,17,18,18,19],[3,11,16,17,18,18,19],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 10 - thymocyte - 04211_PC5 - MPV-ebi-a-GCST90002395
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-2.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-97f4021d5afcf82a6acf" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-97f4021d5afcf82a6acf">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04211_PC5 - MPV-ebi-a-GCST90002395<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[4,34,52,81,83,86,89],[51,98,100,100,100,100,100],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 11 - CD14_positive_monocyte - 04068_PC1 - RBC-ebi-a-GCST90002403
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-3.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-5e55a5507ec837689bd0" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5e55a5507ec837689bd0">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for CD14_positive_monocyte - 04068_PC1 - RBC-ebi-a-GCST90002403<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,0,0,0,0,0,0],[1,28,52,75,88,90,94],["Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 12 - CD14_positive_monocyte - 04924_PC3 - asthma_pmid36778051
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-4.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-cfcd02e72ac20b8c76de" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cfcd02e72ac20b8c76de">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for CD14_positive_monocyte - 04924_PC3 - asthma_pmid36778051<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,0,0,0,0,0,0],[0,1,3,6,8,8,12],["Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 13 - CD14_positive_monocyte - 04390_PC2 - MPV-ebi-a-GCST90002395
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-5.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-41da174dda42004a74af" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-41da174dda42004a74af">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for CD14_positive_monocyte - 04390_PC2 - MPV-ebi-a-GCST90002395<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,0,0,0,0,0,0],[83,100,100,100,100,100,100],["Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 14 - B_cell - 04211_PC3 - T1D-ebi-a-GCST90014023
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-6.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-6019ee703e4f0db556d3" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6019ee703e4f0db556d3">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04211_PC3 - T1D-ebi-a-GCST90014023<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[3,50,81,99,99,100,100],[37,92,99,100,100,100,100],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 15 - B_cell - 04970_PC3 - RBC-ebi-a-GCST90002403
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-7.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-009dd5a4a2833e3f56ce" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-009dd5a4a2833e3f56ce">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04970_PC3 - RBC-ebi-a-GCST90002403<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,2,2,3,7,10,12],[9,31,37,59,71,78,90],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 16 - B_cell - 03010_PC2 - MCH-ebi-a-GCST90002390
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-8.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-09d23fc6535f82194b3c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-09d23fc6535f82194b3c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 03010_PC2 - MCH-ebi-a-GCST90002390<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[5,15,18,31,37,40,45],[5,15,18,31,37,40,45],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 17 - B_cell - 04060_PC2 - HB-ebi-a-GCST90002384
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-9.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-31c57bd681cf7701acc3" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-31c57bd681cf7701acc3">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04060_PC2 - HB-ebi-a-GCST90002384<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[6,15,28,49,55,57,59],[14,36,68,93,98,99,100],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 18 - B_cell - 04060_PC2 - HCT-ebi-a-GCST90002383
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-10.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-e7047e39ca087a558e31" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e7047e39ca087a558e31">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04060_PC2 - HCT-ebi-a-GCST90002383<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,5,11,24,37,46,57],[21,69,93,100,100,100,100],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 19 - B_cell - 04213_PC2 - RET-ebi-a-GCST90002405
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-11.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-d4437ce5bcb1d3024d8a" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d4437ce5bcb1d3024d8a">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04213_PC2 - RET-ebi-a-GCST90002405<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[2,18,29,51,58,62,65],[19,76,95,100,100,100,100],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 20 - B_cell - 04657_PC2 - MPV-ebi-a-GCST90002395
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-12.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-494663bf7c82018a54e4" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-494663bf7c82018a54e4">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04657_PC2 - MPV-ebi-a-GCST90002395<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,2,2,5,9,10,14],[2,4,7,15,25,24,33],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 21 - B_cell - 00510_PC2 - lymphc-ebi-a-GCST90002388
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-13.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-62a97617a832a9edbd8c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-62a97617a832a9edbd8c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 00510_PC2 - lymphc-ebi-a-GCST90002388<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[5,13,25,43,53,61,69],[6,36,58,92,97,97,99],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 22 - B_cell - 04510_PC5 - HCT-ebi-a-GCST90002383
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-14.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-928b98bae3f98e274d5b" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-928b98bae3f98e274d5b">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04510_PC5 - HCT-ebi-a-GCST90002383<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,1,1,1,1,3,3],[3,11,15,21,23,28,30],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 23 - thymocyte - 04962_PC2 - RET-ebi-a-GCST90002405
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-15.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-7f2d707704a9fb899ac0" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7f2d707704a9fb899ac0">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04962_PC2 - RET-ebi-a-GCST90002405<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[20,45,63,89,96,99,100],[20,45,63,89,96,99,100],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 24 - platelet - 04390_PC3 - LDL-ukb-d-30780_irnt
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-16.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-6d2f0dc673920861b7dc" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6d2f0dc673920861b7dc">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04390_PC3 - LDL-ukb-d-30780_irnt<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,2,4,13,20,26,38],[6,38,59,90,94,97,99],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 25 - platelet - 04971_PC4 - HB-ebi-a-GCST90002384
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-17.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-c31230a3ef1628be5daf" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c31230a3ef1628be5daf">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04971_PC4 - HB-ebi-a-GCST90002384<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[4,17,28,45,54,62,63],[4,17,28,45,54,62,63],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 26 - platelet - 04625_PC3 - RET-ebi-a-GCST90002405
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-18.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-ca28f120a43a0f77d99b" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ca28f120a43a0f77d99b">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04625_PC3 - RET-ebi-a-GCST90002405<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[6,16,18,19,19,19,20],[21,47,51,54,58,62,59],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 27 - B_cell - 00640_PC1 - T1D-ebi-a-GCST90014023
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-19.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-8f17cf733d7f83d0063d" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8f17cf733d7f83d0063d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 00640_PC1 - T1D-ebi-a-GCST90014023<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[3,13,20,32,43,51,58],[9,25,39,70,80,85,96],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 28 - B_cell - 03040_PC4 - MCV-ebi-a-GCST90002392
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-20.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-f9e969315d3772953564" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f9e969315d3772953564">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 03040_PC4 - MCV-ebi-a-GCST90002392<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[4,30,52,57,65,66,71],[71,100,100,100,100,100,100],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 29 - B_cell - 03040_PC4 - RBC-ebi-a-GCST90002403
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-21.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-032f9d6587d8bddcbc30" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-032f9d6587d8bddcbc30">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 03040_PC4 - RBC-ebi-a-GCST90002403<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[4,17,30,55,64,72,79],[34,81,93,100,100,100,100],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 30 - B_cell - 04217_PC4 - RBC-ebi-a-GCST90002403
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-22.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-56d4f7ca586da2c2022e" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-56d4f7ca586da2c2022e">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for B_cell - 04217_PC4 - RBC-ebi-a-GCST90002403<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,1,1,7,12,12,19],[19,54,88,100,100,100,100],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 31 - thymocyte - 04024_PC4 - MCV-ebi-a-GCST90002392
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-23.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-a6f89104e89bf56f2f76" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a6f89104e89bf56f2f76">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04024_PC4 - MCV-ebi-a-GCST90002392<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[56,100,100,100,100,100,100],[95,100,100,100,100,100,100],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 32 - thymocyte - 04926_PC4 - MCV-ebi-a-GCST90002392
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-24.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-b0be0404915701026d71" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b0be0404915701026d71">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04926_PC4 - MCV-ebi-a-GCST90002392<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[53,98,100,100,100,100,100],[81,100,100,100,100,100,100],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 33 - thymocyte - 04146_PC3 - IBD-ebi-a-GCST003043
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-25.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-c7f0f6dad5d16c2c11bd" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c7f0f6dad5d16c2c11bd">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04146_PC3 - IBD-ebi-a-GCST003043<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[4,6,7,13,21,23,27],[10,19,25,38,47,48,54],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 34 - thymocyte - 04929_PC1 - MPV-ebi-a-GCST90002395
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-26.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-14e435e91f1ff1b1fc2d" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-14e435e91f1ff1b1fc2d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04929_PC1 - MPV-ebi-a-GCST90002395<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[22,53,57,68,71,72,76],[22,53,57,68,71,72,76],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 35 - thymocyte - 04215_PC2 - lymphc-ebi-a-GCST90002388
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-27.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-e94cc58f92900c19328f" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e94cc58f92900c19328f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for thymocyte - 04215_PC2 - lymphc-ebi-a-GCST90002388<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,8,24,55,76,82,93],[14,72,95,100,100,100,100],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 36 - platelet - 00270_PC3 - HB-ebi-a-GCST90002384
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-28.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-1f1c7f318ae0fe829191" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1f1c7f318ae0fe829191">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 00270_PC3 - HB-ebi-a-GCST90002384<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,1,4,14,22,31,39],[8,23,36,65,82,92,97],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 37 - platelet - 04918_PC4 - MCV-ebi-a-GCST90002392
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-29.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-7fd38a877ddb232fe03f" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7fd38a877ddb232fe03f">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04918_PC4 - MCV-ebi-a-GCST90002392<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[2,12,16,19,19,19,22],[20,55,62,71,75,71,75],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 38 - platelet - 00562_PC5 - HB-ebi-a-GCST90002384
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-30.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-57cf70d65317d7a07a90" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-57cf70d65317d7a07a90">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 00562_PC5 - HB-ebi-a-GCST90002384<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[11,18,25,39,44,46,48],[11,18,25,39,44,46,48],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 39 - platelet - 04062_PC2 - WBC-ebi-a-GCST90002407
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-31.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-bcbeb2bff4e1e9729a04" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bcbeb2bff4e1e9729a04">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04062_PC2 - WBC-ebi-a-GCST90002407<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[4,22,41,64,67,69,70],[4,22,41,64,67,69,70],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 40 - platelet - 04611_PC5 - HCT-ebi-a-GCST90002383
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-32.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-f2a1e16af2aa2c89da28" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f2a1e16af2aa2c89da28">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04611_PC5 - HCT-ebi-a-GCST90002383<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,8,24,57,67,71,84],[13,53,66,97,100,100,100],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 41 - platelet - 04611_PC4 - WBC-ebi-a-GCST90002407
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-33.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-f1ddc8b40061b9dd265c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f1ddc8b40061b9dd265c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04611_PC4 - WBC-ebi-a-GCST90002407<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[0,21,47,87,93,98,99],[13,67,91,100,100,100,100],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 42 - platelet - 04022_PC4 - WBC-ebi-a-GCST90002407
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-34.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-062c9748532a7930b7d0" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-062c9748532a7930b7d0">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04022_PC4 - WBC-ebi-a-GCST90002407<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[4,14,18,30,42,46,50],[9,32,53,82,88,93,98],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 43 - platelet - 04910_PC1 - HB-ebi-a-GCST90002384
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-35.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-654c45f5cb0d6897bf3a" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-654c45f5cb0d6897bf3a">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04910_PC1 - HB-ebi-a-GCST90002384<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[1,41,78,100,100,100,100],[22,80,97,100,100,100,100],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 44 - platelet - 04371_PC5 - HCT-ebi-a-GCST90002383
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-36.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-09cdc38cc124fd2d8f97" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-09cdc38cc124fd2d8f97">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 04371_PC5 - HCT-ebi-a-GCST90002383<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[1,6,15,27,32,41,50],[11,50,78,98,99,99,100],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<h3 style="margin-top:1px">
Top 45 - platelet - 03013_PC5 - UC-ebi-a-GCST003045
</h3>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-10-37.png" width="1152" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-e29a9b91a67f79fa3993" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e29a9b91a67f79fa3993">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Power for platelet - 03013_PC5 - UC-ebi-a-GCST003045<\/caption>","data":[["1","2","3","4","5","6","7"],[0.01,0.05,0.1,0.3,0.5,0.7,1],[3,11,16,24,27,29,33],[3,11,16,24,27,29,33],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n      <th>celltype<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tidyr_1.3.0       dplyr_1.1.4       htmltools_0.5.2   knitr_1.39       
[5] DT_0.22           data.table_1.14.2 gridExtra_2.3     ggplot2_4.0.0    

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0   xfun_0.41          bslib_0.3.1        purrr_1.0.2       
 [5] vctrs_0.6.5        generics_0.1.2     yaml_2.3.5         utf8_1.2.2        
 [9] rlang_1.1.2        jquerylib_0.1.4    later_1.3.0        pillar_1.9.0      
[13] glue_1.6.2         withr_2.5.0        RColorBrewer_1.1-3 S7_0.2.0          
[17] lifecycle_1.0.4    stringr_1.5.1      gtable_0.3.6       workflowr_1.7.0   
[21] htmlwidgets_1.5.4  evaluate_0.15      labeling_0.4.2     fastmap_1.1.0     
[25] httpuv_1.6.5       crosstalk_1.2.0    fansi_1.0.3        highr_0.9         
[29] Rcpp_1.0.12        promises_1.2.0.1   scales_1.4.0       jsonlite_1.8.0    
[33] farver_2.1.0       fs_1.5.2           digest_0.6.29      stringi_1.7.6     
[37] grid_4.2.0         rprojroot_2.0.3    cli_3.6.1          tools_4.2.0       
[41] magrittr_2.0.3     sass_0.4.1         tibble_3.2.1       dichromat_2.0-0.1 
[45] whisker_0.4        pkgconfig_2.0.3    rmarkdown_2.25     rstudioapi_0.13   
[49] R6_2.5.1           git2r_0.30.1       compiler_4.2.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
