<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-10-14" />

<title>Examine old/new blood cell traits</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis_new</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis_new">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Examine old/new blood cell traits</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-10-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis_new/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree9009455e4da7b6470c12a54c79184dba5230effetargetblank9009455a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis_new/tree/9009455e4da7b6470c12a54c79184dba5230effe" target="_blank">9009455</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree9009455e4da7b6470c12a54c79184dba5230effetargetblank9009455a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis_new/tree/9009455e4da7b6470c12a54c79184dba5230effe" target="_blank">9009455</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/revision_examine_oldnew_bloodcelltraits.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/revision_examine_oldnew_bloodcelltraits.Rmd</code>) and
HTML (<code>docs/revision_examine_oldnew_bloodcelltraits.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/ea8d0f34f5de215ec3252a17c26cb8dd831a26bd/analysis/revision_examine_oldnew_bloodcelltraits.Rmd" target="_blank">ea8d0f3</a>
</td>
<td>
XSun
</td>
<td>
2025-10-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/ea8d0f34f5de215ec3252a17c26cb8dd831a26bd/docs/revision_examine_oldnew_bloodcelltraits.html" target="_blank">ea8d0f3</a>
</td>
<td>
XSun
</td>
<td>
2025-10-24
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(data.table)
library(patchwork)
library(ggplot2)
library(dplyr)

plot_ECT_effect_scatter &lt;- function(dat, ECT_p, fdr_cutoff = 0.2) {

  # 1. Filter SNPs by FDR threshold
  snp_select &lt;- dat %&gt;%
    filter(FDR.exposure &lt; fdr_cutoff) %&gt;%
    mutate(
      chr = as.numeric(as.character(chr.exposure)),
      chr = factor(chr.exposure, levels = sort(unique(chr.exposure)))
    )
  
  if (nrow(snp_select) == 0) {
    #stop(&quot;No SNPs passed the FDR cutoff.&quot;)
    p &lt;- NULL
  }else{
    # 2. Linear regression
    fit &lt;- lm(beta.outcome ~ 0 + beta.exposure, data = snp_select)
    
    
    # 4. Create the plot
    p &lt;- ggplot(snp_select, aes(x = beta.exposure, y = beta.outcome, color = chr)) +
      geom_point(size = 4) +
      geom_errorbar(aes(ymin = beta.outcome - se.outcome,
                        ymax = beta.outcome + se.outcome),
                    width = 0.01, alpha = 0.6) +
      geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
      geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
      geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
      geom_label(
        aes(x = Inf, y = Inf),
        hjust = 1, vjust = 1,
        label = paste0(
          &quot;R² = &quot;, signif(summary(fit)$r.squared, 3),
          &quot;\nECT p = &quot;, signif(ECT_p, 3)
        ),
        color = &quot;black&quot;, family = &quot;Times&quot;, size = 3, label.size = NA, fill = NA
      ) +
      xlab(expression(hat(beta)[factor])) +
      ylab(expression(hat(beta)[gwas])) +
      theme_bw(base_line_size = 0.3) +
      theme(
        axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = &quot;black&quot;),
        axis.title.y = element_text(size = 16),
        axis.text.y = element_text(size = 14, color = &quot;black&quot;),
        text = element_text(family = &quot;Times&quot;),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)
      )
  }
  
  
  
  return(p)
}</code></pre>
<p>We examine some blood cell traits here. To check why some pairs were
reported by old GWAS but not by new.</p>
<div id="platelet---ret-initial-submission" class="section level2">
<h2>Platelet - Ret (initial submission)</h2>
<pre class="r"><code>ECT_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/platelet-RET-ebi-a-GCST004622_ECT.RDS&quot;)
ECT_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/platelet-RET-ebi-a-GCST90002405_ECT.RDS&quot;)

factor &lt;- &quot;04062_PC4&quot;

cat(&quot;Old GWAS&quot;)</code></pre>
<pre><code>Old GWAS</code></pre>
<pre class="r"><code>print(ECT_old[ECT_old$factor_name == factor,])</code></pre>
<pre><code>    factor_name                p_std     r_std   p_ECT num_supp_SNP
149   04062_PC4 2.02523374839567e-05 0.9796313 8.5e-05            6</code></pre>
<pre class="r"><code>cat(&quot;New GWAS&quot;)</code></pre>
<pre><code>New GWAS</code></pre>
<pre class="r"><code>print(ECT_new[ECT_new$factor_name == factor,])</code></pre>
<pre><code>    factor_name                     p_std r_std p_ECT num_supp_SNP
149   04062_PC4 No enough supporting SNPs    NA    NA            0</code></pre>
<pre class="r"><code>snp_selected_old &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/RET-ebi-a-GCST004622_prunedsnps.txt&quot;)
snp_selected_new &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/RET-ebi-a-GCST90002405_prunedsnps.txt&quot;)

cat(paste0(&quot;number of independent loci selected -- old GWAS: &quot;, nrow(snp_selected_old)))</code></pre>
<pre><code>number of independent loci selected -- old GWAS: 155</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of independent loci selected -- new GWAS: &quot;, nrow(snp_selected_new)))</code></pre>
<pre><code>number of independent loci selected -- new GWAS: 413</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of overlaps: &quot;, sum(snp_selected_old$SNP %in% snp_selected_new$SNP)))</code></pre>
<pre><code>number of overlaps: 44</code></pre>
<pre class="r"><code>harmo_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize//platelet-RET-ebi-a-GCST004622_harmo.RDS&quot;)
harmo_old_factor &lt;- harmo_old[[factor]]
harmo_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize//platelet-RET-ebi-a-GCST90002405_harmo.RDS&quot;)
harmo_new_factor &lt;- harmo_new[[factor]]

supp_snp_old &lt;- harmo_old_factor$SNP[harmo_old_factor$FDR.exposure &lt; 0.2]
supp_snp_new &lt;- harmo_new_factor$SNP[harmo_new_factor$FDR.exposure &lt; 0.2]

df_supp_snp_old &lt;- harmo_old_factor[harmo_old_factor$FDR.exposure &lt; 0.2 |harmo_old_factor$SNP %in% supp_snp_new, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]
df_supp_snp_new &lt;- harmo_new_factor[harmo_new_factor$FDR.exposure &lt; 0.2 |harmo_new_factor$SNP %in% supp_snp_old, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]


df_merged &lt;- merge(df_supp_snp_old, df_supp_snp_new,
                   by = &quot;SNP&quot;,
                   suffixes = c(&quot;.old&quot;, &quot;.new&quot;),
                   all = TRUE)

DT::datatable(df_merged,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Supporting SNPs in old and new GWAS &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-0c923cdc0df2ab45b6f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c923cdc0df2ab45b6f5">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Supporting SNPs in old and new GWAS <\/caption>","data":[["1","2","3","4","5","6"],["rs10954464","rs11021221","rs11678584","rs11913573","rs41282676","rs7187994"],[2.01187039514943e-21,3.93595387190936e-10,1.09299959241713e-10,6.13606568533649e-09,5.82907978867233e-24,1.94101995387022e-09],[0.002378,0.007174,0.006558,0.007687,0.001528,0.005937],[0.184295,0.198580833333333,0.198580833333333,0.198580833333333,0.184295,0.198580833333333],[null,5.40007759066591e-21,2.49976970217851e-21,null,null,null],[null,0.007174,0.006558,null,null,null],[null,0.47023,0.47023,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SNP<\/th>\n      <th>pval.outcome.old<\/th>\n      <th>pval.exposure.old<\/th>\n      <th>FDR.exposure.old<\/th>\n      <th>pval.outcome.new<\/th>\n      <th>pval.exposure.new<\/th>\n      <th>FDR.exposure.new<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>plot_old &lt;- plot_ECT_effect_scatter(dat = harmo_old_factor,ECT_p = ECT_old[ECT_old$factor_name == factor,]$p_ECT)
plot_new &lt;- plot_ECT_effect_scatter(dat = harmo_new_factor,ECT_p = ECT_new[ECT_new$factor_name == factor,]$p_ECT)

if (!is.null(plot_old) &amp; !is.null(plot_new)) {
  # both exist → plot side by side
  (plot_old | plot_new) +
    plot_annotation(title = paste0(&quot;Comparison for &quot;, factor))

} else if (!is.null(plot_old) &amp; is.null(plot_new)) {
  # only old exists
  plot_old +
    plot_annotation(title = paste0(&quot;Old dataset only: &quot;, factor))

} else if (is.null(plot_old) &amp; !is.null(plot_new)) {
  # only new exists
  plot_new +
    plot_annotation(title = paste0(&quot;New dataset only: &quot;, factor))

} else {
  # both null
  print(&quot;No supporting SNPs for either dataset.&quot;)
}</code></pre>
<p><img src="figure/revision_examine_oldnew_bloodcelltraits.Rmd/unnamed-chunk-2-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="b-cell---mpv-initial-submission" class="section level2">
<h2>B cell - MPV (initial submission)</h2>
<pre class="r"><code>ECT_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/B_cell-mpv-ebi-a-GCST004599_ECT.RDS&quot;)
ECT_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/B_cell-MPV-ebi-a-GCST90002395_ECT.RDS&quot;)

factor &lt;- &quot;04620_PC4&quot;

cat(&quot;Old GWAS&quot;)</code></pre>
<pre><code>Old GWAS</code></pre>
<pre class="r"><code>print(ECT_old[ECT_old$factor_name == factor,])</code></pre>
<pre><code>    factor_name                p_std     r_std     p_ECT num_supp_SNP
574   04620_PC4 0.000665117617629492 0.9865446 0.0019998            4</code></pre>
<pre class="r"><code>cat(&quot;New GWAS&quot;)</code></pre>
<pre><code>New GWAS</code></pre>
<pre class="r"><code>print(ECT_new[ECT_new$factor_name == factor,])</code></pre>
<pre><code>    factor_name               p_std     r_std     p_ECT num_supp_SNP
574   04620_PC4 0.00310994722296906 0.5311117 0.0087991           13</code></pre>
<pre class="r"><code>snp_selected_old &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/mpv-ebi-a-GCST004599_prunedsnps.txt&quot;)
snp_selected_new &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/MPV-ebi-a-GCST90002395_prunedsnps.txt&quot;)

cat(paste0(&quot;number of independent loci selected -- old GWAS: &quot;, nrow(snp_selected_old)))</code></pre>
<pre><code>number of independent loci selected -- old GWAS: 213</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of independent loci selected -- new GWAS: &quot;, nrow(snp_selected_new)))</code></pre>
<pre><code>number of independent loci selected -- new GWAS: 465</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of overlaps: &quot;, sum(snp_selected_old$SNP %in% snp_selected_new$SNP)))</code></pre>
<pre><code>number of overlaps: 52</code></pre>
<pre class="r"><code>harmo_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize//B_cell-mpv-ebi-a-GCST004599_harmo.RDS&quot;)
harmo_old_factor &lt;- harmo_old[[factor]]
harmo_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize//B_cell-MPV-ebi-a-GCST90002395_harmo.RDS&quot;)
harmo_new_factor &lt;- harmo_new[[factor]]

supp_snp_old &lt;- harmo_old_factor$SNP[harmo_old_factor$FDR.exposure &lt; 0.2]
supp_snp_new &lt;- harmo_new_factor$SNP[harmo_new_factor$FDR.exposure &lt; 0.2]

df_supp_snp_old &lt;- harmo_old_factor[harmo_old_factor$FDR.exposure &lt; 0.2 |harmo_old_factor$SNP %in% supp_snp_new, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]
df_supp_snp_new &lt;- harmo_new_factor[harmo_new_factor$FDR.exposure &lt; 0.2 |harmo_new_factor$SNP %in% supp_snp_old, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]


df_merged &lt;- merge(df_supp_snp_old, df_supp_snp_new,
                   by = &quot;SNP&quot;,
                   suffixes = c(&quot;.old&quot;, &quot;.new&quot;),
                   all = TRUE)

DT::datatable(df_merged,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Supporting SNPs in old and new GWAS &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-ad41175326dc68f35292" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ad41175326dc68f35292">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Supporting SNPs in old and new GWAS <\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17"],["rs10984588","rs11807162","rs11950562","rs12096438","rs12155038","rs1736150","rs17382437","rs1985601","rs2631360","rs3924236","rs4655783","rs4870312","rs55736271","rs56165246","rs57554052","rs7596219","rs76549445"],[null,null,2.70022528688822e-33,5.17011257968655e-25,2.58523483956219e-11,null,null,null,null,null,null,null,null,null,null,2.21003931734919e-12,null],[null,null,0.0003625,0.001138,0.002171,null,null,null,null,null,null,null,null,null,null,0.0009167,null],[null,null,0.0772125,0.080798,0.11560575,null,null,null,null,null,null,null,null,null,null,0.080798,null],[2.99999133271614e-10,4.70002318179801e-10,null,null,null,1.99986186963274e-19,1.19999655704811e-09,1.29987023738861e-22,1e-93,7.29961575729302e-64,1.40000633223953e-10,1.89998424621473e-08,1.99986186963274e-15,6.49994976198144e-10,8.90020479625527e-11,null,1e-11],[0.0008075,0.0003154,null,null,null,0.0003811,0.004775,0.001471,0.0003707,0.002049,0.005553,0.001848,0.00056,0.00507,7.028e-06,null,0.00172],[0.06258125,0.044302875,null,null,null,0.044302875,0.1964625,0.0952785,0.044302875,0.0952785,0.198626538461538,0.0952785,0.05208,0.1964625,0.00326802,null,0.0952785]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SNP<\/th>\n      <th>pval.outcome.old<\/th>\n      <th>pval.exposure.old<\/th>\n      <th>FDR.exposure.old<\/th>\n      <th>pval.outcome.new<\/th>\n      <th>pval.exposure.new<\/th>\n      <th>FDR.exposure.new<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>plot_old &lt;- plot_ECT_effect_scatter(dat = harmo_old_factor,ECT_p = ECT_old[ECT_old$factor_name == factor,]$p_ECT)
plot_new &lt;- plot_ECT_effect_scatter(dat = harmo_new_factor,ECT_p = ECT_new[ECT_new$factor_name == factor,]$p_ECT)


if (!is.null(plot_old) &amp; !is.null(plot_new)) {
  # both exist → plot side by side
  (plot_old | plot_new) +
    plot_annotation(title = paste0(&quot;Comparison for &quot;, factor))

} else if (!is.null(plot_old) &amp; is.null(plot_new)) {
  # only old exists
  plot_old +
    plot_annotation(title = paste0(&quot;Old dataset only: &quot;, factor))

} else if (is.null(plot_old) &amp; !is.null(plot_new)) {
  # only new exists
  plot_new +
    plot_annotation(title = paste0(&quot;New dataset only: &quot;, factor))

} else {
  # both null
  print(&quot;No supporting SNPs for either dataset.&quot;)
}</code></pre>
<p><img src="figure/revision_examine_oldnew_bloodcelltraits.Rmd/unnamed-chunk-3-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="leukocyte-cd15---mch" class="section level2">
<h2>Leukocyte (CD15+) - MCH</h2>
<pre class="r"><code>ECT_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/CD15_positive_leukocyte-MCH-ebi-a-GCST004630_ECT.RDS&quot;)
ECT_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/CD15_positive_leukocyte-MCH-ebi-a-GCST90002390_ECT.RDS&quot;)

factor &lt;- &quot;05168_PC3&quot;

cat(&quot;Old GWAS&quot;)</code></pre>
<pre><code>Old GWAS</code></pre>
<pre class="r"><code>print(ECT_old[ECT_old$factor_name == factor,])</code></pre>
<pre><code>    factor_name                p_std     r_std   p_ECT num_supp_SNP
998   05168_PC3 1.71918694359914e-05 0.8833665 2.3e-05           10</code></pre>
<pre class="r"><code>cat(&quot;New GWAS&quot;)</code></pre>
<pre><code>New GWAS</code></pre>
<pre class="r"><code>print(ECT_new[ECT_new$factor_name == factor,])</code></pre>
<pre><code>    factor_name                     p_std r_std p_ECT num_supp_SNP
998   05168_PC3 No enough supporting SNPs    NA    NA            0</code></pre>
<pre class="r"><code>snp_selected_old &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/MCH-ebi-a-GCST004630_prunedsnps.txt&quot;)
snp_selected_new &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/MCH-ebi-a-GCST90002390_prunedsnps.txt&quot;)

cat(paste0(&quot;number of independent loci selected -- old GWAS: &quot;, nrow(snp_selected_old)))</code></pre>
<pre><code>number of independent loci selected -- old GWAS: 203</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of independent loci selected -- new GWAS: &quot;, nrow(snp_selected_new)))</code></pre>
<pre><code>number of independent loci selected -- new GWAS: 462</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of overlaps: &quot;, sum(snp_selected_old$SNP %in% snp_selected_new$SNP)))</code></pre>
<pre><code>number of overlaps: 63</code></pre>
<pre class="r"><code>harmo_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize//CD15_positive_leukocyte-MCH-ebi-a-GCST004630_harmo.RDS&quot;)
harmo_old_factor &lt;- harmo_old[[factor]]
harmo_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize//CD15_positive_leukocyte-MCH-ebi-a-GCST90002390_harmo.RDS&quot;)
harmo_new_factor &lt;- harmo_new[[factor]]

supp_snp_old &lt;- harmo_old_factor$SNP[harmo_old_factor$FDR.exposure &lt; 0.2]
supp_snp_new &lt;- harmo_new_factor$SNP[harmo_new_factor$FDR.exposure &lt; 0.2]

df_supp_snp_old &lt;- harmo_old_factor[harmo_old_factor$FDR.exposure &lt; 0.2 |harmo_old_factor$SNP %in% supp_snp_new, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]
df_supp_snp_new &lt;- harmo_new_factor[harmo_new_factor$FDR.exposure &lt; 0.2 |harmo_new_factor$SNP %in% supp_snp_old, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]


df_merged &lt;- merge(df_supp_snp_old, df_supp_snp_new, 
                   by = &quot;SNP&quot;, 
                   suffixes = c(&quot;.old&quot;, &quot;.new&quot;), 
                   all = TRUE)

DT::datatable(df_merged,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Supporting SNPs in old and new GWAS &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-66aaeb19fe992dfb51c2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-66aaeb19fe992dfb51c2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Supporting SNPs in old and new GWAS <\/caption>","data":[["1","2","3","4","5","6","7","8","9","10"],["rs12451471","rs12935418","rs17699658","rs2283847","rs2713936","rs4590800","rs68051257","rs73146904","rs79953286","rs826415"],[1.39800910530143e-08,2.57798270819929e-08,6.26181164615688e-12,7.42848071113177e-11,1.59001052821486e-09,2.76375791917036e-16,1.11892248195524e-14,3.57519718745026e-17,3.27717776805172e-14,9.25124111252553e-24],[0.009409,0.001992,0.00528,0.008683,0.002053,0.004891,0.001026,0.003445,0.001315,0.006564],[0.1910027,0.10418975,0.15312,0.1910027,0.10418975,0.15312,0.10418975,0.139867,0.10418975,0.1665615],[1.5999263828686e-13,null,6.00067353864114e-26,4.90004422793237e-24,9.20025786648231e-31,null,null,null,5.30029448835795e-35,null],[0.009409,null,0.00528,0.008683,0.002053,null,null,null,0.001315,null],[0.3828825,null,0.368016,0.3828825,0.2371215,null,null,null,0.212982,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SNP<\/th>\n      <th>pval.outcome.old<\/th>\n      <th>pval.exposure.old<\/th>\n      <th>FDR.exposure.old<\/th>\n      <th>pval.outcome.new<\/th>\n      <th>pval.exposure.new<\/th>\n      <th>FDR.exposure.new<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>plot_old &lt;- plot_ECT_effect_scatter(dat = harmo_old_factor,ECT_p = ECT_old[ECT_old$factor_name == factor,]$p_ECT)
plot_new &lt;- plot_ECT_effect_scatter(dat = harmo_new_factor,ECT_p = ECT_new[ECT_new$factor_name == factor,]$p_ECT)

if (!is.null(plot_old) &amp; !is.null(plot_new)) {
  # both exist → plot side by side
  (plot_old | plot_new) +
    plot_annotation(title = paste0(&quot;Comparison for &quot;, factor))
  
} else if (!is.null(plot_old) &amp; is.null(plot_new)) {
  # only old exists
  plot_old + 
    plot_annotation(title = paste0(&quot;Old dataset only: &quot;, factor))
  
} else if (is.null(plot_old) &amp; !is.null(plot_new)) {
  # only new exists
  plot_new + 
    plot_annotation(title = paste0(&quot;New dataset only: &quot;, factor))
  
} else {
  # both null
  print(&quot;No supporting SNPs for either dataset.&quot;)
}</code></pre>
<p><img src="figure/revision_examine_oldnew_bloodcelltraits.Rmd/unnamed-chunk-4-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="t-cell-cd4---hgb" class="section level2">
<h2>T cell (CD4+) - hgb</h2>
<pre class="r"><code>ECT_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/T_cell-hgb-ebi-a-GCST004615_ECT.RDS&quot;)
ECT_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/T_cell-HB-ebi-a-GCST90002384_ECT.RDS&quot;)

factor &lt;- &quot;04066_PC3&quot;

cat(&quot;Old GWAS&quot;)</code></pre>
<pre><code>Old GWAS</code></pre>
<pre class="r"><code>print(ECT_old[ECT_old$factor_name == factor,])</code></pre>
<pre><code>    factor_name               p_std     r_std   p_ECT num_supp_SNP
333   04066_PC3 0.00205838437514155 0.9958875 2.3e-05            3</code></pre>
<pre class="r"><code>cat(&quot;New GWAS&quot;)</code></pre>
<pre><code>New GWAS</code></pre>
<pre class="r"><code>print(ECT_new[ECT_new$factor_name == factor,])</code></pre>
<pre><code>    factor_name              p_std    r_std     p_ECT num_supp_SNP
333   04066_PC3 0.0548051452703656 0.757874 0.0364964            4</code></pre>
<pre class="r"><code>snp_selected_old &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/hgb-ebi-a-GCST004615_prunedsnps.txt&quot;)
snp_selected_new &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/HB-ebi-a-GCST90002384_prunedsnps.txt&quot;)


cat(paste0(&quot;number of independent loci selected -- old GWAS: &quot;, nrow(snp_selected_old)))</code></pre>
<pre><code>number of independent loci selected -- old GWAS: 123</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of independent loci selected -- new GWAS: &quot;, nrow(snp_selected_new)))</code></pre>
<pre><code>number of independent loci selected -- new GWAS: 438</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of overlaps: &quot;, sum(snp_selected_old$SNP %in% snp_selected_new$SNP)))</code></pre>
<pre><code>number of overlaps: 25</code></pre>
<pre class="r"><code>harmo_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/T_cell-hgb-ebi-a-GCST004615_harmo.RDS&quot;)
harmo_old_factor &lt;- harmo_old[[factor]]
harmo_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/T_cell-HB-ebi-a-GCST90002384_harmo.RDS&quot;)
harmo_new_factor &lt;- harmo_new[[factor]]

supp_snp_old &lt;- harmo_old_factor$SNP[harmo_old_factor$FDR.exposure &lt; 0.2]
supp_snp_new &lt;- harmo_new_factor$SNP[harmo_new_factor$FDR.exposure &lt; 0.2]

df_supp_snp_old &lt;- harmo_old_factor[harmo_old_factor$FDR.exposure &lt; 0.2 |harmo_old_factor$SNP %in% supp_snp_new, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]
df_supp_snp_new &lt;- harmo_new_factor[harmo_new_factor$FDR.exposure &lt; 0.2 |harmo_new_factor$SNP %in% supp_snp_old, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]


df_merged &lt;- merge(df_supp_snp_old, df_supp_snp_new, 
                   by = &quot;SNP&quot;, 
                   suffixes = c(&quot;.old&quot;, &quot;.new&quot;), 
                   all = TRUE)

DT::datatable(df_merged,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Supporting SNPs in old and new GWAS &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-1814506231684700bf53" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1814506231684700bf53">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Supporting SNPs in old and new GWAS <\/caption>","data":[["1","2","3","4","5","6"],["rs116009877","rs12894354","rs13306780","rs142736683","rs71559014","rs9953010"],[null,null,4.78597030869085e-10,2.54800338136199e-13,2.7459966305247e-62,null],[null,null,0.0006857,0.004443,0.004868,null],[null,null,0.0843411,0.199588,0.199588,null],[1e-200,1.79998961724559e-10,4.49987033872149e-29,null,null,8.00000023961727e-09],[0.001821,0.0001138,0.0006857,null,null,0.001254],[0.1993995,0.0498444,0.1501683,null,null,0.183084]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SNP<\/th>\n      <th>pval.outcome.old<\/th>\n      <th>pval.exposure.old<\/th>\n      <th>FDR.exposure.old<\/th>\n      <th>pval.outcome.new<\/th>\n      <th>pval.exposure.new<\/th>\n      <th>FDR.exposure.new<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>plot_old &lt;- plot_ECT_effect_scatter(dat = harmo_old_factor,ECT_p = ECT_old[ECT_old$factor_name == factor,]$p_ECT)
plot_new &lt;- plot_ECT_effect_scatter(dat = harmo_new_factor,ECT_p = ECT_new[ECT_new$factor_name == factor,]$p_ECT)


if (!is.null(plot_old) &amp; !is.null(plot_new)) {
  # both exist → plot side by side
  (plot_old | plot_new) +
    plot_annotation(title = paste0(&quot;Comparison for &quot;, factor))
  
} else if (!is.null(plot_old) &amp; is.null(plot_new)) {
  # only old exists
  plot_old + 
    plot_annotation(title = paste0(&quot;Old dataset only: &quot;, factor))
  
} else if (is.null(plot_old) &amp; !is.null(plot_new)) {
  # only new exists
  plot_new + 
    plot_annotation(title = paste0(&quot;New dataset only: &quot;, factor))
  
} else {
  # both null
  print(&quot;No supporting SNPs for either dataset.&quot;)
}</code></pre>
<p><img src="figure/revision_examine_oldnew_bloodcelltraits.Rmd/unnamed-chunk-5-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="b-cell-cd19---rbc" class="section level2">
<h2>B cell (CD19+) - RBC</h2>
<pre class="r"><code>ECT_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/B_cell-RBC-ebi-a-GCST004601_ECT.RDS&quot;)
ECT_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT/B_cell-RBC-ebi-a-GCST90002403_ECT.RDS&quot;)

factor &lt;- &quot;04145_PC3&quot;

cat(&quot;Old GWAS&quot;)</code></pre>
<pre><code>Old GWAS</code></pre>
<pre class="r"><code>print(ECT_old[ECT_old$factor_name == factor,])</code></pre>
<pre><code>    factor_name                p_std     r_std   p_ECT num_supp_SNP
403   04145_PC3 2.64970608617299e-06 0.9638574 4.4e-05            8</code></pre>
<pre class="r"><code>cat(&quot;New GWAS&quot;)</code></pre>
<pre><code>New GWAS</code></pre>
<pre class="r"><code>print(ECT_new[ECT_new$factor_name == factor,])</code></pre>
<pre><code>    factor_name             p_std     r_std     p_ECT num_supp_SNP
403   04145_PC3 0.505162576014114 0.2448641 0.4295704            3</code></pre>
<pre class="r"><code>snp_selected_old &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/hgb-ebi-a-GCST004615_prunedsnps.txt&quot;)
snp_selected_new &lt;- fread(&quot;/project/xinhe/xsun/pathway_factor/data_v2/indep_loci/HB-ebi-a-GCST90002384_prunedsnps.txt&quot;)


cat(paste0(&quot;number of independent loci selected -- old GWAS: &quot;, nrow(snp_selected_old)))</code></pre>
<pre><code>number of independent loci selected -- old GWAS: 123</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of independent loci selected -- new GWAS: &quot;, nrow(snp_selected_new)))</code></pre>
<pre><code>number of independent loci selected -- new GWAS: 438</code></pre>
<pre class="r"><code>cat(paste0(&quot;number of overlaps: &quot;, sum(snp_selected_old$SNP %in% snp_selected_new$SNP)))</code></pre>
<pre><code>number of overlaps: 25</code></pre>
<pre class="r"><code>harmo_old &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/B_cell-RBC-ebi-a-GCST004601_harmo.RDS&quot;)
harmo_old_factor &lt;- harmo_old[[factor]]
harmo_new &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/B_cell-RBC-ebi-a-GCST90002403_harmo.RDS&quot;)
harmo_new_factor &lt;- harmo_new[[factor]]

supp_snp_old &lt;- harmo_old_factor$SNP[harmo_old_factor$FDR.exposure &lt; 0.2]
supp_snp_new &lt;- harmo_new_factor$SNP[harmo_new_factor$FDR.exposure &lt; 0.2]

df_supp_snp_old &lt;- harmo_old_factor[harmo_old_factor$FDR.exposure &lt; 0.2 |harmo_old_factor$SNP %in% supp_snp_new, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]
df_supp_snp_new &lt;- harmo_new_factor[harmo_new_factor$FDR.exposure &lt; 0.2 |harmo_new_factor$SNP %in% supp_snp_old, c(&quot;SNP&quot;,&quot;pval.outcome&quot;,&quot;pval.exposure&quot;,&quot;FDR.exposure&quot;)]


df_merged &lt;- merge(df_supp_snp_old, df_supp_snp_new, 
                   by = &quot;SNP&quot;, 
                   suffixes = c(&quot;.old&quot;, &quot;.new&quot;), 
                   all = TRUE)

DT::datatable(df_merged,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Supporting SNPs in old and new GWAS &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-43c5071fe8a890b96ecd" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-43c5071fe8a890b96ecd">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Supporting SNPs in old and new GWAS <\/caption>","data":[["1","2","3","4","5","6","7","8","9","10"],["rs1936807","rs1966820","rs2186037","rs2923411","rs4791641","rs6414767","rs6676150","rs7601633","rs7725218","rs78744187"],[4.76540701725134e-11,2.31526123324376e-12,null,8.48008098889692e-12,5.11210776117741e-11,9.08510602548359e-10,9.70957006528823e-11,null,4.91926252434165e-24,4.63873966756034e-51],[0.0005837,0.0007856,null,0.004546,0.003728,0.008988,0.0002746,null,0.0009869,0.002975],[0.0434236,0.0434236,null,0.114299428571429,0.109354666666667,0.197736,0.0434236,null,0.0434236,0.10472],[null,null,7.39945954312723e-30,null,null,null,null,1.29987023738861e-24,6.49980009683024e-64,5.49540873857636e-150],[null,null,0.0005837,null,null,null,null,0.001075,0.0009869,0.002975],[null,null,0.187766666666667,null,null,null,null,0.187766666666667,0.187766666666667,0.2143815]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SNP<\/th>\n      <th>pval.outcome.old<\/th>\n      <th>pval.exposure.old<\/th>\n      <th>FDR.exposure.old<\/th>\n      <th>pval.outcome.new<\/th>\n      <th>pval.exposure.new<\/th>\n      <th>FDR.exposure.new<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>plot_old &lt;- plot_ECT_effect_scatter(dat = harmo_old_factor,ECT_p = ECT_old[ECT_old$factor_name == factor,]$p_ECT)
plot_new &lt;- plot_ECT_effect_scatter(dat = harmo_new_factor,ECT_p = ECT_new[ECT_new$factor_name == factor,]$p_ECT)
if (!is.null(plot_old) &amp; !is.null(plot_new)) {
  # both exist → plot side by side
  (plot_old | plot_new) +
    plot_annotation(title = paste0(&quot;Comparison for &quot;, factor))
  
} else if (!is.null(plot_old) &amp; is.null(plot_new)) {
  # only old exists
  plot_old + 
    plot_annotation(title = paste0(&quot;Old dataset only: &quot;, factor))
  
} else if (is.null(plot_old) &amp; !is.null(plot_new)) {
  # only new exists
  plot_new + 
    plot_annotation(title = paste0(&quot;New dataset only: &quot;, factor))
  
} else {
  # both null
  print(&quot;No supporting SNPs for either dataset.&quot;)
}</code></pre>
<p><img src="figure/revision_examine_oldnew_bloodcelltraits.Rmd/unnamed-chunk-6-2.png" width="768" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.1.4          ggplot2_4.0.0        patchwork_1.3.2.9000
[4] data.table_1.14.2   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12        highr_0.9          RColorBrewer_1.1-3 pillar_1.9.0      
 [5] compiler_4.2.0     bslib_0.3.1        later_1.3.0        jquerylib_0.1.4   
 [9] git2r_0.30.1       workflowr_1.7.0    tools_4.2.0        digest_0.6.29     
[13] gtable_0.3.6       jsonlite_1.8.0     evaluate_0.15      lifecycle_1.0.4   
[17] tibble_3.2.1       pkgconfig_2.0.3    rlang_1.1.2        cli_3.6.1         
[21] rstudioapi_0.13    crosstalk_1.2.0    yaml_2.3.5         xfun_0.41         
[25] fastmap_1.1.0      withr_2.5.0        stringr_1.5.1      knitr_1.39        
[29] htmlwidgets_1.5.4  generics_0.1.2     fs_1.5.2           vctrs_0.6.5       
[33] sass_0.4.1         DT_0.22            tidyselect_1.2.0   grid_4.2.0        
[37] rprojroot_2.0.3    glue_1.6.2         R6_2.5.1           fansi_1.0.3       
[41] rmarkdown_2.25     farver_2.1.0       magrittr_2.0.3     whisker_0.4       
[45] scales_1.4.0       promises_1.2.0.1   htmltools_0.5.2    dichromat_2.0-0.1 
[49] httpuv_1.6.5       labeling_0.4.2     S7_0.2.0           utf8_1.2.2        
[53] stringi_1.7.6     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
