<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-11-05" />

<title>Tables and Figures</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis_new</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis_new">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Tables and Figures</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-11-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis_new/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree63a542fa8268aaec996c390e18dc182798ddf616targetblank63a542fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis_new/tree/63a542fa8268aaec996c390e18dc182798ddf616" target="_blank">63a542f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree63a542fa8268aaec996c390e18dc182798ddf616targetblank63a542fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis_new/tree/63a542fa8268aaec996c390e18dc182798ddf616" target="_blank">63a542f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/revision_figures.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/revision_figures.Rmd</code>) and
HTML (<code>docs/revision_figures.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/ea5a468e892aa77f64e266e737fb247724cf4751/docs/revision_figures.html" target="_blank">ea5a468</a>
</td>
<td>
XSun
</td>
<td>
2025-11-16
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c16754472fd389d0401e7fa5ac368f182bac2c67/analysis/revision_figures.Rmd" target="_blank">c167544</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/analysis/revision_figures.Rmd" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/revision_figures.html" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c15139281f22ae76559bdd7518299034aae30822/analysis/revision_figures.Rmd" target="_blank">c151392</a>
</td>
<td>
XSun
</td>
<td>
2025-11-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/e1a196e3bdf812c2157edeb571555d58e1f09b71/analysis/revision_figures.Rmd" target="_blank">e1a196e</a>
</td>
<td>
XSun
</td>
<td>
2025-11-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/analysis/revision_figures.Rmd" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/revision_figures.html" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/analysis/revision_figures.Rmd" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/revision_figures.html" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/analysis/revision_figures.Rmd" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/revision_figures.html" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/f27fc60209b4226ef8ca5445caeaee77c5b5f83b/analysis/revision_figures.Rmd" target="_blank">f27fc60</a>
</td>
<td>
XSun
</td>
<td>
2025-11-06
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(gridExtra)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(EnsDb.Hsapiens.v75)
library(ggrepel)

celltypes &lt;- c(&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;CD15_positive_leukocyte&quot;,&quot;platelet&quot;,&quot;T_cell&quot;,&quot;thymocyte&quot;)
types &lt;- c(&quot;B cell (CD19+)&quot;,&quot;Monocyte (CD14+)&quot;,&quot;Granulocyte (CD15+)&quot;,&quot;Platelet&quot;,&quot;T cell (CD4+)&quot;,&quot;T cell (CD8+)&quot;)
names(celltypes) &lt;- types
names(types) &lt;- celltypes

source(&quot;/project/xinhe/xsun/pathway_factor/data_v2/traits_finalselection_SLEremoved.R&quot;)

folder_ect_summary &lt;- &quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;

qqplot &lt;- function(pvalues,title=NULL) {
  
  pval &lt;- pvalues[complete.cases(pvalues)]
  title &lt;- title
  plotdata &lt;- data.frame(observed = -log10(sort(as.numeric(pval))),
                         expected = -log10(ppoints(length(pval))))
  
  qq &lt;- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5,color = &quot;steelblue&quot;) +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = &quot;red&quot;) +
    ggtitle(title) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste(&quot;Expected -log&quot;[10],&quot;(p-value)&quot;)),
         y = expression(paste(&quot;Observed -log&quot;[10],&quot;(p-value)&quot;))) +
    
    theme(axis.title.x = element_text(size = 14),
          axis.text.x = element_text(size = 12, color = &quot;black&quot;),
          axis.title.y = element_text(size = 14),
          axis.text.y = element_text(size = 12, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;))
  
  return(qq)
  
}</code></pre>
<div id="main-text-table" class="section level1">
<h1>Main text table</h1>
<div id="table-1" class="section level2">
<h2>Table 1</h2>
<pre class="r"><code>df_all_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  df &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_all_fdr02 &lt;- rbind(df_all_fdr02,df_fdr02)
  
}


df_all_fdr02$celltype &lt;- types[df_all_fdr02$celltype]
df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP),]

df_all_fdr01 &lt;- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP &lt; 0.1,]

DT::datatable(df_all_fdr01,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Pairs with ECT FDR &lt;= 0.1 &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-0c923cdc0df2ab45b6f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c923cdc0df2ab45b6f5">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Pairs with ECT FDR &lt;= 0.1 <\/caption>","data":[["11","14","11791","11792","11793","2","110","210"],["Granulocyte (CD15+)","Platelet","Platelet","Platelet","Platelet","B cell (CD19+)","T cell (CD8+)","T cell (CD8+)"],["MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","Psoriasis-ebi-a-GCST90038681","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043"],["04910_PC4","04064_PC1","04390_PC5","04714_PC4","00230_PC1","04062_PC2","04971_PC2","04670_PC1"],["04910","04064","04390","04714","00230","04062","04971","04670"],["Insulin signaling pathway","NF-kappa B signaling pathway","Hippo signaling pathway","Thermogenesis","Purine metabolism","Chemokine signaling pathway","Gastric acid secretion","Leukocyte transendothelial migration"],[8,5,6,9,8,6,5,10],[0,0,0,0,0,0,1,0],[1e-06,8e-06,6.7e-05,9.4e-05,0.000177,4e-05,5e-05,8.3e-05],[0.001438,0.014296,0.0559927,0.0559927,0.0790748,0.08832,0.0996,0.0996],[137,102,155,231,130,189,75,112],[87,47,41,90,34,109,34,69],["bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","disease_EUR_old","disease_EUR_old"],["-0.245270464051887","-0.41768128699344","-0.219352852012402","-0.797167485258106","0.0490457193697609","1.69054931142635","-0.160017285934662","-0.146323318161613"],["0.381444392107266","0.386094674495389","0.606360167370838","0.132433832841616","0.916858027973131","0.793939736663622","0.272526832466116","0.380080136666012"],[0.0650233,0.0829709,0.000215,0.0024138,0.0388076,0.0350311,0.0009857,0.0415006],[0.0646544,0.0825434,0.000214,0.0024024,0.0386281,0.0313682,0.0913229,0.0412207],[0.5544554,0.4455446,0.009901,0.029703,0.3168317,0.1584158,0.039604,0.2376238],[0.5346535,0.4356436,0.009901,0.029703,0.3168317,0.1287129,0.3960396,0.2376238],[0.2832572,0.4244096,0.0044039,0.0269258,0.254628,0.1065031,0.0089497,0.2015042],[0.2784657,0.4216313,0.004394,0.0268569,0.2531699,0.0972008,0.3708311,0.1987707],[0.2174599,0.3434217,0.0040564,0.0198934,0.1893318,0.153741,0.0094632,0.2052896],[0.2143697,0.3408274,0.0040367,0.0197926,0.1878342,0.1351176,0.3738789,0.2013522]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>num_supp_SNP_MHC<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n      <th>acat_p_correct_noMHC_traitnull<\/th>\n      <th>acat_p_correct_globalnull<\/th>\n      <th>acat_p_correct_noMHC_globalnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,11,15,16,17,18,19,20,21,22]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="main-text-figures" class="section level1">
<h1>Main text figures</h1>
<div id="figure-3" class="section level2">
<h2>Figure 3</h2>
<div id="ret-ebi-a-gcst90002405" class="section level3">
<h3>RET-ebi-a-GCST90002405</h3>
<pre class="r"><code>trait &lt;- &quot;RET-ebi-a-GCST90002405&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 5)

p</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-3-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-3-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/figure/revision_figures.Rmd/unnamed-chunk-3-1.png" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Platelets</li>
</ul>
<p>Activated platelets, which are recruited to sites of vascular injury,
are known to release a variety of chemokines (Y. Chen et al.,2020). The
resulting chemokine-receptor interactions are crucial regulators of
hematopoiesis (Bonavita et al., 2018; Broxmeyer, 2008; Duchene et al.,
2017), which can impact the reticulocyte counts.</p>
<ol style="list-style-type: decimal">
<li>Hippo signaling pathway</li>
</ol>
<p>Hippo signalling pathway plays a crucial role in human
megakaryoblastic cell differentiation and thrombopoiesis.[PMID:
27786336,27903710,27515249]. megakaryocytes’ main function is producing
platelets [PMID 26787738]</p>
<ol start="2" style="list-style-type: decimal">
<li>Purine metabolism</li>
</ol>
<p>purines have been shown to play an important role in the physiology
of platelets[PMID 37108190]. Platelets contain very high levels of
adenine nucleotides (like ADP and ATP), which are stored in granules and
released during activation by stimuli such as thrombin or collagen.
Released ADP promotes aggregation, while other purine metabolites—ATP,
AMP, adenosine, and AP4—inhibit platelet aggregation. [(]Agarwal, K.C.
(1996). Purines and Regulation of Platelet Activation. In: Abd-Elfattah,
AS.A., Wechsler, A.S. (eds) Purines and Myocardial Protection.
Developments in Cardiovascular Medicine, vol 181. Springer, Boston,
MA.]</p>
<ol start="3" style="list-style-type: decimal">
<li>C-type lectin receptor signaling pathway</li>
</ol>
<p>C-type lectin receptor signaling—especially via CLEC-2—drives strong
platelet activation, aggregation, and thrombo-inflammatory responses
through Syk-dependent pathways.[PMID: 25150298]</p>
<ol start="4" style="list-style-type: decimal">
<li>Glutamatergic synapse</li>
</ol>
<p>glutamate activates AMPA receptors on platelet surface. AMPAR
signaling increases platelet activation and promotes thrombosis, while
blocking AMPAR reduces platelet activation. [PMID: 18283118]</p>
<ol start="5" style="list-style-type: decimal">
<li>cAMP signaling pathway</li>
</ol>
<p>The cAMP signaling pathway represents the most potent endogenous
system for the control of platelet activation through its ability to
modulate multiple platelet functions[PMID: 26617518, 24100445,
26617518]</p>
<ol start="6" style="list-style-type: decimal">
<li>mTOR signaling pathway</li>
</ol>
<p>platelets use an mTOR/S6K1 pathway to activate Rac1 and regulate
platelet spreading on fibrinogen [PMID: 21757621]. mTOR regulates
GPVI-mediated platelet activation [PMID: 33971888]</p>
<ol start="7" style="list-style-type: decimal">
<li>PI3K-Akt signaling pathway</li>
</ol>
<p>The PI3K/Akt signaling pathway has been implicated in playing an
important role in platelet activation during hemostasis and thrombosis
[PMID: 30972780]</p>
</div>
<div id="t1d-ebi-a-gcst90014023" class="section level3">
<h3>T1D-ebi-a-GCST90014023</h3>
<pre class="r"><code>trait &lt;- &quot;T1D-ebi-a-GCST90014023&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 5)

p</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-4-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-4-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_figures.Rmd/unnamed-chunk-4-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/figure/revision_figures.Rmd/unnamed-chunk-4-1.png" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="figure-4" class="section level2">
<h2>Figure 4</h2>
<div
id="example-1-nf-kappa-b-signaling-pathway---platelet---hb-ebi-a-gcst90002384"
class="section level3">
<h3>example 1, NF-kappa B signaling pathway - Platelet -
HB-ebi-a-GCST90002384</h3>
<pre class="r"><code>celltype &lt;- &quot;platelet&quot;
trait &lt;- &quot;HB-ebi-a-GCST90002384&quot;
factor &lt;- &quot;04064_PC1&quot;
fdr_cutoff &lt;- 0.2
  
file_harmo &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;,celltype, &quot;-&quot;, trait, &quot;_harmo.RDS&quot;)
dt_harmo &lt;- readRDS(file_harmo)

df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

dat &lt;- dt_harmo[[factor]]
snp_select &lt;- dat[dat$FDR.exposure &lt; fdr_cutoff,]
snp_select$chr &lt;- as.numeric(as.character(snp_select$chr.exposure))
snp_select$chr &lt;- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))

fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)

ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]

pathwayname &lt;- df$pathwayName[df$factor == factor &amp; df$trait ==trait]

eff &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
  geom_point(size = 4) + 
  geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
  theme_bw(base_line_size = 0.3) +
  theme(axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = &quot;black&quot;),
        axis.title.y = element_text(size =16),
        axis.text.y = element_text(size = 14, color = &quot;black&quot;),
        text= element_text(family=&quot;Times&quot;),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12)) + 
  geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
  geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
             label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                           &quot;\nECT p-value =&quot;, ect_p),
             color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
  xlab(expression(hat(beta)[factor])) + 
  ylab(expression(hat(beta)[gwas])) +
  geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)


#snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff &amp; dat$steiger_dir ==T,]
snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff,]

fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
coef &lt;- summary(fit)
p &lt;- round(coef$coefficients[,4], digits = 4)
#ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]

eff_rev &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome)) +
  geom_point(size = 4, color =&quot;steelblue&quot;) + 
  theme_bw(base_line_size = 0.3) +
  theme(axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = &quot;black&quot;),
        axis.title.y = element_text(size =16),
        axis.text.y = element_text(size = 14, color = &quot;black&quot;),
        text= element_text(family=&quot;Times&quot;),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12)) + 
  geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
  geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
             label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                           &quot;\n p-value =&quot;, p),
             color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
  xlab(expression(hat(beta)[factor])) + 
  ylab(expression(hat(beta)[gwas])) +
  geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)



file_gene_loading &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/gene_loading/&quot;,celltype, &quot;_&quot;, factor, &quot;-&quot;, trait, &quot;.RDS&quot;)
gene_loadings_pc &lt;- readRDS(file_gene_loading)

gene_loadings_pc_abs &lt;- abs(gene_loadings_pc)
gene_loadings_pc_abs_sort &lt;- sort(gene_loadings_pc_abs,decreasing = T)
topgenes &lt;- names(gene_loadings_pc_abs_sort)[1:10]

gene_loadings_pc &lt;- as.data.frame(cbind(names(gene_loadings_pc),gene_loadings_pc))
names(gene_loadings_pc) &lt;- c(&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)

# Extract gene information based on the gene symbol
gene_info &lt;- genes(EnsDb.Hsapiens.v75, filter = ~ gene_name ==  gene_loadings_pc$gene_name)
gene_info &lt;- as.data.frame(gene_info)
gene_info &lt;- gene_info[!duplicated(gene_info$gene_name),]

gene_info &lt;- merge(gene_info,gene_loadings_pc, by=&quot;gene_name&quot;)
gene_info &lt;- gene_info[gene_info$seqnames %in%c(1:22),]

loading_plot &lt;- gene_info[,c(&quot;seqnames&quot;,&quot;start&quot;,&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)]
colnames(loading_plot) &lt;- c(&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;gene_name&quot;,&quot;loadings&quot;)
loading_plot$loadings &lt;- round(as.numeric(loading_plot$loadings),digits = 4)

don &lt;- loading_plot %&gt;%
  
  # Compute chromosome size
  group_by(chromosome_name) %&gt;%
  summarise(chr_len=max(start_position)) %&gt;%
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %&gt;%
  dplyr::select(-chr_len) %&gt;%
  
  # Add this info to the initial dataset
  left_join(loading_plot, ., by=c(&quot;chromosome_name&quot;=&quot;chromosome_name&quot;)) %&gt;%
  
  # Add a cumulative position of each SNP
  arrange(chromosome_name, start_position) %&gt;%
  mutate( start_positioncum=start_position+tot) %&gt;%
  
  # Add highlight and annotation information
  #mutate( is_annotate=ifelse(gene_name %in% loading_plot_top10$gene_name, &quot;yes&quot;, &quot;no&quot;))
  mutate( is_annotate=ifelse(gene_name %in% topgenes, &quot;yes&quot;, &quot;no&quot;))
# Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
axisdf = don %&gt;% group_by(chromosome_name) %&gt;% summarize(center=( max(start_positioncum) + min(start_positioncum) ) / 2 )

manhplot_gene &lt;- ggplot(don, aes(x=start_positioncum, y=loadings)) +
  
  # Show all points
  #rasterise(geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3),dpi=80 )+
  geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3) +
  scale_color_manual(values = rep(c(&quot;red&quot;, &quot;steelblue&quot;), 22)) +
  
  # custom X axis:
  scale_x_continuous( label = as.integer(axisdf$chromosome_name), breaks= axisdf$center ) +
  #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  
  #ylab(expression(paste(&quot;-log&quot;[10],&quot;(p-value)&quot;))) +
  #ylim(0,max(-log10(don$loadings)) + 2) +
  #ylim(min(as.numeric(don$loadings)),max(as.numeric(don$loadings))) +
  #ylab(&quot;&quot;) +
  ylab(&quot;Gene Loadings&quot;) +
  xlab(&quot;Chromosome&quot;) +
  
  # Custom the theme:
  theme_bw() +
  #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
  theme(
    legend.position=&quot;none&quot;,
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_text(size = 18),
    axis.text.x = element_text(size = 16, color = &quot;black&quot;),
    axis.title.y = element_text(size = 18),
    axis.text.y = element_text(size = 16, color = &quot;black&quot;),
    text= element_text(family=&quot;Times&quot;),
    #axis.text.y=element_blank(),
    #axis.ticks.y=element_blank()
  ) +
  geom_label_repel( data=subset(don, is_annotate==&quot;yes&quot;), aes(label=gene_name), size=4)


cat(paste0(celltype,&quot;-&quot;, pathwayname, &quot;-&quot;, trait ))</code></pre>
<pre><code>platelet-NF-kappa B signaling pathway-HB-ebi-a-GCST90002384</code></pre>
<pre class="r"><code>all &lt;- grid.arrange(eff, eff_rev, nrow = 1)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-5-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-5-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-5-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_figures.Rmd/unnamed-chunk-5-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/figure/revision_figures.Rmd/unnamed-chunk-5-1.png" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>print(manhplot_gene)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-5-2.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-5-2.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>print(paste0(&quot;Top 10 genes with largest abs loading: &quot;, paste0(topgenes, collapse = &quot;,&quot;)))</code></pre>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: LTB,LY96,TNFRSF1A,PLCG2,CD14,NFKBIA,NFKB1,PTGS2,CFLAR,PARP1&quot;</code></pre>
<ul>
<li>NF-kappa B signaling pathway -&gt; Platelet -&gt; hemoglobin</li>
</ul>
<p>NF-κB activation in platelets promotes SNAP-23–dependent granule
secretion, increases P-selectin surface expression, stimulates TXA₂ and
ADP release, and facilitates inside-out activation of αIIbβ3 integrin to
drive aggregation and thrombus formation. NF-κB activation in
endothelial cells under inflammatory conditions upregulates adhesion
molecules, enhancing platelet binding and promoting a pro-thrombotic
vascular environment. [PMID: 31461836]</p>
<p>Thrombocytopenia increases mucosal and internal bleeding risk,
leading to iron-deficiency anemia and reduced hemoglobin levels [PMID:
15248165]. Inflammatory cytokines such as IL-6 elevate platelet counts
[PMID: 7951005] and induce hepcidin, which decreases iron availability
and lowers hemoglobin [PMID: 15124018]. Platelet-derived S1P maintains
endothelial barrier integrity; when this pathway is disrupted, vascular
leak and red blood cell loss can occur [PMID: 36142188]. Additionally,
conditions characterized by excessive platelet activation (e.g., TTP,
HUS, DIC) produce platelet-rich microthrombi, causing schistocyte
formation and hemolysis [PMID: 32310498], ultimately leading to
hemoglobin reduction.</p>
<p>Also, Activated platelets, which are recruited to sites of vascular
injury, are known to release a variety of chemokines (Y. Chen et
al.,2020). The resulting chemokine-receptor interactions are crucial
regulators of hematopoiesis (Bonavita et al., 2018; Broxmeyer, 2008;
Duchene et al., 2017), leading to the change of hemoglobin.</p>
<ul>
<li>TOP loading genes -&gt; Platelet</li>
</ul>
<ol style="list-style-type: decimal">
<li>LY96</li>
</ol>
<p>LY96 (MD-2) is the LPS-binding component of the TLR4–MD2 receptor
complex. LPS binds directly to MD-2, enabling TLR4 activation and
downstream inflammatory signaling. In platelets, LPS–TLR4–MD2 signaling
promotes platelet activation and secretion [PMID: 19494325].</p>
<ol start="2" style="list-style-type: decimal">
<li>TNFRSF1A</li>
</ol>
<p>TNFRSF1A encodes TNF receptor 1 (TNFR1). TNF-α signaling through
TNFR1 enhances platelet activation and procoagulant activity, indicating
a direct stimulatory effect of TNF-α on platelets [PMID: 26819099;
8679214].</p>
<ol start="3" style="list-style-type: decimal">
<li>PLCG2</li>
</ol>
<p>PLCG2 encodes PLCγ2, the essential phospholipase required for
platelet activation downstream of the collagen receptor GPVI, as well as
integrin αIIbβ3 outside-in signaling. Loss of PLCγ2 severely impairs
platelet activation [PMID: 25793864].</p>
<ol start="4" style="list-style-type: decimal">
<li>CD14</li>
</ol>
<p>CD14 is a co-receptor that facilitates LPS–TLR4 signaling. Although
platelets do not express membrane-bound CD14, soluble CD14 enables
platelet TLR4 responsiveness. Together with LY96/MD-2, CD14 contributes
to LPS-induced platelet activation and secretion [PMID: 24585776].</p>
<ol start="5" style="list-style-type: decimal">
<li>NFKBIA</li>
</ol>
<p>NFKBIA encodes IκBα, the major inhibitory protein that restrains
NF-κB activity. Because NF-κB regulates platelet activation, granule
secretion, and integrin signaling, NFKBIA contributes to the control of
NF-κB–dependent platelet responses, as discussed before.</p>
<ol start="6" style="list-style-type: decimal">
<li>NFKB1</li>
</ol>
<p>NFKB1 encodes the NF-κB subunit p50, a core component of the platelet
NF-κB signaling machinery. NF-κB activation in platelets influences
secretion, aggregation, and thrombus formation, linking NFKB1 to
platelet inflammatory and functional responses, as discussed before.</p>
<ol start="7" style="list-style-type: decimal">
<li>CFLAR</li>
</ol>
<p>CFLAR encodes c-FLIP, a key anti-apoptotic regulator that modulates
caspase-8 activity. c-FLIP affects TNF-α–driven signaling pathways
[PMID: 23070002] and participates in mechanisms that regulate platelet
lifespan and survival.</p>
<ol start="8" style="list-style-type: decimal">
<li>PARP1</li>
</ol>
<p>PARP1 is a major substrate of activated caspases. Caspase signaling
has both apoptotic and non-apoptotic roles in hematopoiesis, including
megakaryocyte differentiation and platelet maturation. [PMID:
21176168].</p>
<!-- 1. LY96 -->
<!-- LY96 (MD-2) is the LPS-binding component of the TLR4 receptor. LPS binds to MD-2, enabling TLR4–MD2 activation and downstream inflammatory signaling. LPS stimulates platelet secretion [PMID: 19494325].  -->
<!-- 2. TNFRSF1A -->
<!-- TNFRSF1A is TNF receptor 1. Increased levels of TNF-α are known to activate platelets[PMID: 26819099, 8679214].  -->
<!-- 3. PLCG2 -->
<!-- PLCG2 encodes PLCγ2, which is required for platelet activation. [PMID: 25793864] -->
<!-- 4. CD14 -->
<!-- CD14 is a coreceptor for LPS signaling through TLR4. Combine with the evidence of LY96, it is related to latelet secretion.[PMID: 24585776] -->
<!-- 5. NFKBIA  -->
<!-- NFKBIA is the gene that encodes IκBα, the major inhibitory protein that keeps NF-κB inactive under resting conditions. We have discussed the role of  NF-κB in platelets before.  -->
<!-- 6. NFKB1 -->
<!-- Critical part of platelet NF-κB machinery. We have discussed the role of  NF-κB in platelets -->
<!-- 7. CFLAR  -->
<!-- CFLAR encodes c-FLIP, is a master anti-apoptotic regulator and resistance factor that suppresses tumor necrosis factor-α (TNF-α) [PMID: 23070002], which has been discussed earlier.  -->
<!-- 8. PARP1  -->
<!-- PARP1 is a key substrate of activated caspases, which are central regulators of apoptosis and also have important non-apoptotic roles in hematopoiesis, including platelet maturation. [PMID: 21176168] -->
</div>
<div
id="example-2-leukocyte-transendothelial-migration---t-cell-cd-8---ibd-ebi-a-gcst003043"
class="section level3">
<h3>example 2, Leukocyte transendothelial migration - T cell (CD 8+) -
IBD-ebi-a-GCST003043</h3>
<pre class="r"><code>celltype &lt;- &quot;thymocyte&quot;
trait &lt;- &quot;IBD-ebi-a-GCST003043&quot;
factor &lt;- &quot;04670_PC1&quot;
fdr_cutoff &lt;- 0.2
  
file_harmo &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;,celltype, &quot;-&quot;, trait, &quot;_harmo.RDS&quot;)
dt_harmo &lt;- readRDS(file_harmo)

df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

dat &lt;- dt_harmo[[factor]]
snp_select &lt;- dat[dat$FDR.exposure &lt; fdr_cutoff,]
snp_select$chr &lt;- as.numeric(as.character(snp_select$chr.exposure))
snp_select$chr &lt;- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))

fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)

ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]

pathwayname &lt;- df$pathwayName[df$factor == factor &amp; df$trait ==trait]

eff &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
  geom_point(size = 4) + 
  geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
  theme_bw(base_line_size = 0.3) +
  theme(axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = &quot;black&quot;),
        axis.title.y = element_text(size =16),
        axis.text.y = element_text(size = 14, color = &quot;black&quot;),
        text= element_text(family=&quot;Times&quot;),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12)) + 
  geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
  geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
             label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                           &quot;\nECT p-value =&quot;, ect_p),
             color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
  xlab(expression(hat(beta)[factor])) + 
  ylab(expression(hat(beta)[gwas])) +
  geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)


#snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff &amp; dat$steiger_dir ==T,]
snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff,]

fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
coef &lt;- summary(fit)
p &lt;- round(coef$coefficients[,4], digits = 4)
#ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]

eff_rev &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome)) +
  geom_point(size = 4, color =&quot;steelblue&quot;) + 
  theme_bw(base_line_size = 0.3) +
  theme(axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = &quot;black&quot;),
        axis.title.y = element_text(size =16),
        axis.text.y = element_text(size = 14, color = &quot;black&quot;),
        text= element_text(family=&quot;Times&quot;),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12)) + 
  geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
  geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
             label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                           &quot;\n p-value =&quot;, p),
             color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
  xlab(expression(hat(beta)[factor])) + 
  ylab(expression(hat(beta)[gwas])) +
  geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)



file_gene_loading &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/gene_loading/&quot;,celltype, &quot;_&quot;, factor, &quot;-&quot;, trait, &quot;.RDS&quot;)
gene_loadings_pc &lt;- readRDS(file_gene_loading)

gene_loadings_pc_abs &lt;- abs(gene_loadings_pc)
gene_loadings_pc_abs_sort &lt;- sort(gene_loadings_pc_abs,decreasing = T)
topgenes &lt;- names(gene_loadings_pc_abs_sort)[1:10]

gene_loadings_pc &lt;- as.data.frame(cbind(names(gene_loadings_pc),gene_loadings_pc))
names(gene_loadings_pc) &lt;- c(&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)

# Extract gene information based on the gene symbol
gene_info &lt;- genes(EnsDb.Hsapiens.v75, filter = ~ gene_name ==  gene_loadings_pc$gene_name)
gene_info &lt;- as.data.frame(gene_info)
gene_info &lt;- gene_info[!duplicated(gene_info$gene_name),]

gene_info &lt;- merge(gene_info,gene_loadings_pc, by=&quot;gene_name&quot;)
gene_info &lt;- gene_info[gene_info$seqnames %in%c(1:22),]

loading_plot &lt;- gene_info[,c(&quot;seqnames&quot;,&quot;start&quot;,&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)]
colnames(loading_plot) &lt;- c(&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;gene_name&quot;,&quot;loadings&quot;)
loading_plot$loadings &lt;- round(as.numeric(loading_plot$loadings),digits = 4)

don &lt;- loading_plot %&gt;%
  
  # Compute chromosome size
  group_by(chromosome_name) %&gt;%
  summarise(chr_len=max(start_position)) %&gt;%
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %&gt;%
  dplyr::select(-chr_len) %&gt;%
  
  # Add this info to the initial dataset
  left_join(loading_plot, ., by=c(&quot;chromosome_name&quot;=&quot;chromosome_name&quot;)) %&gt;%
  
  # Add a cumulative position of each SNP
  arrange(chromosome_name, start_position) %&gt;%
  mutate( start_positioncum=start_position+tot) %&gt;%
  
  # Add highlight and annotation information
  #mutate( is_annotate=ifelse(gene_name %in% loading_plot_top10$gene_name, &quot;yes&quot;, &quot;no&quot;))
  mutate( is_annotate=ifelse(gene_name %in% topgenes, &quot;yes&quot;, &quot;no&quot;))
# Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
axisdf = don %&gt;% group_by(chromosome_name) %&gt;% summarize(center=( max(start_positioncum) + min(start_positioncum) ) / 2 )

manhplot_gene &lt;- ggplot(don, aes(x=start_positioncum, y=loadings)) +
  
  # Show all points
  #rasterise(geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3),dpi=80 )+
  geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3) +
  scale_color_manual(values = rep(c(&quot;red&quot;, &quot;steelblue&quot;), 22)) +
  
  # custom X axis:
  scale_x_continuous( label = as.integer(axisdf$chromosome_name), breaks= axisdf$center ) +
  #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  
  #ylab(expression(paste(&quot;-log&quot;[10],&quot;(p-value)&quot;))) +
  #ylim(0,max(-log10(don$loadings)) + 2) +
  #ylim(min(as.numeric(don$loadings)),max(as.numeric(don$loadings))) +
  #ylab(&quot;&quot;) +
  ylab(&quot;Gene Loadings&quot;) +
  xlab(&quot;Chromosome&quot;) +
  
  # Custom the theme:
  theme_bw() +
  #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
  theme(
    legend.position=&quot;none&quot;,
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_text(size = 18),
    axis.text.x = element_text(size = 16, color = &quot;black&quot;),
    axis.title.y = element_text(size = 18),
    axis.text.y = element_text(size = 16, color = &quot;black&quot;),
    text= element_text(family=&quot;Times&quot;),
    #axis.text.y=element_blank(),
    #axis.ticks.y=element_blank()
  ) +
  geom_label_repel( data=subset(don, is_annotate==&quot;yes&quot;), aes(label=gene_name), size=4)


cat(paste0(celltype,&quot;-&quot;, pathwayname, &quot;-&quot;, trait ))</code></pre>
<pre><code>thymocyte-Leukocyte transendothelial migration-IBD-ebi-a-GCST003043</code></pre>
<pre class="r"><code>all &lt;- grid.arrange(eff, eff_rev, nrow = 1)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-6-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-6-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-6-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_figures.Rmd/unnamed-chunk-6-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/figure/revision_figures.Rmd/unnamed-chunk-6-1.png" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>print(manhplot_gene)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-6-2.png" width="739.2" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(paste0(&quot;Top 10 genes with largest abs loading: &quot;, paste0(topgenes, collapse = &quot;,&quot;)))</code></pre>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: CTNNA1,ACTB,PIK3R2,ITGAL,ACTN4,ITGB2,RAC2,MSN,MAPK13,SIPA1&quot;</code></pre>
<ul>
<li>Leukocyte transendothelial migration -&gt; CD8⁺ T cells -&gt;
IBD</li>
</ul>
<p>Leukocyte transendothelial migration (TEM) is the process by which
white blood cells (leukocytes) exit the bloodstream and enter tissues
[PMID: 21073340, 33855018]. Because CD8⁺ T cells are leukocytes, TEM
directly enables their recruitment to inflammatory sites, allowing them
to exert their cytotoxic and cytokine-mediated functions.</p>
<p>Emerging evidence shows that CD8⁺ T cells play important roles in
IBD. Chronic intestinal infection can drive epithelial barrier leakage
through type I IFN signaling and CD8⁺ T-cell activity, and blocking CD8⁺
T cells prevents this barrier breakdown [PMID: 32880630]. At the same
time, CD8⁺ T cells in IBD are highly heterogeneous: cytotoxic (Tc1) and
IL-17–producing (Tc17) subsets promote inflammation, whereas other CD8⁺
subsets may be neutral or even protective [PMID: 34707610].</p>
<ul>
<li>TOP loading genes -&gt; CD8⁺ T cells</li>
</ul>
<p>1 &amp; 2 . ITGAL &amp; ITGB2</p>
<!-- ITGAL is essential for CD8⁺ T-cell activation, where its product, the CD11a α-subunit of LFA-1, supports magnesium-dependent immune synapse formation and antigen-specific cytotoxicity [PMID: 39605903].  -->
<p>ITGAL and ITGB2 form LFA-1, which has been shown to facilitate T-cell
activation by functioning as a scavenger receptor on CD8⁺ dendritic
cells, enabling them to acquire antigen from antigen-bearing DCs [PMID:
17641014].</p>
<ol start="3" style="list-style-type: decimal">
<li>RAC2</li>
</ol>
<p>RAC2 is a member of the Rac subfamily of Rho GTPases. It plays
essential roles in T-cell development, cytoskeletal remodeling, and
survival[PMID: 19088377, 11581314, 18579797]. <strong>In the absence of
Rac2, the motility of T cells in response to CXCL12, CCL19, and CCL21
was partially reduced [PMID: 20870900]</strong></p>
<ol start="4" style="list-style-type: decimal">
<li>SIPA1</li>
</ol>
<p>Sipa1 was shown to inhibit thymocyte development [PMID: 37545505]</p>
<ol start="5" style="list-style-type: decimal">
<li>PIK3R2</li>
</ol>
<p>PIK3R2 is a key component of the PI3K–Akt signaling pathway, which
plays a central role in T-cell exhaustion[PMID 17164340]</p>
<ol start="6" style="list-style-type: decimal">
<li>MAPK13</li>
</ol>
<p>MAPK13 phosphorylates TCF1, leading to increase of stem-like T cells
[PMID 38272565].</p>
<ol start="7" style="list-style-type: decimal">
<li>MSN</li>
</ol>
<p>MSN (Moesin) are implicated in promoting T cell activation and
polarity [PMID: 19124745]</p>
<ol start="8" style="list-style-type: decimal">
<li>ACTN4</li>
</ol>
<p><strong>ACTN4 has been shown to play important roles in cytoskeleton
organization, cell adhesion, and cell migration [PMID:
25885339]</strong></p>
</div>
</div>
</div>
<div id="si-figures" class="section level1">
<h1>SI figures</h1>
<p>The main change for figure S1-S3 is: more factors are included since
we decreased the gene num cutoff from 40 to 20.</p>
<div id="s1-hsq-and-se" class="section level2">
<h2>S1 hsq and SE</h2>
<pre class="r"><code>p1 &lt;- list()
p2 &lt;- list()
for (celltype in celltypes){
  
  file_hsq &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/7.analysis_for_factors/results/heritability/hsq_summary_&quot;, celltype,&quot;.txt&quot;)
  
  H &lt;- read.table(file_hsq, header = T)
  sum_sorted &lt;- H[order(as.numeric(H$h2)),]
  sum_sorted$order &lt;- seq(1,nrow(H),by=1)
  
  p1[[celltype]] &lt;- ggplot(sum_sorted, aes(x=order, y=h2)) + theme_bw(base_line_size =0.3) +
    geom_errorbar(aes(ymin=h2-SE, ymax=h2+SE), width=0.1, color = &quot;grey&quot;,alpha=0.6) +
    geom_point(size = 0.8, color = &quot;steelblue&quot;) + 
    ggtitle(types[celltype]) + theme(plot.title = element_text(hjust = 0.5)) + 
    xlab(&quot;PCs sorted by heritability&quot;) + ylab(&quot;Heritability ± SE&quot;) + 
    ylim(-2,3)  + 
    theme(axis.title.x = element_text(size = 14),
          axis.text.x = element_text(size = 12, color = &quot;black&quot;),
          axis.title.y = element_text(size = 14),
          axis.text.y = element_text(size = 12, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;)) 
  
  plotdata &lt;- data.frame(observed = -log10(sort(H$Pval)),
                         expected = -log10(ppoints(nrow(H))))
  
  
  p2[[celltype]] &lt;- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5, color = &quot;steelblue&quot;) +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = &quot;red&quot;) +
    ggtitle(types[celltype]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste(&quot;Expected -log&quot;[10],&quot;(p-value)&quot;)),
         y = expression(paste(&quot;Observed -log&quot;[10],&quot;(p-value)&quot;))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = &quot;black&quot;),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;)) 
  
  
  
  
}


all_interval &lt;- grid.arrange(p1[[1]],p1[[2]],p1[[3]],p1[[4]],p1[[5]],p1[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-7-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-7-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-7-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_figures.Rmd/unnamed-chunk-7-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/figure/revision_figures.Rmd/unnamed-chunk-7-1.png" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="s2-hsq-qqplot-for-p-values" class="section level2">
<h2>S2 hsq qqplot for p-values</h2>
<pre class="r"><code>all_qq &lt;- grid.arrange(p2[[1]],p2[[2]],p2[[3]],p2[[4]],p2[[5]],p2[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-8-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-8-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-8-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_figures.Rmd/unnamed-chunk-8-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/figure/revision_figures.Rmd/unnamed-chunk-8-1.png" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="s3-number-of-independent-signals-for-factor-gwas-factors-filtered-by-hsq-0.9-hsq-pvalue-0.05"
class="section level2">
<h2>S3 Number of independent signals for factor gwas (factors filtered
by hsq &gt;0.9 &amp; hsq pvalue &lt; 0.05)</h2>
<pre class="r"><code>p1 &lt;- list()
for (celltype in celltypes){

  file &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/7.analysis_for_factors/results/num_gwas_signals_&quot;, celltype, &quot;.RDS&quot;)
  num_signal_all &lt;- readRDS(file)

  num_signal_all &lt;- as.data.frame(num_signal_all)
  colnames(num_signal_all) &lt;- &quot;num&quot;
  #p1[[i]] &lt;-  ggplot(num_signal_all, aes(x=num)) + geom_histogram(breaks = seq(0, max(num_signal_all), by = 1),color=&quot;white&quot;,fill = &quot;steelblue&quot;)+
  p1[[celltype]] &lt;-  ggplot(num_signal_all, aes(x=num)) + geom_histogram(bins = 30,color=&quot;white&quot;,fill = &quot;steelblue&quot;)+
    theme_bw(base_line_size =0.3) +
    ggtitle(types[celltype]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = &quot;# of independent signals&quot;,
         y = &quot;Count&quot;) +

    theme(axis.title.x = element_text(size = 14),
          axis.text.x = element_text(size = 12, color = &quot;black&quot;),
          axis.title.y = element_text(size = 14),
          axis.text.y = element_text(size = 12, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;)) +

    annotate(geom = &quot;text&quot;, x = Inf, y = Inf, label = paste0(&quot;The total number of tested factor = &quot;,nrow(num_signal_all)),
             hjust = 1, vjust = 1,  size = 2,  family = &quot;Times&quot;,   color = &quot;black&quot;)


}

all1 &lt;- grid.arrange(p1[[1]],p1[[2]],p1[[3]],p1[[4]],p1[[5]],p1[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-9-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-9-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-9-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_figures.Rmd/unnamed-chunk-9-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/figure/revision_figures.Rmd/unnamed-chunk-9-1.png" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="s4-2-examples-for-acat" class="section level2">
<h2>S4 2 examples for ACAT</h2>
<p>same examples with initial submission. but the snps are a bit
different</p>
<pre class="r"><code>folder_harmo &lt;- &quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;


celltype &lt;- &quot;CD14_positive_monocyte&quot;
trait &lt;- &quot;WBC-ebi-a-GCST90002407&quot;
factor &lt;- &quot;04141_PC1&quot;

file_harmo &lt;- paste0(folder_harmo,celltype,&quot;-&quot;,trait,&quot;_harmo.RDS&quot;)
list_harmo &lt;- readRDS(file_harmo)

df_harmo &lt;- list_harmo[[factor]]

p1 &lt;- qqplot(df_harmo$pval.exposure, title = &quot;1st PC of Protein processing in \nendoplasmic reticulum - WBC&quot;)


celltype &lt;- &quot;platelet&quot;
trait &lt;- &quot;RET-ebi-a-GCST90002405&quot;
factor &lt;- &quot;04390_PC5&quot;

file_harmo &lt;- paste0(folder_harmo,celltype,&quot;-&quot;,trait,&quot;_harmo.RDS&quot;)
list_harmo &lt;- readRDS(file_harmo)

df_harmo &lt;- list_harmo[[factor]]

p2 &lt;- qqplot(df_harmo$pval.exposure, title = &quot;5th PC of Hippo signaling pathway \n - Reticulocyte Count&quot;)


all &lt;- grid.arrange(p1,p2, nrow = 1)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-10-1.png" width="528" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-10-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-10-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_figures.Rmd/unnamed-chunk-10-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/33e06f736eed2fbc1d073176551806e5cd0c37eb/docs/figure/revision_figures.Rmd/unnamed-chunk-10-1.png" target="_blank">33e06f7</a>
</td>
<td>
XSun
</td>
<td>
2025-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="s5-number-of-supporting-snps-per-pair" class="section level2">
<h2>S5 Number of supporting SNPs per pair</h2>
<p>difference: the current cutoff is &gt;=5SNPs and &gt;=20genes</p>
<div id="eur" class="section level3">
<h3>EUR</h3>
<pre class="r"><code>p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR.RDS&quot;))
  df &lt;- df[df$trait %in% traits_EUR,]
  
  summary &lt;- df %&gt;%
    count(num_supp_SNP, name = &quot;Freq&quot;)
  
  # Plot
  p[[celltype]] &lt;- ggplot(summary, aes(x = num_supp_SNP, y = Freq)) +
    geom_bar(stat = &quot;identity&quot;, fill = &quot;steelblue&quot;) +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = &quot;# of supporting SNPs in each pair&quot;, y = &quot;Count&quot;) +
    geom_text(
      aes(label = ifelse(num_supp_SNP %% 5 == 0, as.character(Freq), &quot;&quot;)),
      colour = &quot;black&quot;, size = 2, vjust = -0.1, family = &quot;Times&quot;
    ) +
    annotate(
      geom = &quot;text&quot;,
      x = Inf, y = Inf,
      label = paste0(&quot;# of candidate pairs = &quot;, sum(summary$Freq[summary$num_supp_SNP &gt;= 5])),
      hjust = 1.1, vjust = 1.5, size = 3, family = &quot;Times&quot;, color = &quot;black&quot;
    ) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = &quot;black&quot;),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = &quot;black&quot;),
      text = element_text(family = &quot;Times&quot;)
    ) +
    scale_x_continuous(breaks = seq(0, max(summary$num_supp_SNP), 5))
  
}


all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-11-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-11-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-11-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_figures.Rmd/unnamed-chunk-11-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="eas" class="section level3">
<h3>EAS</h3>
<pre class="r"><code>p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EAS.RDS&quot;))
  df &lt;- df[df$trait %in% traits_EAS,]
  
  summary &lt;- df %&gt;%
    count(num_supp_SNP, name = &quot;Freq&quot;)
  
  # Plot
  p[[celltype]] &lt;- ggplot(summary, aes(x = num_supp_SNP, y = Freq)) +
    geom_bar(stat = &quot;identity&quot;, fill = &quot;steelblue&quot;) +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = &quot;# of supporting SNPs in each pair&quot;, y = &quot;Count&quot;) +
    geom_text(
      aes(label = ifelse(num_supp_SNP %% 5 == 0, as.character(Freq), &quot;&quot;)),
      colour = &quot;black&quot;, size = 2, vjust = -0.1, family = &quot;Times&quot;
    ) +
    annotate(
      geom = &quot;text&quot;,
      x = Inf, y = Inf,
      label = paste0(&quot;# of candidate pairs = &quot;, sum(summary$Freq[summary$num_supp_SNP &gt;= 5])),
      hjust = 1.1, vjust = 1.5, size = 3, family = &quot;Times&quot;, color = &quot;black&quot;
    ) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = &quot;black&quot;),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = &quot;black&quot;),
      text = element_text(family = &quot;Times&quot;)
    ) +
    scale_x_continuous(breaks = seq(0, max(summary$num_supp_SNP), 5))
  
}


all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-12-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-12-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-12-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="s6-ect-p-values" class="section level2">
<h2>S6 ECT p-values</h2>
<div id="eur-1" class="section level3">
<h3>EUR</h3>
<pre class="r"><code>p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR.RDS&quot;))
  df &lt;- df[df$trait %in% traits_EUR,]
  
  df &lt;- df[complete.cases(df$p_ECT),]
  
  summary_p &lt;- df %&gt;%
    mutate(bin = cut(p_ECT, breaks = seq(0, 1, by = 0.05), include.lowest = TRUE)) %&gt;%
    count(bin, name = &quot;Freq&quot;)
  
  # Correctly handle both [ and ( at the start of interval labels
  summary_p &lt;- summary_p %&gt;%
    mutate(
      bin_low  = as.numeric(sub(&quot;[^0-9\\.]*([0-9\\.]+),.*&quot;, &quot;\\1&quot;, bin)),
      bin_high = as.numeric(sub(&quot;.*,([0-9\\.]+)\\]&quot;, &quot;\\1&quot;, bin)),
      bin_mid  = (bin_low + bin_high) / 2
    )
  
  p[[celltype]] &lt;- ggplot(summary_p, aes(x = bin_mid, y = Freq)) +
    geom_bar(stat = &quot;identity&quot;, fill = &quot;steelblue&quot;) +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = &quot;ECT p-value&quot;, y = &quot;Count&quot;) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = &quot;black&quot;),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = &quot;black&quot;),
      text = element_text(family = &quot;Times&quot;)
    ) +
    scale_x_continuous(breaks = seq(0, 1, 0.1), limits = c(0, 1))
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-13-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-13-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/docs/figure/revision_figures.Rmd/unnamed-chunk-13-1.png" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR.RDS&quot;))
  df &lt;- df[df$trait %in% traits_EUR,]
  
  df &lt;- df[complete.cases(df$p_ECT),]
  
  p[[celltype]] &lt;- qqplot(pvalues = df$p_ECT, title = types[celltype])
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-13-2.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-2">
Past versions of unnamed-chunk-13-2.png
</button>
</p>
<div id="fig-unnamed-chunk-13-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-13-2.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="eas-1" class="section level3">
<h3>EAS</h3>
<pre class="r"><code>p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EAS.RDS&quot;))
  df &lt;- df[df$trait %in% traits_EAS,]
  
  df &lt;- df[complete.cases(df$p_ECT),]
  
  summary_p &lt;- df %&gt;%
    mutate(bin = cut(p_ECT, breaks = seq(0, 1, by = 0.05), include.lowest = TRUE)) %&gt;%
    count(bin, name = &quot;Freq&quot;)
  
  # Correctly handle both [ and ( at the start of interval labels
  summary_p &lt;- summary_p %&gt;%
    mutate(
      bin_low  = as.numeric(sub(&quot;[^0-9\\.]*([0-9\\.]+),.*&quot;, &quot;\\1&quot;, bin)),
      bin_high = as.numeric(sub(&quot;.*,([0-9\\.]+)\\]&quot;, &quot;\\1&quot;, bin)),
      bin_mid  = (bin_low + bin_high) / 2
    )
  
  p[[celltype]] &lt;- ggplot(summary_p, aes(x = bin_mid, y = Freq)) +
    geom_bar(stat = &quot;identity&quot;, fill = &quot;steelblue&quot;) +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = &quot;ECT p-value&quot;, y = &quot;Count&quot;) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = &quot;black&quot;),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = &quot;black&quot;),
      text = element_text(family = &quot;Times&quot;)
    ) +
    scale_x_continuous(breaks = seq(0, 1, 0.1), limits = c(0, 1))
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-14-1.png" width="739.2" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-14-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EAS.RDS&quot;))
  df &lt;- df[df$trait %in% traits_EAS,]
  
  df &lt;- df[complete.cases(df$p_ECT),]
  
  p[[celltype]] &lt;- qqplot(pvalues = df$p_ECT, title = types[celltype])
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-14-2.png" width="739.2" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="s7-ect-storey-pi1" class="section level2">
<h2>S7 ECT storey pi1</h2>
<div id="eur-2" class="section level3">
<h3>EUR</h3>
<pre class="r"><code>load(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/ECT_story_EUR.rdata&quot;)

values &lt;- pi1_all
labels &lt;- c(&quot;B cell \n(CD19+)&quot;,&quot;Monocyte \n(CD14+)&quot;,&quot;Leukocyte \n(CD15+)&quot;,&quot;Platelet&quot;,&quot;T cell \n(CD4+)&quot;,&quot;T cell \n(CD8+)&quot;)
data &lt;- data.frame(labels, values)

# Create the bar plot
p &lt;- ggplot(data, aes(x=labels, y=values, fill=labels)) +
  geom_bar(stat=&quot;identity&quot;, color=&quot;steelblue&quot;, fill=&quot;steelblue&quot;) +
  labs(x=&quot;Cell Types&quot;, y=&quot;Storey&#39;s Pi1&quot;) +
  theme_minimal()  +
  
  theme(axis.title.x = element_text(size = 14),
        axis.text.x = element_text(size = 12, color = &quot;black&quot;, angle = 45, hjust = 1,vjust = 1.1),
        axis.title.y = element_text(size = 14),
        axis.text.y = element_text(size = 12, color = &quot;black&quot;),
        text= element_text(family=&quot;Times&quot;)) 

p</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-15-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_figures.Rmd/unnamed-chunk-15-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="eas-2" class="section level3">
<h3>EAS</h3>
<pre class="r"><code>load(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/ECT_story_EAS.rdata&quot;)

values &lt;- pi1_all
labels &lt;- c(&quot;B cell \n(CD19+)&quot;,&quot;Monocyte \n(CD14+)&quot;,&quot;Leukocyte \n(CD15+)&quot;,&quot;Platelet&quot;,&quot;T cell \n(CD4+)&quot;,&quot;T cell \n(CD8+)&quot;)
data &lt;- data.frame(labels, values)

# Create the bar plot
p &lt;- ggplot(data, aes(x=labels, y=values, fill=labels)) +
  geom_bar(stat=&quot;identity&quot;, color=&quot;steelblue&quot;, fill=&quot;steelblue&quot;) +
  labs(x=&quot;Cell Types&quot;, y=&quot;Storey&#39;s Pi1&quot;) +
  theme_minimal()  +
  
  theme(axis.title.x = element_text(size = 14),
        axis.text.x = element_text(size = 12, color = &quot;black&quot;, angle = 45, hjust = 1,vjust = 1.1),
        axis.title.y = element_text(size = 14),
        axis.text.y = element_text(size = 12, color = &quot;black&quot;),
        text= element_text(family=&quot;Times&quot;)) 

p</code></pre>
<p><img src="figure/revision_figures.Rmd/unnamed-chunk-16-1.png" width="480" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggrepel_0.9.1             EnsDb.Hsapiens.v75_2.99.0
 [3] ensembldb_2.20.2          AnnotationFilter_1.20.0  
 [5] GenomicFeatures_1.48.3    AnnotationDbi_1.58.0     
 [7] Biobase_2.56.0            GenomicRanges_1.48.0     
 [9] GenomeInfoDb_1.39.9       IRanges_2.30.0           
[11] S4Vectors_0.34.0          BiocGenerics_0.42.0      
[13] forcats_0.5.1             stringr_1.5.1            
[15] purrr_1.0.2               readr_2.1.2              
[17] tidyr_1.3.0               tibble_3.2.1             
[19] tidyverse_1.3.1           dplyr_1.1.4              
[21] ggplot2_4.0.0             gridExtra_2.3            

loaded via a namespace (and not attached):
  [1] rjson_0.2.21                ellipsis_0.3.2             
  [3] rprojroot_2.0.3             XVector_0.36.0             
  [5] fs_1.5.2                    dichromat_2.0-0.1          
  [7] rstudioapi_0.13             farver_2.1.0               
  [9] DT_0.22                     bit64_4.0.5                
 [11] fansi_1.0.3                 lubridate_1.8.0            
 [13] xml2_1.3.3                  codetools_0.2-18           
 [15] cachem_1.0.6                knitr_1.39                 
 [17] jsonlite_1.8.0              workflowr_1.7.0            
 [19] Rsamtools_2.12.0            broom_0.8.0                
 [21] dbplyr_2.1.1                png_0.1-7                  
 [23] compiler_4.2.0              httr_1.4.3                 
 [25] backports_1.4.1             lazyeval_0.2.2             
 [27] Matrix_1.5-3                assertthat_0.2.1           
 [29] fastmap_1.1.0               cli_3.6.1                  
 [31] later_1.3.0                 htmltools_0.5.2            
 [33] prettyunits_1.1.1           tools_4.2.0                
 [35] gtable_0.3.6                glue_1.6.2                 
 [37] GenomeInfoDbData_1.2.8      rappdirs_0.3.3             
 [39] Rcpp_1.0.12                 cellranger_1.1.0           
 [41] jquerylib_0.1.4             vctrs_0.6.5                
 [43] Biostrings_2.64.0           rtracklayer_1.56.0         
 [45] crosstalk_1.2.0             xfun_0.41                  
 [47] rvest_1.0.2                 lifecycle_1.0.4            
 [49] restfulr_0.0.14             XML_3.99-0.14              
 [51] zlibbioc_1.42.0             scales_1.4.0               
 [53] ProtGenerics_1.28.0         MatrixGenerics_1.8.0       
 [55] hms_1.1.1                   promises_1.2.0.1           
 [57] SummarizedExperiment_1.26.1 parallel_4.2.0             
 [59] RColorBrewer_1.1-3          yaml_2.3.5                 
 [61] curl_4.3.2                  memoise_2.0.1              
 [63] sass_0.4.1                  biomaRt_2.54.1             
 [65] stringi_1.7.6               RSQLite_2.3.1              
 [67] highr_0.9                   BiocIO_1.6.0               
 [69] filelock_1.0.2              BiocParallel_1.30.3        
 [71] matrixStats_0.62.0          rlang_1.1.2                
 [73] pkgconfig_2.0.3             bitops_1.0-7               
 [75] lattice_0.20-45             evaluate_0.15              
 [77] labeling_0.4.2              htmlwidgets_1.5.4          
 [79] GenomicAlignments_1.32.0    bit_4.0.4                  
 [81] tidyselect_1.2.0            magrittr_2.0.3             
 [83] R6_2.5.1                    generics_0.1.2             
 [85] DelayedArray_0.22.0         DBI_1.2.2                  
 [87] pillar_1.9.0                haven_2.5.0                
 [89] whisker_0.4                 withr_2.5.0                
 [91] KEGGREST_1.36.3             RCurl_1.98-1.7             
 [93] modelr_0.1.8                crayon_1.5.1               
 [95] utf8_1.2.2                  BiocFileCache_2.4.0        
 [97] tzdb_0.4.0                  rmarkdown_2.25             
 [99] progress_1.2.2              grid_4.2.0                 
[101] readxl_1.4.0                blob_1.2.3                 
[103] git2r_0.30.1                S7_0.2.0                   
[105] reprex_2.0.1                digest_0.6.29              
[107] httpuv_1.6.5                bslib_0.3.1                </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
