<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-11-16" />

<title>Power analysis</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis_new</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis_new">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Power analysis</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-11-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis_new/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree0fb1c37548b1b850b3036a1870bc5bed17db3e20targetblank0fb1c37a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis_new/tree/0fb1c37548b1b850b3036a1870bc5bed17db3e20" target="_blank">0fb1c37</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree0fb1c37548b1b850b3036a1870bc5bed17db3e20targetblank0fb1c37a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis_new/tree/0fb1c37548b1b850b3036a1870bc5bed17db3e20" target="_blank">0fb1c37</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/revision_power_simulate.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/revision_power_simulate.Rmd</code>) and HTML
(<code>docs/revision_power_simulate.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0fb1c37548b1b850b3036a1870bc5bed17db3e20/analysis/revision_power_simulate.Rmd" target="_blank">0fb1c37</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/0fb1c37548b1b850b3036a1870bc5bed17db3e20/docs/revision_power_simulate.html" target="_blank">0fb1c37</a>
</td>
<td>
XSun
</td>
<td>
2025-11-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/bb3beb06fe4f5af31ddb1995851a83492ca15718/analysis/revision_power_simulate.Rmd" target="_blank">bb3beb0</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/bb3beb06fe4f5af31ddb1995851a83492ca15718/docs/revision_power_simulate.html" target="_blank">bb3beb0</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/d42343d6b42ab1e52cb3aa4c75d6170cded5cc3c/analysis/revision_power_simulate.Rmd" target="_blank">d42343d</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/977037c765dc7f77a1fec555ab6788033e846f25/analysis/revision_power_simulate.Rmd" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/977037c765dc7f77a1fec555ab6788033e846f25/docs/revision_power_simulate.html" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/analysis/revision_power_simulate.Rmd" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/6920c6bf856ac9c848365efe0a92384b867c3177/docs/revision_power_simulate.html" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/analysis/revision_power_simulate.Rmd" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/revision_power_simulate.html" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/analysis/revision_power_simulate.Rmd" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/revision_power_simulate.html" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/23a11ad02272616669af2010c84af8abe09401d9/analysis/revision_power_simulate.Rmd" target="_blank">23a11ad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>To evaluate the statistical power of the ECT under reduced GWAS
sample sizes, we performed a simulation procedure for some factor-trait
pairs.</p>
<ul>
<li>Estimating the true effect size of factor on trait</li>
</ul>
<p>ECT models the relationship between SNP effects on the factor
(exposure) and on the trait (outcome) using a no-intercept
regression:</p>
<p><span class="math display">\[
\beta^{\text{GWAS}} = \gamma \cdot \beta^{\text{factor}} + \varepsilon ,
\]</span></p>
<p>where <span class="math inline">\(\gamma\)</span> represents the
causal effect of the factor on the trait.</p>
<p>For each factor–trait pair, we extracted supporting SNPs with <span
class="math inline">\(\mathrm{FDR}_{\text{exposure}} &lt; 0.2\)</span>
and estimated <span class="math inline">\(\gamma\)</span> by
fitting:</p>
<p><span class="math display">\[
\beta^{\text{GWAS}}_j = \gamma \, \beta^{\text{factor}}_j +
\varepsilon_j .
\]</span> The fitted slope <span
class="math inline">\(\hat{\gamma}\)</span> was taken as the estimated
true effect.</p>
<p>For each SNP <span class="math inline">\(j\)</span>, the expected
(true) GWAS effect is therefore:</p>
<p><span class="math display">\[
\beta^{\text{true}}_j = \hat{\gamma} \cdot \beta^{\text{factor}}_j .
\]</span></p>
<ul>
<li>Simulating down-sampled GWAS effect estimates</li>
</ul>
<p>Assuming the standard error of a GWAS effect estimate scales with
sample size as <span class="math inline">\(SE \propto N^{-1/2}\)</span>,
a down-sampled GWAS using a proportion <span
class="math inline">\(d\)</span> of the original sample has:</p>
<p><span class="math display">\[
SE^{\text{new}}_j = \frac{SE_j}{\sqrt{d}} .
\]</span></p>
<p>For each supporting SNP, we simulated GWAS estimates under the
down-sampled scenario according to:</p>
<p><span class="math display">\[
\hat{\beta}_j \sim N\!\left(
      \beta^{\text{true}}_j,\
      SE^{\text{new}}_j
\right).
\]</span></p>
<p>We generated 100 independent simulations per SNP, yielding 100
complete GWAS-like datasets.</p>
<ul>
<li>Re-running ECT on simulated datasets</li>
</ul>
<p>For each simulated dataset, we replaced the observed GWAS effect
sizes with the simulated values <span
class="math inline">\(\hat{\beta}_j\)</span>, retained the original
exposure effect estimates and <span
class="math inline">\(p\)</span>-values, and re-ran ECT for the
corresponding factor–trait pair.</p>
<p>This procedure produced 100 ECT results, each representing the
inference obtained under a GWAS with reduced sample size.</p>
<ul>
<li>Power estimation</li>
</ul>
<p>Power was estimated as the proportion of simulation rounds in which
the ECT result remained statistically significant under the specified
down-sampling proportion.</p>
<pre class="r"><code>suppressMessages(library(ggplot2))
suppressMessages(library(gridExtra))

celltypes &lt;- c(&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;CD15_positive_leukocyte&quot;,&quot;platelet&quot;,&quot;T_cell&quot;,&quot;thymocyte&quot;)
types &lt;- c(&quot;B cell (CD19+)&quot;,&quot;Monocyte (CD14+)&quot;,&quot;Granulocyte (CD15+)&quot;,&quot;Platelet&quot;,&quot;T cell (CD4+)&quot;,&quot;T cell (CD8+)&quot;)
names(celltypes) &lt;- types
names(types) &lt;- celltypes</code></pre>
</div>
<div id="summary-of-the-power" class="section level1">
<h1>Summary of the power</h1>
<div
id="highlighted-example-1-nf-kappa-b-signaling-pathway---platelet---hb-ebi-a-gcst90002384"
class="section level2">
<h2>Highlighted example 1: NF-kappa B signaling pathway - Platelet -
HB-ebi-a-GCST90002384</h2>
<pre class="r"><code>trait=&quot;HB-ebi-a-GCST90002384&quot;
factor=&quot;04064_PC1&quot;
celltype=&quot;platelet&quot;

df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

df$celltype &lt;- types[df$celltype]

dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
p_obs &lt;- dt_pair$p_ECT


df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]

DT::datatable(df_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;pairs with FDR &lt; 0.2&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-0c923cdc0df2ab45b6f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c923cdc0df2ab45b6f5">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">pairs with FDR &lt; 0.2<\/caption>","data":[["1","11791","11792","11793","5","11794","4","6","8","11","18","25","11797","11798","11799","11802","11803","11804","11805","11806","11807","11808","11809","11810","11811","20","21","22","9","12","15","17","13","19","23","24","27"],["Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet"],["HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","LDL-ukb-d-30780_irnt","UC-ebi-a-GCST003045","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","UC-ebi-a-GCST003045","UC-ebi-a-GCST003045","UC-ebi-a-GCST003045","MCV-ebi-a-GCST90002392","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","MCH-ebi-a-GCST90002390"],["04064_PC1","04390_PC5","04714_PC4","00230_PC1","04971_PC4","04625_PC3","04390_PC3","04270_PC4","00270_PC3","00562_PC5","04910_PC1","04145_PC1","04724_PC4","04261_PC4","04726_PC4","04728_PC5","04912_PC1","03018_PC2","04024_PC1","04725_PC1","04218_PC1","04144_PC4","04150_PC2","04151_PC2","04216_PC3","03013_PC5","04260_PC5","04726_PC3","04918_PC4","04062_PC2","04611_PC4","04022_PC4","04611_PC5","04371_PC5","00480_PC4","04217_PC3","04061_PC2"],["04064","04390","04714","00230","04971","04625","04390","04270","00270","00562","04910","04145","04724","04261","04726","04728","04912","03018","04024","04725","04218","04144","04150","04151","04216","03013","04260","04726","04918","04062","04611","04022","04611","04371","00480","04217","04061"],["NF-kappa B signaling pathway","Hippo signaling pathway","Thermogenesis","Purine metabolism","Gastric acid secretion","C-type lectin receptor signaling pathway","Hippo signaling pathway","Vascular smooth muscle contraction","Cysteine and methionine metabolism","Inositol phosphate metabolism","Insulin signaling pathway","Phagosome","Glutamatergic synapse","Adrenergic signaling in cardiomyocytes","Serotonergic synapse","Dopaminergic synapse","GnRH signaling pathway","RNA degradation","cAMP signaling pathway","Cholinergic synapse","Cellular senescence","Endocytosis","mTOR signaling pathway","PI3K-Akt signaling pathway","Ferroptosis","RNA transport","Cardiac muscle contraction","Serotonergic synapse","Thyroid hormone synthesis","Chemokine signaling pathway","Platelet activation","cGMP-PKG signaling pathway","Platelet activation","Apelin signaling pathway","Glutathione metabolism","Necroptosis","Viral protein interaction with cytokine and cytokine receptor"],[5,6,9,8,5,6,8,5,9,5,16,8,10,5,8,7,5,6,8,7,20,15,8,8,6,5,14,5,6,5,10,6,9,7,8,15,7],[0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,2,1,0,0,0,0,0,0,0,0,0],[8e-06,6.7e-05,9.4e-05,0.000177,0.000677,0.000595,0.00064,0.000803,0.0013999,0.0017998,0.0027997,0.0038996,0.0016998,0.0017998,0.0018998,0.0023998,0.0024998,0.0027997,0.0027997,0.0031997,0.0036996,0.0037996,0.0037996,0.0037996,0.0039996,0.0032997,0.0033997,0.0035996,0.0013999,0.0018998,0.0021998,0.0024998,0.0019998,0.0032997,0.0037996,0.0037996,0.0039996],[0.014296,0.0559927,0.0559927,0.0790748,0.1728284,0.1728284,0.1728284,0.1793701,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699],[102,155,231,130,75,104,155,132,49,74,137,152,114,149,115,132,93,79,216,112,160,249,153,354,40,179,86,115,74,189,124,167,124,137,56,162,100],[47,41,90,34,28,54,41,55,23,29,62,81,36,56,38,56,37,37,71,41,78,111,55,126,22,71,34,38,24,88,70,67,70,56,27,69,38],["bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new"],["-0.41768128699344","-0.219352852012402","-0.797167485258106","0.0490457193697609","0.0109467259031775","-0.131943556293009","0.165289903927005","-0.0910772003692204","0.5259942054983","0.560386923702547","0.549513227634144","-0.134040508002528","-0.54428879927892","-0.405271170683785","0.392211512031683","-0.539877365042058","0.13958108952101","0.0584426773316338","0.0441347782270358","0.147938467588175","0.296463052202976","0.22793051722002","-0.0416611070894055","0.0323660276705429","-0.165866865073491","-0.133814949928015","0.156927059687846","0.10608763221134","0.378866542149566","-0.39756014039602","0.0454542778745604","0.486738685474682","-0.659405280342488","-0.227839601219098","0.776218730681591","0.0150066929027473","-0.413012718373063"],["0.386094674495389","0.606360167370838","0.132433832841616","0.916858027973131","0.981300380495821","0.670683188315133","0.84410453151432","0.603283698260543","0.257978185483141","0.234557683466446","0.388446580296467","0.865495007923532","0.231930532914844","0.386464726940903","0.231288726314972","0.258818812447651","0.681373709410221","0.888791317157249","0.906030570268332","0.68939993487819","0.392331848308167","0.58136440984346","0.91726350363736","0.921461676690888","0.660935938097067","0.537255583958351","0.482016680200454","0.632751968554825","0.124962375643065","0.365943356400005","0.946348478214342","0.410045414016918","0.386842509399878","0.698338609981389","0.189242100754593","0.978206000196248","0.185519835265756"],[0.0829709,0.000215,0.0024138,0.0388076,0.0428586,0.0143981,0.0532078,0.248685,0.0142615,0.0404826,0.0100246,3.4e-05,0.0002787,0.0001895,0.0052506,0.0141543,0.0161583,0.0768882,0.0571586,0.0876472,0.0375206,1,0.0158862,0.0417202,0.0914741,0.0132481,0.0210123,0.0432386,0.0132213,0.0551095,0.054751,0.0265939,0.0352653,0.0137847,0.0440503,0.0386792,0.0011242],[0.0825434,0.000214,0.0024024,0.0386281,0.0426652,0.0143427,0.0573591,0.2742167,0.0141772,0.040178,0.0099601,3.38e-05,0.0002773,0.0001886,0.0052318,0.0140832,0.0160997,0.0768847,0.056906,0.0871026,0.0373917,1,0.0158038,0.0414849,0.09343,0.0129032,0.0213711,0.0458239,0.0131166,0.0547758,0.0542514,0.026447,0.0353031,0.0137237,0.0440581,0.0385244,0.0011168],[0.4455446,0.009901,0.029703,0.3168317,0.1386139,0.1782178,0.2079208,0.6435644,0.0792079,0.1485149,0.0891089,0.009901,0.0594059,0.009901,0.0990099,0.0594059,0.2376238,0.4554455,0.4257426,0.7029703,0.3168317,0.980198,0.2178218,0.6336634,0.4158416,0.1089109,0.0990099,0.1881188,0.1386139,0.5445545,0.3960396,0.1683168,0.1782178,0.0693069,0.1683168,0.2178218,0.029703],[0.4356436,0.009901,0.029703,0.3168317,0.1386139,0.1782178,0.2079208,0.6534653,0.0792079,0.1485149,0.0792079,0.009901,0.0594059,0.009901,0.0990099,0.0594059,0.2376238,0.4554455,0.4257426,0.7029703,0.3069307,0.980198,0.2178218,0.6237624,0.4257426,0.0990099,0.0990099,0.1881188,0.1386139,0.5445545,0.3861386,0.1683168,0.1782178,0.0693069,0.1683168,0.2178218,0.029703],[0.4244096,0.0044039,0.0269258,0.254628,0.2497611,0.1130334,0.2374262,0.5958365,0.0957725,0.2377415,0.0716742,0.0020591,0.0053004,0.003931,0.0502557,0.1113684,0.1240776,0.4242027,0.3430508,0.4627245,0.2478695,0.9838228,0.1225111,0.2698003,0.4752663,0.0617629,0.0936148,0.1758406,0.1159496,0.3227259,0.321051,0.1726091,0.2079585,0.0891518,0.2522734,0.2250421,0.0189259],[0.4216313,0.004394,0.0268569,0.2531699,0.2477907,0.1125014,0.2511108,0.6215998,0.0949646,0.2351898,0.0711422,0.0020197,0.0053004,0.0039014,0.0500488,0.1107477,0.123516,0.4231288,0.3410607,0.4595127,0.2467857,0.9838819,0.1217722,0.2680171,0.4803598,0.0604526,0.0950631,0.1846583,0.115122,0.3201446,0.3174944,0.17122,0.2071605,0.0885509,0.2510123,0.2230224,0.0187387],[0.3434217,0.0040564,0.0198934,0.1893318,0.2056802,0.0821536,0.245055,0.6261945,0.0814921,0.1961002,0.0610268,0.0013918,0.0047532,0.0037532,0.0362535,0.0809852,0.0904661,0.3251109,0.2593917,0.3569712,0.1840178,0.99418,0.0891885,0.2011472,0.3676138,0.0767031,0.1130407,0.2071902,0.0765714,0.2519909,0.2506752,0.1379422,0.1747671,0.0792396,0.2103738,0.188806,0.0116274],[0.3408274,0.0040367,0.0197926,0.1878342,0.2040814,0.0816399,0.2590845,0.6478327,0.0808607,0.1940953,0.0605387,0.0013851,0.004734,0.0037286,0.0360797,0.0804303,0.0898737,0.3238001,0.2574916,0.3539837,0.1827326,0.9939772,0.0885096,0.1993779,0.371488,0.0748696,0.1142306,0.2164159,0.0758759,0.2497921,0.2479108,0.1367523,0.1742355,0.0787196,0.2095377,0.1874043,0.011558]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>num_supp_SNP_MHC<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n      <th>acat_p_correct_noMHC_traitnull<\/th>\n      <th>acat_p_correct_globalnull<\/th>\n      <th>acat_p_correct_noMHC_globalnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,11,15,16,17,18,19,20,21,22]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)

p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)</code></pre>
<pre class="r"><code>power_all &lt;- c()
p &lt;- list()
for (downsample_size in c(0.01, 0.1, 0.3, 0.5, 0.7, 1)){
  
  file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
  ECT_res_allround_df&lt;- readRDS(file_power)
  
  tmp &lt;- c(downsample_size, sum(ECT_res_allround_df$p_ECT &lt; p_cutoff_fdr01), sum(ECT_res_allround_df$p_ECT &lt; p_cutoff_fdr02))
  power_all &lt;- rbind(power_all, tmp)
  
  num_leq_fdr01 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01)
  num_leq_fdr02 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
  
  p[[as.character(downsample_size)]] &lt;- ggplot(ECT_res_allround_df, aes(x = -log10(p_ECT))) +
    geom_histogram(bins = 30, color = &quot;white&quot;, fill = &quot;steelblue&quot;) +
    
    # cutoff lines
    geom_vline(xintercept = -log10(p_cutoff_fdr01), color = &quot;red&quot;, size = 1) +
    geom_vline(xintercept = -log10(p_cutoff_fdr02), color = &quot;red&quot;, linetype = &quot;dashed&quot;, size = 1) +
    
    # stacked annotation block
    annotate(
      &quot;label&quot;,
      x = max(-log10(ECT_res_allround_df$p_ECT)),
      y = Inf,
      hjust = 1, vjust = 1.2,
      label = paste0(
        &quot;Count p ≤ FDR 0.1 cutoff: &quot;, num_leq_fdr01, &quot;\n&quot;,
        &quot;Count p ≤ FDR 0.2 cutoff: &quot;, num_leq_fdr02
      ),
      size = 4,
      fill = &quot;white&quot;,
      label.size = 0.3
    ) +
    
    theme_bw(base_size = 13) +
    labs(
      x = &quot;-log10(ECT p-value)&quot;,
      y = &quot;Count&quot;,
      title = paste0(&quot;downsample size &quot;, downsample_size)
    )

  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-3-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
rownames(power_all) &lt;- NULL

DT::datatable(power_all ,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;power&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-ad41175326dc68f35292" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ad41175326dc68f35292">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">power<\/caption>","data":[[0.01,0.1,0.3,0.5,0.7,1],[5,24,44,57,67,72],[5,24,44,57,67,72]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[0,1,2]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div
id="highlighted-example-2-leukocyte-transendothelial-migration---t-cell-cd-8---ibd-ebi-a-gcst003043"
class="section level2">
<h2>Highlighted example 2: Leukocyte transendothelial migration - T cell
(CD 8+) - IBD-ebi-a-GCST003043</h2>
<pre class="r"><code>celltype &lt;- &quot;thymocyte&quot;
trait &lt;- &quot;IBD-ebi-a-GCST003043&quot;
factor &lt;- &quot;04670_PC1&quot;

df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

df$celltype &lt;- types[df$celltype]

dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
p_obs &lt;- dt_pair$p_ECT

df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]

DT::datatable(df_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;pairs with FDR &lt; 0.2&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-66aaeb19fe992dfb51c2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-66aaeb19fe992dfb51c2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">pairs with FDR &lt; 0.2<\/caption>","data":[["1","2","4","15751","15752","9","10","16","29","31","15753","15754","15758","15760","15761","15763","15756","15762","14","15","19","20","22","33","34","35","13","26","32","17","18","28","36","24","17501","38","15765","15766","15767","41","42","39","46","15768","44","17502","47"],["T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"],["IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","PLT-ebi-a-GCST90002402","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","RBC-ebi-a-GCST90002403","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","HB-ebi-a-GCST90002384","lymphc-ebi-a-GCST90002388","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","HB-ebi-a-GCST90002384","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","MCV-ebi-a-GCST90002392","lymphc-ebi-a-GCST90002388","MCV-ebi-a-GCST90002392"],["04971_PC2","04670_PC1","04971_PC1","04211_PC5","04962_PC2","04390_PC4","04060_PC2","04217_PC5","04010_PC5","04929_PC2","04151_PC4","04730_PC1","04612_PC3","04935_PC1","04727_PC2","03008_PC2","04925_PC3","04520_PC4","04014_PC3","04370_PC3","03040_PC3","00071_PC2","03440_PC3","03018_PC3","04550_PC3","04919_PC4","04070_PC2","04911_PC3","04340_PC4","04714_PC4","04972_PC3","00514_PC5","03060_PC3","04216_PC2","04218_PC5","04918_PC3","04962_PC1","00280_PC3","04010_PC3","04071_PC2","04714_PC4","04072_PC3","04146_PC3","04929_PC1","04926_PC4","04215_PC2","04024_PC4"],["04971","04670","04971","04211","04962","04390","04060","04217","04010","04929","04151","04730","04612","04935","04727","03008","04925","04520","04014","04370","03040","00071","03440","03018","04550","04919","04070","04911","04340","04714","04972","00514","03060","04216","04218","04918","04962","00280","04010","04071","04714","04072","04146","04929","04926","04215","04024"],["Gastric acid secretion","Leukocyte transendothelial migration","Gastric acid secretion","Longevity regulating pathway","Vasopressin-regulated water reabsorption","Hippo signaling pathway","Cytokine-cytokine receptor interaction","Necroptosis","MAPK signaling pathway","GnRH secretion","PI3K-Akt signaling pathway","Long-term depression","Antigen processing and presentation","Growth hormone synthesis, secretion and action","GABAergic synapse","Ribosome biogenesis in eukaryotes","Aldosterone synthesis and secretion","Adherens junction","Ras signaling pathway","VEGF signaling pathway","Spliceosome","Fatty acid degradation","Homologous recombination","RNA degradation","Signaling pathways regulating pluripotency of stem cells","Thyroid hormone signaling pathway","Phosphatidylinositol signaling system","Insulin secretion","Hedgehog signaling pathway","Thermogenesis","Pancreatic secretion","Other types of O-glycan biosynthesis","Protein export","Ferroptosis","Cellular senescence","Thyroid hormone synthesis","Vasopressin-regulated water reabsorption","Valine, leucine and isoleucine degradation","MAPK signaling pathway","Sphingolipid signaling pathway","Thermogenesis","Phospholipase D signaling pathway","Peroxisome","GnRH secretion","Relaxin signaling pathway","Apoptosis - multiple species","cAMP signaling pathway"],[5,10,5,9,5,8,5,7,6,8,9,5,9,5,6,7,10,5,8,11,5,14,5,6,5,11,6,6,10,16,6,5,5,21,16,7,7,12,5,5,10,5,6,5,20,9,15],[1,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,2,0,0,0,0,0,0,0,0,0,0,0,0],[5e-05,8.3e-05,0.000189,0.00018,0.000346,0.000436,0.000527,0.0011999,0.0021998,0.0022998,0.0011999,0.0011999,0.0016998,0.0018998,0.0020998,0.0025997,0.0013999,0.0020998,0.0010999,0.0011999,0.0013999,0.001478,0.0017998,0.0023998,0.0024998,0.0025997,0.0010999,0.0019998,0.0023998,0.0012999,0.0012999,0.0020998,0.0025997,0.0019998,0.001328,0.0028997,0.0027997,0.0029997,0.0031997,0.0031997,0.0031997,0.0029997,0.0037996,0.0036996,0.0037996,0.0037996,0.0038996],[0.0996,0.0996,0.1134,0.1134,0.16608,0.1744,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.18284,0.18284,0.18284,0.18284,0.18284,0.18284,0.18284,0.19824,0.19824,0.19824,0.19824,0.1991285],[75,112,75,89,44,155,294,162,295,64,354,60,78,119,89,105,98,71,232,59,149,44,41,79,140,119,99,86,50,231,102,46,23,40,160,74,44,48,295,119,231,148,83,64,129,32,216],[34,69,34,57,33,75,118,97,168,30,178,30,65,71,35,61,49,44,119,39,112,28,28,67,63,84,67,31,24,130,40,25,20,33,117,36,33,39,168,81,130,81,59,30,69,25,94],["disease_EUR_old","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","disease_EUR_old","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new"],["-0.160017285934662","-0.146323318161613","-0.0195316446390257","-0.280029290491469","-0.369079521944207","0.272708295051201","-0.0495864361159254","-0.0139165601476714","-0.148981738791476","0.201944929329823","-0.167618721127787","0.0900331861451125","0.650380512841018","-0.0640671509178383","0.0601787103416731","0.0178089277936049","0.0915678874980503","0.586873863031775","-0.284482997416838","0.0116704298005137","-0.346552058826978","0.373332837053078","0.138848375442602","0.609834657956357","0.306469104611784","0.341146698851871","0.400944087090976","0.17112589085453","-0.0814124936249649","0.908184857186706","-0.0427098263051655","0.132673406068839","-0.173591571192809","0.186204196970328","-0.226259051798917","-0.018084141884111","0.384601143357574","-0.0498490589289173","-0.566552792213228","0.124554307321948","-0.537867543275294","0.149081328940241","0.150491568198327","0.355452720582586","-0.0194109476464144","0.175664192729935","-0.0497481840979672"],["0.272526832466116","0.380080136666012","0.878437186997933","0.226486849411033","0.193315258562892","0.187412601065013","0.666382327665147","0.937378747053766","0.339101948874325","0.169171895778485","0.398004857735557","0.558453875836253","0.0120705732822415","0.750408941399446","0.749755390391455","0.921748880203971","0.788672156956791","0.130446334859127","0.294726350408341","0.96720958554113","0.234718940896182","0.176864788775393","0.603525221729898","0.0338619202031168","0.263384075532353","0.240923912300581","0.0946000828443417","0.493810098956484","0.791365014934603","0.130430359113488","0.924512828343235","0.749491860203692","0.717262597195071","0.598502916408624","0.695014070209916","0.858900654427318","0.0383060556657068","0.902885205156944","0.105475680291036","0.607938158577543","0.220017018873353","0.670031725380746","0.323319147212933","0.0329821476481283","0.94134129987067","0.696576899217406","0.839044609865298"],[0.0009857,0.0415006,0.0355491,0.0131546,0.0136082,0.0026897,0.0002286,0.0113243,0.1272579,0.0136204,0.0496565,0.0123531,0.0138749,0.0317571,0.002806,0.0182166,0.0311182,0.0046882,0.0077831,0.0725959,0.0649782,0.0574167,0.6487035,0.0285295,0.0699494,0.0133443,0.0318237,0.0579279,0.034016,0.015001,0.0377493,0.0182108,0.0036896,0.0238865,0.0012331,0.0507942,0.012418,0.0497995,0.0807982,0.0690528,0.0376975,0.0709253,0.0008969,0.0602225,0.0525288,0.0390251,0.0082523],[0.0913229,0.0412207,0.0348279,0.0138185,0.0135827,0.0026764,0.0002237,0.0110714,0.1139942,0.0133486,0.0509742,0.0122997,0.0138079,0.0316375,0.0027889,0.0181087,0.0309314,0.0046648,0.0445922,0.0723547,0.064518,0.0565423,0.6289994,0.0292198,0.0696152,0.0132553,0.0315655,0.0574937,0.0337729,0.0150468,0.0378506,0.0180983,0.0131945,0.0237279,0.0012354,0.0497635,0.0123661,0.048638,0.0804363,0.0685495,0.0371428,0.0701786,0.0008776,0.0613372,0.0520201,0.0387636,0.0081855],[0.039604,0.2376238,0.2475248,0.0990099,0.1683168,0.039604,0.009901,0.0990099,0.7623762,0.0990099,0.7821782,0.1089109,0.1089109,0.3267327,0.049505,0.2277228,0.2079208,0.1089109,0.1485149,0.2277228,0.6831683,0.3762376,0.9108911,0.1584158,0.3960396,0.1683168,0.2673267,0.1782178,0.0891089,0.0990099,0.0990099,0.0594059,0.009901,0.2079208,0.009901,0.4257426,0.1287129,0.3564356,0.6732673,0.5049505,0.2772277,0.4059406,0.009901,0.4752475,0.4752475,0.2475248,0.0594059],[0.3960396,0.2376238,0.2475248,0.0990099,0.1683168,0.039604,0.009901,0.0990099,0.7227723,0.0990099,0.7821782,0.1089109,0.1089109,0.3267327,0.049505,0.2277228,0.2079208,0.1089109,0.6039604,0.2277228,0.6831683,0.3663366,0.9108911,0.1683168,0.3861386,0.1683168,0.2673267,0.1782178,0.0891089,0.0990099,0.0990099,0.0594059,0.0594059,0.1881188,0.009901,0.4059406,0.1188119,0.3564356,0.6732673,0.5049505,0.2574257,0.4059406,0.009901,0.4851485,0.4851485,0.2475248,0.0594059],[0.0089497,0.2015042,0.1775894,0.0917676,0.0978216,0.0252353,0.002888,0.0676288,0.4643362,0.0793353,0.2968162,0.0866171,0.0961074,0.2001066,0.0238145,0.1219373,0.202122,0.0396136,0.0604474,0.428877,0.3945143,0.357727,0.9008039,0.1934734,0.4168694,0.0967869,0.2216585,0.3639509,0.2352337,0.1119142,0.2550868,0.1345781,0.0357295,0.1568637,0.0107728,0.237365,0.0869877,0.3025228,0.4361511,0.4144987,0.2561525,0.3999506,0.0082934,0.3483989,0.3364453,0.2357237,0.0655285],[0.3708311,0.1987707,0.1729794,0.0952116,0.0952348,0.0226408,0.002139,0.0652273,0.4314716,0.0765168,0.3023529,0.0857677,0.0951267,0.1987089,0.0234052,0.1206168,0.1985699,0.0374592,0.2852488,0.4235643,0.3882673,0.3491633,0.8953522,0.1946008,0.4110625,0.0929645,0.2161528,0.3570938,0.2296507,0.1079451,0.2520521,0.1307712,0.0984934,0.1525162,0.0091245,0.2317048,0.0861847,0.2940286,0.4318191,0.4076494,0.2488707,0.3929545,0.00722,0.3526305,0.3295959,0.2290834,0.0622235],[0.0094632,0.2052896,0.180176,0.0764364,0.0786913,0.0205547,0.0034589,0.0672065,0.4649765,0.078752,0.2379927,0.0724126,0.080039,0.163671,0.0212729,0.1012138,0.160853,0.0320718,0.0488796,0.3203948,0.2946937,0.267504,0.8477691,0.1492509,0.3115883,0.0773604,0.1639595,0.2694322,0.1735832,0.0856094,0.1895302,0.1011868,0.0264438,0.1279833,0.0111987,0.2424477,0.0727369,0.238564,0.3464201,0.3085854,0.1893198,0.3148328,0.0088157,0.2778171,0.2491219,0.1948954,0.051361],[0.3738789,0.2013522,0.1743951,0.0777068,0.0765316,0.0189952,0.0023395,0.064028,0.4316933,0.0753494,0.2401239,0.0701581,0.0776538,0.1606032,0.0196771,0.0985012,0.157529,0.0303092,0.2150033,0.3161928,0.2898453,0.261147,0.8388315,0.1499114,0.3071393,0.0749074,0.1602849,0.2646751,0.1699469,0.0837638,0.1872655,0.0984478,0.0746098,0.1248859,0.009863,0.2353778,0.0704968,0.2310539,0.3418405,0.3035737,0.1842918,0.3090313,0.0074001,0.2786153,0.2441316,0.1910304,0.049252]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>num_supp_SNP_MHC<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n      <th>acat_p_correct_noMHC_traitnull<\/th>\n      <th>acat_p_correct_globalnull<\/th>\n      <th>acat_p_correct_noMHC_globalnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,11,15,16,17,18,19,20,21,22]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)

p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)</code></pre>
<pre class="r"><code>power_all &lt;- c()
p &lt;- list()
for (downsample_size in c(0.01, 0.1, 0.3, 0.5, 0.7, 1)){
  
  file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
  ECT_res_allround_df&lt;- readRDS(file_power)
  
  tmp &lt;- c(downsample_size, sum(ECT_res_allround_df$p_ECT &lt; p_cutoff_fdr01), sum(ECT_res_allround_df$p_ECT &lt; p_cutoff_fdr02))
  power_all &lt;- rbind(power_all, tmp)
  
  num_leq_fdr01 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01)
  num_leq_fdr02 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
  
  p[[as.character(downsample_size)]] &lt;- ggplot(ECT_res_allround_df, aes(x = -log10(p_ECT))) +
    geom_histogram(bins = 30, color = &quot;white&quot;, fill = &quot;steelblue&quot;) +
    
    # cutoff lines
    geom_vline(xintercept = -log10(p_cutoff_fdr01), color = &quot;red&quot;, size = 1) +
    geom_vline(xintercept = -log10(p_cutoff_fdr02), color = &quot;red&quot;, linetype = &quot;dashed&quot;, size = 1) +
    
    # stacked annotation block
    annotate(
      &quot;label&quot;,
      x = max(-log10(ECT_res_allround_df$p_ECT)),
      y = Inf,
      hjust = 1, vjust = 1.2,
      label = paste0(
        &quot;Count p ≤ FDR 0.1 cutoff: &quot;, num_leq_fdr01, &quot;\n&quot;,
        &quot;Count p ≤ FDR 0.2 cutoff: &quot;, num_leq_fdr02
      ),
      size = 4,
      fill = &quot;white&quot;,
      label.size = 0.3
    ) +
    
    theme_bw(base_size = 13) +
    labs(
      x = &quot;-log10(ECT p-value)&quot;,
      y = &quot;Count&quot;,
      title = paste0(&quot;downsample size &quot;, downsample_size)
    )

  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-5-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
rownames(power_all) &lt;- NULL

DT::datatable(power_all ,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;power&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-1814506231684700bf53" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1814506231684700bf53">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">power<\/caption>","data":[[0.01,0.1,0.3,0.5,0.7,1],[4,73,96,98,100,100],[26,94,100,100,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[0,1,2]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div
id="not-highlighted-but-ect-fdr-0.2-example-1-b_cell---rbc-ebi-a-gcst90002403---04217_pc4"
class="section level2">
<h2>Not highlighted but ECT FDR &lt; 0.2 example 1: B_cell -
RBC-ebi-a-GCST90002403 - 04217_PC4</h2>
<pre class="r"><code>celltype &lt;- &quot;B_cell&quot;
trait &lt;- &quot;RBC-ebi-a-GCST90002403&quot;
factor &lt;- &quot;04217_PC4&quot;

df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

df$celltype &lt;- types[df$celltype]

dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
p_obs &lt;- dt_pair$p_ECT

df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]

DT::datatable(df_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;pairs with FDR &lt; 0.2&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-43c5071fe8a890b96ecd" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-43c5071fe8a890b96ecd">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">pairs with FDR &lt; 0.2<\/caption>","data":[["2","4","6","7","8","9","16436","16437","18166","10","16","24","22","23","26","25","16438","16439","16444","16445","16447","16440","16441","16442","16443","16446","18168","20","27","28","18169","31","16448","32","33","16451","16452"],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"],["Psoriasis-ebi-a-GCST90038681","T1D-ebi-a-GCST90014023","RBC-ebi-a-GCST90002403","MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","HCT-ebi-a-GCST90002383","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","HCT-ebi-a-GCST90002383","T1D-ebi-a-GCST90014023","T1D-ebi-a-GCST90014023","RBC-ebi-a-GCST90002403","RBC-ebi-a-GCST90002403","MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","HB-ebi-a-GCST90002384","lymphc-ebi-a-GCST90002388","MCV-ebi-a-GCST90002392","MPV-ebi-a-GCST90002395","WBC-ebi-a-GCST90002407","asthma_pmid36778051","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395"],["04062_PC2","04211_PC3","04970_PC3","03010_PC2","04060_PC2","04060_PC2","04213_PC2","04657_PC2","00510_PC2","04510_PC5","00640_PC1","04144_PC4","03040_PC4","04217_PC4","04730_PC5","04151_PC5","04917_PC2","04380_PC3","03050_PC3","04650_PC3","04668_PC2","04640_PC2","04722_PC4","04015_PC4","04659_PC4","04010_PC4","00630_PC4","03040_PC4","04110_PC5","04612_PC5","04612_PC1","04210_PC2","04620_PC5","04068_PC3","04921_PC3","04728_PC4","04918_PC2"],["04062","04211","04970","03010","04060","04060","04213","04657","00510","04510","00640","04144","03040","04217","04730","04151","04917","04380","03050","04650","04668","04640","04722","04015","04659","04010","00630","03040","04110","04612","04612","04210","04620","04068","04921","04728","04918"],["Chemokine signaling pathway","Longevity regulating pathway","Salivary secretion","Ribosome","Cytokine-cytokine receptor interaction","Cytokine-cytokine receptor interaction","Longevity regulating pathway - multiple species","IL-17 signaling pathway","N-Glycan biosynthesis","Focal adhesion","Propanoate metabolism","Endocytosis","Spliceosome","Necroptosis","Long-term depression","PI3K-Akt signaling pathway","Prolactin signaling pathway","Osteoclast differentiation","Proteasome","Natural killer cell mediated cytotoxicity","TNF signaling pathway","Hematopoietic cell lineage","Neurotrophin signaling pathway","Rap1 signaling pathway","Th17 cell differentiation","MAPK signaling pathway","Glyoxylate and dicarboxylate metabolism","Spliceosome","Cell cycle","Antigen processing and presentation","Antigen processing and presentation","Apoptosis","Toll-like receptor signaling pathway","FoxO signaling pathway","Oxytocin signaling pathway","Dopaminergic synapse","Thyroid hormone synthesis"],[6,12,7,5,6,8,7,6,6,7,6,11,7,9,11,9,6,5,5,6,5,9,7,14,8,7,7,8,8,6,7,5,5,7,5,5,6],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0],[4e-05,0.000195,0.00035,0.00045,0.000489,0.000565,0.000221,0.000519,0.000278,0.000709,0.0012999,0.0019998,0.0017998,0.0019998,0.0020998,0.0020998,0.001053,0.001153,0.0018998,0.0019998,0.0021998,0.0012999,0.001372,0.0013999,0.001527,0.0021998,0.0018998,0.0016998,0.0022998,0.0023998,0.0024998,0.0026997,0.0028997,0.0029997,0.0029997,0.0030997,0.0031997],[0.08832,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1565472,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.175102,0.1766253,0.1780503,0.1862793,0.1892382,0.1892382,0.1892382,0.1901149,0.1909443],[189,89,90,153,294,294,62,94,50,199,34,249,149,162,60,354,70,128,45,134,112,99,119,210,107,295,30,149,124,78,78,136,104,131,153,132,74],[109,56,34,126,101,101,36,46,39,105,29,175,111,93,31,175,42,86,40,79,69,62,89,104,79,162,23,111,92,57,57,104,63,87,79,76,31],["disease_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new"],["1.69054931142635","0.0673449539000475","-0.128493235792523","-0.226096753153129","0.233290435680774","0.356193457802385","-0.27071396926541","-0.0983783761147528","0.803780051926971","-0.319389509893942","-0.0400493575356977","-0.0133601370358113","0.0345362910612599","0.506296610377211","-0.255109506096205","-0.390877583722192","-0.15053842633599","-0.0830551548728864","-0.417405016126126","-0.32927676223193","-0.46363696898361","-0.643224676525653","0.00675380298541528","-0.206185838472806","-0.348348769804703","-0.0201372579675425","-0.298272254009567","0.38668824064209","0.677674795102813","0.807887694724516","0.177658176432276","-0.227886836956831","0.14373227924004","0.0697980284727306","0.00693293256160454","0.246432050632072","0.177562763063755"],["0.793939736663622","0.419586506203072","0.696491658215512","0.652614467038651","0.714976796119257","0.553028372038752","0.315566916417708","0.57743030781489","0.213549053167502","0.518149015164986","0.502309526364542","0.870945912245638","0.92292114024808","0.341137939355012","0.462912348171951","0.42444858566781","0.649473464532669","0.825651549584431","0.136673185020984","0.346333133638904","0.32468447926906","0.0384225144339296","0.977036848092291","0.375685874442427","0.262400957242923","0.927089640139701","0.481583736109852","0.155265962575607","0.0513939859196159","0.275333129922457","0.731412966083133","0.400970678595854","0.573284618882391","0.850217531548008","0.991924642896619","0.397134232450949","0.336094738609318"],[0.0350311,0.0655913,0.0442804,0.0273526,0.0664088,0.0452325,0.0145661,0.1063782,1,0.0040001,0.8828099,0.0436977,0.0488894,0.0092223,0.0868603,0.0783865,0.0463272,0.0358707,0.0097742,0.1385439,0.0081822,0.0946554,0.059218,0.0376386,0.0423516,1,0.0173718,0.046462,0.0065683,0.0001711,1,0.0644015,0.0445163,0.0462309,0.0188757,0.1788443,0.0600319],[0.0313682,0.0638589,0.043997,0.0271653,0.0528417,0.0381385,0.0144865,0.1061518,1,0.0039809,0.887038,0.0421872,0.0486338,0.0091719,0.1009872,0.0761631,0.0461265,0.0357014,0.0097286,0.1397131,0.0081416,0.0936901,0.0587943,0.0374142,0.0420664,1,0.0172194,0.0465073,0.0065155,0.023752,1,0.0640073,0.0441792,0.0463672,0.0182766,0.17847,0.0595836],[0.1584158,0.4059406,0.1683168,0.1188119,0.3465347,0.2079208,0.039604,0.3663366,0.970297,0.009901,0.9405941,0.3663366,0.2475248,0.1188119,0.2277228,0.5247525,0.1287129,0.3267327,0.1287129,0.5841584,0.049505,0.6435644,0.5247525,0.3267327,0.3366337,1,0.039604,0.2574257,0.0891089,0.2178218,1,0.5346535,0.1584158,0.5841584,0.1386139,0.6831683,0.3069307],[0.1287129,0.3861386,0.1584158,0.1188119,0.2970297,0.1584158,0.029703,0.3663366,0.970297,0.009901,0.9405941,0.3465347,0.2475248,0.1188119,0.2376238,0.5049505,0.1287129,0.3168317,0.1287129,0.5841584,0.049505,0.6336634,0.5049505,0.3267327,0.3267327,1,0.039604,0.2574257,0.0891089,0.0594059,1,0.5346535,0.1584158,0.5841584,0.1287129,0.6633663,0.3069307],[0.1065031,0.239223,0.2452307,0.1513399,0.3356563,0.2312773,0.0866195,0.4848734,0.9808451,0.0240076,0.9300858,0.1703398,0.2687421,0.0569375,0.4274075,0.3830823,0.2560988,0.2027736,0.0592941,0.5799335,0.0499919,0.445713,0.3020008,0.1936643,0.2184781,0.9827476,0.0962473,0.2669902,0.0444338,0.0047829,0.9808451,0.3533849,0.2298794,0.2650212,0.0825032,0.6471287,0.3056904],[0.0972008,0.2318199,0.2350679,0.1445338,0.2677189,0.1917194,0.08148,0.4826723,0.9808994,0.0200696,0.931706,0.1625879,0.2588197,0.0500694,0.4699964,0.3679584,0.24975,0.1965489,0.0547128,0.5774839,0.0455268,0.4409704,0.2991425,0.1913603,0.2158969,0.9827713,0.0898598,0.2613468,0.0395268,0.1233479,0.9808994,0.3459586,0.2271003,0.2593778,0.0777823,0.6452126,0.3024362],[0.153741,0.2680301,0.1902521,0.1220424,0.2708697,0.1940119,0.0681952,0.3892895,0.9911896,0.0222065,0.9320428,0.1879381,0.2080315,0.0452248,0.3357594,0.3099569,0.198259,0.157135,0.047656,0.4618508,0.0406736,0.3580625,0.2457721,0.1641882,0.1826878,0.9915307,0.0801693,0.1987697,0.0336215,0.00303,0.9911896,0.2639268,0.1911931,0.197889,0.0866328,0.5319118,0.2486738],[0.1351176,0.258217,0.1853793,0.1177695,0.2189993,0.1624272,0.0646281,0.3847648,0.9915025,0.0193718,0.9335951,0.1782484,0.2032874,0.0419839,0.3713158,0.2989834,0.1936821,0.1527679,0.0444143,0.4603596,0.0375096,0.3513451,0.2403932,0.1595929,0.177791,0.9915826,0.0762244,0.1951437,0.0304804,0.1036659,0.9915025,0.2587468,0.1860896,0.1946023,0.0807533,0.5277866,0.2431922]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>num_supp_SNP_MHC<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n      <th>acat_p_correct_noMHC_traitnull<\/th>\n      <th>acat_p_correct_globalnull<\/th>\n      <th>acat_p_correct_noMHC_globalnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,11,15,16,17,18,19,20,21,22]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)

p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)</code></pre>
<pre class="r"><code>power_all &lt;- c()
p &lt;- list()
for (downsample_size in c(0.01, 0.1, 0.3, 0.5, 0.7, 1)){
  
  file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
  ECT_res_allround_df&lt;- readRDS(file_power)
  
  tmp &lt;- c(downsample_size, sum(ECT_res_allround_df$p_ECT &lt; p_cutoff_fdr01), sum(ECT_res_allround_df$p_ECT &lt; p_cutoff_fdr02))
  power_all &lt;- rbind(power_all, tmp)
  
  num_leq_fdr01 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01)
  num_leq_fdr02 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
  
  p[[as.character(downsample_size)]] &lt;- ggplot(ECT_res_allround_df, aes(x = -log10(p_ECT))) +
    geom_histogram(bins = 30, color = &quot;white&quot;, fill = &quot;steelblue&quot;) +
    
    # cutoff lines
    geom_vline(xintercept = -log10(p_cutoff_fdr01), color = &quot;red&quot;, size = 1) +
    geom_vline(xintercept = -log10(p_cutoff_fdr02), color = &quot;red&quot;, linetype = &quot;dashed&quot;, size = 1) +
    
    # stacked annotation block
    annotate(
      &quot;label&quot;,
      x = max(-log10(ECT_res_allround_df$p_ECT)),
      y = Inf,
      hjust = 1, vjust = 1.2,
      label = paste0(
        &quot;Count p ≤ FDR 0.1 cutoff: &quot;, num_leq_fdr01, &quot;\n&quot;,
        &quot;Count p ≤ FDR 0.2 cutoff: &quot;, num_leq_fdr02
      ),
      size = 4,
      fill = &quot;white&quot;,
      label.size = 0.3
    ) +
    
    theme_bw(base_size = 13) +
    labs(
      x = &quot;-log10(ECT p-value)&quot;,
      y = &quot;Count&quot;,
      title = paste0(&quot;downsample size &quot;, downsample_size)
    )

  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/977037c765dc7f77a1fec555ab6788033e846f25/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-7-1.png" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
rownames(power_all) &lt;- NULL

DT::datatable(power_all ,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;power&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-3f22443f93eeeda03f7b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3f22443f93eeeda03f7b">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">power<\/caption>","data":[[0.01,0.1,0.3,0.5,0.7,1],[0,1,7,11,12,19],[18,88,100,100,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[0,1,2]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div
id="not-highlighted-but-ect-fdr-0.2-example-1-b_cell---rbc-ebi-a-gcst90002403---03040_pc4"
class="section level2">
<h2>Not highlighted but ECT FDR &lt; 0.2 example 1: B_cell -
RBC-ebi-a-GCST90002403 - 03040_PC4</h2>
<pre class="r"><code>celltype &lt;- &quot;B_cell&quot;
trait &lt;- &quot;RBC-ebi-a-GCST90002403&quot;
factor &lt;- &quot;03040_PC4&quot;

df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))

df$celltype &lt;- types[df$celltype]

dt_pair &lt;- df[df$trait == trait &amp; df$factor == factor,]
p_obs &lt;- dt_pair$p_ECT

df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]

DT::datatable(df_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;pairs with FDR &lt; 0.2&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-198b30f56771bc600b2c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-198b30f56771bc600b2c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">pairs with FDR &lt; 0.2<\/caption>","data":[["2","4","6","7","8","9","16436","16437","18166","10","16","24","22","23","26","25","16438","16439","16444","16445","16447","16440","16441","16442","16443","16446","18168","20","27","28","18169","31","16448","32","33","16451","16452"],["B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)"],["Psoriasis-ebi-a-GCST90038681","T1D-ebi-a-GCST90014023","RBC-ebi-a-GCST90002403","MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","HCT-ebi-a-GCST90002383","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","HCT-ebi-a-GCST90002383","T1D-ebi-a-GCST90014023","T1D-ebi-a-GCST90014023","RBC-ebi-a-GCST90002403","RBC-ebi-a-GCST90002403","MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","HB-ebi-a-GCST90002384","lymphc-ebi-a-GCST90002388","MCV-ebi-a-GCST90002392","MPV-ebi-a-GCST90002395","WBC-ebi-a-GCST90002407","asthma_pmid36778051","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395"],["04062_PC2","04211_PC3","04970_PC3","03010_PC2","04060_PC2","04060_PC2","04213_PC2","04657_PC2","00510_PC2","04510_PC5","00640_PC1","04144_PC4","03040_PC4","04217_PC4","04730_PC5","04151_PC5","04917_PC2","04380_PC3","03050_PC3","04650_PC3","04668_PC2","04640_PC2","04722_PC4","04015_PC4","04659_PC4","04010_PC4","00630_PC4","03040_PC4","04110_PC5","04612_PC5","04612_PC1","04210_PC2","04620_PC5","04068_PC3","04921_PC3","04728_PC4","04918_PC2"],["04062","04211","04970","03010","04060","04060","04213","04657","00510","04510","00640","04144","03040","04217","04730","04151","04917","04380","03050","04650","04668","04640","04722","04015","04659","04010","00630","03040","04110","04612","04612","04210","04620","04068","04921","04728","04918"],["Chemokine signaling pathway","Longevity regulating pathway","Salivary secretion","Ribosome","Cytokine-cytokine receptor interaction","Cytokine-cytokine receptor interaction","Longevity regulating pathway - multiple species","IL-17 signaling pathway","N-Glycan biosynthesis","Focal adhesion","Propanoate metabolism","Endocytosis","Spliceosome","Necroptosis","Long-term depression","PI3K-Akt signaling pathway","Prolactin signaling pathway","Osteoclast differentiation","Proteasome","Natural killer cell mediated cytotoxicity","TNF signaling pathway","Hematopoietic cell lineage","Neurotrophin signaling pathway","Rap1 signaling pathway","Th17 cell differentiation","MAPK signaling pathway","Glyoxylate and dicarboxylate metabolism","Spliceosome","Cell cycle","Antigen processing and presentation","Antigen processing and presentation","Apoptosis","Toll-like receptor signaling pathway","FoxO signaling pathway","Oxytocin signaling pathway","Dopaminergic synapse","Thyroid hormone synthesis"],[6,12,7,5,6,8,7,6,6,7,6,11,7,9,11,9,6,5,5,6,5,9,7,14,8,7,7,8,8,6,7,5,5,7,5,5,6],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0],[4e-05,0.000195,0.00035,0.00045,0.000489,0.000565,0.000221,0.000519,0.000278,0.000709,0.0012999,0.0019998,0.0017998,0.0019998,0.0020998,0.0020998,0.001053,0.001153,0.0018998,0.0019998,0.0021998,0.0012999,0.001372,0.0013999,0.001527,0.0021998,0.0018998,0.0016998,0.0022998,0.0023998,0.0024998,0.0026997,0.0028997,0.0029997,0.0029997,0.0030997,0.0031997],[0.08832,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1565472,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.175102,0.1766253,0.1780503,0.1862793,0.1892382,0.1892382,0.1892382,0.1901149,0.1909443],[189,89,90,153,294,294,62,94,50,199,34,249,149,162,60,354,70,128,45,134,112,99,119,210,107,295,30,149,124,78,78,136,104,131,153,132,74],[109,56,34,126,101,101,36,46,39,105,29,175,111,93,31,175,42,86,40,79,69,62,89,104,79,162,23,111,92,57,57,104,63,87,79,76,31],["disease_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new"],["1.69054931142635","0.0673449539000475","-0.128493235792523","-0.226096753153129","0.233290435680774","0.356193457802385","-0.27071396926541","-0.0983783761147528","0.803780051926971","-0.319389509893942","-0.0400493575356977","-0.0133601370358113","0.0345362910612599","0.506296610377211","-0.255109506096205","-0.390877583722192","-0.15053842633599","-0.0830551548728864","-0.417405016126126","-0.32927676223193","-0.46363696898361","-0.643224676525653","0.00675380298541528","-0.206185838472806","-0.348348769804703","-0.0201372579675425","-0.298272254009567","0.38668824064209","0.677674795102813","0.807887694724516","0.177658176432276","-0.227886836956831","0.14373227924004","0.0697980284727306","0.00693293256160454","0.246432050632072","0.177562763063755"],["0.793939736663622","0.419586506203072","0.696491658215512","0.652614467038651","0.714976796119257","0.553028372038752","0.315566916417708","0.57743030781489","0.213549053167502","0.518149015164986","0.502309526364542","0.870945912245638","0.92292114024808","0.341137939355012","0.462912348171951","0.42444858566781","0.649473464532669","0.825651549584431","0.136673185020984","0.346333133638904","0.32468447926906","0.0384225144339296","0.977036848092291","0.375685874442427","0.262400957242923","0.927089640139701","0.481583736109852","0.155265962575607","0.0513939859196159","0.275333129922457","0.731412966083133","0.400970678595854","0.573284618882391","0.850217531548008","0.991924642896619","0.397134232450949","0.336094738609318"],[0.0350311,0.0655913,0.0442804,0.0273526,0.0664088,0.0452325,0.0145661,0.1063782,1,0.0040001,0.8828099,0.0436977,0.0488894,0.0092223,0.0868603,0.0783865,0.0463272,0.0358707,0.0097742,0.1385439,0.0081822,0.0946554,0.059218,0.0376386,0.0423516,1,0.0173718,0.046462,0.0065683,0.0001711,1,0.0644015,0.0445163,0.0462309,0.0188757,0.1788443,0.0600319],[0.0313682,0.0638589,0.043997,0.0271653,0.0528417,0.0381385,0.0144865,0.1061518,1,0.0039809,0.887038,0.0421872,0.0486338,0.0091719,0.1009872,0.0761631,0.0461265,0.0357014,0.0097286,0.1397131,0.0081416,0.0936901,0.0587943,0.0374142,0.0420664,1,0.0172194,0.0465073,0.0065155,0.023752,1,0.0640073,0.0441792,0.0463672,0.0182766,0.17847,0.0595836],[0.1584158,0.4059406,0.1683168,0.1188119,0.3465347,0.2079208,0.039604,0.3663366,0.970297,0.009901,0.9405941,0.3663366,0.2475248,0.1188119,0.2277228,0.5247525,0.1287129,0.3267327,0.1287129,0.5841584,0.049505,0.6435644,0.5247525,0.3267327,0.3366337,1,0.039604,0.2574257,0.0891089,0.2178218,1,0.5346535,0.1584158,0.5841584,0.1386139,0.6831683,0.3069307],[0.1287129,0.3861386,0.1584158,0.1188119,0.2970297,0.1584158,0.029703,0.3663366,0.970297,0.009901,0.9405941,0.3465347,0.2475248,0.1188119,0.2376238,0.5049505,0.1287129,0.3168317,0.1287129,0.5841584,0.049505,0.6336634,0.5049505,0.3267327,0.3267327,1,0.039604,0.2574257,0.0891089,0.0594059,1,0.5346535,0.1584158,0.5841584,0.1287129,0.6633663,0.3069307],[0.1065031,0.239223,0.2452307,0.1513399,0.3356563,0.2312773,0.0866195,0.4848734,0.9808451,0.0240076,0.9300858,0.1703398,0.2687421,0.0569375,0.4274075,0.3830823,0.2560988,0.2027736,0.0592941,0.5799335,0.0499919,0.445713,0.3020008,0.1936643,0.2184781,0.9827476,0.0962473,0.2669902,0.0444338,0.0047829,0.9808451,0.3533849,0.2298794,0.2650212,0.0825032,0.6471287,0.3056904],[0.0972008,0.2318199,0.2350679,0.1445338,0.2677189,0.1917194,0.08148,0.4826723,0.9808994,0.0200696,0.931706,0.1625879,0.2588197,0.0500694,0.4699964,0.3679584,0.24975,0.1965489,0.0547128,0.5774839,0.0455268,0.4409704,0.2991425,0.1913603,0.2158969,0.9827713,0.0898598,0.2613468,0.0395268,0.1233479,0.9808994,0.3459586,0.2271003,0.2593778,0.0777823,0.6452126,0.3024362],[0.153741,0.2680301,0.1902521,0.1220424,0.2708697,0.1940119,0.0681952,0.3892895,0.9911896,0.0222065,0.9320428,0.1879381,0.2080315,0.0452248,0.3357594,0.3099569,0.198259,0.157135,0.047656,0.4618508,0.0406736,0.3580625,0.2457721,0.1641882,0.1826878,0.9915307,0.0801693,0.1987697,0.0336215,0.00303,0.9911896,0.2639268,0.1911931,0.197889,0.0866328,0.5319118,0.2486738],[0.1351176,0.258217,0.1853793,0.1177695,0.2189993,0.1624272,0.0646281,0.3847648,0.9915025,0.0193718,0.9335951,0.1782484,0.2032874,0.0419839,0.3713158,0.2989834,0.1936821,0.1527679,0.0444143,0.4603596,0.0375096,0.3513451,0.2403932,0.1595929,0.177791,0.9915826,0.0762244,0.1951437,0.0304804,0.1036659,0.9915025,0.2587468,0.1860896,0.1946023,0.0807533,0.5277866,0.2431922]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>num_supp_SNP_MHC<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n      <th>acat_p_correct_noMHC_traitnull<\/th>\n      <th>acat_p_correct_globalnull<\/th>\n      <th>acat_p_correct_noMHC_globalnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,11,15,16,17,18,19,20,21,22]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
p_cutoff_fdr01 &lt;- max(df_fdr01$p_ECT)

p_cutoff_fdr02 &lt;- max(df_fdr02$p_ECT)</code></pre>
<pre class="r"><code>power_all &lt;- c()
p &lt;- list()
for (downsample_size in c(0.01, 0.1, 0.3, 0.5, 0.7, 1)){
  
  file_power &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/power_res/simulation2/&quot;, celltype,&quot;-&quot;, factor, &quot;-&quot;, trait,&quot;_downsample_&quot;,downsample_size,&quot;_simulated.RDS&quot;)
  ECT_res_allround_df&lt;- readRDS(file_power)
  
  tmp &lt;- c(downsample_size, sum(ECT_res_allround_df$p_ECT &lt; p_cutoff_fdr01), sum(ECT_res_allround_df$p_ECT &lt; p_cutoff_fdr02))
  power_all &lt;- rbind(power_all, tmp)
  
  num_leq_fdr01 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr01)
  num_leq_fdr02 &lt;- sum(ECT_res_allround_df$p_ECT &lt;= p_cutoff_fdr02)
  
  p[[as.character(downsample_size)]] &lt;- ggplot(ECT_res_allround_df, aes(x = -log10(p_ECT))) +
    geom_histogram(bins = 30, color = &quot;white&quot;, fill = &quot;steelblue&quot;) +
    
    # cutoff lines
    geom_vline(xintercept = -log10(p_cutoff_fdr01), color = &quot;red&quot;, size = 1) +
    geom_vline(xintercept = -log10(p_cutoff_fdr02), color = &quot;red&quot;, linetype = &quot;dashed&quot;, size = 1) +
    
    # stacked annotation block
    annotate(
      &quot;label&quot;,
      x = max(-log10(ECT_res_allround_df$p_ECT)),
      y = Inf,
      hjust = 1, vjust = 1.2,
      label = paste0(
        &quot;Count p ≤ FDR 0.1 cutoff: &quot;, num_leq_fdr01, &quot;\n&quot;,
        &quot;Count p ≤ FDR 0.2 cutoff: &quot;, num_leq_fdr02
      ),
      size = 4,
      fill = &quot;white&quot;,
      label.size = 0.3
    ) +
    
    theme_bw(base_size = 13) +
    labs(
      x = &quot;-log10(ECT p-value)&quot;,
      y = &quot;Count&quot;,
      title = paste0(&quot;downsample size &quot;, downsample_size)
    )

  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/977037c765dc7f77a1fec555ab6788033e846f25/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">977037c</a>
</td>
<td>
XSun
</td>
<td>
2025-11-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6920c6bf856ac9c848365efe0a92384b867c3177/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">6920c6b</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/64ce668514ccfb856f49cdbcdb44f60dbb6127b2/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">64ce668</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/aa8e1ff98c4973ac339b20d1c40f779adca87417/docs/figure/revision_power_simulate.Rmd/unnamed-chunk-9-1.png" target="_blank">aa8e1ff</a>
</td>
<td>
XSun
</td>
<td>
2025-11-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>colnames(power_all) &lt;- c(&quot;downsample_size&quot;, &quot;power_fdr01&quot;, &quot;power_fdr02&quot;)
rownames(power_all) &lt;- NULL

DT::datatable(power_all ,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;power&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-b544a055166914a18edc" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b544a055166914a18edc">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">power<\/caption>","data":[[0.01,0.1,0.3,0.5,0.7,1],[4,30,55,64,72,79],[33,93,100,100,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>downsample_size<\/th>\n      <th>power_fdr01<\/th>\n      <th>power_fdr02<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[0,1,2]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gridExtra_2.3 ggplot2_4.0.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12        highr_0.9          RColorBrewer_1.1-3 pillar_1.9.0      
 [5] compiler_4.2.0     bslib_0.3.1        later_1.3.0        jquerylib_0.1.4   
 [9] git2r_0.30.1       workflowr_1.7.0    tools_4.2.0        digest_0.6.29     
[13] gtable_0.3.6       jsonlite_1.8.0     evaluate_0.15      lifecycle_1.0.4   
[17] tibble_3.2.1       pkgconfig_2.0.3    rlang_1.1.2        cli_3.6.1         
[21] rstudioapi_0.13    crosstalk_1.2.0    yaml_2.3.5         xfun_0.41         
[25] fastmap_1.1.0      withr_2.5.0        dplyr_1.1.4        stringr_1.5.1     
[29] knitr_1.39         htmlwidgets_1.5.4  generics_0.1.2     fs_1.5.2          
[33] vctrs_0.6.5        sass_0.4.1         DT_0.22            tidyselect_1.2.0  
[37] rprojroot_2.0.3    grid_4.2.0         glue_1.6.2         R6_2.5.1          
[41] fansi_1.0.3        rmarkdown_2.25     farver_2.1.0       magrittr_2.0.3    
[45] whisker_0.4        scales_1.4.0       promises_1.2.0.1   htmltools_0.5.2   
[49] dichromat_2.0-0.1  httpuv_1.6.5       labeling_0.4.2     S7_0.2.0          
[53] utf8_1.2.2         stringi_1.7.6     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
