<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-11-06" />

<title>Summarizing ECT results – ≥ 5 supporting SNPs &amp; ≥ 20 genes, SLE removed</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis_new</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis_new">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Summarizing ECT results – ≥ 5 supporting
SNPs &amp; ≥ 20 genes, SLE removed</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-11-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis_new/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree33e06f736eed2fbc1d073176551806e5cd0c37ebtargetblank33e06f7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis_new/tree/33e06f736eed2fbc1d073176551806e5cd0c37eb" target="_blank">33e06f7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree33e06f736eed2fbc1d073176551806e5cd0c37ebtargetblank33e06f7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis_new/tree/33e06f736eed2fbc1d073176551806e5cd0c37eb" target="_blank">33e06f7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/revision_figures.Rmd
    Modified:   analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd</code>)
and HTML
(<code>docs/revision_summarizing_ECT_setting_SLEremoved.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/revision_summarizing_ECT_setting_SLEremoved.html" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/c81644744954e1781983b8656df97d40ea6925f1/docs/revision_summarizing_ECT_setting_SLEremoved.html" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>We applied cutoffs of + - <code>supporting SNP ≥ 3</code> and -
<code>gene number in pathway ≥ 20</code></p>
<p>to filter the pairs before performing FDR control. All results
presented below are based on this setting.</p>
<pre class="r"><code>library(dplyr)
library(ggplot2)
library(gridExtra)
library(forcats)
library(tidyr)

library(ensembldb)
library(EnsDb.Hsapiens.v75)
library(ggrepel)
library(ggrastr)

celltypes &lt;- c(&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;CD15_positive_leukocyte&quot;,&quot;platelet&quot;,&quot;T_cell&quot;,&quot;thymocyte&quot;)
types &lt;- c(&quot;B cell (CD19+)&quot;,&quot;Monocyte (CD14+)&quot;,&quot;Leukocyte (CD15+)&quot;,&quot;Platelet&quot;,&quot;T cell (CD4+)&quot;,&quot;T cell (CD8+)&quot;)
names(celltypes) &lt;- types
names(types) &lt;- celltypes

folder_ect_summary &lt;- &quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;</code></pre>
<div id="eur" class="section level1">
<h1>EUR</h1>
<div id="detailed-results-for-all-pairs-at-ect-fdr-0.2"
class="section level2">
<h2>Detailed results for all pairs at ECT FDR &lt; 0.2</h2>
<pre class="r"><code>df_all_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  df &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_all_fdr02 &lt;- rbind(df_all_fdr02,df_fdr02)
  
}


df_all_fdr02$celltype &lt;- types[df_all_fdr02$celltype]
df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP),]

df_all_fdr01 &lt;- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP &lt; 0.1,]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Pairs with ECT FDR &lt;= 0.2 &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-0c923cdc0df2ab45b6f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c923cdc0df2ab45b6f5">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Pairs with ECT FDR &lt;= 0.2 <\/caption>","data":[["11","14","11794","11795","11796","2","110","210","43","15754","1","21","16721","4","6","7","8","9","16436","16437","18166","10","15755","41","5","11797","16","20","22","23","24","25","26","16438","16439","16440","16441","16442","16443","16444","16445","16446","16447","18168","92","27","28","18169","101","131","141","151","161","171","181","191","202","222","242","261","281","29","311","321","331","34","35","36","15756","15757","15759","15761","15763","15764","15765","15766","17504","61","38","39","411","42","15768","15769","15770","31","32","33","16448","16451","16452","81","91","111","12","13","15","17","18","19","201","211","221","231","241","251","271","11800","11801","11802","11805","11806","11807","11808","11809","11810","11811","11812","11813","11814","44","49","15771","17505","50"],["Leukocyte (CD15+)","Platelet","Platelet","Platelet","Platelet","B cell (CD19+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","T cell (CD8+)","Platelet","Platelet","Platelet","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","T cell (CD8+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","Platelet","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"],["MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","Psoriasis-ebi-a-GCST90038681","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","RBC-ebi-a-GCST90002403","asthma_pmid36778051","MPV-ebi-a-GCST90002395","T1D-ebi-a-GCST90014023","RBC-ebi-a-GCST90002403","MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","HCT-ebi-a-GCST90002383","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","HCT-ebi-a-GCST90002383","RET-ebi-a-GCST90002405","LDL-ukb-d-30780_irnt","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","T1D-ebi-a-GCST90014023","MCV-ebi-a-GCST90002392","RBC-ebi-a-GCST90002403","RBC-ebi-a-GCST90002403","T1D-ebi-a-GCST90014023","HB-ebi-a-GCST90002384","MCH-ebi-a-GCST90002390","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","lymphc-ebi-a-GCST90002388","PLT-ebi-a-GCST90002402","MCV-ebi-a-GCST90002392","HB-ebi-a-GCST90002384","lymphc-ebi-a-GCST90002388","IBD-ebi-a-GCST003043","MCV-ebi-a-GCST90002392","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","IBD-ebi-a-GCST003043","RBC-ebi-a-GCST90002403","WBC-ebi-a-GCST90002407","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","HB-ebi-a-GCST90002384","MCV-ebi-a-GCST90002392","WBC-ebi-a-GCST90002407","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","MCV-ebi-a-GCST90002392","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","WBC-ebi-a-GCST90002407","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","UC-ebi-a-GCST003045","IBD-ebi-a-GCST003043","HB-ebi-a-GCST90002384","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MCV-ebi-a-GCST90002392","WBC-ebi-a-GCST90002407","asthma_pmid36778051","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","HB-ebi-a-GCST90002384","MCV-ebi-a-GCST90002392","HB-ebi-a-GCST90002384","WBC-ebi-a-GCST90002407","HCT-ebi-a-GCST90002383","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","HB-ebi-a-GCST90002384","HCT-ebi-a-GCST90002383","UC-ebi-a-GCST003045","UC-ebi-a-GCST003045","UC-ebi-a-GCST003045","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","HB-ebi-a-GCST90002384","MCH-ebi-a-GCST90002390","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MCV-ebi-a-GCST90002392","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","MCV-ebi-a-GCST90002392"],["04910_PC4","04064_PC1","04390_PC5","04714_PC4","00230_PC1","04062_PC2","04971_PC2","04670_PC1","04971_PC1","04211_PC5","04068_PC1","04924_PC3","04390_PC2","04211_PC3","04970_PC3","03010_PC2","04060_PC2","04060_PC2","04213_PC2","04657_PC2","00510_PC2","04510_PC5","04962_PC2","04390_PC3","04971_PC4","04625_PC3","00640_PC1","03040_PC4","03040_PC4","04217_PC4","04144_PC4","04151_PC5","04730_PC5","04917_PC2","04380_PC3","04640_PC2","04722_PC4","04015_PC4","04659_PC4","03050_PC3","04650_PC3","04010_PC4","04668_PC2","00630_PC4","04390_PC4","04110_PC5","04612_PC5","04612_PC1","04060_PC2","04070_PC2","04014_PC3","04370_PC3","04217_PC5","04714_PC4","04972_PC3","03040_PC3","00071_PC2","03440_PC3","04216_PC2","04911_PC3","00514_PC5","04010_PC5","04929_PC2","04340_PC4","03018_PC3","04550_PC3","04919_PC4","03060_PC3","04151_PC4","04730_PC1","04925_PC3","04612_PC3","04935_PC1","04727_PC2","04520_PC4","03008_PC2","04218_PC5","04270_PC4","04918_PC3","04072_PC3","04071_PC2","04714_PC4","04962_PC1","00280_PC3","04010_PC3","04210_PC2","04068_PC3","04921_PC3","04620_PC5","04728_PC4","04918_PC2","00270_PC3","04918_PC4","00562_PC5","04062_PC2","04611_PC5","04611_PC4","04022_PC4","04910_PC1","04371_PC5","03013_PC5","04260_PC5","04726_PC3","00480_PC4","04217_PC3","04145_PC1","04061_PC2","04724_PC4","04261_PC4","04726_PC4","04728_PC5","04912_PC1","03018_PC2","04024_PC1","04725_PC1","04218_PC1","04144_PC4","04150_PC2","04151_PC2","04216_PC3","04926_PC4","04146_PC3","04929_PC1","04215_PC2","04024_PC4"],["04910","04064","04390","04714","00230","04062","04971","04670","04971","04211","04068","04924","04390","04211","04970","03010","04060","04060","04213","04657","00510","04510","04962","04390","04971","04625","00640","03040","03040","04217","04144","04151","04730","04917","04380","04640","04722","04015","04659","03050","04650","04010","04668","00630","04390","04110","04612","04612","04060","04070","04014","04370","04217","04714","04972","03040","00071","03440","04216","04911","00514","04010","04929","04340","03018","04550","04919","03060","04151","04730","04925","04612","04935","04727","04520","03008","04218","04270","04918","04072","04071","04714","04962","00280","04010","04210","04068","04921","04620","04728","04918","00270","04918","00562","04062","04611","04611","04022","04910","04371","03013","04260","04726","00480","04217","04145","04061","04724","04261","04726","04728","04912","03018","04024","04725","04218","04144","04150","04151","04216","04926","04146","04929","04215","04024"],["Insulin signaling pathway","NF-kappa B signaling pathway","Hippo signaling pathway","Thermogenesis","Purine metabolism","Chemokine signaling pathway","Gastric acid secretion","Leukocyte transendothelial migration","Gastric acid secretion","Longevity regulating pathway","FoxO signaling pathway","Renin secretion","Hippo signaling pathway","Longevity regulating pathway","Salivary secretion","Ribosome","Cytokine-cytokine receptor interaction","Cytokine-cytokine receptor interaction","Longevity regulating pathway - multiple species","IL-17 signaling pathway","N-Glycan biosynthesis","Focal adhesion","Vasopressin-regulated water reabsorption","Hippo signaling pathway","Gastric acid secretion","C-type lectin receptor signaling pathway","Propanoate metabolism","Spliceosome","Spliceosome","Necroptosis","Endocytosis","PI3K-Akt signaling pathway","Long-term depression","Prolactin signaling pathway","Osteoclast differentiation","Hematopoietic cell lineage","Neurotrophin signaling pathway","Rap1 signaling pathway","Th17 cell differentiation","Proteasome","Natural killer cell mediated cytotoxicity","MAPK signaling pathway","TNF signaling pathway","Glyoxylate and dicarboxylate metabolism","Hippo signaling pathway","Cell cycle","Antigen processing and presentation","Antigen processing and presentation","Cytokine-cytokine receptor interaction","Phosphatidylinositol signaling system","Ras signaling pathway","VEGF signaling pathway","Necroptosis","Thermogenesis","Pancreatic secretion","Spliceosome","Fatty acid degradation","Homologous recombination","Ferroptosis","Insulin secretion","Other types of O-glycan biosynthesis","MAPK signaling pathway","GnRH secretion","Hedgehog signaling pathway","RNA degradation","Signaling pathways regulating pluripotency of stem cells","Thyroid hormone signaling pathway","Protein export","PI3K-Akt signaling pathway","Long-term depression","Aldosterone synthesis and secretion","Antigen processing and presentation","Growth hormone synthesis, secretion and action","GABAergic synapse","Adherens junction","Ribosome biogenesis in eukaryotes","Cellular senescence","Vascular smooth muscle contraction","Thyroid hormone synthesis","Phospholipase D signaling pathway","Sphingolipid signaling pathway","Thermogenesis","Vasopressin-regulated water reabsorption","Valine, leucine and isoleucine degradation","MAPK signaling pathway","Apoptosis","FoxO signaling pathway","Oxytocin signaling pathway","Toll-like receptor signaling pathway","Dopaminergic synapse","Thyroid hormone synthesis","Cysteine and methionine metabolism","Thyroid hormone synthesis","Inositol phosphate metabolism","Chemokine signaling pathway","Platelet activation","Platelet activation","cGMP-PKG signaling pathway","Insulin signaling pathway","Apelin signaling pathway","RNA transport","Cardiac muscle contraction","Serotonergic synapse","Glutathione metabolism","Necroptosis","Phagosome","Viral protein interaction with cytokine and cytokine receptor","Glutamatergic synapse","Adrenergic signaling in cardiomyocytes","Serotonergic synapse","Dopaminergic synapse","GnRH signaling pathway","RNA degradation","cAMP signaling pathway","Cholinergic synapse","Cellular senescence","Endocytosis","mTOR signaling pathway","PI3K-Akt signaling pathway","Ferroptosis","Relaxin signaling pathway","Peroxisome","GnRH secretion","Apoptosis - multiple species","cAMP signaling pathway"],[8,5,6,9,8,6,5,10,5,9,9,6,16,12,7,5,6,8,7,6,6,7,5,8,5,6,6,8,7,9,11,9,11,6,5,9,7,14,8,5,6,7,5,7,8,8,6,7,5,6,8,11,7,16,6,5,14,5,21,6,5,6,8,10,6,5,11,5,9,5,10,9,5,6,5,7,16,5,7,5,5,10,7,12,5,5,7,5,5,5,6,9,6,5,5,9,10,6,16,7,5,14,5,8,15,8,7,10,5,8,7,5,6,8,7,20,15,8,8,6,20,6,5,9,15],[1e-06,8e-06,6.7e-05,9.4e-05,0.000177,4e-05,5e-05,8.3e-05,0.000189,0.00018,7.1e-05,0.000105,4.3e-05,0.000195,0.00035,0.00045,0.000489,0.000565,0.000221,0.000519,0.000278,0.000709,0.000346,0.00064,0.000677,0.000595,0.0012999,0.0016998,0.0017998,0.0019998,0.0019998,0.0020998,0.0020998,0.001053,0.001153,0.0012999,0.001372,0.0013999,0.001527,0.0018998,0.0019998,0.0021998,0.0021998,0.0018998,0.000436,0.0022998,0.0023998,0.0024998,0.000527,0.0010999,0.0010999,0.0011999,0.0011999,0.0012999,0.0012999,0.0013999,0.001478,0.0017998,0.0019998,0.0019998,0.0020998,0.0021998,0.0022998,0.0023998,0.0023998,0.0024998,0.0025997,0.0025997,0.0011999,0.0011999,0.0013999,0.0016998,0.0018998,0.0020998,0.0020998,0.0025997,0.001328,0.000803,0.0028997,0.0029997,0.0031997,0.0031997,0.0027997,0.0029997,0.0031997,0.0026997,0.0029997,0.0029997,0.0028997,0.0030997,0.0031997,0.0013999,0.0013999,0.0017998,0.0018998,0.0019998,0.0021998,0.0024998,0.0027997,0.0032997,0.0032997,0.0033997,0.0035996,0.0037996,0.0037996,0.0038996,0.0039996,0.0016998,0.0017998,0.0018998,0.0023998,0.0024998,0.0027997,0.0027997,0.0031997,0.0036996,0.0037996,0.0037996,0.0037996,0.0039996,0.0037996,0.0037996,0.0036996,0.0037996,0.0038996],[0.001438,0.014296,0.0559927,0.0559927,0.0790748,0.08832,0.0996,0.0996,0.1134,0.1134,0.129815,0.129815,0.129815,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1565472,0.16608,0.1728284,0.1728284,0.1728284,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1744,0.175102,0.1766253,0.1780503,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1793701,0.18284,0.18284,0.18284,0.18284,0.18284,0.18284,0.18284,0.1862793,0.1892382,0.1892382,0.1892382,0.1901149,0.1909443,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.19824,0.19824,0.19824,0.19824,0.1991285],[137,102,155,231,130,189,75,112,75,89,131,69,155,89,90,153,294,294,62,94,50,199,44,155,75,104,34,149,149,162,249,354,60,70,128,99,119,210,107,45,134,295,112,30,155,124,78,78,294,99,232,59,162,231,102,149,44,41,40,86,46,295,64,50,79,140,119,23,354,60,98,78,119,89,71,105,160,132,74,148,119,231,44,48,295,136,131,153,104,132,74,49,74,74,189,124,124,167,137,137,179,86,115,56,162,152,100,114,149,115,132,93,79,216,112,160,249,153,354,40,129,83,64,32,216],[87,47,41,90,34,109,34,69,34,57,85,30,70,56,34,126,101,101,36,46,39,105,33,41,28,54,29,111,111,93,175,175,31,42,86,62,89,104,79,40,79,162,69,23,75,92,57,57,118,67,119,39,97,130,40,112,28,28,33,31,25,168,30,24,67,63,84,20,178,30,49,65,71,35,44,61,117,55,36,81,81,130,33,39,168,104,87,79,63,76,31,23,24,29,88,70,70,67,62,56,71,34,38,27,69,81,38,36,56,38,56,37,37,71,41,78,111,55,126,22,69,59,30,25,94],["bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","disease_EUR_old","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new"],["-0.245270464051887","-0.41768128699344","-0.219352852012402","-0.797167485258106","0.0490457193697609","1.69054931142635","-0.160017285934662","-0.146323318161613","-0.0195316446390257","-0.280029290491469","0.219067049860872","0.0574186991373062","-0.056669509978619","0.0673449539000475","-0.128493235792523","-0.226096753153129","0.233290435680774","0.356193457802385","-0.27071396926541","-0.0983783761147528","0.803780051926971","-0.319389509893942","-0.369079521944207","0.165289903927005","0.0109467259031775","-0.131943556293009","-0.0400493575356977","0.38668824064209","0.0345362910612599","0.506296610377211","-0.0133601370358113","-0.390877583722192","-0.255109506096205","-0.15053842633599","-0.0830551548728864","-0.643224676525653","0.00675380298541528","-0.206185838472806","-0.348348769804703","-0.417405016126126","-0.32927676223193","-0.0201372579675425","-0.46363696898361","-0.298272254009567","0.272708295051201","0.677674795102813","0.807887694724516","0.177658176432276","-0.0495864361159254","0.400944087090976","-0.284482997416838","0.0116704298005137","-0.0139165601476714","0.908184857186706","-0.0427098263051655","-0.346552058826978","0.373332837053078","0.138848375442602","0.186204196970328","0.17112589085453","0.132673406068839","-0.148981738791476","0.201944929329823","-0.0814124936249649","0.609834657956357","0.306469104611784","0.341146698851871","-0.173591571192809","-0.167618721127787","0.0900331861451125","0.0915678874980503","0.650380512841018","-0.0640671509178383","0.0601787103416731","0.586873863031775","0.0178089277936049","-0.226259051798917","-0.0910772003692204","-0.018084141884111","0.149081328940241","0.124554307321948","-0.537867543275294","0.384601143357574","-0.0498490589289173","-0.566552792213228","-0.227886836956831","0.0697980284727306","0.00693293256160454","0.14373227924004","0.246432050632072","0.177562763063755","0.5259942054983","0.378866542149566","0.560386923702547","-0.39756014039602","-0.659405280342488","0.0454542778745604","0.486738685474682","0.549513227634144","-0.227839601219098","-0.133814949928015","0.156927059687846","0.10608763221134","0.776218730681591","0.0150066929027473","-0.134040508002528","-0.413012718373063","-0.54428879927892","-0.405271170683785","0.392211512031683","-0.539877365042058","0.13958108952101","0.0584426773316338","0.0441347782270358","0.147938467588175","0.296463052202976","0.22793051722002","-0.0416611070894055","0.0323660276705429","-0.165866865073491","-0.0194109476464144","0.150491568198327","0.355452720582586","0.175664192729935","-0.0497481840979672"],["0.381444392107266","0.386094674495389","0.606360167370838","0.132433832841616","0.916858027973131","0.793939736663622","0.272526832466116","0.380080136666012","0.878437186997933","0.226486849411033","0.556808502315391","0.919996273137037","0.772695185086866","0.419586506203072","0.696491658215512","0.652614467038651","0.714976796119257","0.553028372038752","0.315566916417708","0.57743030781489","0.213549053167502","0.518149015164986","0.193315258562892","0.84410453151432","0.981300380495821","0.670683188315133","0.502309526364542","0.155265962575607","0.92292114024808","0.341137939355012","0.870945912245638","0.42444858566781","0.462912348171951","0.649473464532669","0.825651549584431","0.0384225144339296","0.977036848092291","0.375685874442427","0.262400957242923","0.136673185020984","0.346333133638904","0.927089640139701","0.32468447926906","0.481583736109852","0.187412601065013","0.0513939859196159","0.275333129922457","0.731412966083133","0.666382327665147","0.0946000828443417","0.294726350408341","0.96720958554113","0.937378747053766","0.130430359113488","0.924512828343235","0.234718940896182","0.176864788775393","0.603525221729898","0.598502916408624","0.493810098956484","0.749491860203692","0.339101948874325","0.169171895778485","0.791365014934603","0.0338619202031168","0.263384075532353","0.240923912300581","0.717262597195071","0.398004857735557","0.558453875836253","0.788672156956791","0.0120705732822415","0.750408941399446","0.749755390391455","0.130446334859127","0.921748880203971","0.695014070209916","0.603283698260543","0.858900654427318","0.670031725380746","0.607938158577543","0.220017018873353","0.0383060556657068","0.902885205156944","0.105475680291036","0.400970678595854","0.850217531548008","0.991924642896619","0.573284618882391","0.397134232450949","0.336094738609318","0.257978185483141","0.124962375643065","0.234557683466446","0.365943356400005","0.386842509399878","0.946348478214342","0.410045414016918","0.388446580296467","0.698338609981389","0.537255583958351","0.482016680200454","0.632751968554825","0.189242100754593","0.978206000196248","0.865495007923532","0.185519835265756","0.231930532914844","0.386464726940903","0.231288726314972","0.258818812447651","0.681373709410221","0.888791317157249","0.906030570268332","0.68939993487819","0.392331848308167","0.58136440984346","0.91726350363736","0.921461676690888","0.660935938097067","0.94134129987067","0.323319147212933","0.0329821476481283","0.696576899217406","0.839044609865298"],[0.0650233,0.0829709,null,null,null,0.0350311,null,null,null,null,0.0776991,0.0925372,null,0.0655913,0.0442804,0.0273526,0.0664088,0.0452325,null,null,null,0.0040001,null,null,0.0428586,null,0.8828099,0.046462,0.0488894,0.0092223,0.0436977,0.0783865,0.0868603,null,null,null,null,null,null,null,null,null,null,null,null,0.0065683,0.0001711,null,null,0.0318237,null,null,null,0.015001,0.0377493,null,null,null,0.0238865,0.0579279,0.0182108,null,null,0.034016,null,null,null,0.0036896,null,null,null,null,null,null,null,null,null,null,null,0.0709253,0.0690528,0.0376975,null,null,null,0.0644015,0.0462309,0.0188757,null,null,null,0.0142615,0.0132213,0.0404826,0.0551095,0.0352653,0.054751,0.0265939,0.0100246,0.0137847,null,null,null,0.0440503,0.0386792,3.4e-05,0.0011242,null,null,null,null,null,null,null,null,null,null,null,null,null,0.0525288,null,null,null,0.0082523],[0.0646544,0.0825434,null,null,null,0.0313682,null,null,null,null,0.0771593,0.0907451,null,0.0638589,0.043997,0.0271653,0.0528417,0.0381385,null,null,null,0.0039809,null,null,0.0426652,null,0.887038,0.0465073,0.0486338,0.0091719,0.0421872,0.0761631,0.1009872,null,null,null,null,null,null,null,null,null,null,null,null,0.0065155,0.023752,null,null,0.0315655,null,null,null,0.0150468,0.0378506,null,null,null,0.0237279,0.0574937,0.0180983,null,null,0.0337729,null,null,null,0.0131945,null,null,null,null,null,null,null,null,null,null,null,0.0701786,0.0685495,0.0371428,null,null,null,0.0640073,0.0463672,0.0182766,null,null,null,0.0141772,0.0131166,0.040178,0.0547758,0.0353031,0.0542514,0.026447,0.0099601,0.0137237,null,null,null,0.0440581,0.0385244,3.38e-05,0.0011168,null,null,null,null,null,null,null,null,null,null,null,null,null,0.0520201,null,null,null,0.0081855],[0.4554455,0.5643564,null,null,null,0.8514851,null,null,null,null,0.3861386,0.6633663,null,0.6039604,0.8415842,0.8910891,0.6633663,0.8019802,null,null,null,1,null,null,0.8712871,null,0.0693069,0.7524752,0.7623762,0.8910891,0.6435644,0.4851485,0.7821782,null,null,null,null,null,null,null,null,null,null,null,null,0.9207921,0.7920792,null,null,0.7425743,null,null,null,0.9108911,0.9108911,null,null,null,0.8019802,0.8316832,0.950495,null,null,0.9207921,null,null,null,1,null,null,null,null,null,null,null,null,null,null,null,0.6039604,0.5049505,0.7326733,null,null,null,0.4752475,0.4257426,0.8712871,null,null,null,0.9306931,0.8712871,0.8613861,0.4653465,0.8316832,0.6138614,0.8415842,0.9207921,0.9405941,null,null,null,0.8415842,0.7920792,1,0.980198,null,null,null,null,null,null,null,null,null,null,null,null,null,0.5346535,null,null,null,0.950495],[0.4752475,0.5742574,null,null,null,0.8811881,null,null,null,null,0.3861386,0.6633663,null,0.6237624,0.8514851,0.8910891,0.7128713,0.8514851,null,null,null,1,null,null,0.8712871,null,0.0693069,0.7524752,0.7623762,0.8910891,0.6633663,0.5049505,0.7722772,null,null,null,null,null,null,null,null,null,null,null,null,0.9207921,0.950495,null,null,0.7425743,null,null,null,0.9108911,0.9108911,null,null,null,0.8217822,0.8316832,0.950495,null,null,0.9207921,null,null,null,0.950495,null,null,null,null,null,null,null,null,null,null,null,0.6039604,0.5049505,0.7524752,null,null,null,0.4752475,0.4257426,0.8811881,null,null,null,0.9306931,0.8712871,0.8613861,0.4653465,0.8316832,0.6237624,0.8415842,0.9306931,0.9405941,null,null,null,0.8415842,0.7920792,1,0.980198,null,null,null,null,null,null,null,null,null,null,null,null,null,0.5247525,null,null,null,0.950495],[0.7167513,0.5756002,null,null,null,0.8935047,null,null,null,null,0.4974161,0.6241937,null,0.7607848,0.7547771,0.8486678,0.6643514,0.7687305,null,null,null,0.9760002,null,null,0.7502488,null,0.0699219,0.7330176,0.7312656,0.9430702,0.829668,0.6169255,0.5726002,null,null,null,null,null,null,null,null,null,null,null,null,0.955574,0.9952248,null,null,0.7783492,null,null,null,0.8880935,0.7449209,null,null,null,0.8431441,0.6360569,0.8654296,null,null,0.764774,null,null,null,0.9642783,null,null,null,null,null,null,null,null,null,null,null,0.6000571,0.585509,0.7438553,null,null,null,0.6466229,0.7349866,0.9175045,null,null,null,0.9042374,0.8840603,0.7622684,0.677284,0.7920513,0.6789588,0.8274007,0.9283357,0.910858,null,null,null,0.7477365,0.7749677,0.9979508,0.9810839,null,null,null,null,null,null,null,null,null,null,null,null,null,0.6635624,null,null,null,0.9344793]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,14,15,16,17,18]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-trait"
class="section level2">
<h2>Number of significant pairs at different cutoffs, for each
trait</h2>
<div id="ect-fdr-0.1" class="section level3">
<h3>ECT FDR &lt; 0.1</h3>
<pre class="r"><code>FDRcutoff &lt;- 0.1
MBEcutoff &lt;- 0.05

p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  df_summary &lt;- df %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_FDR_3suppSNP = sum(ECT_FDR_5suppSNP &lt; FDRcutoff, na.rm = TRUE)
    ) %&gt;%
    arrange(desc(n_sig_FDR_3suppSNP))
  
  df_summary$trait[df_summary$trait ==&quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  df_summary &lt;- df_summary %&gt;%
    mutate(trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait))
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_FDR_3suppSNP),
                       y = n_sig_FDR_3suppSNP,
                       fill = n_sig_FDR_3suppSNP)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7, color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;lightblue&quot;, high = &quot;steelblue&quot;) +
  labs(
    x = &quot;Trait&quot;,
    y = paste0(&quot;# of pairs at FDR &lt; &quot;, FDRcutoff),
    title = types[celltype]
  ) +
  theme_bw(base_size = 14) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
  )
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-1.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- list()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype, &quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric if it’s stored as character
  df &lt;- df %&gt;%
    mutate(mbe_pval_reverse_pair = suppressWarnings(as.numeric(mbe_pval_reverse_pair)))
  
  # Keep only FDR-significant pairs
  df_sig &lt;- df %&gt;%
    dplyr::filter(ECT_FDR_5suppSNP &lt; FDRcutoff)
  
  # Fix naming if needed
  df_sig$trait[df_sig$trait == &quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  # Summarize counts per trait
  df_summary &lt;- df_sig %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_pairs = n(),
      n_mbe_sig = sum(!is.na(mbe_pval_reverse_pair) &amp; mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    # Keep traits with at least one FDR-significant pair
    dplyr::filter(n_sig_pairs &gt;= 1) %&gt;%
    mutate(
      trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait)
    ) %&gt;%
    arrange(desc(n_sig_pairs))
  
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_pairs))) +
    geom_bar(aes(y = n_sig_pairs), stat = &quot;identity&quot;, fill = &quot;grey80&quot;, color = &quot;black&quot;, width = 0.7) +
    geom_bar(aes(y = n_mbe_sig), stat = &quot;identity&quot;, fill = &quot;darkturquoise&quot;, width = 0.7) +
    geom_text(aes(y = n_mbe_sig, label = n_mbe_sig), vjust = -0.4, size = 3.5) +
    labs(
      x = &quot;Trait&quot;,
      y = paste0(&quot;# of FDR&lt;&quot;, FDRcutoff, &quot; (blue: MBE p&lt;&quot;, MBEcutoff, &quot;)&quot;),
      title = types[celltype]
    ) +
    theme_bw(base_size = 14) +
    theme(
      legend.position = &quot;none&quot;,
      axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
      axis.text.y = element_text(size = 10, color = &quot;black&quot;),
      plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
    )
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2, top = &quot;Number of pairs with MBE p &lt; 0.05, for the sig pairs&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-2.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ect-fdr-0.2" class="section level3">
<h3>ECT FDR &lt; 0.2</h3>
<pre class="r"><code>FDRcutoff &lt;- 0.2
MBEcutoff &lt;- 0.05

p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  df_summary &lt;- df %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_FDR_3suppSNP = sum(ECT_FDR_5suppSNP &lt; FDRcutoff, na.rm = TRUE)
    ) %&gt;%
    arrange(desc(n_sig_FDR_3suppSNP))
  
  df_summary$trait[df_summary$trait ==&quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  df_summary &lt;- df_summary %&gt;%
    mutate(trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait))
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_FDR_3suppSNP),
                       y = n_sig_FDR_3suppSNP,
                       fill = n_sig_FDR_3suppSNP)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7, color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;lightblue&quot;, high = &quot;steelblue&quot;) +
  labs(
    x = &quot;Trait&quot;,
    y = paste0(&quot;# of pairs at FDR &lt; &quot;, FDRcutoff),
    title = types[celltype]
  ) +
  theme_bw(base_size = 14) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
  )
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-1.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- list()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype, &quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric if it’s stored as character
  df &lt;- df %&gt;%
    mutate(mbe_pval_reverse_pair = suppressWarnings(as.numeric(mbe_pval_reverse_pair)))
  
  # Keep only FDR-significant pairs
  df_sig &lt;- df %&gt;%
    dplyr::filter(ECT_FDR_5suppSNP &lt; FDRcutoff)
  
    # Fix naming if needed
  df_sig$trait[df_sig$trait == &quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  # Summarize counts per trait
  df_summary &lt;- df_sig %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_pairs = n(),
      n_mbe_sig = sum(!is.na(mbe_pval_reverse_pair) &amp; mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    # Keep traits with at least one FDR-significant pair
    dplyr::filter(n_sig_pairs &gt;= 1) %&gt;%
    mutate(
      trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait)
    ) %&gt;%
    arrange(desc(n_sig_pairs))
  

  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_pairs))) +
    geom_bar(aes(y = n_sig_pairs), stat = &quot;identity&quot;, fill = &quot;grey80&quot;, color = &quot;black&quot;, width = 0.7) +
    geom_bar(aes(y = n_mbe_sig), stat = &quot;identity&quot;, fill = &quot;darkturquoise&quot;, width = 0.7) +
    geom_text(aes(y = n_mbe_sig, label = n_mbe_sig), vjust = -0.4, size = 3.5) +
    labs(
      x = &quot;Trait&quot;,
      y = paste0(&quot;# of FDR&lt;&quot;, FDRcutoff, &quot; (blue: MBE p&lt;&quot;, MBEcutoff, &quot;)&quot;),
      title = types[celltype]
    ) +
    theme_bw(base_size = 14) +
    theme(
      legend.position = &quot;none&quot;,
      axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
      axis.text.y = element_text(size = 10, color = &quot;black&quot;),
      plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
    )
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2, top = &quot;Number of pairs with MBE p &lt; 0.05, for the sig pairs&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-2.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-2.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-cell-type"
class="section level2">
<h2>Number of significant pairs at different cutoffs, for each cell
type</h2>
<pre class="r"><code>num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2 &amp; df$ECT_FDR_5suppSNP &gt;= 0.1,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))
  
}

sum &lt;- data.frame(celltype = types,
                  num_fdr01 = num_fdr01,
                  num_fdr02 = num_fdr02)


df &lt;- sum %&gt;%
  pivot_longer(cols = starts_with(&quot;num_fdr&quot;), names_to = &quot;event&quot;, values_to = &quot;total&quot;) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# Plot
p1 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(x = &quot;Cell types&quot;, y = &quot;Count&quot;) +
  ggtitle(&quot;# of pairs&quot;) +
  ylim(0, 60) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, 
                       celltype, &quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric safely (ignore &quot;0/1snp&quot; or NA)
  df$mbe_pval_reverse_pair &lt;- suppressWarnings(as.numeric(df$mbe_pval_reverse_pair))
  df$mbe_pval_reverse_pair[is.na(df$mbe_pval_reverse_pair)] &lt;- 0.5
  
  # --- FDR &lt; 0.1 ---
  df_fdr01 &lt;- df %&gt;%
    dplyr::filter(
      ECT_FDR_5suppSNP &lt; 0.1,
      !(mbe_pval_reverse_pair &lt; MBEcutoff)  # remove reverse-causal pairs
    ) %&gt;%
    dplyr::filter(!is.na(p_ECT))
  
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))
  
  # --- 0.1 &lt; FDR &lt; 0.2 ---
  df_fdr02 &lt;- df %&gt;%
    dplyr::filter(
      ECT_FDR_5suppSNP &lt; 0.2,
      ECT_FDR_5suppSNP &gt;= 0.1,
      !(mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    dplyr::filter(!is.na(p_ECT))
  
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))
}

# summary table
sum &lt;- data.frame(
  celltype = types,
  num_fdr01 = num_fdr01,
  num_fdr02 = num_fdr02
)



df &lt;- sum %&gt;%
  pivot_longer(
    cols = starts_with(&quot;num_fdr&quot;),
    names_to = &quot;event&quot;,
    values_to = &quot;total&quot;
  ) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;,
                                        &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# ---- Plot ----
p2 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(
    x = &quot;Cell types&quot;,
    y = &quot;Count (excluding MBE p &lt; 0.05)&quot;,
    title = &quot;# of pairs after removing reverse-causal signals&quot;
  ) +
  ylim(0, 60) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

all &lt;- grid.arrange(p1, p2, nrow = 1)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-5-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-5-1.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-5-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="traits-and-pathways-identified" class="section level2">
<h2>Traits and pathways identified</h2>
<div id="ibd-ebi-a-gcst003043" class="section level3">
<h3>IBD-ebi-a-GCST003043</h3>
<pre class="r"><code>trait &lt;- &quot;IBD-ebi-a-GCST003043&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  dplyr::mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-6-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>supp_snps_all &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/supp_snps_ECTFDR02.RDS&quot;)
supp_snps_trait &lt;- supp_snps_all[supp_snps_all$trait == trait,]

supp_snps_trait &lt;- supp_snps_trait[supp_snps_trait$factor %in% df_trait_fdr01$factor,]

DT::datatable(supp_snps_trait,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&quot;supporting snps for &quot;, trait)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-ad41175326dc68f35292" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ad41175326dc68f35292">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">supporting snps for IBD-ebi-a-GCST003043<\/caption>","data":[["604","605","606","607","608","609","610","611","612","613","614","615","616","617","618"],["thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte"],["IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043"],["04971_PC2","04971_PC2","04971_PC2","04971_PC2","04971_PC2","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1"],["04971","04971","04971","04971","04971","04670","04670","04670","04670","04670","04670","04670","04670","04670","04670"],["Gastric acid secretion","Gastric acid secretion","Gastric acid secretion","Gastric acid secretion","Gastric acid secretion","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration"],["5e-05","5e-05","5e-05","5e-05","5e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05"],["0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996"],["5","5","5","5","5","10","10","10","10","10","10","10","10","10","10"],["rs1388585","rs2024092","rs2497318","rs35256947","rs7738430","rs1292053","rs13107612","rs1388585","rs259964","rs3776414","rs395157","rs648541","rs6740462","rs7097656","rs740287"],["12","19","10","2","6","17","4","12","20","5","5","11","2","10","7"],["40531691","1124031","94432000","231161026","31508836","57963537","102739980","40531691","57824309","10689562","38867732","114429934","65667272","82250831","107584780"],["0.001531","0.005837","0.003773","0.006124","7.166e-06","0.01326","0.0008946","0.001019","0.007458","0.001132","0.0116","0.001247","0.00909","0.001859","0.01426"],["0.1064045","0.1702472","0.1702472","0.1702472","0.000996074","0.198214","0.04333325","0.04333325","0.172777","0.04333325","0.198214","0.04333325","0.180501428571429","0.0516802","0.198214"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pwyID<\/th>\n      <th>pwyNAME<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>Num_supp_snp<\/th>\n      <th>SNP<\/th>\n      <th>chr<\/th>\n      <th>bp<\/th>\n      <th>pval_SNP_factor<\/th>\n      <th>fdr_SNP_factor<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-6-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-3">
Past versions of unnamed-chunk-6-3.png
</button>
</p>
<div id="fig-unnamed-chunk-6-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-6-3.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<table>
<colgroup>
<col width="7%" />
<col width="6%" />
<col width="86%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Pathway</th>
<th align="center">Evidence for role in IBD?</th>
<th align="center">Notes / rationale</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Gastric acid secretion</td>
<td align="center">Weak / indirect</td>
<td align="center">There are associations: e.g., acid‐suppressive
medications (proton-pump inhibitors, H2 R antagonists) that reduce
gastric acid secretion have been associated with worse outcomes in IBD.
(Karger Publishers) Also acid‐base / electrolyte disturbances are
observed in IBD. (PMC) But the pathway of “gastric acid secretion” per
se is not usually considered a core mechanistic immune/epithelial
pathway in IBD. So I’d mark it as possible but not a strong hit.</td>
</tr>
<tr class="even">
<td align="center">Leukocyte transendothelial migration</td>
<td align="center">Strong evidence</td>
<td align="center">Leukocyte recruitment and migration (including across
endothelium) is central to intestinal inflammation in IBD. For example,
a review of therapies targeting leukocyte trafficking in the gut is
available. (PMC) Also the KEGG pathway “Leukocyte transendothelial
migration” is listed and overlaps immune‐system processes. (KEGG) So
this pathway is well supported.</td>
</tr>
<tr class="odd">
<td align="center">Cytokine-cytokine receptor interaction</td>
<td align="center">Strong evidence</td>
<td align="center">Cytokines and their receptors are central to IBD
pathogenesis (many therapies target these). Reviews explicitly mention
pathways of cytokine-cytokine receptor interaction in IBD. (PMC) Also
gene-set analyses pick this pathway out in IBD datasets. So this is a
good candidate to include.</td>
</tr>
<tr class="even">
<td align="center">Necroptosis</td>
<td align="center">Good evidence / emerging</td>
<td align="center">Necroptosis (a programmed necrosis cell-death
pathway) has been implicated in IBD: for example “The Function of
Necroptosis and Its Treatment Target in IBD” review. (PMC) And more
recent work links epithelial death via RIPK3/MLKL to colitis. (JCI
Insight) So it’s not as classical as leukocyte migration or cytokine
signalling, but it is increasingly recognised.</td>
</tr>
<tr class="odd">
<td align="center">MAPK signalling pathway</td>
<td align="center">Strong evidence</td>
<td align="center">The MAPK family (ERK/JNK/p38) has been reviewed in
the context of IBD, noting their activation in mucosa, links to
inflammatory signals, and explored as therapeutic targets. (PubMed) Also
experimental colitis models show involvement. (PLOS) So yes this pathway
is relevant.</td>
</tr>
<tr class="even">
<td align="center">GnRH secretion</td>
<td align="center">Weak / no strong direct evidence</td>
<td align="center">The “GnRH secretion” endocrine pathway is not one I
found strong direct links to IBD pathogenesis in the literature. It
appears more in endocrine/reproductive biology rather than gut
inflammation. So likely less relevant (unless you have a specific
hypothesis linking it).</td>
</tr>
<tr class="odd">
<td align="center">Thyroid hormone synthesis</td>
<td align="center">Weak / no strong direct evidence</td>
<td align="center">Similar to above: thyroid hormone synthesis is an
endocrine/metabolic pathway. I did not find strong IBD-pathogenic links
in the core IBD immunology/epithelial literature. Could be tangential
(via metabolic/inflammatory cross-talk) but not a primary pathway.</td>
</tr>
<tr class="even">
<td align="center">Peroxisome</td>
<td align="center">Moderate evidence</td>
<td align="center">The “peroxisome” pathway (and peroxisome
proliferator‐activated receptors PPARs) has been implicated in
intestinal inflammation and IBD. For example, the review “The Versatile
Role of Peroxisome Proliferator-Activated …” in IBD and gut
inflammation. (PMC) So while not as high-profile as cytokine signalling,
it is a plausible pathway to consider.</td>
</tr>
</tbody>
</table>
</div>
<div id="ret-ebi-a-gcst90002405" class="section level3">
<h3>RET-ebi-a-GCST90002405</h3>
<pre class="r"><code>trait &lt;- &quot;RET-ebi-a-GCST90002405&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-2.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-2.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<table>
<colgroup>
<col width="20%" />
<col width="13%" />
<col width="65%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Pathway</th>
<th align="center">Relevance to Reticulocyte Count</th>
<th align="center">Rationale</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Hippo signaling pathway</td>
<td align="center">Moderate</td>
<td align="center">Hippo regulates cell proliferation and organ size;
some evidence in hematopoietic stem/progenitor cell regulation, so may
influence erythroid progenitors.</td>
</tr>
<tr class="even">
<td align="center">Thermogenesis</td>
<td align="center">Low</td>
<td align="center">Primarily metabolic/thermoregulatory; little direct
evidence linking to reticulocyte production.</td>
</tr>
<tr class="odd">
<td align="center">Purine metabolism</td>
<td align="center">Strong</td>
<td align="center">Nucleotide supply is critical for rapidly dividing
erythroid progenitors; metabolism affects DNA/RNA synthesis in
erythropoiesis.</td>
</tr>
<tr class="even">
<td align="center">Longevity regulating pathway (multiple species)</td>
<td align="center">Low/Moderate</td>
<td align="center">Overlaps with mTOR/PI3K/insulin signalling;
indirectly may affect HSC/erythroid ageing, but not a major direct
driver of retic-count.</td>
</tr>
<tr class="odd">
<td align="center">Vasopressin-regulated water reabsorption</td>
<td align="center">Low</td>
<td align="center">Renal water handling pathway; no clear link to
reticulocyte production.</td>
</tr>
<tr class="even">
<td align="center">C-type lectin receptor signaling pathway</td>
<td align="center">Low/Moderate</td>
<td align="center">Innate immune receptor signalling; could affect
marrow niche via inflammation, but less direct on retic production.</td>
</tr>
<tr class="odd">
<td align="center">Prolactin signaling pathway</td>
<td align="center">Low</td>
<td align="center">Hormone signalling, limited evidence for direct
erythropoiesis/retic effect.</td>
</tr>
<tr class="even">
<td align="center">Osteoclast differentiation</td>
<td align="center">Moderate</td>
<td align="center">Bone marrow micro-environment matters for
erythropoiesis; osteoclast/osteoblast activity interplay may influence
niche and reticulocyte output.</td>
</tr>
<tr class="odd">
<td align="center">Proteasome</td>
<td align="center">Strong</td>
<td align="center">Proteostasis and protein turnover are critical in
late erythroid maturation/reticulocyte formation.</td>
</tr>
<tr class="even">
<td align="center">Natural killer cell mediated cytotoxicity</td>
<td align="center">Low/Moderate</td>
<td align="center">Immune cytotoxic cells may impact marrow environment
under stress, but not a primary regulator of reticulocyte count.</td>
</tr>
<tr class="odd">
<td align="center">TNF signaling pathway</td>
<td align="center">Moderate</td>
<td align="center">Inflammatory cytokine (TNF) can suppress
erythropoiesis in chronic disease → may reduce reticulocyte output.</td>
</tr>
<tr class="even">
<td align="center">Aldosterone synthesis and secretion</td>
<td align="center">Low</td>
<td align="center">Mineralocorticoid hormone system; limited direct
erythroid link.</td>
</tr>
<tr class="odd">
<td align="center">Adherens junction</td>
<td align="center">Moderate</td>
<td align="center">Cell-cell adhesion in marrow niche/erythroblastic
islands may affect erythroid maturation and release of
reticulocytes.</td>
</tr>
<tr class="even">
<td align="center">Valine, leucine and isoleucine degradation</td>
<td align="center">Moderate</td>
<td align="center">Amino acid metabolism for rapidly dividing cells;
BCAA degradation could influence erythroid progenitor growth.</td>
</tr>
<tr class="odd">
<td align="center">MAPK signaling pathway</td>
<td align="center">Strong</td>
<td align="center">MAPK pathways known to mediate erythroid progenitor
proliferation/differentiation.</td>
</tr>
<tr class="even">
<td align="center">Glutamatergic synapse</td>
<td align="center">Low</td>
<td align="center">Neurotransmission pathway; little direct reticulocyte
relevance.</td>
</tr>
<tr class="odd">
<td align="center">Adrenergic signaling in cardiomyocytes</td>
<td align="center">Low</td>
<td align="center">Heart muscle signalling; unlikely to directly
regulate retic count.</td>
</tr>
<tr class="even">
<td align="center">Serotonergic synapse</td>
<td align="center">Low</td>
<td align="center">Gut/brain neurotransmission; minimal direct evidence
for reticulocyte regulation.</td>
</tr>
<tr class="odd">
<td align="center">Dopaminergic synapse</td>
<td align="center">Low</td>
<td align="center">Same reasoning as above.</td>
</tr>
<tr class="even">
<td align="center">GnRH signaling pathway</td>
<td align="center">Low</td>
<td align="center">Reproductive hormone axis; little direct
erythropoiesis evidence.</td>
</tr>
<tr class="odd">
<td align="center">RNA degradation</td>
<td align="center">Moderate</td>
<td align="center">RNA clearance and turnover likely important in
reticulocyte maturation (they lose residual RNA) so plausible.</td>
</tr>
<tr class="even">
<td align="center">cAMP signaling pathway</td>
<td align="center">Moderate</td>
<td align="center">cAMP is a common second messenger in many cell types;
could regulate erythroid progenitors/reticulocyte release.</td>
</tr>
<tr class="odd">
<td align="center">Cholinergic synapse</td>
<td align="center">Low</td>
<td align="center">Neurotransmission; minimal direct link to
reticulocyte production.</td>
</tr>
<tr class="even">
<td align="center">Cellular senescence</td>
<td align="center">Moderate</td>
<td align="center">Senescence in marrow niche/HSC compartment may impair
erythropoiesis → influences reticulocyte count.</td>
</tr>
<tr class="odd">
<td align="center">Endocytosis</td>
<td align="center">Strong</td>
<td align="center">Iron uptake (transferrin receptor endocytosis) and
receptor internalisation critical during erythroid maturation and
reticulocyte release.</td>
</tr>
<tr class="even">
<td align="center">mTOR signaling pathway</td>
<td align="center">Strong</td>
<td align="center">mTOR integrates nutrient/growth signals for cell
proliferation including erythroid cells; very plausible effect on
reticulocyte count.</td>
</tr>
<tr class="odd">
<td align="center">PI3K-Akt signaling pathway</td>
<td align="center">Strong</td>
<td align="center">Downstream of growth factor signalling (e.g., EPO) in
erythroid progenitors; influences survival/proliferation leading to
changes in retics.</td>
</tr>
<tr class="even">
<td align="center">Ferroptosis</td>
<td align="center">Moderate</td>
<td align="center">Iron-dependent cell death; excessive ferroptosis
could impair erythroid maturation → lowering reticulocyte output.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="plots-for-all-pairs-at-ect-fdr-0.1" class="section level2">
<h2>Plots for all pairs at ECT FDR &lt; 0.1</h2>
<div id="plots" class="section level3">
<h3>Plots</h3>
<pre class="r"><code># candidate_pairs &lt;- data.frame(celltype = c(&quot;B_cell&quot;,&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;),
#                               trait = c(&quot;Psoriasis-ebi-a-GCST90038681&quot;,&quot;RA-ukb-a-105&quot;,&quot;SLE-ebi-a-GCST90018917&quot;,&quot;SLE-ebi-a-GCST90018917&quot;,
#                                         &quot;SLE-ebi-a-GCST90018917&quot;,&quot;IBD-ebi-a-GCST003043&quot;,&quot;IBD-ebi-a-GCST003043&quot;),
#                               factor = c(&quot;04062_PC2&quot;,&quot;04071_PC4&quot;,&quot;04120_PC1&quot;,&quot;04015_PC2&quot;,&quot;04022_PC4&quot;,&quot;00230_PC2&quot;,&quot;04924_PC1&quot;)
#                               )

fdr_cutoff &lt;- 0.2
df_all_fdr01$celltype &lt;- celltypes[df_all_fdr01$celltype]

for (i in 1:nrow(df_all_fdr01)){
  
  celltype &lt;- df_all_fdr01$celltype[i]
  trait &lt;- df_all_fdr01$trait[i]
  factor &lt;- df_all_fdr01$factor[i]
  
  file_harmo &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;,celltype, &quot;-&quot;, trait, &quot;_harmo.RDS&quot;)
  dt_harmo &lt;- readRDS(file_harmo)
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved.RDS&quot;))
  
  dat &lt;- dt_harmo[[factor]]
  snp_select &lt;- dat[dat$FDR.exposure &lt; fdr_cutoff,]
  snp_select$chr &lt;- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr &lt;- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]
  
  pathwayname &lt;- df$pathwayName[df$factor == factor &amp; df$trait ==trait]
  
  eff &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\nECT p-value =&quot;, ect_p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)
  
  
  #snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff &amp; dat$steiger_dir ==T,]
  snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff,]
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  coef &lt;- summary(fit)
  p &lt;- round(coef$coefficients[,4], digits = 4)
  #ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]
  
  eff_rev &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome)) +
    geom_point(size = 4, color =&quot;steelblue&quot;) + 
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\n p-value =&quot;, p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)
  

  
  file_gene_loading &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/gene_loading/&quot;,celltype, &quot;_&quot;, factor, &quot;-&quot;, trait, &quot;.RDS&quot;)
  gene_loadings_pc &lt;- readRDS(file_gene_loading)
  
  gene_loadings_pc_abs &lt;- abs(gene_loadings_pc)
  gene_loadings_pc_abs_sort &lt;- sort(gene_loadings_pc_abs,decreasing = T)
  topgenes &lt;- names(gene_loadings_pc_abs_sort)[1:10]
  
  gene_loadings_pc &lt;- as.data.frame(cbind(names(gene_loadings_pc),gene_loadings_pc))
  names(gene_loadings_pc) &lt;- c(&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)
  
  # Extract gene information based on the gene symbol
  gene_info &lt;- genes(EnsDb.Hsapiens.v75, filter = ~ gene_name ==  gene_loadings_pc$gene_name)
  gene_info &lt;- as.data.frame(gene_info)
  gene_info &lt;- gene_info[!duplicated(gene_info$gene_name),]
  
  gene_info &lt;- merge(gene_info,gene_loadings_pc, by=&quot;gene_name&quot;)
  gene_info &lt;- gene_info[gene_info$seqnames %in%c(1:22),]
  
  loading_plot &lt;- gene_info[,c(&quot;seqnames&quot;,&quot;start&quot;,&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)]
  colnames(loading_plot) &lt;- c(&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;gene_name&quot;,&quot;loadings&quot;)
  loading_plot$loadings &lt;- round(as.numeric(loading_plot$loadings),digits = 4)
  
  don &lt;- loading_plot %&gt;%
    
    # Compute chromosome size
    group_by(chromosome_name) %&gt;%
    summarise(chr_len=max(start_position)) %&gt;%
    
    # Calculate cumulative position of each chromosome
    mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %&gt;%
    dplyr::select(-chr_len) %&gt;%
    
    # Add this info to the initial dataset
    left_join(loading_plot, ., by=c(&quot;chromosome_name&quot;=&quot;chromosome_name&quot;)) %&gt;%
    
    # Add a cumulative position of each SNP
    arrange(chromosome_name, start_position) %&gt;%
    mutate( start_positioncum=start_position+tot) %&gt;%
    
    # Add highlight and annotation information
    #mutate( is_annotate=ifelse(gene_name %in% loading_plot_top10$gene_name, &quot;yes&quot;, &quot;no&quot;))
    mutate( is_annotate=ifelse(gene_name %in% topgenes, &quot;yes&quot;, &quot;no&quot;))
  # Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
  axisdf = don %&gt;% group_by(chromosome_name) %&gt;% summarize(center=( max(start_positioncum) + min(start_positioncum) ) / 2 )
  
  manhplot_gene &lt;- ggplot(don, aes(x=start_positioncum, y=loadings)) +
    
    # Show all points
    #rasterise(geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3),dpi=80 )+
    geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3) +
    scale_color_manual(values = rep(c(&quot;red&quot;, &quot;steelblue&quot;), 22)) +
    
    # custom X axis:
    scale_x_continuous( label = as.integer(axisdf$chromosome_name), breaks= axisdf$center ) +
    #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
    
    #ylab(expression(paste(&quot;-log&quot;[10],&quot;(p-value)&quot;))) +
    #ylim(0,max(-log10(don$loadings)) + 2) +
    #ylim(min(as.numeric(don$loadings)),max(as.numeric(don$loadings))) +
    #ylab(&quot;&quot;) +
    ylab(&quot;Gene Loadings&quot;) +
    xlab(&quot;Chromosome&quot;) +
    
    # Custom the theme:
    theme_bw() +
    #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
    theme(
      legend.position=&quot;none&quot;,
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank(),
      axis.title.x = element_text(size = 18),
      axis.text.x = element_text(size = 16, color = &quot;black&quot;),
      axis.title.y = element_text(size = 18),
      axis.text.y = element_text(size = 16, color = &quot;black&quot;),
      text= element_text(family=&quot;Times&quot;),
      #axis.text.y=element_blank(),
      #axis.ticks.y=element_blank()
    ) +
    geom_label_repel( data=subset(don, is_annotate==&quot;yes&quot;), aes(label=gene_name), size=4)
  
    
    cat(paste0(celltype,&quot;-&quot;, pathwayname, &quot;-&quot;, trait ))
  
  
    all &lt;- grid.arrange(eff, eff_rev, nrow = 1)
    
    print(manhplot_gene)
    print(paste0(&quot;Top 10 genes with largest abs loading: &quot;, paste0(topgenes, collapse = &quot;,&quot;)))
  
}</code></pre>
<pre><code>CD15_positive_leukocyte-Insulin signaling pathway-MCH-ebi-a-GCST90002390</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-2.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: IKBKB,TSC1,CBL,PHKA2,SOCS4,PHKB,MAPK3,RPS6,RAF1,INPPL1&quot;
platelet-NF-kappa B signaling pathway-HB-ebi-a-GCST90002384</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-3">
Past versions of unnamed-chunk-8-3.png
</button>
</p>
<div id="fig-unnamed-chunk-8-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-3.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-4">
Past versions of unnamed-chunk-8-4.png
</button>
</p>
<div id="fig-unnamed-chunk-8-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-4.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: LTB,LY96,TNFRSF1A,PLCG2,CD14,NFKBIA,NFKB1,PTGS2,CFLAR,PARP1&quot;
platelet-Hippo signaling pathway-RET-ebi-a-GCST90002405</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-5">
Past versions of unnamed-chunk-8-5.png
</button>
</p>
<div id="fig-unnamed-chunk-8-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-5.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-6">
Past versions of unnamed-chunk-8-6.png
</button>
</p>
<div id="fig-unnamed-chunk-8-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-6.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: ACTB,BIRC2,CTNNB1,FZD7,TEAD4,FZD1,PPP1CA,PPP2R2D,PPP2R2B,NF2&quot;
platelet-Thermogenesis-RET-ebi-a-GCST90002405</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-7">
Past versions of unnamed-chunk-8-7.png
</button>
</p>
<div id="fig-unnamed-chunk-8-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-7.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-8">
Past versions of unnamed-chunk-8-8.png
</button>
</p>
<div id="fig-unnamed-chunk-8-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-8.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: ACTG1,PRKAA1,GRB2,COX4I1,GNAS,KDM3B,PRKACB,SLC25A20,MGLL,NDUFB9&quot;
platelet-Purine metabolism-RET-ebi-a-GCST90002405</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-9">
Past versions of unnamed-chunk-8-9.png
</button>
</p>
<div id="fig-unnamed-chunk-8-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-9.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-10">
Past versions of unnamed-chunk-8-10.png
</button>
</p>
<div id="fig-unnamed-chunk-8-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-10.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: NT5C2,AK3,NT5C,PDE4D,PDE5A,ADCY3,DGUOK,NME1-NME2,RRM2,PAPSS1&quot;
B_cell-Chemokine signaling pathway-Psoriasis-ebi-a-GCST90038681</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-11">
Past versions of unnamed-chunk-8-11.png
</button>
</p>
<div id="fig-unnamed-chunk-8-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-11.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-12">
Past versions of unnamed-chunk-8-12.png
</button>
</p>
<div id="fig-unnamed-chunk-8-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-12.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3R2,RHOA,PTK2,PRKCD,DOCK2,NFKB1,ELMO1,IKBKG,VAV2,CDC42&quot;
thymocyte-Gastric acid secretion-IBD-ebi-a-GCST003043</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-13">
Past versions of unnamed-chunk-8-13.png
</button>
</p>
<div id="fig-unnamed-chunk-8-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-13.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-14">
Past versions of unnamed-chunk-8-14.png
</button>
</p>
<div id="fig-unnamed-chunk-8-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-14.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PLCB1,CALM2,ATP1B3,KCNJ10,GNAI3,ITPR1,PRKCA,CALM3,ATP1A1,ACTB&quot;
thymocyte-Leukocyte transendothelial migration-IBD-ebi-a-GCST003043</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-15">
Past versions of unnamed-chunk-8-15.png
</button>
</p>
<div id="fig-unnamed-chunk-8-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-15.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-16">
Past versions of unnamed-chunk-8-16.png
</button>
</p>
<div id="fig-unnamed-chunk-8-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-16.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: CTNNA1,ACTB,PIK3R2,ITGAL,ACTN4,ITGB2,RAC2,MSN,MAPK13,SIPA1&quot;</code></pre>
</div>
<div id="gene-function-on-the-cell-types-nots-from-chatgpt"
class="section level3">
<h3>Gene function on the cell types , nots from chatgpt</h3>
<ul>
<li>B cell (CD19+), Psoriasis-ebi-a-GCST90038681, 04062_PC2, Chemokine
signaling pathway</li>
</ul>
<table>
<colgroup>
<col width="4%" />
<col width="67%" />
<col width="27%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Gene</th>
<th align="center">Evidence for B-cell activation involvement</th>
<th align="center">Notes / uncertainties</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">IKBKB(IKKβ)</td>
<td align="center">There is strong evidence: A nonsense mutation in
IKBKB caused absence of IKKβ and a lack of T- and B-cell activation
through their antigen receptors. (PMC) Also, canonical NF-κB pathway
downstream of BCR or CD40 in B cells involves IKK complex. (JACI
Online)</td>
<td align="center">This is a good “hit” for B-cell activation.</td>
</tr>
<tr class="even">
<td align="center">RAF1 (C-RAF)</td>
<td align="center">Evidence that RAF family (including RAF-1) are
involved in B cell activation/differentiation: e.g., a thesis shows that
B-cell specific knockout of B-Raf and Raf-1 impacted B cell activation,
expansion/plasma cell differentiation. (LMU Munich Theses) However, the
study indicates that in certain mature B-cells, ERK phosphorylation
could happen independent of Raf-1. (Science)</td>
<td align="center">So RAF1 is involved, but perhaps not strictly
essential in all contexts. It likely participates in the
BCR→Ras→Raf→MEK→ERK axis in B cells.</td>
</tr>
<tr class="odd">
<td align="center">MAPK3(ERK1)</td>
<td align="center">More general MAPK/ERK signalling is known to be
important in B cell activation: e.g., BCR-induced cell spreading is
inhibited by ERK inhibitors. (MicroPublication) Because MAPK3 is a core
ERK1, it likely participates downstream of B-cell activation
signals.</td>
<td align="center">Good candidate to include.</td>
</tr>
<tr class="even">
<td align="center">INPPL1(SHIP2)</td>
<td align="center">Some relevant evidence: INPPL1 expression is
annotated for “B-cell function” in a cell‐type summary. (Human Protein
Atlas) Also SHIP family (primarily SHIP1) are established regulators of
BCR signalling (often negative regulators) via PI3K signalling in B
cells. Though direct literature for SHIP2 (INPPL1) in B cell activation
is sparse.</td>
<td align="center">Likely involved (perhaps via regulatory/inhibitory
signalling) but evidence is modest.</td>
</tr>
<tr class="odd">
<td align="center">RPS6(Ribosomal protein S6)</td>
<td align="center">There is evidence for involvement in translation
regulation in immune cells, but I did not identify a direct paper
linking RPS6 to B-cell activation signalling specifically.</td>
<td align="center">Probably a very indirect player (translation/protein
synthesis) rather than direct activation signalling; you may decide to
exclude or mark as secondary.</td>
</tr>
</tbody>
</table>
<ul>
<li>Platelet, HB-ebi-a-GCST90002384, 04064_PC1, NF-kappa B signaling
pathway</li>
</ul>
<table>
<colgroup>
<col width="5%" />
<col width="68%" />
<col width="26%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Gene</th>
<th align="center">Evidence for involvement in platelet activation</th>
<th align="center">Notes / uncertainties</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">PLCG2(Phospholipase C γ2)</td>
<td align="center">Strong evidence: The literature shows that platelet
activation (especially via collagen receptor GP VI or integrins)
involves activation of PLCγ2. (PubMed) For example: “Agonist‐induced
platelet activation … involves different signalling pathways leading to
the activation of phospholipase C … γ2.” (PubMed)</td>
<td align="center">This is one of the clearest “hits” on your list for
platelet activation.</td>
</tr>
<tr class="even">
<td align="center">NFKB1 (NF-κB p50 subunit)</td>
<td align="center">Yes: There is evidence that NF-κB family members in
platelets (despite them being anuclear) are engaged in platelet
activation/priming and aggregation. (PMC) For example: “NF-κB inhibitors
prevent platelet activation following stimulation with several agonists,
suggesting that NF‐κB in platelets functions rather non-genomically.”
(PMC)</td>
<td align="center">Good candidate. Although the mechanistic depth may
vary, the role is plausible.</td>
</tr>
<tr class="odd">
<td align="center">NFKBIA (IκBα)</td>
<td align="center">Yes: Since IκBα is the inhibitor of NF-κB, and NF-κB
signalling is present and functional in platelets (see above), there is
support for involvement of the pathway including IκBα in platelet
activation. (PubMed)</td>
<td align="center">While the direct role of IκBα specifically in
platelet activation may be less studied, given the pathway it is likely
involved.</td>
</tr>
<tr class="even">
<td align="center">TNFRSF1A(TNF-receptor 1)</td>
<td align="center">Moderate evidence: There is study evidence that
platelets express TNF receptors (including TNFR1) and that TNF-α can
trigger platelet activation/aggregation. (PubMed) For example: “TNF-α
behaves as a trigger of platelet activation through stimulation of the
arachidonic acid pathway.” (PubMed)</td>
<td align="center">Good candidate for inclusion as a regulator/modulator
of platelet activation (rather than core immediate signalling).</td>
</tr>
</tbody>
</table>
<ul>
<li>T cell (CD8+), IBD-ebi-a-GCST003043, 04670_PC1, Leukocyte
transendothelial migration</li>
</ul>
<table>
<colgroup>
<col width="5%" />
<col width="68%" />
<col width="26%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Gene</th>
<th align="center">Evidence for involvement in platelet activation</th>
<th align="center">Notes / uncertainties</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">PLCG2(Phospholipase C γ2)</td>
<td align="center">Strong evidence: The literature shows that platelet
activation (especially via collagen receptor GP VI or integrins)
involves activation of PLCγ2. (PubMed) For example: “Agonist‐induced
platelet activation … involves different signalling pathways leading to
the activation of phospholipase C … γ2.” (PubMed)</td>
<td align="center">This is one of the clearest “hits” on your list for
platelet activation.</td>
</tr>
<tr class="even">
<td align="center">NFKB1 (NF-κB p50 subunit)</td>
<td align="center">Yes: There is evidence that NF-κB family members in
platelets (despite them being anuclear) are engaged in platelet
activation/priming and aggregation. (PMC) For example: “NF-κB inhibitors
prevent platelet activation following stimulation with several agonists,
suggesting that NF‐κB in platelets functions rather non-genomically.”
(PMC)</td>
<td align="center">Good candidate. Although the mechanistic depth may
vary, the role is plausible.</td>
</tr>
<tr class="odd">
<td align="center">NFKBIA (IκBα)</td>
<td align="center">Yes: Since IκBα is the inhibitor of NF-κB, and NF-κB
signalling is present and functional in platelets (see above), there is
support for involvement of the pathway including IκBα in platelet
activation. (PubMed)</td>
<td align="center">While the direct role of IκBα specifically in
platelet activation may be less studied, given the pathway it is likely
involved.</td>
</tr>
<tr class="even">
<td align="center">TNFRSF1A(TNF-receptor 1)</td>
<td align="center">Moderate evidence: There is study evidence that
platelets express TNF receptors (including TNFR1) and that TNF-α can
trigger platelet activation/aggregation. (PubMed) For example: “TNF-α
behaves as a trigger of platelet activation through stimulation of the
arachidonic acid pathway.” (PubMed)</td>
<td align="center">Good candidate for inclusion as a regulator/modulator
of platelet activation (rather than core immediate signalling).</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="eas" class="section level1">
<h1>EAS</h1>
<div id="detailed-results-for-all-pairs-at-ect-fdr-0.2-1"
class="section level2">
<h2>Detailed results for all pairs at ECT FDR &lt; 0.2</h2>
<pre class="r"><code>df_all_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  df &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_all_fdr02 &lt;- rbind(df_all_fdr02,df_fdr02)
  
}


df_all_fdr02$celltype &lt;- types[df_all_fdr02$celltype]
df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP),]

df_all_fdr01 &lt;- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP &lt; 0.1,]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Pairs with ECT FDR &lt;= 0.2 &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-66aaeb19fe992dfb51c2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-66aaeb19fe992dfb51c2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Pairs with ECT FDR &lt;= 0.2 <\/caption>","data":[["31","42","5","61","7","10","11","12","14","15","2","17","18","19","20","3","4","23","41","25","6","27","28","30","32","37","411","46","47","48","49","53","9"],["Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Monocyte (CD14+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","B cell (CD19+)","B cell (CD19+)","Leukocyte (CD15+)","Monocyte (CD14+)","Leukocyte (CD15+)","B cell (CD19+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","Leukocyte (CD15+)","B cell (CD19+)"],["height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","ATD-bbj-a-90","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","BMI-ebi-a-GCST90018727","MCV-ebi-a-GCST90018746","height-ebi-a-GCST90018739","BMI-ebi-a-GCST90018727","ATD-bbj-a-90","height-ebi-a-GCST90018739","MCV-ebi-a-GCST90018746","MCV-ebi-a-GCST90018746","height-ebi-a-GCST90018739","MCV-ebi-a-GCST90018746","height-ebi-a-GCST90018739","MCV-ebi-a-GCST90018746","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","ATD-bbj-a-90","RA-bbj-a-151","height-ebi-a-GCST90018739","BMI-ebi-a-GCST90018727","ATD-bbj-a-90","WBC-ebi-a-GCST90018758"],["05167_PC3","05161_PC3","04935_PC3","05200_PC3","05165_PC3","03015_PC3","04972_PC4","04726_PC4","04550_PC2","04360_PC3","04931_PC1","05163_PC3","05016_PC5","04070_PC3","04360_PC3","04611_PC3","04810_PC4","00513_PC3","05200_PC1","05131_PC3","04921_PC4","04630_PC2","04150_PC3","04261_PC3","04730_PC4","04110_PC3","05226_PC3","04910_PC3","04750_PC3","04151_PC4","04151_PC4","04150_PC3","00270_PC2"],["05167","05161","04935","05200","05165","03015","04972","04726","04550","04360","04931","05163","05016","04070","04360","04611","04810","00513","05200","05131","04921","04630","04150","04261","04730","04110","05226","04910","04750","04151","04151","04150","00270"],["Kaposi sarcoma-associated herpesvirus infection","Hepatitis B","Growth hormone synthesis, secretion and action","Pathways in cancer","Human papillomavirus infection","mRNA surveillance pathway","Pancreatic secretion","Serotonergic synapse","Signaling pathways regulating pluripotency of stem cells","Axon guidance","Insulin resistance","Human cytomegalovirus infection","Huntington disease","Phosphatidylinositol signaling system","Axon guidance","Platelet activation","Regulation of actin cytoskeleton","Various types of N-glycan biosynthesis","Pathways in cancer","Shigellosis","Oxytocin signaling pathway","JAK-STAT signaling pathway","mTOR signaling pathway","Adrenergic signaling in cardiomyocytes","Long-term depression","Cell cycle","Gastric cancer","Insulin signaling pathway","Inflammatory mediator regulation of TRP channels","PI3K-Akt signaling pathway","PI3K-Akt signaling pathway","mTOR signaling pathway","Cysteine and methionine metabolism"],[14,6,8,7,12,5,6,6,11,5,7,7,5,5,6,8,5,11,5,11,6,7,6,5,8,9,9,5,5,12,10,5,9],[1e-05,1.3e-05,1.4e-05,1.5e-05,0.000215,0.000337,0.000369,0.00039,0.000672,0.000689,3.1e-05,0.0010999,0.0011999,0.0012999,0.0012999,5.6e-05,0.000107,0.0020998,0.000106,0.0025997,0.000285,0.0037996,0.0038996,0.0043996,0.0044996,0.0049995,0.0063994,0.0077992,0.0077992,0.0078992,0.0081992,0.0089991,0.000762],[0.00222,0.00222,0.00222,0.00222,0.025456,0.02886,0.02886,0.02886,0.0407888,0.0407888,0.049445,0.0549672,0.0549672,0.0549672,0.0549672,0.056068,0.056068,0.0828721,0.084535,0.0961889,0.09956,0.1282535,0.1282535,0.1331882,0.1331882,0.1409383,0.172202,0.1866895,0.1866895,0.1866895,0.1866895,0.1973136,0.199644],[186,162,119,531,330,91,102,115,140,181,108,225,264,99,181,124,213,39,531,236,153,162,153,149,60,124,149,137,100,354,354,153,49],[119,105,65,255,174,64,35,39,51,74,69,136,146,63,74,71,120,24,290,150,79,73,81,70,28,69,60,87,47,156,156,81,32],["disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","bloodcell_EAS_new","disease_EAS","disease_EAS","disease_EAS","disease_EAS","bloodcell_EAS_new","bloodcell_EAS_new","disease_EAS","bloodcell_EAS_new","disease_EAS","bloodcell_EAS_new","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","bloodcell_EAS_new"],["-0.335320189440536","0.620471254657609","0.774097800142423","-0.924999067074685","-0.628082585903604","0/1snp","-0.0331521299069109","0.0320183424815683","0.012336281274246","-0.523110577590398","-0.0780203996195739","0.382195178073134","0.568640271452703","0/1snp","-0.312450227158299","-0.920969134117914","0.320301620905627","0.302813271864685","0.0289598358094416","-0.140027545018815","-0.77540583139736","-0.0676601639868393","0.86014429611751","-0.579958541338593","0.421358059436136","0.828820548905881","-0.763514895581253","0/1snp","0.0504123419360684","-0.395194415278412","0.380136725276964","0/1snp","0.431538751405459"],["0.696464692880394","0.430496336025231","0.359173780334774","0.296237509071482","0.474956008235431","0/1snp","0.965903671216125","0.968750524709777","0.987883789277398","0.548349402716188","0.784298192893093","0.661729006909189","0.675589994882587","0/1snp","0.700367394570114","0.107689727067654","0.479354160984548","0.672074254887795","0.899252620378613","0.88745417970635","0.0783774426584705","0.933647714663954","0.312849596924828","0.585654893451985","0.523797160148899","0.298773729511107","0.367999370227373","0/1snp","0.595598480652005","0.60041055366391","0.807294644487019","0/1snp","0.642968949046111"],[null,null,null,null,null,0.054235,null,null,null,null,0.122148,null,null,0.0277911,null,0.0279252,0.0506612,null,0.0808219,null,0.0526402,null,null,null,null,null,null,null,null,null,null,null,0.0697099],[null,null,null,null,null,0.1001975,null,null,null,null,0.1176445,null,null,0.0685909,null,0.0270048,0.0494326,null,0.0788072,null,0.0519174,null,null,null,null,null,null,null,null,null,null,null,0.0664139],[null,null,null,null,null,0.8316832,null,null,null,null,0.5445545,null,null,0.9306931,null,0.8415842,0.9108911,null,0.4752475,null,0.7128713,null,null,null,null,null,null,null,null,null,null,null,0.7425743],[null,null,null,null,null,0.7425743,null,null,null,null,0.5544554,null,null,0.8712871,null,0.8514851,0.9108911,null,0.4950495,null,0.7128713,null,null,null,null,null,null,null,null,null,null,null,0.7425743],[null,null,null,null,null,0.8872095,null,null,null,null,0.5307593,null,null,0.9321453,null,0.872187,0.7852962,null,0.6484683,null,0.778273,null,null,null,null,null,null,null,null,null,null,null,0.7709708]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,14,15,16,17,18]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-trait-1"
class="section level2">
<h2>Number of significant pairs at different cutoffs, for each
trait</h2>
<div id="ect-fdr-0.1-1" class="section level3">
<h3>ECT FDR &lt; 0.1</h3>
<pre class="r"><code>FDRcutoff &lt;- 0.1
MBEcutoff &lt;- 0.05

p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_summary &lt;- df %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_FDR_3suppSNP = sum(ECT_FDR_5suppSNP &lt; FDRcutoff, na.rm = TRUE)
    ) %&gt;%
    arrange(desc(n_sig_FDR_3suppSNP))
  
  df_summary$trait[df_summary$trait ==&quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  df_summary &lt;- df_summary %&gt;%
    mutate(trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait))
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_FDR_3suppSNP),
                       y = n_sig_FDR_3suppSNP,
                       fill = n_sig_FDR_3suppSNP)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7, color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;lightblue&quot;, high = &quot;steelblue&quot;) +
  labs(
    x = &quot;Trait&quot;,
    y = paste0(&quot;# of pairs at FDR &lt; &quot;, FDRcutoff),
    title = types[celltype]
  ) +
  theme_bw(base_size = 14) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
  )
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-10-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p &lt;- list()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype, &quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric if it’s stored as character
  df &lt;- df %&gt;%
    mutate(mbe_pval_reverse_pair = suppressWarnings(as.numeric(mbe_pval_reverse_pair)))
  
  # Keep only FDR-significant pairs
  df_sig &lt;- df %&gt;%
    dplyr::filter(ECT_FDR_5suppSNP &lt; FDRcutoff)
  
  # Fix naming if needed
  df_sig$trait[df_sig$trait == &quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  # Summarize counts per trait
  df_summary &lt;- df_sig %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_pairs = n(),
      n_mbe_sig = sum(!is.na(mbe_pval_reverse_pair) &amp; mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    # Keep traits with at least one FDR-significant pair
    dplyr::filter(n_sig_pairs &gt;= 1) %&gt;%
    mutate(
      trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait)
    ) %&gt;%
    arrange(desc(n_sig_pairs))
  
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_pairs))) +
    geom_bar(aes(y = n_sig_pairs), stat = &quot;identity&quot;, fill = &quot;grey80&quot;, color = &quot;black&quot;, width = 0.7) +
    geom_bar(aes(y = n_mbe_sig), stat = &quot;identity&quot;, fill = &quot;darkturquoise&quot;, width = 0.7) +
    geom_text(aes(y = n_mbe_sig, label = n_mbe_sig), vjust = -0.4, size = 3.5) +
    labs(
      x = &quot;Trait&quot;,
      y = paste0(&quot;# of FDR&lt;&quot;, FDRcutoff, &quot; (blue: MBE p&lt;&quot;, MBEcutoff, &quot;)&quot;),
      title = types[celltype]
    ) +
    theme_bw(base_size = 14) +
    theme(
      legend.position = &quot;none&quot;,
      axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
      axis.text.y = element_text(size = 10, color = &quot;black&quot;),
      plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
    )
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2, top = &quot;Number of pairs with MBE p &lt; 0.05, for the sig pairs&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-10-2.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="ect-fdr-0.2-1" class="section level3">
<h3>ECT FDR &lt; 0.2</h3>
<pre class="r"><code>FDRcutoff &lt;- 0.2
MBEcutoff &lt;- 0.05

p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_summary &lt;- df %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_FDR_3suppSNP = sum(ECT_FDR_5suppSNP &lt; FDRcutoff, na.rm = TRUE)
    ) %&gt;%
    arrange(desc(n_sig_FDR_3suppSNP))
  
  df_summary$trait[df_summary$trait ==&quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  df_summary &lt;- df_summary %&gt;%
    mutate(trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait))
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_FDR_3suppSNP),
                       y = n_sig_FDR_3suppSNP,
                       fill = n_sig_FDR_3suppSNP)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7, color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;lightblue&quot;, high = &quot;steelblue&quot;) +
  labs(
    x = &quot;Trait&quot;,
    y = paste0(&quot;# of pairs at FDR &lt; &quot;, FDRcutoff),
    title = types[celltype]
  ) +
  theme_bw(base_size = 14) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
  )
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p &lt;- list()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype, &quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric if it’s stored as character
  df &lt;- df %&gt;%
    mutate(mbe_pval_reverse_pair = suppressWarnings(as.numeric(mbe_pval_reverse_pair)))
  
  # Keep only FDR-significant pairs
  df_sig &lt;- df %&gt;%
    dplyr::filter(ECT_FDR_5suppSNP &lt; FDRcutoff)
  
    # Fix naming if needed
  df_sig$trait[df_sig$trait == &quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  # Summarize counts per trait
  df_summary &lt;- df_sig %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_pairs = n(),
      n_mbe_sig = sum(!is.na(mbe_pval_reverse_pair) &amp; mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    # Keep traits with at least one FDR-significant pair
    dplyr::filter(n_sig_pairs &gt;= 1) %&gt;%
    mutate(
      trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait)
    ) %&gt;%
    arrange(desc(n_sig_pairs))
  

  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_pairs))) +
    geom_bar(aes(y = n_sig_pairs), stat = &quot;identity&quot;, fill = &quot;grey80&quot;, color = &quot;black&quot;, width = 0.7) +
    geom_bar(aes(y = n_mbe_sig), stat = &quot;identity&quot;, fill = &quot;darkturquoise&quot;, width = 0.7) +
    geom_text(aes(y = n_mbe_sig, label = n_mbe_sig), vjust = -0.4, size = 3.5) +
    labs(
      x = &quot;Trait&quot;,
      y = paste0(&quot;# of FDR&lt;&quot;, FDRcutoff, &quot; (blue: MBE p&lt;&quot;, MBEcutoff, &quot;)&quot;),
      title = types[celltype]
    ) +
    theme_bw(base_size = 14) +
    theme(
      legend.position = &quot;none&quot;,
      axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
      axis.text.y = element_text(size = 10, color = &quot;black&quot;),
      plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
    )
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2, top = &quot;Number of pairs with MBE p &lt; 0.05, for the sig pairs&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-2.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-cell-type-1"
class="section level2">
<h2>Number of significant pairs at different cutoffs, for each cell
type</h2>
<pre class="r"><code>num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2 &amp; df$ECT_FDR_5suppSNP &gt;= 0.1,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))
  
}

sum &lt;- data.frame(celltype = types,
                  num_fdr01 = num_fdr01,
                  num_fdr02 = num_fdr02)


df &lt;- sum %&gt;%
  pivot_longer(cols = starts_with(&quot;num_fdr&quot;), names_to = &quot;event&quot;, values_to = &quot;total&quot;) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# Plot
p1 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(x = &quot;Cell types&quot;, y = &quot;Count&quot;) +
  ggtitle(&quot;# of pairs&quot;) +
  ylim(0, 60) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, 
                       celltype, &quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric safely (ignore &quot;0/1snp&quot; or NA)
  df$mbe_pval_reverse_pair &lt;- suppressWarnings(as.numeric(df$mbe_pval_reverse_pair))
  df$mbe_pval_reverse_pair[is.na(df$mbe_pval_reverse_pair)] &lt;- 0.5
  
  # --- FDR &lt; 0.1 ---
  df_fdr01 &lt;- df %&gt;%
    dplyr::filter(
      ECT_FDR_5suppSNP &lt; 0.1,
      !(mbe_pval_reverse_pair &lt; MBEcutoff)  # remove reverse-causal pairs
    ) %&gt;%
    dplyr::filter(!is.na(p_ECT))
  
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))
  
  # --- 0.1 &lt; FDR &lt; 0.2 ---
  df_fdr02 &lt;- df %&gt;%
    dplyr::filter(
      ECT_FDR_5suppSNP &lt; 0.2,
      ECT_FDR_5suppSNP &gt;= 0.1,
      !(mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    dplyr::filter(!is.na(p_ECT))
  
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))
}

# summary table
sum &lt;- data.frame(
  celltype = types,
  num_fdr01 = num_fdr01,
  num_fdr02 = num_fdr02
)



df &lt;- sum %&gt;%
  pivot_longer(
    cols = starts_with(&quot;num_fdr&quot;),
    names_to = &quot;event&quot;,
    values_to = &quot;total&quot;
  ) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;,
                                        &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# ---- Plot ----
p2 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(
    x = &quot;Cell types&quot;,
    y = &quot;Count (excluding MBE p &lt; 0.05)&quot;,
    title = &quot;# of pairs after removing reverse-causal signals&quot;
  ) +
  ylim(0, 60) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

all &lt;- grid.arrange(p1, p2, nrow = 1)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-12-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="traits-and-pathways-identified-1" class="section level2">
<h2>Traits and pathways identified</h2>
<div id="height-ebi-a-gcst90018739" class="section level3">
<h3>height-ebi-a-GCST90018739</h3>
<pre class="r"><code>trait &lt;- &quot;height-ebi-a-GCST90018739&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  dplyr::mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>supp_snps_all &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/supp_snps_ECTFDR02.RDS&quot;)
supp_snps_trait &lt;- supp_snps_all[supp_snps_all$trait == trait,]

supp_snps_trait &lt;- supp_snps_trait[supp_snps_trait$factor %in% df_trait_fdr01$factor,]

DT::datatable(supp_snps_trait,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&quot;supporting snps for &quot;, trait)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-1814506231684700bf53" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1814506231684700bf53">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">supporting snps for height-ebi-a-GCST90018739<\/caption>","data":[[],[],[],[],[],[],[],[],[],[],[],[],[],[]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pwyID<\/th>\n      <th>pwyNAME<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>Num_supp_snp<\/th>\n      <th>SNP<\/th>\n      <th>chr<\/th>\n      <th>bp<\/th>\n      <th>pval_SNP_factor<\/th>\n      <th>fdr_SNP_factor<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="mcv-ebi-a-gcst90018746" class="section level3">
<h3>MCV-ebi-a-GCST90018746</h3>
<pre class="r"><code>trait &lt;- &quot;MCV-ebi-a-GCST90018746&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-2.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plots-for-all-pairs-at-ect-fdr-0.1-1" class="section level2">
<h2>Plots for all pairs at ECT FDR &lt; 0.1</h2>
<div id="plots-1" class="section level3">
<h3>Plots</h3>
<pre class="r"><code># candidate_pairs &lt;- data.frame(celltype = c(&quot;B_cell&quot;,&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;),
#                               trait = c(&quot;Psoriasis-ebi-a-GCST90038681&quot;,&quot;RA-ukb-a-105&quot;,&quot;SLE-ebi-a-GCST90018917&quot;,&quot;SLE-ebi-a-GCST90018917&quot;,
#                                         &quot;SLE-ebi-a-GCST90018917&quot;,&quot;IBD-ebi-a-GCST003043&quot;,&quot;IBD-ebi-a-GCST003043&quot;),
#                               factor = c(&quot;04062_PC2&quot;,&quot;04071_PC4&quot;,&quot;04120_PC1&quot;,&quot;04015_PC2&quot;,&quot;04022_PC4&quot;,&quot;00230_PC2&quot;,&quot;04924_PC1&quot;)
#                               )

fdr_cutoff &lt;- 0.2
df_all_fdr01$celltype &lt;- celltypes[df_all_fdr01$celltype]

for (i in 1:nrow(df_all_fdr01)){
  
  celltype &lt;- df_all_fdr01$celltype[i]
  trait &lt;- df_all_fdr01$trait[i]
  factor &lt;- df_all_fdr01$factor[i]
  
  file_harmo &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;,celltype, &quot;-&quot;, trait, &quot;_harmo.RDS&quot;)
  dt_harmo &lt;- readRDS(file_harmo)
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  dat &lt;- dt_harmo[[factor]]
  snp_select &lt;- dat[dat$FDR.exposure &lt; fdr_cutoff,]
  snp_select$chr &lt;- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr &lt;- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]
  
  pathwayname &lt;- df$pathwayName[df$factor == factor &amp; df$trait ==trait]
  
  eff &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\nECT p-value =&quot;, ect_p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)
  
  
  #snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff &amp; dat$steiger_dir ==T,]
  snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff,]
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  coef &lt;- summary(fit)
  p &lt;- round(coef$coefficients[,4], digits = 4)
  #ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]
  
  eff_rev &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome)) +
    geom_point(size = 4, color =&quot;steelblue&quot;) + 
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\n p-value =&quot;, p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)
  

  
  file_gene_loading &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/gene_loading/&quot;,celltype, &quot;_&quot;, factor, &quot;-&quot;, trait, &quot;.RDS&quot;)
  gene_loadings_pc &lt;- readRDS(file_gene_loading)
  
  gene_loadings_pc_abs &lt;- abs(gene_loadings_pc)
  gene_loadings_pc_abs_sort &lt;- sort(gene_loadings_pc_abs,decreasing = T)
  topgenes &lt;- names(gene_loadings_pc_abs_sort)[1:10]
  
  gene_loadings_pc &lt;- as.data.frame(cbind(names(gene_loadings_pc),gene_loadings_pc))
  names(gene_loadings_pc) &lt;- c(&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)
  
  # Extract gene information based on the gene symbol
  gene_info &lt;- genes(EnsDb.Hsapiens.v75, filter = ~ gene_name ==  gene_loadings_pc$gene_name)
  gene_info &lt;- as.data.frame(gene_info)
  gene_info &lt;- gene_info[!duplicated(gene_info$gene_name),]
  
  gene_info &lt;- merge(gene_info,gene_loadings_pc, by=&quot;gene_name&quot;)
  gene_info &lt;- gene_info[gene_info$seqnames %in%c(1:22),]
  
  loading_plot &lt;- gene_info[,c(&quot;seqnames&quot;,&quot;start&quot;,&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)]
  colnames(loading_plot) &lt;- c(&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;gene_name&quot;,&quot;loadings&quot;)
  loading_plot$loadings &lt;- round(as.numeric(loading_plot$loadings),digits = 4)
  
  don &lt;- loading_plot %&gt;%
    
    # Compute chromosome size
    group_by(chromosome_name) %&gt;%
    summarise(chr_len=max(start_position)) %&gt;%
    
    # Calculate cumulative position of each chromosome
    mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %&gt;%
    dplyr::select(-chr_len) %&gt;%
    
    # Add this info to the initial dataset
    left_join(loading_plot, ., by=c(&quot;chromosome_name&quot;=&quot;chromosome_name&quot;)) %&gt;%
    
    # Add a cumulative position of each SNP
    arrange(chromosome_name, start_position) %&gt;%
    mutate( start_positioncum=start_position+tot) %&gt;%
    
    # Add highlight and annotation information
    #mutate( is_annotate=ifelse(gene_name %in% loading_plot_top10$gene_name, &quot;yes&quot;, &quot;no&quot;))
    mutate( is_annotate=ifelse(gene_name %in% topgenes, &quot;yes&quot;, &quot;no&quot;))
  # Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
  axisdf = don %&gt;% group_by(chromosome_name) %&gt;% summarize(center=( max(start_positioncum) + min(start_positioncum) ) / 2 )
  
  manhplot_gene &lt;- ggplot(don, aes(x=start_positioncum, y=loadings)) +
    
    # Show all points
    #rasterise(geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3),dpi=80 )+
    geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3) +
    scale_color_manual(values = rep(c(&quot;red&quot;, &quot;steelblue&quot;), 22)) +
    
    # custom X axis:
    scale_x_continuous( label = as.integer(axisdf$chromosome_name), breaks= axisdf$center ) +
    #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
    
    #ylab(expression(paste(&quot;-log&quot;[10],&quot;(p-value)&quot;))) +
    #ylim(0,max(-log10(don$loadings)) + 2) +
    #ylim(min(as.numeric(don$loadings)),max(as.numeric(don$loadings))) +
    #ylab(&quot;&quot;) +
    ylab(&quot;Gene Loadings&quot;) +
    xlab(&quot;Chromosome&quot;) +
    
    # Custom the theme:
    theme_bw() +
    #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
    theme(
      legend.position=&quot;none&quot;,
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank(),
      axis.title.x = element_text(size = 18),
      axis.text.x = element_text(size = 16, color = &quot;black&quot;),
      axis.title.y = element_text(size = 18),
      axis.text.y = element_text(size = 16, color = &quot;black&quot;),
      text= element_text(family=&quot;Times&quot;),
      #axis.text.y=element_blank(),
      #axis.ticks.y=element_blank()
    ) +
    geom_label_repel( data=subset(don, is_annotate==&quot;yes&quot;), aes(label=gene_name), size=4)
  
    
    cat(paste0(celltype,&quot;-&quot;, pathwayname, &quot;-&quot;, trait ))
  
  
    all &lt;- grid.arrange(eff, eff_rev, nrow = 1)
    
    print(manhplot_gene)
    print(paste0(&quot;Top 10 genes with largest abs loading: &quot;, paste0(topgenes, collapse = &quot;,&quot;)))
  
}</code></pre>
<pre><code>CD15_positive_leukocyte-Kaposi sarcoma-associated herpesvirus infection-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,GNG2,JAK1,RB1,JAK2,MAP2K4,RCAN1,MAP2K1,GNG10,IFNAR2&quot;
CD15_positive_leukocyte-Hepatitis B-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,TRAF6,JAK2,MAP2K4,JAK1,RB1,MAP2K1,ARAF,TYK2,NRAS&quot;
CD15_positive_leukocyte-Growth hormone synthesis, secretion and action-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,JAK2,GSK3B,AKT1,MAP2K4,GNAQ,RAF1,EP300,MAPK3,ITPR2&quot;
CD15_positive_leukocyte-Pathways in cancer-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,GNG2,TRAF6,JAK1,JAK2,RB1,GNAQ,MAP2K1,HDAC2,IFNAR2&quot;
CD15_positive_leukocyte-Human papillomavirus infection-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,PPP2R2D,JAK1,RB1,PPP2CA,MAP2K1,HDAC2,CDC42,NRAS,TBPL1&quot;
CD15_positive_leukocyte-mRNA surveillance pathway-ATD-bbj-a-90</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PABPC1L,PNN,SMG1,PELO,WDR33,NXF1,UPF2,PABPN1,CPSF4,CPSF2&quot;
CD15_positive_leukocyte-Pancreatic secretion-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PLCB2,RAB11A,ITPR1,RAP1A,GNAQ,ITPR2,TPCN2,AMY1C,PLCB1,KCNQ1&quot;
CD15_positive_leukocyte-Serotonergic synapse-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PLA2G4B,MAPK3,GNAI2,PTGS1,ITPR2,ITPR1,GNG5,GNB2,GNB1,MAPK1&quot;
CD15_positive_leukocyte-Signaling pathways regulating pluripotency of stem cells-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,ACVR1,JAK2,JAK1,NRAS,MAP2K1,GSK3B,CTNNB1,AKT1,RIF1&quot;
CD15_positive_leukocyte-Axon guidance-BMI-ebi-a-GCST90018727</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,NRAS,RAF1,EPHB1,GSK3B,MAPK3,CDC42,PIK3R1,LIMK2,MAPK1&quot;
CD14_positive_monocyte-Insulin resistance-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-21.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-22.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PPP1CA,NR1H2,PPP1CC,RPS6KB1,G6PC3,PCK2,RELA,STAT3,PIK3CA,PTEN&quot;
CD15_positive_leukocyte-Human cytomegalovirus infection-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-23.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-24.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,GNG2,JAK1,GNAQ,RB1,RPS6KB2,GNA13,GNG10,MAP2K1,GNG5&quot;
CD15_positive_leukocyte-Huntington disease-BMI-ebi-a-GCST90018727</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-25.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-26.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: DNAH1,NDUFB3,ULK1,TBP,NDUFS7,NDUFS3,TUBA1C,CASP9,CYC1,NRBF2&quot;
CD15_positive_leukocyte-Phosphatidylinositol signaling system-ATD-bbj-a-90</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-27.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-28.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,IPMK,MTM1,ITPR2,PI4KB,IP6K2,PIK3R1,CALM2,MTMR4,PRKCB&quot;
CD15_positive_leukocyte-Axon guidance-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-29.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-30.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,NRAS,RAF1,EPHB1,GSK3B,MAPK3,CDC42,PIK3R1,LIMK2,MAPK1&quot;
B_cell-Platelet activation-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-31.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-32.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: VWF,PTGS1,GP9,TLN1,MYLK,GP1BA,ITGB3,GP6,P2RY12,ADCY3&quot;
B_cell-Regulation of actin cytoskeleton-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-33.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-34.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: ITGB5,ACTN1,MYLK,MAPK3,ITGB3,NRAS,NCKAP1L,PIP4K2A,ITGA2B,MYL9&quot;
CD15_positive_leukocyte-Various types of N-glycan biosynthesis-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-35.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-36.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: STT3B,MAN2A2,B4GALT2,STT3A,DDOST,MGAT2,MGAT1,MGAT4B,ALG12,MGAT4A&quot;
CD14_positive_monocyte-Pathways in cancer-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-37.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-38.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: FH,HDAC2,CDKN1B,CTBP1,CSF3R,RASSF5,ARHGEF1,NFKB2,BAX,RELA&quot;
CD15_positive_leukocyte-Shigellosis-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-39.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-40.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,UBE2D2,BCL10,UBE2N,TRAF6,FBXW11,UBE2V2,RBCK1,CDC42,MAPK9&quot;
B_cell-Oxytocin signaling pathway-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-41.png" width="672" style="display: block; margin: auto;" /><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-42.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: MAPK3,CALM3,CDKN1A,MAP2K2,MYL6B,CAMK2B,CALML4,PIK3CG,RHOA,PLA2G4B&quot;</code></pre>
<!-- ### Gene function on the cell types , nots from chatgpt  -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggrastr_1.0.1             ggrepel_0.9.1            
 [3] EnsDb.Hsapiens.v75_2.99.0 ensembldb_2.20.2         
 [5] AnnotationFilter_1.20.0   GenomicFeatures_1.48.3   
 [7] AnnotationDbi_1.58.0      Biobase_2.56.0           
 [9] GenomicRanges_1.48.0      GenomeInfoDb_1.39.9      
[11] IRanges_2.30.0            S4Vectors_0.34.0         
[13] BiocGenerics_0.42.0       tidyr_1.3.0              
[15] forcats_0.5.1             gridExtra_2.3            
[17] ggplot2_4.0.0             dplyr_1.1.4              

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              fs_1.5.2                   
  [7] dichromat_2.0-0.1           rstudioapi_0.13            
  [9] farver_2.1.0                DT_0.22                    
 [11] bit64_4.0.5                 fansi_1.0.3                
 [13] xml2_1.3.3                  codetools_0.2-18           
 [15] cachem_1.0.6                knitr_1.39                 
 [17] jsonlite_1.8.0              workflowr_1.7.0            
 [19] Rsamtools_2.12.0            dbplyr_2.1.1               
 [21] png_0.1-7                   compiler_4.2.0             
 [23] httr_1.4.3                  assertthat_0.2.1           
 [25] Matrix_1.5-3                fastmap_1.1.0              
 [27] lazyeval_0.2.2              cli_3.6.1                  
 [29] later_1.3.0                 htmltools_0.5.2            
 [31] prettyunits_1.1.1           tools_4.2.0                
 [33] gtable_0.3.6                glue_1.6.2                 
 [35] GenomeInfoDbData_1.2.8      rappdirs_0.3.3             
 [37] Rcpp_1.0.12                 jquerylib_0.1.4            
 [39] vctrs_0.6.5                 Biostrings_2.64.0          
 [41] rtracklayer_1.56.0          crosstalk_1.2.0            
 [43] xfun_0.41                   stringr_1.5.1              
 [45] lifecycle_1.0.4             restfulr_0.0.14            
 [47] XML_3.99-0.14               zlibbioc_1.42.0            
 [49] scales_1.4.0                hms_1.1.1                  
 [51] promises_1.2.0.1            MatrixGenerics_1.8.0       
 [53] ProtGenerics_1.28.0         parallel_4.2.0             
 [55] SummarizedExperiment_1.26.1 RColorBrewer_1.1-3         
 [57] yaml_2.3.5                  curl_4.3.2                 
 [59] memoise_2.0.1               sass_0.4.1                 
 [61] biomaRt_2.54.1              stringi_1.7.6              
 [63] RSQLite_2.3.1               highr_0.9                  
 [65] BiocIO_1.6.0                filelock_1.0.2             
 [67] BiocParallel_1.30.3         rlang_1.1.2                
 [69] pkgconfig_2.0.3             bitops_1.0-7               
 [71] matrixStats_0.62.0          evaluate_0.15              
 [73] lattice_0.20-45             purrr_1.0.2                
 [75] labeling_0.4.2              GenomicAlignments_1.32.0   
 [77] htmlwidgets_1.5.4           bit_4.0.4                  
 [79] tidyselect_1.2.0            magrittr_2.0.3             
 [81] R6_2.5.1                    generics_0.1.2             
 [83] DelayedArray_0.22.0         DBI_1.2.2                  
 [85] pillar_1.9.0                whisker_0.4                
 [87] withr_2.5.0                 KEGGREST_1.36.3            
 [89] RCurl_1.98-1.7              tibble_3.2.1               
 [91] crayon_1.5.1                utf8_1.2.2                 
 [93] BiocFileCache_2.4.0         rmarkdown_2.25             
 [95] progress_1.2.2              grid_4.2.0                 
 [97] blob_1.2.3                  git2r_0.30.1               
 [99] S7_0.2.0                    digest_0.6.29              
[101] httpuv_1.6.5                beeswarm_0.4.0             
[103] vipor_0.4.5                 bslib_0.3.1                </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
