<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2026-01-07" />

<title>Summarizing ECT results – ≥ 5 supporting SNPs &amp; ≥ 20 genes, CEDAR, bin AF</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis_new</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis_new">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Summarizing ECT results – ≥ 5 supporting
SNPs &amp; ≥ 20 genes, CEDAR, bin AF</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2026-01-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis_new/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree311ea6901f3ef029110252c4d51af89a0b56727etargetblank311ea69a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis_new/tree/311ea6901f3ef029110252c4d51af89a0b56727e" target="_blank">311ea69</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree311ea6901f3ef029110252c4d51af89a0b56727etargetblank311ea69a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis_new/tree/311ea6901f3ef029110252c4d51af89a0b56727e" target="_blank">311ea69</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/revision_summarizing_ECT_CEDAR_binAF.Rmd

Staged changes:
    Modified:   analysis/revision_num_pair_permut_ECT.Rmd
    Modified:   analysis/revision_num_pair_permut_ECT_binAF.Rmd
    Deleted:    analysis/revision_summarizing_ECT_CEDAR_binAF.knit.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/revision_summarizing_ECT_CEDAR_binAF.Rmd</code>) and
HTML (<code>docs/revision_summarizing_ECT_CEDAR_binAF.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/311ea6901f3ef029110252c4d51af89a0b56727e/analysis/revision_summarizing_ECT_CEDAR_binAF.Rmd" target="_blank">311ea69</a>
</td>
<td>
XSun
</td>
<td>
2026-01-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/311ea6901f3ef029110252c4d51af89a0b56727e/docs/revision_summarizing_ECT_CEDAR_binAF.html" target="_blank">311ea69</a>
</td>
<td>
XSun
</td>
<td>
2026-01-08
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/768f8ff3767501763e71e0d18d2533cc380b1f7b/analysis/revision_summarizing_ECT_CEDAR_binAF.Rmd" target="_blank">768f8ff</a>
</td>
<td>
XSun
</td>
<td>
2026-01-07
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/768f8ff3767501763e71e0d18d2533cc380b1f7b/docs/revision_summarizing_ECT_CEDAR_binAF.html" target="_blank">768f8ff</a>
</td>
<td>
XSun
</td>
<td>
2026-01-07
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>We applied cutoffs of + - <code>supporting SNP ≥ 5</code> and -
<code>gene number in pathway ≥ 20</code></p>
<p>to filter the pairs before performing FDR control. All results
presented below are based on this setting.</p>
<p><strong>ECT strategy: When generating “random” GWAS effects of a SNP,
we sample from GWAS SNPs with comparable AF with that SNP.</strong></p>
<pre class="r"><code>library(dplyr)
library(ggplot2)
library(gridExtra)
library(forcats)
library(tidyr)

# library(ensembldb)
# library(EnsDb.Hsapiens.v75)
# library(ggrepel)
# library(ggrastr)

celltypes &lt;- c(&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;CD15_positive_leukocyte&quot;,&quot;platelet&quot;,&quot;T_cell&quot;,&quot;thymocyte&quot;)
types &lt;- c(&quot;B cell (CD19+)&quot;,&quot;Monocyte (CD14+)&quot;,&quot;Leukocyte (CD15+)&quot;,&quot;Platelet&quot;,&quot;T cell (CD4+)&quot;,&quot;T cell (CD8+)&quot;)
names(types) &lt;- celltypes
names(celltypes) &lt;- types

source(&quot;/project/xinhe/xsun/pathway_factor/data_v2/traits_finalselection_SLEremoved.R&quot;)

folder_ect_summary &lt;- &quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_binAF/&quot;

DT::datatable(matrix())</code></pre>
<div id="htmlwidget-0c923cdc0df2ab45b6f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c923cdc0df2ab45b6f5">{"x":{"filter":"none","vertical":false,"data":[[null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="number-of-supporting-variants" class="section level1">
<h1>Number of supporting variants</h1>
<pre class="r"><code>p &lt;- list()
for(celltype in celltypes){

  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_poolalltraits_selected.RDS&quot;))
  df &lt;- df[df$trait_id %in% traits_EUR,]

  summary &lt;- df %&gt;%
    count(num_supp_SNP, name = &quot;Freq&quot;)

  # Plot
  p[[celltype]] &lt;- ggplot(summary, aes(x = num_supp_SNP, y = Freq)) +
    geom_bar(stat = &quot;identity&quot;, fill = &quot;steelblue&quot;) +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = &quot;# of supporting SNPs in each pair&quot;, y = &quot;# of factor–trait pairs&quot;) +
    geom_text(
      aes(label = ifelse(num_supp_SNP %% 5 == 0, as.character(Freq), &quot;&quot;)),
      colour = &quot;black&quot;, size = 2, vjust = -0.1, family = &quot;Times&quot;
    ) +
    annotate(
      geom = &quot;text&quot;,
      x = Inf, y = Inf,
      label = paste0(&quot;# of candidate pairs = &quot;, sum(summary$Freq[summary$num_supp_SNP &gt;= 5])),
      hjust = 1.1, vjust = 1.5, size = 3, family = &quot;Times&quot;, color = &quot;black&quot;
    ) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = &quot;black&quot;),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = &quot;black&quot;),
      text = element_text(family = &quot;Times&quot;)
    ) +
    scale_x_continuous(breaks = seq(0, max(summary$num_supp_SNP), 5))

}


all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-2-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/311ea6901f3ef029110252c4d51af89a0b56727e/docs/figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-2-1.png" target="_blank">311ea69</a>
</td>
<td>
XSun
</td>
<td>
2026-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/768f8ff3767501763e71e0d18d2533cc380b1f7b/docs/figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-2-1.png" target="_blank">768f8ff</a>
</td>
<td>
XSun
</td>
<td>
2026-01-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="histogram-for-ect-p-values" class="section level1">
<h1>Histogram for ECT p-values</h1>
<pre class="r"><code>p &lt;- list()
for(celltype in celltypes){

  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_poolalltraits_selected.RDS&quot;))
  df &lt;- df[df$trait_id %in% traits_EUR,]

  df &lt;- df[complete.cases(df$p_ECT),]

  summary_p &lt;- df %&gt;%
    mutate(bin = cut(p_ECT, breaks = seq(0, 1, by = 0.05), include.lowest = TRUE)) %&gt;%
    count(bin, name = &quot;Freq&quot;)

  # Correctly handle both [ and ( at the start of interval labels
  summary_p &lt;- summary_p %&gt;%
    mutate(
      bin_low  = as.numeric(sub(&quot;[^0-9\\.]*([0-9\\.]+),.*&quot;, &quot;\\1&quot;, bin)),
      bin_high = as.numeric(sub(&quot;.*,([0-9\\.]+)\\]&quot;, &quot;\\1&quot;, bin)),
      bin_mid  = (bin_low + bin_high) / 2
    )

  p[[celltype]] &lt;- ggplot(summary_p, aes(x = bin_mid, y = Freq)) +
    geom_bar(stat = &quot;identity&quot;, fill = &quot;steelblue&quot;) +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = &quot;ECT p-value&quot;, y = &quot;Count&quot;) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = &quot;black&quot;),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = &quot;black&quot;),
      text = element_text(family = &quot;Times&quot;)
    ) +
    scale_x_continuous(breaks = seq(0, 1, 0.1), limits = c(0, 1))

}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-3-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/311ea6901f3ef029110252c4d51af89a0b56727e/docs/figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-3-1.png" target="_blank">311ea69</a>
</td>
<td>
XSun
</td>
<td>
2026-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/768f8ff3767501763e71e0d18d2533cc380b1f7b/docs/figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-3-1.png" target="_blank">768f8ff</a>
</td>
<td>
XSun
</td>
<td>
2026-01-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-cell-type"
class="section level1">
<h1>Number of significant pairs at different cutoffs, for each cell
type</h1>
<pre class="r"><code>num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for(celltype in celltypes){

  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_poolalltraits_selected.RDS&quot;))

  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))

  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2 &amp; df$ECT_FDR_5suppSNP &gt;= 0.1,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))

}

sum &lt;- data.frame(celltype = types,
                  num_fdr01 = num_fdr01,
                  num_fdr02 = num_fdr02)


df &lt;- sum %&gt;%
  pivot_longer(cols = starts_with(&quot;num_fdr&quot;), names_to = &quot;event&quot;, values_to = &quot;total&quot;) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# Plot
p1 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(x = &quot;Cell types&quot;, y = &quot;Count&quot;) +
  ggtitle(&quot;# of pairs&quot;) +
  ylim(0, 30) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

p1</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-4-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/768f8ff3767501763e71e0d18d2533cc380b1f7b/docs/figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-4-1.png" target="_blank">768f8ff</a>
</td>
<td>
XSun
</td>
<td>
2026-01-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="detailed-results-for-all-pairs-at-ect-fdr-0.2"
class="section level1">
<h1>Detailed results for all pairs at ECT FDR &lt; 0.2</h1>
<pre class="r"><code>df_all_fdr02 &lt;- c()
df_largeECTp &lt;- c()
for(celltype in celltypes){

  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_poolalltraits_selected.RDS&quot;))
  df &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]

  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_all_fdr02 &lt;- rbind(df_all_fdr02,df_fdr02)
  
  df_sort &lt;- df[order(df$p_ECT, decreasing = T),]
  df_largeECTp &lt;- rbind(df_largeECTp , df_sort[1,])

}

rownames(df_all_fdr02) &lt;- NULL

#df_all_fdr02$celltype &lt;- types[df_all_fdr02$celltype]
df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP),]

df_all_fdr01 &lt;- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP &lt; 0.1,]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Pairs with ECT FDR &lt;= 0.2 &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-ad41175326dc68f35292" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ad41175326dc68f35292">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Pairs with ECT FDR &lt;= 0.2 <\/caption>","data":[["1","2","3"],["B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)"],["00600_PC1","04911_PC2","04910_PC4"],["00600","04911","04910"],["Sphingolipid metabolism","Insulin secretion","Insulin signaling pathway"],["AS-finn-b-M13_ANKYLOSPON","RA-ukb-a-105","MCH-ebi-a-GCST90002390"],["Ankylosing spondylitis ","Rheumatoid arthritis ","Mean Corpuscular Hemoglobin "],[6,5,8],[1e-06,1e-06,1.1e-05],[0.002208,0.003709,0.015818],[47,86,137],[24,37,87]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>trait_id<\/th>\n      <th>trait_name<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="effect-size-plots-for-the-top-pairs-ect-fdr-0.1"
class="section level2">
<h2>effect size plots for the top pairs (ECT FDR &lt; 0.1)</h2>
<pre class="r"><code>df_all_fdr01 &lt;- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP &lt; 0.1,]
df_all_fdr01$celltype &lt;- celltypes[df_all_fdr01$celltype]

fdr_cutoff &lt;- 0.2

for (i in 1:nrow(df_all_fdr01)){
  
  celltype &lt;- df_all_fdr01$celltype[i]
  trait &lt;- df_all_fdr01$trait_id[i]
  factor &lt;- df_all_fdr01$factor[i]
  pwy &lt;- df_all_fdr01$pathwayName[i]
  
  file_harmo &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;,celltype, &quot;-&quot;, trait, &quot;_harmo.RDS&quot;)
  dt_harmo &lt;- readRDS(file_harmo)
  
  dat &lt;- dt_harmo[[factor]]
  snp_select &lt;- dat[dat$FDR.exposure &lt; fdr_cutoff,]
  snp_select$chr &lt;- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr &lt;- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p &lt;- df_all_fdr01$p_ECT[i]
  
  pathwayname &lt;- df$pathwayName[df$factor == factor &amp; df$trait ==trait]
  
  eff_chr &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\nECT p-value =&quot;, ect_p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    ggtitle(paste0(pwy, &quot;-&quot;, trait))
  
  
  eaf_breaks &lt;- c(0, 0.01, 0.05, 0.10, 0.20, 0.30, 0.50)
  eaf_labels &lt;- c(&quot;&lt;0.01&quot;,&quot;0.01-0.05&quot;, &quot;0.05-0.1&quot;, &quot;0.1-0.2&quot;,
                  &quot;0.2-0.3&quot;, &quot;0.3-0.5&quot;)
  
  snp_select$AF_bin &lt;- cut(
    snp_select[[&quot;eaf.exposure&quot;]],
    breaks = eaf_breaks, labels = eaf_labels,
    include.lowest = TRUE, right = TRUE
  )
  
  eff_AF &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=AF_bin)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=AF_bin), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\nECT p-value =&quot;, ect_p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    ggtitle(paste0(pwy, &quot;-&quot;, trait))
  
  all &lt;- grid.arrange(eff_chr, eff_AF, nrow = 1)
  print(all)
}</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/311ea6901f3ef029110252c4d51af89a0b56727e/docs/figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-6-1.png" target="_blank">311ea69</a>
</td>
<td>
XSun
</td>
<td>
2026-01-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-6-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-6-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
</div>
<div id="effect-size-plots-for-the-pairs-with-large-ect-p-values"
class="section level2">
<h2>effect size plots for the pairs with large ECT p-values</h2>
<pre class="r"><code>df_largeECTp$celltype &lt;- celltypes[df_largeECTp$celltype]

fdr_cutoff &lt;- 0.2

for (i in 1:nrow(df_largeECTp)){
  
  celltype &lt;- df_largeECTp$celltype[i]
  trait &lt;- df_largeECTp$trait_id[i]
  factor &lt;- df_largeECTp$factor[i]
  pwy &lt;- df_largeECTp$pathwayName[i]
  
  file_harmo &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;,celltype, &quot;-&quot;, trait, &quot;_harmo.RDS&quot;)
  dt_harmo &lt;- readRDS(file_harmo)
  
  dat &lt;- dt_harmo[[factor]]
  snp_select &lt;- dat[dat$FDR.exposure &lt; fdr_cutoff,]
  snp_select$chr &lt;- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr &lt;- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p &lt;- df_largeECTp$p_ECT[i]
  
  pathwayname &lt;- df$pathwayName[df$factor == factor &amp; df$trait ==trait]
  
  eff_chr &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\nECT p-value =&quot;, ect_p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    ggtitle(paste0(pwy, &quot;-&quot;, trait))
  
  
  eaf_breaks &lt;- c(0, 0.01, 0.05, 0.10, 0.20, 0.30, 0.50)
  eaf_labels &lt;- c(&quot;&lt;0.01&quot;,&quot;0.01-0.05&quot;, &quot;0.05-0.1&quot;, &quot;0.1-0.2&quot;,
                  &quot;0.2-0.3&quot;, &quot;0.3-0.5&quot;)
  
  snp_select$AF_bin &lt;- cut(
    snp_select[[&quot;eaf.exposure&quot;]],
    breaks = eaf_breaks, labels = eaf_labels,
    include.lowest = TRUE, right = TRUE
  )
  
  eff_AF &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=AF_bin)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=AF_bin), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\nECT p-value =&quot;, ect_p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    ggtitle(paste0(pwy, &quot;-&quot;, trait))
  
  all &lt;- grid.arrange(eff_chr, eff_AF, nrow = 1)
  print(all)
}</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-7-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-7-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-7-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-7-4.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-7-5.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-7-6.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>TableGrob (1 x 2) &quot;arrange&quot;: 2 grobs
  z     cells    name           grob
1 1 (1-1,1-1) arrange gtable[layout]
2 2 (1-1,2-2) arrange gtable[layout]</code></pre>
</div>
</div>
<div id="comparing-the-ect-p-values-from-old-and-new-ect-strategy"
class="section level1">
<h1>Comparing the ECT p-values from old and new ECT strategy</h1>
<pre class="r"><code>library(DT)
library(htmltools)
library(knitr)

p &lt;- list()
tables &lt;- list()
for(celltype in celltypes){
  
  df_new &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_poolalltraits_selected.RDS&quot;))
  df_new$id &lt;- paste0(df_new$factor,&quot;-&quot;, df_new$trait_id)
  df_new &lt;- df_new[df_new$num_supp_SNP &gt;=5,]
  df_new &lt;- df_new[c(&quot;id&quot;,&quot;p_ECT&quot;,&quot;ECT_FDR_5suppSNP&quot;)]
  
  df_old &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_originalECT/&quot;, celltype,&quot;_ECT_summary_poolalltraits_selected.RDS&quot;))
  df_old$id &lt;- paste0(df_old$factor,&quot;-&quot;, df_old$trait_id)
  df_old &lt;- df_old[df_old$num_supp_SNP &gt;=5,]
  df_old &lt;- df_old[c(&quot;id&quot;,&quot;p_ECT&quot;,&quot;ECT_FDR_5suppSNP&quot;,&quot;pathwayName&quot;,&quot;trait_id&quot;,&quot;trait_name&quot;)]
  
  df_merge &lt;- merge(df_new,df_old, by = &quot;id&quot;)
  colnames(df_merge)[1:5] &lt;- c(&quot;id&quot;,&quot;p_ECT_new&quot;,&quot;FDR_ECT_new&quot; ,&quot;p_ECT_old&quot;,&quot;FDR_ECT_old&quot;)
  
  df_plot &lt;- df_merge %&gt;%
    filter(!is.na(p_ECT_new), !is.na(p_ECT_old)) %&gt;%
    mutate(
      logp_old = -log10(p_ECT_old),
      logp_new = -log10(p_ECT_new)
    )
  
  p[[celltype]] &lt;- ggplot(df_plot, aes(x = logp_old, y = logp_new)) +
    geom_point(alpha = 0.6, size = 2) +
    geom_abline(slope = 1, intercept = 0,
                linetype = &quot;dashed&quot;, color = &quot;red&quot;) +
    ggtitle(types[celltype]) +
    labs(
      x = expression(-log[10](p[ECT~old_strategy])),
      y = expression(-log[10](p[ECT~new_binAF]))
    ) +
    theme_classic()
  
  df_show &lt;- df_merge[order(df_merge$p_ECT_new, decreasing = F),]
  df_show &lt;- df_show[1:20,]
  
  tables[[celltype]] &lt;- DT::datatable(
    df_show,
    caption = tags$caption(
      style = &#39;caption-side: left; text-align: left; color:black; font-size:150%;&#39;,
      paste0(&#39;Pairs with lowest new ECT p-values - &#39;, types[celltype])
    ),
    options = list(pageLength = 10)
  )
  
  tables[[celltype]] &lt;- datatable(
    df_show,
    caption = tags$caption(
      style = &#39;caption-side:left; text-align:left; color:black; font-size:140%;&#39;,
      paste0(&#39;Pairs with lowest new ECT p-values - &#39;, types[celltype])
    ),
    options = list(pageLength = 10)
  )
  
  # print(
  #   DT::datatable(
  #     df_show,
  #     caption = tags$caption(
  #       style = &#39;caption-side: left; text-align: left; color:black; font-size:150%;&#39;,
  #       paste0(&#39;Pairs with lowest new ECT p-values - &#39;, types[celltype])
  #     ),
  #     options = list(pageLength = 10)
  #   )
  # )
  # 
  # cat(&quot;\n\n&quot;)
  
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_CEDAR_binAF.Rmd/unnamed-chunk-8-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for(celltype in celltypes) {
 
  cat(knit_print(tables[[celltype]]))
}</code></pre>
<div id="htmlwidget-66aaeb19fe992dfb51c2" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-66aaeb19fe992dfb51c2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Pairs with lowest new ECT p-values - B cell (CD19+)<\/caption>","data":[["167","992","1005","122","219","1072","1213","186","182","543","316","511","619","318","328","1846","1659","1965","510","1836"],["00600_PC1-AS-finn-b-M13_ANKYLOSPON","04211_PC3-T1D-ebi-a-GCST90014023","04213_PC2-RET-ebi-a-GCST90002405","00510_PC2-lymphc-ebi-a-GCST90002388","03010_PC2-MCH-ebi-a-GCST90002390","04218_PC4-WBC-ebi-a-GCST90002407","04380_PC3-RET-ebi-a-GCST90002405","00640_PC1-T1D-ebi-a-GCST90014023","00630_PC4-lymphc-ebi-a-GCST90002388","04062_PC2-Psoriasis-ebi-a-GCST90038681","03040_PC4-MCV-ebi-a-GCST90002392","04060_PC2-HCT-ebi-a-GCST90002383","04068_PC3-WBC-ebi-a-GCST90002407","03040_PC4-RBC-ebi-a-GCST90002403","03050_PC3-RET-ebi-a-GCST90002405","04730_PC5-MCH-ebi-a-GCST90002390","04668_PC2-RET-ebi-a-GCST90002405","04917_PC2-RET-ebi-a-GCST90002405","04060_PC2-HB-ebi-a-GCST90002384","04728_PC4-MPV-ebi-a-GCST90002395"],[1e-06,0.000203,0.000616,0.000883,0.0011999,0.0012999,0.0015998,0.0017998,0.0024998,0.0028997,0.0030997,0.0036996,0.0038996,0.0040996,0.0043996,0.0044996,0.0050995,0.0051995,0.0059994,0.0060994],[0.002208,0.224112,0.453376,0.4783632,0.4783632,0.4783632,0.4967448,0.4967448,0.613284266666667,0.6209448,0.6209448,0.6209448,0.6209448,0.6209448,0.6209448,0.6209448,0.629623466666667,0.629623466666667,0.629623466666667,0.629623466666667],[0.0458954,0.000195,0.000221,0.000278,0.00045,0.3866134,0.001153,0.0012999,0.0018998,4e-05,0.0016998,0.000565,0.0029997,0.0017998,0.0018998,0.0020998,0.0021998,0.001053,0.000489,0.0030997],[0.37685869010989,0.138613333333333,0.138613333333333,0.138613333333333,0.138613333333333,0.706728471676301,0.173469942857143,0.173469942857143,0.173469942857143,0.08832,0.173469942857143,0.138613333333333,0.189238217142857,0.173469942857143,0.173469942857143,0.173469942857143,0.173469942857143,0.173469942857143,0.138613333333333,0.190114933333333],["Sphingolipid metabolism","Longevity regulating pathway","Longevity regulating pathway - multiple species","N-Glycan biosynthesis","Ribosome","Cellular senescence","Osteoclast differentiation","Propanoate metabolism","Glyoxylate and dicarboxylate metabolism","Chemokine signaling pathway","Spliceosome","Cytokine-cytokine receptor interaction","FoxO signaling pathway","Spliceosome","Proteasome","Long-term depression","TNF signaling pathway","Prolactin signaling pathway","Cytokine-cytokine receptor interaction","Dopaminergic synapse"],["AS-finn-b-M13_ANKYLOSPON","T1D-ebi-a-GCST90014023","RET-ebi-a-GCST90002405","lymphc-ebi-a-GCST90002388","MCH-ebi-a-GCST90002390","WBC-ebi-a-GCST90002407","RET-ebi-a-GCST90002405","T1D-ebi-a-GCST90014023","lymphc-ebi-a-GCST90002388","Psoriasis-ebi-a-GCST90038681","MCV-ebi-a-GCST90002392","HCT-ebi-a-GCST90002383","WBC-ebi-a-GCST90002407","RBC-ebi-a-GCST90002403","RET-ebi-a-GCST90002405","MCH-ebi-a-GCST90002390","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","HB-ebi-a-GCST90002384","MPV-ebi-a-GCST90002395"],["Ankylosing spondylitis ","Type 1 diabetes ","Reticulocyte count","Lymphocyte count  lymphc","Mean Corpuscular Hemoglobin ","White blood cell count ","Reticulocyte count","Type 1 diabetes ","Lymphocyte count  lymphc","Psoriasis","Mean Corpuscular Volume","Hematocrit","White blood cell count ","Red blood cell count ","Reticulocyte count","Mean Corpuscular Hemoglobin ","Reticulocyte count","Reticulocyte count","Hemoglobin ","Mean Platelet Volume"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>p_ECT_new<\/th>\n      <th>FDR_ECT_new<\/th>\n      <th>p_ECT_old<\/th>\n      <th>FDR_ECT_old<\/th>\n      <th>pathwayName<\/th>\n      <th>trait_id<\/th>\n      <th>trait_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-1814506231684700bf53" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1814506231684700bf53">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Pairs with lowest new ECT p-values - Monocyte (CD14+)<\/caption>","data":[["3184","1047","1089","3449","1283","2002","3709","2446","1463","2141","1328","3154","2707","911","2354","3707","2533","2543","1739","117"],["04911_PC2-RA-ukb-a-105","04068_PC1-RBC-ebi-a-GCST90002403","04070_PC2-asthma_pmid36778051","04924_PC3-asthma_pmid36778051","04120_PC2-asthma_pmid36778051","04340_PC4-celiac-ieu-a-1058","04979_PC5-lymphc-ebi-a-GCST90002388","04622_PC1-MCH-ebi-a-GCST90002390","04142_PC3-WBC-ebi-a-GCST90002407","04390_PC2-MPV-ebi-a-GCST90002395","04136_PC3-UC-ebi-a-GCST003045","04810_PC5-UC-ebi-a-GCST003045","04664_PC1-asthma_pmid36778051","04020_PC5-MCH-ebi-a-GCST90002390","04611_PC1-HCT-ebi-a-GCST90002383","04979_PC1-lymphc-ebi-a-GCST90002388","04630_PC5-lymphc-ebi-a-GCST90002388","04650_PC1-RBC-ebi-a-GCST90002403","04211_PC4-MPV-ebi-a-GCST90002395","00230_PC3-CD-ebi-a-GCST003044"],[1e-06,0.000189,0.000228,0.000231,0.000406,0.000822,0.00101,0.0010999,0.0011999,0.0015998,0.0018998,0.0018998,0.0023998,0.0025997,0.0026997,0.0027997,0.0029997,0.0031997,0.0036996,0.0037996],[0.003709,0.21419475,0.21419475,0.21419475,0.3011708,0.494492122222222,0.494492122222222,0.494492122222222,0.494492122222222,0.587196516666667,0.587196516666667,0.587196516666667,0.64900545625,0.64900545625,0.64900545625,0.64900545625,0.654463958823529,0.659315961111111,0.70463582,0.70463582],[0.0183982,7.1e-05,0.000309,0.000105,0.000361,0.000659,0.000527,0.0010999,0.00046,4.3e-05,0.000761,0.0018998,0.0022998,0.000895,0.0027997,0.0012999,0.0017998,0.0027997,0.000349,0.0028997],[0.454606147741936,0.129815,0.201345714285714,0.129815,0.201345714285714,0.222202818181818,0.2121548,0.226640505555556,0.2121548,0.129815,0.226640505555556,0.250881123529412,0.250881123529412,0.226640505555556,0.256071126190476,0.247235758333333,0.247239192592593,0.256071126190476,0.201345714285714,0.256071126190476],["Insulin secretion","FoxO signaling pathway","Phosphatidylinositol signaling system","Renin secretion","Ubiquitin mediated proteolysis","Hedgehog signaling pathway","Cholesterol metabolism","RIG-I-like receptor signaling pathway","Lysosome","Hippo signaling pathway","Autophagy - other","Regulation of actin cytoskeleton","Fc epsilon RI signaling pathway","Calcium signaling pathway","Platelet activation","Cholesterol metabolism","JAK-STAT signaling pathway","Natural killer cell mediated cytotoxicity","Longevity regulating pathway","Purine metabolism"],["RA-ukb-a-105","RBC-ebi-a-GCST90002403","asthma_pmid36778051","asthma_pmid36778051","asthma_pmid36778051","celiac-ieu-a-1058","lymphc-ebi-a-GCST90002388","MCH-ebi-a-GCST90002390","WBC-ebi-a-GCST90002407","MPV-ebi-a-GCST90002395","UC-ebi-a-GCST003045","UC-ebi-a-GCST003045","asthma_pmid36778051","MCH-ebi-a-GCST90002390","HCT-ebi-a-GCST90002383","lymphc-ebi-a-GCST90002388","lymphc-ebi-a-GCST90002388","RBC-ebi-a-GCST90002403","MPV-ebi-a-GCST90002395","CD-ebi-a-GCST003044"],["Rheumatoid arthritis ","Red blood cell count ","Asthma","Asthma","Asthma","Celiac disease","Lymphocyte count  lymphc","Mean Corpuscular Hemoglobin ","White blood cell count ","Mean Platelet Volume","Ulcerative colitis ","Ulcerative colitis ","Asthma","Mean Corpuscular Hemoglobin ","Hematocrit","Lymphocyte count  lymphc","Lymphocyte count  lymphc","Red blood cell count ","Mean Platelet Volume","Crohn<U+2019>s disease "]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>p_ECT_new<\/th>\n      <th>FDR_ECT_new<\/th>\n      <th>p_ECT_old<\/th>\n      <th>FDR_ECT_old<\/th>\n      <th>pathwayName<\/th>\n      <th>trait_id<\/th>\n      <th>trait_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-43c5071fe8a890b96ecd" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-43c5071fe8a890b96ecd">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Pairs with lowest new ECT p-values - Leukocyte (CD15+)<\/caption>","data":[["1202","913","351","751","239","1412","823","946","74","1313","306","126","128","345","648","688","548","115","1044","835"],["04910_PC4-MCH-ebi-a-GCST90002390","04630_PC2-WBC-ebi-a-GCST90002407","04068_PC2-lymphc-ebi-a-GCST90002388","04371_PC4-MCH-ebi-a-GCST90002390","04010_PC2-MCV-ebi-a-GCST90002392","04971_PC1-MCV-ebi-a-GCST90002392","04550_PC2-lymphc-ebi-a-GCST90002388","04658_PC4-HB-ebi-a-GCST90002384","00510_PC3-LDL-ukb-d-30780_irnt","04922_PC2-MCV-ebi-a-GCST90002392","04024_PC4-RBC-ebi-a-GCST90002403","03008_PC2-allergy-ebi-a-GCST005038","03008_PC4-LDL-ukb-d-30780_irnt","04068_PC1-celiac-ieu-a-1058","04217_PC4-MCH-ebi-a-GCST90002390","04270_PC1-HT-ebi-a-GCST90018855","04146_PC3-HT-ebi-a-GCST90018855","00590_PC5-CD-ebi-a-GCST003044","04714_PC4-IBD-ebi-a-GCST003043","04611_PC3-lymphc-ebi-a-GCST90002388"],[1.1e-05,0.0022998,0.0026997,0.0026997,0.0050995,0.0059994,0.0063994,0.0064994,0.0074993,0.0077992,0.0107989,0.0114989,0.0122988,0.0132987,0.0140986,0.0145985,0.0169983,0.0170983,0.0178982,0.020298],[0.015818,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528,0.914792517924528],[1e-06,0.0010999,0.0048995,0.000528,0.0012999,0.0013999,0.0026997,0.0011999,0.0028997,0.0109989,0.0023998,0.0091991,0.0063994,0.0105989,0.0047995,0.039996,0.0035996,0.0051995,0.0105989,0.0044996],[0.001438,0.287579457142857,0.302976328571429,0.287579457142857,0.287579457142857,0.287579457142857,0.302976328571429,0.287579457142857,0.302976328571429,0.337457514285714,0.302976328571429,0.337457514285714,0.306744573333333,0.337457514285714,0.302976328571429,0.456462285714286,0.302976328571429,0.302976328571429,0.337457514285714,0.302976328571429],["Insulin signaling pathway","JAK-STAT signaling pathway","FoxO signaling pathway","Apelin signaling pathway","MAPK signaling pathway","Gastric acid secretion","Signaling pathways regulating pluripotency of stem cells","Th1 and Th2 cell differentiation","N-Glycan biosynthesis","Glucagon signaling pathway","cAMP signaling pathway","Ribosome biogenesis in eukaryotes","Ribosome biogenesis in eukaryotes","FoxO signaling pathway","Necroptosis","Vascular smooth muscle contraction","Peroxisome","Arachidonic acid metabolism","Thermogenesis","Platelet activation"],["MCH-ebi-a-GCST90002390","WBC-ebi-a-GCST90002407","lymphc-ebi-a-GCST90002388","MCH-ebi-a-GCST90002390","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","lymphc-ebi-a-GCST90002388","HB-ebi-a-GCST90002384","LDL-ukb-d-30780_irnt","MCV-ebi-a-GCST90002392","RBC-ebi-a-GCST90002403","allergy-ebi-a-GCST005038","LDL-ukb-d-30780_irnt","celiac-ieu-a-1058","MCH-ebi-a-GCST90002390","HT-ebi-a-GCST90018855","HT-ebi-a-GCST90018855","CD-ebi-a-GCST003044","IBD-ebi-a-GCST003043","lymphc-ebi-a-GCST90002388"],["Mean Corpuscular Hemoglobin ","White blood cell count ","Lymphocyte count  lymphc","Mean Corpuscular Hemoglobin ","Mean Corpuscular Volume","Mean Corpuscular Volume","Lymphocyte count  lymphc","Hemoglobin ","LDL cholesterol ","Mean Corpuscular Volume","Red blood cell count ","Allergy","LDL cholesterol ","Celiac disease","Mean Corpuscular Hemoglobin ","Hashimoto thyroiditis","Hashimoto thyroiditis","Crohn<U+2019>s disease ","Inflammatory bowel disease ","Lymphocyte count  lymphc"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>p_ECT_new<\/th>\n      <th>FDR_ECT_new<\/th>\n      <th>p_ECT_old<\/th>\n      <th>FDR_ECT_old<\/th>\n      <th>pathwayName<\/th>\n      <th>trait_id<\/th>\n      <th>trait_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-3f22443f93eeeda03f7b" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3f22443f93eeeda03f7b">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Pairs with lowest new ECT p-values - Platelet<\/caption>","data":[["935","36","931","1308","1767","821","1137","736","1572","136","129","905","274","1460","728","1541","801","708","326","311"],["04390_PC5-RET-ebi-a-GCST90002405","00230_PC1-RET-ebi-a-GCST90002405","04390_PC3-LDL-ukb-d-30780_irnt","04714_PC4-RET-ebi-a-GCST90002405","04971_PC4-HB-ebi-a-GCST90002384","04270_PC4-UC-ebi-a-GCST003045","04625_PC3-RET-ebi-a-GCST90002405","04217_PC3-HCT-ebi-a-GCST90002383","04918_PC4-MCV-ebi-a-GCST90002392","03018_PC2-RET-ebi-a-GCST90002405","03013_PC5-UC-ebi-a-GCST003045","04371_PC5-HCT-ebi-a-GCST90002383","04022_PC4-WBC-ebi-a-GCST90002407","04750_PC3-WBC-ebi-a-GCST90002407","04216_PC3-RET-ebi-a-GCST90002405","04915_PC5-WBC-ebi-a-GCST90002407","04261_PC4-RET-ebi-a-GCST90002405","04211_PC5-MCH-ebi-a-GCST90002390","04062_PC2-WBC-ebi-a-GCST90002407","04061_PC2-MCH-ebi-a-GCST90002390"],[0.000574,0.000784,0.000785,0.0011999,0.0021998,0.0022998,0.0029997,0.0032997,0.0052995,0.0059994,0.0069993,0.0070993,0.0085991,0.0089991,0.009599,0.009699,0.010099,0.0121988,0.0123988,0.0132987],[0.478326666666667,0.478326666666667,0.478326666666667,0.5483543,0.7006724,0.7006724,0.75398145,0.75398145,0.8396540975,0.8396540975,0.8396540975,0.8396540975,0.8396540975,0.8396540975,0.8396540975,0.8396540975,0.8396540975,0.8396540975,0.8396540975,0.8396540975],[6.7e-05,0.000177,0.00064,9.4e-05,0.000677,0.000803,0.000595,0.0040996,0.0013999,0.0027997,0.0032997,0.0042996,0.0024998,0.0056994,0.0039996,0.010199,0.0017998,0.0079992,0.0018998,0.0039996],[0.0559926666666667,0.07907475,0.172828428571429,0.0559926666666667,0.172828428571429,0.179370125,0.172828428571429,0.201591928205128,0.201591928205128,0.201591928205128,0.201591928205128,0.201591928205128,0.201591928205128,0.2165844,0.201591928205128,0.239810697368421,0.201591928205128,0.2165844,0.201591928205128,0.201591928205128],["Hippo signaling pathway","Purine metabolism","Hippo signaling pathway","Thermogenesis","Gastric acid secretion","Vascular smooth muscle contraction","C-type lectin receptor signaling pathway","Necroptosis","Thyroid hormone synthesis","RNA degradation","RNA transport","Apelin signaling pathway","cGMP-PKG signaling pathway","Inflammatory mediator regulation of TRP channels","Ferroptosis","Estrogen signaling pathway","Adrenergic signaling in cardiomyocytes","Longevity regulating pathway","Chemokine signaling pathway","Viral protein interaction with cytokine and cytokine receptor"],["RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","LDL-ukb-d-30780_irnt","RET-ebi-a-GCST90002405","HB-ebi-a-GCST90002384","UC-ebi-a-GCST003045","RET-ebi-a-GCST90002405","HCT-ebi-a-GCST90002383","MCV-ebi-a-GCST90002392","RET-ebi-a-GCST90002405","UC-ebi-a-GCST003045","HCT-ebi-a-GCST90002383","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","RET-ebi-a-GCST90002405","WBC-ebi-a-GCST90002407","RET-ebi-a-GCST90002405","MCH-ebi-a-GCST90002390","WBC-ebi-a-GCST90002407","MCH-ebi-a-GCST90002390"],["Reticulocyte count","Reticulocyte count","LDL cholesterol ","Reticulocyte count","Hemoglobin ","Ulcerative colitis ","Reticulocyte count","Hematocrit","Mean Corpuscular Volume","Reticulocyte count","Ulcerative colitis ","Hematocrit","White blood cell count ","White blood cell count ","Reticulocyte count","White blood cell count ","Reticulocyte count","Mean Corpuscular Hemoglobin ","White blood cell count ","Mean Corpuscular Hemoglobin "]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>p_ECT_new<\/th>\n      <th>FDR_ECT_new<\/th>\n      <th>p_ECT_old<\/th>\n      <th>FDR_ECT_old<\/th>\n      <th>pathwayName<\/th>\n      <th>trait_id<\/th>\n      <th>trait_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-198b30f56771bc600b2c" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-198b30f56771bc600b2c">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Pairs with lowest new ECT p-values - T cell (CD4+)<\/caption>","data":[["1859","66","1825","441","1011","2040","858","640","1826","337","1575","2275","969","1299","2401","2244","2501","779","2342","444"],["04666_PC2-RBC-ebi-a-GCST90002403","00190_PC5-MPV-ebi-a-GCST90002395","04662_PC4-lymphc-ebi-a-GCST90002388","03040_PC5-PLT-ebi-a-GCST90002402","04142_PC5-WBC-ebi-a-GCST90002407","04724_PC1-IBD-ebi-a-GCST003043","04110_PC3-MCV-ebi-a-GCST90002392","04022_PC5-WBC-ebi-a-GCST90002407","04662_PC5-CD-ebi-a-GCST003044","03013_PC3-MCV-ebi-a-GCST90002392","04611_PC2-lymphc-ebi-a-GCST90002388","04919_PC3-CD-ebi-a-GCST003044","04140_PC5-RBC-ebi-a-GCST90002403","04261_PC3-UC-ebi-a-GCST003045","04927_PC4-IBD-ebi-a-GCST003043","04917_PC2-CD-ebi-a-GCST003044","04971_PC1-IBD-ebi-a-GCST003043","04068_PC2-PLT-ebi-a-GCST90002402","04923_PC1-WBC-ebi-a-GCST90002407","03050_PC3-WBC-ebi-a-GCST90002407"],[0.000781,0.0013999,0.0016998,0.0017998,0.0019998,0.0022998,0.0024998,0.0029997,0.0031997,0.0032997,0.0037996,0.0037996,0.0038996,0.0048995,0.0051995,0.0055994,0.0056994,0.0058994,0.0065993,0.0068993],[0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182,0.666888181818182],[0.000212,0.000216,0.0010999,0.0013999,0.0189981,0.0014999,0.000402,0.0028997,0.0016998,0.0016998,0.000445,0.0046995,0.0023998,0.0035996,0.0012999,0.0025997,0.0043996,0.0061994,0.0044996,0.0011999],[0.255469941176471,0.255469941176471,0.255469941176471,0.255469941176471,0.369225474452555,0.255469941176471,0.255469941176471,0.266827166666667,0.255469941176471,0.255469941176471,0.255469941176471,0.266827166666667,0.255478708333333,0.266827166666667,0.255469941176471,0.265297442307692,0.266827166666667,0.311212936363636,0.266827166666667,0.255469941176471],["Fc gamma R-mediated phagocytosis","Oxidative phosphorylation","B cell receptor signaling pathway","Spliceosome","Lysosome","Glutamatergic synapse","Cell cycle","cGMP-PKG signaling pathway","B cell receptor signaling pathway","RNA transport","Platelet activation","Thyroid hormone signaling pathway","Autophagy - animal","Adrenergic signaling in cardiomyocytes","Cortisol synthesis and secretion","Prolactin signaling pathway","Gastric acid secretion","FoxO signaling pathway","Regulation of lipolysis in adipocytes","Proteasome"],["RBC-ebi-a-GCST90002403","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","PLT-ebi-a-GCST90002402","WBC-ebi-a-GCST90002407","IBD-ebi-a-GCST003043","MCV-ebi-a-GCST90002392","WBC-ebi-a-GCST90002407","CD-ebi-a-GCST003044","MCV-ebi-a-GCST90002392","lymphc-ebi-a-GCST90002388","CD-ebi-a-GCST003044","RBC-ebi-a-GCST90002403","UC-ebi-a-GCST003045","IBD-ebi-a-GCST003043","CD-ebi-a-GCST003044","IBD-ebi-a-GCST003043","PLT-ebi-a-GCST90002402","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407"],["Red blood cell count ","Mean Platelet Volume","Lymphocyte count  lymphc","Platelet count ","White blood cell count ","Inflammatory bowel disease ","Mean Corpuscular Volume","White blood cell count ","Crohn<U+2019>s disease ","Mean Corpuscular Volume","Lymphocyte count  lymphc","Crohn<U+2019>s disease ","Red blood cell count ","Ulcerative colitis ","Inflammatory bowel disease ","Crohn<U+2019>s disease ","Inflammatory bowel disease ","Platelet count ","White blood cell count ","White blood cell count "]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>p_ECT_new<\/th>\n      <th>FDR_ECT_new<\/th>\n      <th>p_ECT_old<\/th>\n      <th>FDR_ECT_old<\/th>\n      <th>pathwayName<\/th>\n      <th>trait_id<\/th>\n      <th>trait_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-b544a055166914a18edc" style="width:960px;height:500px;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b544a055166914a18edc">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side:left; text-align:left; color:black; font-size:140%;\">Pairs with lowest new ECT p-values - T cell (CD8+)<\/caption>","data":[["1768","1833","2377","1161","2251","2075","1177","2313","1121","2344","2363","1136","1349","625","1255","1485","977","524","610","1878"],["04670_PC1-IBD-ebi-a-GCST003043","04714_PC4-RBC-ebi-a-GCST90002403","04972_PC3-WBC-ebi-a-GCST90002407","04217_PC5-IBD-ebi-a-GCST003043","04925_PC3-RET-ebi-a-GCST90002405","04911_PC3-MCV-ebi-a-GCST90002392","04218_PC5-lymphc-ebi-a-GCST90002388","04935_PC1-MPV-ebi-a-GCST90002395","04211_PC5-MPV-ebi-a-GCST90002395","04962_PC2-RET-ebi-a-GCST90002405","04971_PC1-IBD-ebi-a-GCST003043","04215_PC2-lymphc-ebi-a-GCST90002388","04390_PC4-PLT-ebi-a-GCST90002402","04060_PC2-IBD-ebi-a-GCST003043","04340_PC4-MCV-ebi-a-GCST90002392","04612_PC3-MPV-ebi-a-GCST90002395","04144_PC3-MCV-ebi-a-GCST90002392","04014_PC3-PLT-ebi-a-GCST90002402","04024_PC4-MCV-ebi-a-GCST90002392","04722_PC3-MCV-ebi-a-GCST90002392"],[0.000554,0.0023998,0.0026997,0.0027997,0.0028997,0.0029997,0.0037996,0.0042996,0.0047995,0.0047995,0.0050995,0.0052995,0.0056994,0.0058994,0.0062994,0.0070993,0.0071993,0.0072993,0.0073993,0.0078992],[0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431,0.775869396078431],[8.3e-05,0.0012999,0.0012999,0.0011999,0.0013999,0.0019998,0.001328,0.0018998,0.00018,0.000346,0.000189,0.0037996,0.000436,0.000527,0.0023998,0.0016998,0.010099,0.0010999,0.0038996,0.0044996],[0.1001395,0.179230745714286,0.179230745714286,0.179230745714286,0.179230745714286,0.179230745714286,0.179230745714286,0.179230745714286,0.11401425,0.1669796,0.11401425,0.1993138,0.175344666666667,0.179230745714286,0.179230745714286,0.179230745714286,0.282875712643678,0.179230745714286,0.200207123404255,0.212892839215686],["Leukocyte transendothelial migration","Thermogenesis","Pancreatic secretion","Necroptosis","Aldosterone synthesis and secretion","Insulin secretion","Cellular senescence","Growth hormone synthesis, secretion and action","Longevity regulating pathway","Vasopressin-regulated water reabsorption","Gastric acid secretion","Apoptosis - multiple species","Hippo signaling pathway","Cytokine-cytokine receptor interaction","Hedgehog signaling pathway","Antigen processing and presentation","Endocytosis","Ras signaling pathway","cAMP signaling pathway","Neurotrophin signaling pathway"],["IBD-ebi-a-GCST003043","RBC-ebi-a-GCST90002403","WBC-ebi-a-GCST90002407","IBD-ebi-a-GCST003043","RET-ebi-a-GCST90002405","MCV-ebi-a-GCST90002392","lymphc-ebi-a-GCST90002388","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","IBD-ebi-a-GCST003043","lymphc-ebi-a-GCST90002388","PLT-ebi-a-GCST90002402","IBD-ebi-a-GCST003043","MCV-ebi-a-GCST90002392","MPV-ebi-a-GCST90002395","MCV-ebi-a-GCST90002392","PLT-ebi-a-GCST90002402","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392"],["Inflammatory bowel disease ","Red blood cell count ","White blood cell count ","Inflammatory bowel disease ","Reticulocyte count","Mean Corpuscular Volume","Lymphocyte count  lymphc","Mean Platelet Volume","Mean Platelet Volume","Reticulocyte count","Inflammatory bowel disease ","Lymphocyte count  lymphc","Platelet count ","Inflammatory bowel disease ","Mean Corpuscular Volume","Mean Platelet Volume","Mean Corpuscular Volume","Platelet count ","Mean Corpuscular Volume","Mean Corpuscular Volume"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>id<\/th>\n      <th>p_ECT_new<\/th>\n      <th>FDR_ECT_new<\/th>\n      <th>p_ECT_old<\/th>\n      <th>FDR_ECT_old<\/th>\n      <th>pathwayName<\/th>\n      <th>trait_id<\/th>\n      <th>trait_name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.39      htmltools_0.5.2 DT_0.22         tidyr_1.3.0    
[5] forcats_0.5.1   gridExtra_2.3   ggplot2_4.0.0   dplyr_1.1.4    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12        highr_0.9          RColorBrewer_1.1-3 pillar_1.9.0      
 [5] compiler_4.2.0     bslib_0.3.1        later_1.3.0        jquerylib_0.1.4   
 [9] git2r_0.30.1       workflowr_1.7.0    tools_4.2.0        digest_0.6.29     
[13] gtable_0.3.6       jsonlite_1.8.0     evaluate_0.15      lifecycle_1.0.4   
[17] tibble_3.2.1       pkgconfig_2.0.3    rlang_1.1.2        cli_3.6.1         
[21] rstudioapi_0.13    crosstalk_1.2.0    yaml_2.3.5         xfun_0.41         
[25] fastmap_1.1.0      withr_2.5.0        stringr_1.5.1      htmlwidgets_1.5.4 
[29] generics_0.1.4     fs_1.5.2           vctrs_0.6.5        sass_0.4.1        
[33] grid_4.2.0         rprojroot_2.0.3    tidyselect_1.2.0   glue_1.6.2        
[37] R6_2.5.1           fansi_1.0.3        rmarkdown_2.25     purrr_1.0.2       
[41] farver_2.1.0       magrittr_2.0.3     whisker_0.4        scales_1.4.0      
[45] promises_1.2.0.1   dichromat_2.0-0.1  httpuv_1.6.5       labeling_0.4.2    
[49] S7_0.2.0           utf8_1.2.2         stringi_1.7.6     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
