<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-11-06" />

<title>Summarizing ECT results – ≥ 5 supporting SNPs &amp; ≥ 20 genes, SLE removed</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis_new</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis_new">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Summarizing ECT results – ≥ 5 supporting
SNPs &amp; ≥ 20 genes, SLE removed</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-11-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis_new/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtreedcc84263a7f25929dbc8f217a0cab74fa2fc21ddtargetblankdcc8426a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis_new/tree/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd" target="_blank">dcc8426</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtreedcc84263a7f25929dbc8f217a0cab74fa2fc21ddtargetblankdcc8426a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis_new/tree/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd" target="_blank">dcc8426</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/revision_acat_specific_pairs.Rmd

Unstaged changes:
    Modified:   analysis/revision.Rmd
    Modified:   analysis/revision_acat_selectedtraits.Rmd
    Modified:   analysis/revision_figures.Rmd
    Modified:   analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd</code>)
and HTML
(<code>docs/revision_summarizing_ECT_setting_SLEremoved.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/revision_summarizing_ECT_setting_SLEremoved.html" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/4fb0b06135482a4a2eccb77f03ac45bb5ba75ea8/analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd" target="_blank">4fb0b06</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/revision_summarizing_ECT_setting_SLEremoved.html" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/revision_summarizing_ECT_setting_SLEremoved.html" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/revision_summarizing_ECT_setting_SLEremoved.html" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/analysis/revision_summarizing_ECT_setting_SLEremoved.Rmd" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/c81644744954e1781983b8656df97d40ea6925f1/docs/revision_summarizing_ECT_setting_SLEremoved.html" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>We applied cutoffs of + - <code>supporting SNP ≥ 5</code> and -
<code>gene number in pathway ≥ 20</code></p>
<p>to filter the pairs before performing FDR control. All results
presented below are based on this setting.</p>
<pre class="r"><code>library(dplyr)
library(ggplot2)
library(gridExtra)
library(forcats)
library(tidyr)

library(ensembldb)
library(EnsDb.Hsapiens.v75)
library(ggrepel)
library(ggrastr)

celltypes &lt;- c(&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;CD15_positive_leukocyte&quot;,&quot;platelet&quot;,&quot;T_cell&quot;,&quot;thymocyte&quot;)
types &lt;- c(&quot;B cell (CD19+)&quot;,&quot;Monocyte (CD14+)&quot;,&quot;Granulocyte (CD15+)&quot;,&quot;Platelet&quot;,&quot;T cell (CD4+)&quot;,&quot;T cell (CD8+)&quot;)
names(celltypes) &lt;- types
names(types) &lt;- celltypes

folder_ect_summary &lt;- &quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;</code></pre>
<div id="eur" class="section level1">
<h1>EUR</h1>
<div id="detailed-results-for-all-pairs-at-ect-fdr-0.2"
class="section level2">
<h2>Detailed results for all pairs at ECT FDR &lt; 0.2</h2>
<pre class="r"><code>df_all_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  df &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_all_fdr02 &lt;- rbind(df_all_fdr02,df_fdr02)
  
}


df_all_fdr02$celltype &lt;- types[df_all_fdr02$celltype]
df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP),]

df_all_fdr01 &lt;- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP &lt; 0.1,]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Pairs with ECT FDR &lt;= 0.2 &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-0c923cdc0df2ab45b6f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c923cdc0df2ab45b6f5">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Pairs with ECT FDR &lt;= 0.2 <\/caption>","data":[["11","14","11791","11792","11793","2","110","210","43","15751","1","21","16721","4","6","7","8","9","16436","16437","18166","10","15752","5","11794","41","16","24","22","23","26","25","16438","16439","16444","16445","16447","16440","16441","16442","16443","16446","18168","20","92","27","28","18169","101","161","29","311","15753","15754","15758","15760","15761","15763","15756","15762","141","151","191","202","222","331","34","35","131","261","321","171","181","281","36","242","17501","61","38","15765","15766","15767","411","42","39","31","16448","32","33","16451","16452","81","111","18","251","11797","11798","11799","11802","11803","11804","11805","11806","11807","11808","11809","11810","11811","201","211","221","91","12","15","17","13","19","231","241","271","46","15768","44","17502","47"],["Granulocyte (CD15+)","Platelet","Platelet","Platelet","Platelet","B cell (CD19+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","Monocyte (CD14+)","Monocyte (CD14+)","Monocyte (CD14+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","T cell (CD8+)","Platelet","Platelet","Platelet","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","T cell (CD8+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","Platelet","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","B cell (CD19+)","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","Platelet","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)","T cell (CD8+)"],["MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","Psoriasis-ebi-a-GCST90038681","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","RBC-ebi-a-GCST90002403","asthma_pmid36778051","MPV-ebi-a-GCST90002395","T1D-ebi-a-GCST90014023","RBC-ebi-a-GCST90002403","MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","HCT-ebi-a-GCST90002383","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","HCT-ebi-a-GCST90002383","RET-ebi-a-GCST90002405","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","LDL-ukb-d-30780_irnt","T1D-ebi-a-GCST90014023","T1D-ebi-a-GCST90014023","RBC-ebi-a-GCST90002403","RBC-ebi-a-GCST90002403","MCH-ebi-a-GCST90002390","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","lymphc-ebi-a-GCST90002388","MCV-ebi-a-GCST90002392","PLT-ebi-a-GCST90002402","MCV-ebi-a-GCST90002392","HB-ebi-a-GCST90002384","lymphc-ebi-a-GCST90002388","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","PLT-ebi-a-GCST90002402","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","RBC-ebi-a-GCST90002403","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","HB-ebi-a-GCST90002384","lymphc-ebi-a-GCST90002388","UC-ebi-a-GCST003045","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","MCV-ebi-a-GCST90002392","MCV-ebi-a-GCST90002392","HB-ebi-a-GCST90002384","MCV-ebi-a-GCST90002392","MPV-ebi-a-GCST90002395","WBC-ebi-a-GCST90002407","asthma_pmid36778051","MPV-ebi-a-GCST90002395","MPV-ebi-a-GCST90002395","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","RET-ebi-a-GCST90002405","UC-ebi-a-GCST003045","UC-ebi-a-GCST003045","UC-ebi-a-GCST003045","MCV-ebi-a-GCST90002392","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","WBC-ebi-a-GCST90002407","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","MCH-ebi-a-GCST90002390","IBD-ebi-a-GCST003043","MPV-ebi-a-GCST90002395","MCV-ebi-a-GCST90002392","lymphc-ebi-a-GCST90002388","MCV-ebi-a-GCST90002392"],["04910_PC4","04064_PC1","04390_PC5","04714_PC4","00230_PC1","04062_PC2","04971_PC2","04670_PC1","04971_PC1","04211_PC5","04068_PC1","04924_PC3","04390_PC2","04211_PC3","04970_PC3","03010_PC2","04060_PC2","04060_PC2","04213_PC2","04657_PC2","00510_PC2","04510_PC5","04962_PC2","04971_PC4","04625_PC3","04390_PC3","00640_PC1","04144_PC4","03040_PC4","04217_PC4","04730_PC5","04151_PC5","04917_PC2","04380_PC3","03050_PC3","04650_PC3","04668_PC2","04640_PC2","04722_PC4","04015_PC4","04659_PC4","04010_PC4","00630_PC4","03040_PC4","04390_PC4","04110_PC5","04612_PC5","04612_PC1","04060_PC2","04217_PC5","04010_PC5","04929_PC2","04151_PC4","04730_PC1","04612_PC3","04935_PC1","04727_PC2","03008_PC2","04925_PC3","04520_PC4","04014_PC3","04370_PC3","03040_PC3","00071_PC2","03440_PC3","03018_PC3","04550_PC3","04919_PC4","04070_PC2","04911_PC3","04340_PC4","04714_PC4","04972_PC3","00514_PC5","03060_PC3","04216_PC2","04218_PC5","04270_PC4","04918_PC3","04962_PC1","00280_PC3","04010_PC3","04071_PC2","04714_PC4","04072_PC3","04210_PC2","04620_PC5","04068_PC3","04921_PC3","04728_PC4","04918_PC2","00270_PC3","00562_PC5","04910_PC1","04145_PC1","04724_PC4","04261_PC4","04726_PC4","04728_PC5","04912_PC1","03018_PC2","04024_PC1","04725_PC1","04218_PC1","04144_PC4","04150_PC2","04151_PC2","04216_PC3","03013_PC5","04260_PC5","04726_PC3","04918_PC4","04062_PC2","04611_PC4","04022_PC4","04611_PC5","04371_PC5","00480_PC4","04217_PC3","04061_PC2","04146_PC3","04929_PC1","04926_PC4","04215_PC2","04024_PC4"],["04910","04064","04390","04714","00230","04062","04971","04670","04971","04211","04068","04924","04390","04211","04970","03010","04060","04060","04213","04657","00510","04510","04962","04971","04625","04390","00640","04144","03040","04217","04730","04151","04917","04380","03050","04650","04668","04640","04722","04015","04659","04010","00630","03040","04390","04110","04612","04612","04060","04217","04010","04929","04151","04730","04612","04935","04727","03008","04925","04520","04014","04370","03040","00071","03440","03018","04550","04919","04070","04911","04340","04714","04972","00514","03060","04216","04218","04270","04918","04962","00280","04010","04071","04714","04072","04210","04620","04068","04921","04728","04918","00270","00562","04910","04145","04724","04261","04726","04728","04912","03018","04024","04725","04218","04144","04150","04151","04216","03013","04260","04726","04918","04062","04611","04022","04611","04371","00480","04217","04061","04146","04929","04926","04215","04024"],["Insulin signaling pathway","NF-kappa B signaling pathway","Hippo signaling pathway","Thermogenesis","Purine metabolism","Chemokine signaling pathway","Gastric acid secretion","Leukocyte transendothelial migration","Gastric acid secretion","Longevity regulating pathway","FoxO signaling pathway","Renin secretion","Hippo signaling pathway","Longevity regulating pathway","Salivary secretion","Ribosome","Cytokine-cytokine receptor interaction","Cytokine-cytokine receptor interaction","Longevity regulating pathway - multiple species","IL-17 signaling pathway","N-Glycan biosynthesis","Focal adhesion","Vasopressin-regulated water reabsorption","Gastric acid secretion","C-type lectin receptor signaling pathway","Hippo signaling pathway","Propanoate metabolism","Endocytosis","Spliceosome","Necroptosis","Long-term depression","PI3K-Akt signaling pathway","Prolactin signaling pathway","Osteoclast differentiation","Proteasome","Natural killer cell mediated cytotoxicity","TNF signaling pathway","Hematopoietic cell lineage","Neurotrophin signaling pathway","Rap1 signaling pathway","Th17 cell differentiation","MAPK signaling pathway","Glyoxylate and dicarboxylate metabolism","Spliceosome","Hippo signaling pathway","Cell cycle","Antigen processing and presentation","Antigen processing and presentation","Cytokine-cytokine receptor interaction","Necroptosis","MAPK signaling pathway","GnRH secretion","PI3K-Akt signaling pathway","Long-term depression","Antigen processing and presentation","Growth hormone synthesis, secretion and action","GABAergic synapse","Ribosome biogenesis in eukaryotes","Aldosterone synthesis and secretion","Adherens junction","Ras signaling pathway","VEGF signaling pathway","Spliceosome","Fatty acid degradation","Homologous recombination","RNA degradation","Signaling pathways regulating pluripotency of stem cells","Thyroid hormone signaling pathway","Phosphatidylinositol signaling system","Insulin secretion","Hedgehog signaling pathway","Thermogenesis","Pancreatic secretion","Other types of O-glycan biosynthesis","Protein export","Ferroptosis","Cellular senescence","Vascular smooth muscle contraction","Thyroid hormone synthesis","Vasopressin-regulated water reabsorption","Valine, leucine and isoleucine degradation","MAPK signaling pathway","Sphingolipid signaling pathway","Thermogenesis","Phospholipase D signaling pathway","Apoptosis","Toll-like receptor signaling pathway","FoxO signaling pathway","Oxytocin signaling pathway","Dopaminergic synapse","Thyroid hormone synthesis","Cysteine and methionine metabolism","Inositol phosphate metabolism","Insulin signaling pathway","Phagosome","Glutamatergic synapse","Adrenergic signaling in cardiomyocytes","Serotonergic synapse","Dopaminergic synapse","GnRH signaling pathway","RNA degradation","cAMP signaling pathway","Cholinergic synapse","Cellular senescence","Endocytosis","mTOR signaling pathway","PI3K-Akt signaling pathway","Ferroptosis","RNA transport","Cardiac muscle contraction","Serotonergic synapse","Thyroid hormone synthesis","Chemokine signaling pathway","Platelet activation","cGMP-PKG signaling pathway","Platelet activation","Apelin signaling pathway","Glutathione metabolism","Necroptosis","Viral protein interaction with cytokine and cytokine receptor","Peroxisome","GnRH secretion","Relaxin signaling pathway","Apoptosis - multiple species","cAMP signaling pathway"],[8,5,6,9,8,6,5,10,5,9,9,6,16,12,7,5,6,8,7,6,6,7,5,5,6,8,6,11,7,9,11,9,6,5,5,6,5,9,7,14,8,7,7,8,8,8,6,7,5,7,6,8,9,5,9,5,6,7,10,5,8,11,5,14,5,6,5,11,6,6,10,16,6,5,5,21,16,5,7,7,12,5,5,10,5,5,5,7,5,5,6,9,5,16,8,10,5,8,7,5,6,8,7,20,15,8,8,6,5,14,5,6,5,10,6,9,7,8,15,7,6,5,20,9,15],[0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,1,0,0,1,0,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[1e-06,8e-06,6.7e-05,9.4e-05,0.000177,4e-05,5e-05,8.3e-05,0.000189,0.00018,7.1e-05,0.000105,4.3e-05,0.000195,0.00035,0.00045,0.000489,0.000565,0.000221,0.000519,0.000278,0.000709,0.000346,0.000677,0.000595,0.00064,0.0012999,0.0019998,0.0017998,0.0019998,0.0020998,0.0020998,0.001053,0.001153,0.0018998,0.0019998,0.0021998,0.0012999,0.001372,0.0013999,0.001527,0.0021998,0.0018998,0.0016998,0.000436,0.0022998,0.0023998,0.0024998,0.000527,0.0011999,0.0021998,0.0022998,0.0011999,0.0011999,0.0016998,0.0018998,0.0020998,0.0025997,0.0013999,0.0020998,0.0010999,0.0011999,0.0013999,0.001478,0.0017998,0.0023998,0.0024998,0.0025997,0.0010999,0.0019998,0.0023998,0.0012999,0.0012999,0.0020998,0.0025997,0.0019998,0.001328,0.000803,0.0028997,0.0027997,0.0029997,0.0031997,0.0031997,0.0031997,0.0029997,0.0026997,0.0028997,0.0029997,0.0029997,0.0030997,0.0031997,0.0013999,0.0017998,0.0027997,0.0038996,0.0016998,0.0017998,0.0018998,0.0023998,0.0024998,0.0027997,0.0027997,0.0031997,0.0036996,0.0037996,0.0037996,0.0037996,0.0039996,0.0032997,0.0033997,0.0035996,0.0013999,0.0018998,0.0021998,0.0024998,0.0019998,0.0032997,0.0037996,0.0037996,0.0039996,0.0037996,0.0036996,0.0037996,0.0037996,0.0038996],[0.001438,0.014296,0.0559927,0.0559927,0.0790748,0.08832,0.0996,0.0996,0.1134,0.1134,0.129815,0.129815,0.129815,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1386133,0.1565472,0.16608,0.1728284,0.1728284,0.1728284,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1734699,0.1744,0.175102,0.1766253,0.1780503,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1782651,0.1793701,0.18284,0.18284,0.18284,0.18284,0.18284,0.18284,0.18284,0.1862793,0.1892382,0.1892382,0.1892382,0.1901149,0.1909443,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.1931699,0.19824,0.19824,0.19824,0.19824,0.1991285],[137,102,155,231,130,189,75,112,75,89,131,69,155,89,90,153,294,294,62,94,50,199,44,75,104,155,34,249,149,162,60,354,70,128,45,134,112,99,119,210,107,295,30,149,155,124,78,78,294,162,295,64,354,60,78,119,89,105,98,71,232,59,149,44,41,79,140,119,99,86,50,231,102,46,23,40,160,132,74,44,48,295,119,231,148,136,104,131,153,132,74,49,74,137,152,114,149,115,132,93,79,216,112,160,249,153,354,40,179,86,115,74,189,124,167,124,137,56,162,100,83,64,129,32,216],[87,47,41,90,34,109,34,69,34,57,85,30,70,56,34,126,101,101,36,46,39,105,33,28,54,41,29,175,111,93,31,175,42,86,40,79,69,62,89,104,79,162,23,111,75,92,57,57,118,97,168,30,178,30,65,71,35,61,49,44,119,39,112,28,28,67,63,84,67,31,24,130,40,25,20,33,117,55,36,33,39,168,81,130,81,104,63,87,79,76,31,23,29,62,81,36,56,38,56,37,37,71,41,78,111,55,126,22,71,34,38,24,88,70,67,70,56,27,69,38,59,30,69,25,94],["bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","disease_EUR_old","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","disease_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","disease_EUR_old","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","disease_EUR_old","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","disease_EUR_old","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new","bloodcell_EUR_new"],["-0.245270464051887","-0.41768128699344","-0.219352852012402","-0.797167485258106","0.0490457193697609","1.69054931142635","-0.160017285934662","-0.146323318161613","-0.0195316446390257","-0.280029290491469","0.219067049860872","0.0574186991373062","-0.056669509978619","0.0673449539000475","-0.128493235792523","-0.226096753153129","0.233290435680774","0.356193457802385","-0.27071396926541","-0.0983783761147528","0.803780051926971","-0.319389509893942","-0.369079521944207","0.0109467259031775","-0.131943556293009","0.165289903927005","-0.0400493575356977","-0.0133601370358113","0.0345362910612599","0.506296610377211","-0.255109506096205","-0.390877583722192","-0.15053842633599","-0.0830551548728864","-0.417405016126126","-0.32927676223193","-0.46363696898361","-0.643224676525653","0.00675380298541528","-0.206185838472806","-0.348348769804703","-0.0201372579675425","-0.298272254009567","0.38668824064209","0.272708295051201","0.677674795102813","0.807887694724516","0.177658176432276","-0.0495864361159254","-0.0139165601476714","-0.148981738791476","0.201944929329823","-0.167618721127787","0.0900331861451125","0.650380512841018","-0.0640671509178383","0.0601787103416731","0.0178089277936049","0.0915678874980503","0.586873863031775","-0.284482997416838","0.0116704298005137","-0.346552058826978","0.373332837053078","0.138848375442602","0.609834657956357","0.306469104611784","0.341146698851871","0.400944087090976","0.17112589085453","-0.0814124936249649","0.908184857186706","-0.0427098263051655","0.132673406068839","-0.173591571192809","0.186204196970328","-0.226259051798917","-0.0910772003692204","-0.018084141884111","0.384601143357574","-0.0498490589289173","-0.566552792213228","0.124554307321948","-0.537867543275294","0.149081328940241","-0.227886836956831","0.14373227924004","0.0697980284727306","0.00693293256160454","0.246432050632072","0.177562763063755","0.5259942054983","0.560386923702547","0.549513227634144","-0.134040508002528","-0.54428879927892","-0.405271170683785","0.392211512031683","-0.539877365042058","0.13958108952101","0.0584426773316338","0.0441347782270358","0.147938467588175","0.296463052202976","0.22793051722002","-0.0416611070894055","0.0323660276705429","-0.165866865073491","-0.133814949928015","0.156927059687846","0.10608763221134","0.378866542149566","-0.39756014039602","0.0454542778745604","0.486738685474682","-0.659405280342488","-0.227839601219098","0.776218730681591","0.0150066929027473","-0.413012718373063","0.150491568198327","0.355452720582586","-0.0194109476464144","0.175664192729935","-0.0497481840979672"],["0.381444392107266","0.386094674495389","0.606360167370838","0.132433832841616","0.916858027973131","0.793939736663622","0.272526832466116","0.380080136666012","0.878437186997933","0.226486849411033","0.556808502315391","0.919996273137037","0.772695185086866","0.419586506203072","0.696491658215512","0.652614467038651","0.714976796119257","0.553028372038752","0.315566916417708","0.57743030781489","0.213549053167502","0.518149015164986","0.193315258562892","0.981300380495821","0.670683188315133","0.84410453151432","0.502309526364542","0.870945912245638","0.92292114024808","0.341137939355012","0.462912348171951","0.42444858566781","0.649473464532669","0.825651549584431","0.136673185020984","0.346333133638904","0.32468447926906","0.0384225144339296","0.977036848092291","0.375685874442427","0.262400957242923","0.927089640139701","0.481583736109852","0.155265962575607","0.187412601065013","0.0513939859196159","0.275333129922457","0.731412966083133","0.666382327665147","0.937378747053766","0.339101948874325","0.169171895778485","0.398004857735557","0.558453875836253","0.0120705732822415","0.750408941399446","0.749755390391455","0.921748880203971","0.788672156956791","0.130446334859127","0.294726350408341","0.96720958554113","0.234718940896182","0.176864788775393","0.603525221729898","0.0338619202031168","0.263384075532353","0.240923912300581","0.0946000828443417","0.493810098956484","0.791365014934603","0.130430359113488","0.924512828343235","0.749491860203692","0.717262597195071","0.598502916408624","0.695014070209916","0.603283698260543","0.858900654427318","0.0383060556657068","0.902885205156944","0.105475680291036","0.607938158577543","0.220017018873353","0.670031725380746","0.400970678595854","0.573284618882391","0.850217531548008","0.991924642896619","0.397134232450949","0.336094738609318","0.257978185483141","0.234557683466446","0.388446580296467","0.865495007923532","0.231930532914844","0.386464726940903","0.231288726314972","0.258818812447651","0.681373709410221","0.888791317157249","0.906030570268332","0.68939993487819","0.392331848308167","0.58136440984346","0.91726350363736","0.921461676690888","0.660935938097067","0.537255583958351","0.482016680200454","0.632751968554825","0.124962375643065","0.365943356400005","0.946348478214342","0.410045414016918","0.386842509399878","0.698338609981389","0.189242100754593","0.978206000196248","0.185519835265756","0.323319147212933","0.0329821476481283","0.94134129987067","0.696576899217406","0.839044609865298"],[0.0650233,0.0829709,0.000215,0.0024138,0.0388076,0.0350311,0.0009857,0.0415006,0.0355491,0.0131546,0.0776991,0.0925372,0.0313349,0.0655913,0.0442804,0.0273526,0.0664088,0.0452325,0.0145661,0.1063782,1,0.0040001,0.0136082,0.0428586,0.0143981,0.0532078,0.8828099,0.0436977,0.0488894,0.0092223,0.0868603,0.0783865,0.0463272,0.0358707,0.0097742,0.1385439,0.0081822,0.0946554,0.059218,0.0376386,0.0423516,1,0.0173718,0.046462,0.0026897,0.0065683,0.0001711,1,0.0002286,0.0113243,0.1272579,0.0136204,0.0496565,0.0123531,0.0138749,0.0317571,0.002806,0.0182166,0.0311182,0.0046882,0.0077831,0.0725959,0.0649782,0.0574167,0.6487035,0.0285295,0.0699494,0.0133443,0.0318237,0.0579279,0.034016,0.015001,0.0377493,0.0182108,0.0036896,0.0238865,0.0012331,0.248685,0.0507942,0.012418,0.0497995,0.0807982,0.0690528,0.0376975,0.0709253,0.0644015,0.0445163,0.0462309,0.0188757,0.1788443,0.0600319,0.0142615,0.0404826,0.0100246,3.4e-05,0.0002787,0.0001895,0.0052506,0.0141543,0.0161583,0.0768882,0.0571586,0.0876472,0.0375206,1,0.0158862,0.0417202,0.0914741,0.0132481,0.0210123,0.0432386,0.0132213,0.0551095,0.054751,0.0265939,0.0352653,0.0137847,0.0440503,0.0386792,0.0011242,0.0008969,0.0602225,0.0525288,0.0390251,0.0082523],[0.0646544,0.0825434,0.000214,0.0024024,0.0386281,0.0313682,0.0913229,0.0412207,0.0348279,0.0138185,0.0771593,0.0907451,0.0313154,0.0638589,0.043997,0.0271653,0.0528417,0.0381385,0.0144865,0.1061518,1,0.0039809,0.0135827,0.0426652,0.0143427,0.0573591,0.887038,0.0421872,0.0486338,0.0091719,0.1009872,0.0761631,0.0461265,0.0357014,0.0097286,0.1397131,0.0081416,0.0936901,0.0587943,0.0374142,0.0420664,1,0.0172194,0.0465073,0.0026764,0.0065155,0.023752,1,0.0002237,0.0110714,0.1139942,0.0133486,0.0509742,0.0122997,0.0138079,0.0316375,0.0027889,0.0181087,0.0309314,0.0046648,0.0445922,0.0723547,0.064518,0.0565423,0.6289994,0.0292198,0.0696152,0.0132553,0.0315655,0.0574937,0.0337729,0.0150468,0.0378506,0.0180983,0.0131945,0.0237279,0.0012354,0.2742167,0.0497635,0.0123661,0.048638,0.0804363,0.0685495,0.0371428,0.0701786,0.0640073,0.0441792,0.0463672,0.0182766,0.17847,0.0595836,0.0141772,0.040178,0.0099601,3.38e-05,0.0002773,0.0001886,0.0052318,0.0140832,0.0160997,0.0768847,0.056906,0.0871026,0.0373917,1,0.0158038,0.0414849,0.09343,0.0129032,0.0213711,0.0458239,0.0131166,0.0547758,0.0542514,0.026447,0.0353031,0.0137237,0.0440581,0.0385244,0.0011168,0.0008776,0.0613372,0.0520201,0.0387636,0.0081855],[0.5544554,0.4455446,0.009901,0.029703,0.3168317,0.1584158,0.039604,0.2376238,0.2475248,0.0990099,0.6237624,0.3465347,0.3465347,0.4059406,0.1683168,0.1188119,0.3465347,0.2079208,0.039604,0.3663366,0.970297,0.009901,0.1683168,0.1386139,0.1782178,0.2079208,0.9405941,0.3663366,0.2475248,0.1188119,0.2277228,0.5247525,0.1287129,0.3267327,0.1287129,0.5841584,0.049505,0.6435644,0.5247525,0.3267327,0.3366337,1,0.039604,0.2574257,0.039604,0.0891089,0.2178218,1,0.009901,0.0990099,0.7623762,0.0990099,0.7821782,0.1089109,0.1089109,0.3267327,0.049505,0.2277228,0.2079208,0.1089109,0.1485149,0.2277228,0.6831683,0.3762376,0.9108911,0.1584158,0.3960396,0.1683168,0.2673267,0.1782178,0.0891089,0.0990099,0.0990099,0.0594059,0.009901,0.2079208,0.009901,0.6435644,0.4257426,0.1287129,0.3564356,0.6732673,0.5049505,0.2772277,0.4059406,0.5346535,0.1584158,0.5841584,0.1386139,0.6831683,0.3069307,0.0792079,0.1485149,0.0891089,0.009901,0.0594059,0.009901,0.0990099,0.0594059,0.2376238,0.4554455,0.4257426,0.7029703,0.3168317,0.980198,0.2178218,0.6336634,0.4158416,0.1089109,0.0990099,0.1881188,0.1386139,0.5445545,0.3960396,0.1683168,0.1782178,0.0693069,0.1683168,0.2178218,0.029703,0.009901,0.4752475,0.4752475,0.2475248,0.0594059],[0.5346535,0.4356436,0.009901,0.029703,0.3168317,0.1287129,0.3960396,0.2376238,0.2475248,0.0990099,0.6237624,0.3465347,0.3366337,0.3861386,0.1584158,0.1188119,0.2970297,0.1584158,0.029703,0.3663366,0.970297,0.009901,0.1683168,0.1386139,0.1782178,0.2079208,0.9405941,0.3465347,0.2475248,0.1188119,0.2376238,0.5049505,0.1287129,0.3168317,0.1287129,0.5841584,0.049505,0.6336634,0.5049505,0.3267327,0.3267327,1,0.039604,0.2574257,0.039604,0.0891089,0.0594059,1,0.009901,0.0990099,0.7227723,0.0990099,0.7821782,0.1089109,0.1089109,0.3267327,0.049505,0.2277228,0.2079208,0.1089109,0.6039604,0.2277228,0.6831683,0.3663366,0.9108911,0.1683168,0.3861386,0.1683168,0.2673267,0.1782178,0.0891089,0.0990099,0.0990099,0.0594059,0.0594059,0.1881188,0.009901,0.6534653,0.4059406,0.1188119,0.3564356,0.6732673,0.5049505,0.2574257,0.4059406,0.5346535,0.1584158,0.5841584,0.1287129,0.6633663,0.3069307,0.0792079,0.1485149,0.0792079,0.009901,0.0594059,0.009901,0.0990099,0.0594059,0.2376238,0.4554455,0.4257426,0.7029703,0.3069307,0.980198,0.2178218,0.6237624,0.4257426,0.0990099,0.0990099,0.1881188,0.1386139,0.5445545,0.3861386,0.1683168,0.1782178,0.0693069,0.1683168,0.2178218,0.029703,0.009901,0.4851485,0.4851485,0.2475248,0.0594059],[0.2832572,0.4244096,0.0044039,0.0269258,0.254628,0.1065031,0.0089497,0.2015042,0.1775894,0.0917676,0.5025916,0.3758139,0.2109602,0.239223,0.2452307,0.1513399,0.3356563,0.2312773,0.0866195,0.4848734,0.9808451,0.0240076,0.0978216,0.2497611,0.1130334,0.2374262,0.9300858,0.1703398,0.2687421,0.0569375,0.4274075,0.3830823,0.2560988,0.2027736,0.0592941,0.5799335,0.0499919,0.445713,0.3020008,0.1936643,0.2184781,0.9827476,0.0962473,0.2669902,0.0252353,0.0444338,0.0047829,0.9808451,0.002888,0.0676288,0.4643362,0.0793353,0.2968162,0.0866171,0.0961074,0.2001066,0.0238145,0.1219373,0.202122,0.0396136,0.0604474,0.428877,0.3945143,0.357727,0.9008039,0.1934734,0.4168694,0.0967869,0.2216585,0.3639509,0.2352337,0.1119142,0.2550868,0.1345781,0.0357295,0.1568637,0.0107728,0.5958365,0.237365,0.0869877,0.3025228,0.4361511,0.4144987,0.2561525,0.3999506,0.3533849,0.2298794,0.2650212,0.0825032,0.6471287,0.3056904,0.0957725,0.2377415,0.0716742,0.0020591,0.0053004,0.003931,0.0502557,0.1113684,0.1240776,0.4242027,0.3430508,0.4627245,0.2478695,0.9838228,0.1225111,0.2698003,0.4752663,0.0617629,0.0936148,0.1758406,0.1159496,0.3227259,0.321051,0.1726091,0.2079585,0.0891518,0.2522734,0.2250421,0.0189259,0.0082934,0.3483989,0.3364453,0.2357237,0.0655285],[0.2784657,0.4216313,0.004394,0.0268569,0.2531699,0.0972008,0.3708311,0.1987707,0.1729794,0.0952116,0.4946527,0.3672033,0.2105175,0.2318199,0.2350679,0.1445338,0.2677189,0.1917194,0.08148,0.4826723,0.9808994,0.0200696,0.0952348,0.2477907,0.1125014,0.2511108,0.931706,0.1625879,0.2588197,0.0500694,0.4699964,0.3679584,0.24975,0.1965489,0.0547128,0.5774839,0.0455268,0.4409704,0.2991425,0.1913603,0.2158969,0.9827713,0.0898598,0.2613468,0.0226408,0.0395268,0.1233479,0.9808994,0.002139,0.0652273,0.4314716,0.0765168,0.3023529,0.0857677,0.0951267,0.1987089,0.0234052,0.1206168,0.1985699,0.0374592,0.2852488,0.4235643,0.3882673,0.3491633,0.8953522,0.1946008,0.4110625,0.0929645,0.2161528,0.3570938,0.2296507,0.1079451,0.2520521,0.1307712,0.0984934,0.1525162,0.0091245,0.6215998,0.2317048,0.0861847,0.2940286,0.4318191,0.4076494,0.2488707,0.3929545,0.3459586,0.2271003,0.2593778,0.0777823,0.6452126,0.3024362,0.0949646,0.2351898,0.0711422,0.0020197,0.0053004,0.0039014,0.0500488,0.1107477,0.123516,0.4231288,0.3410607,0.4595127,0.2467857,0.9838819,0.1217722,0.2680171,0.4803598,0.0604526,0.0950631,0.1846583,0.115122,0.3201446,0.3174944,0.17122,0.2071605,0.0885509,0.2510123,0.2230224,0.0187387,0.00722,0.3526305,0.3295959,0.2290834,0.0622235],[0.2174599,0.3434217,0.0040564,0.0198934,0.1893318,0.153741,0.0094632,0.2052896,0.180176,0.0764364,0.3744971,0.4216839,0.1761671,0.2680301,0.1902521,0.1220424,0.2708697,0.1940119,0.0681952,0.3892895,0.9911896,0.0222065,0.0786913,0.2056802,0.0821536,0.245055,0.9320428,0.1879381,0.2080315,0.0452248,0.3357594,0.3099569,0.198259,0.157135,0.047656,0.4618508,0.0406736,0.3580625,0.2457721,0.1641882,0.1826878,0.9915307,0.0801693,0.1987697,0.0205547,0.0336215,0.00303,0.9911896,0.0034589,0.0672065,0.4649765,0.078752,0.2379927,0.0724126,0.080039,0.163671,0.0212729,0.1012138,0.160853,0.0320718,0.0488796,0.3203948,0.2946937,0.267504,0.8477691,0.1492509,0.3115883,0.0773604,0.1639595,0.2694322,0.1735832,0.0856094,0.1895302,0.1011868,0.0264438,0.1279833,0.0111987,0.6261945,0.2424477,0.0727369,0.238564,0.3464201,0.3085854,0.1893198,0.3148328,0.2639268,0.1911931,0.197889,0.0866328,0.5319118,0.2486738,0.0814921,0.1961002,0.0610268,0.0013918,0.0047532,0.0037532,0.0362535,0.0809852,0.0904661,0.3251109,0.2593917,0.3569712,0.1840178,0.99418,0.0891885,0.2011472,0.3676138,0.0767031,0.1130407,0.2071902,0.0765714,0.2519909,0.2506752,0.1379422,0.1747671,0.0792396,0.2103738,0.188806,0.0116274,0.0088157,0.2778171,0.2491219,0.1948954,0.051361],[0.2143697,0.3408274,0.0040367,0.0197926,0.1878342,0.1351176,0.3738789,0.2013522,0.1743951,0.0777068,0.3696112,0.4132034,0.1733487,0.258217,0.1853793,0.1177695,0.2189993,0.1624272,0.0646281,0.3847648,0.9915025,0.0193718,0.0765316,0.2040814,0.0816399,0.2590845,0.9335951,0.1782484,0.2032874,0.0419839,0.3713158,0.2989834,0.1936821,0.1527679,0.0444143,0.4603596,0.0375096,0.3513451,0.2403932,0.1595929,0.177791,0.9915826,0.0762244,0.1951437,0.0189952,0.0304804,0.1036659,0.9915025,0.0023395,0.064028,0.4316933,0.0753494,0.2401239,0.0701581,0.0776538,0.1606032,0.0196771,0.0985012,0.157529,0.0303092,0.2150033,0.3161928,0.2898453,0.261147,0.8388315,0.1499114,0.3071393,0.0749074,0.1602849,0.2646751,0.1699469,0.0837638,0.1872655,0.0984478,0.0746098,0.1248859,0.009863,0.6478327,0.2353778,0.0704968,0.2310539,0.3418405,0.3035737,0.1842918,0.3090313,0.2587468,0.1860896,0.1946023,0.0807533,0.5277866,0.2431922,0.0808607,0.1940953,0.0605387,0.0013851,0.004734,0.0037286,0.0360797,0.0804303,0.0898737,0.3238001,0.2574916,0.3539837,0.1827326,0.9939772,0.0885096,0.1993779,0.371488,0.0748696,0.1142306,0.2164159,0.0758759,0.2497921,0.2479108,0.1367523,0.1742355,0.0787196,0.2095377,0.1874043,0.011558,0.0074001,0.2786153,0.2441316,0.1910304,0.049252]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>num_supp_SNP_MHC<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n      <th>acat_p_correct_noMHC_traitnull<\/th>\n      <th>acat_p_correct_globalnull<\/th>\n      <th>acat_p_correct_noMHC_globalnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,11,15,16,17,18,19,20,21,22]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-trait"
class="section level2">
<h2>Number of significant pairs at different cutoffs, for each
trait</h2>
<div id="ect-fdr-0.1" class="section level3">
<h3>ECT FDR &lt; 0.1</h3>
<pre class="r"><code>FDRcutoff &lt;- 0.1
MBEcutoff &lt;- 0.05

p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_summary &lt;- df %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_FDR_3suppSNP = sum(ECT_FDR_5suppSNP &lt; FDRcutoff, na.rm = TRUE)
    ) %&gt;%
    arrange(desc(n_sig_FDR_3suppSNP))
  
  df_summary$trait[df_summary$trait ==&quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  df_summary &lt;- df_summary %&gt;%
    mutate(trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait))
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_FDR_3suppSNP),
                       y = n_sig_FDR_3suppSNP,
                       fill = n_sig_FDR_3suppSNP)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7, color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;lightblue&quot;, high = &quot;steelblue&quot;) +
  labs(
    x = &quot;Trait&quot;,
    y = paste0(&quot;# of pairs at FDR &lt; &quot;, FDRcutoff),
    title = types[celltype]
  ) +
  theme_bw(base_size = 14) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
  )
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-1.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- list()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype, &quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric if it’s stored as character
  df &lt;- df %&gt;%
    mutate(mbe_pval_reverse_pair = suppressWarnings(as.numeric(mbe_pval_reverse_pair)))
  
  # Keep only FDR-significant pairs
  df_sig &lt;- df %&gt;%
    dplyr::filter(ECT_FDR_5suppSNP &lt; FDRcutoff)
  
  # Fix naming if needed
  df_sig$trait[df_sig$trait == &quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  # Summarize counts per trait
  df_summary &lt;- df_sig %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_pairs = n(),
      n_mbe_sig = sum(!is.na(mbe_pval_reverse_pair) &amp; mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    # Keep traits with at least one FDR-significant pair
    dplyr::filter(n_sig_pairs &gt;= 1) %&gt;%
    mutate(
      trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait)
    ) %&gt;%
    arrange(desc(n_sig_pairs))
  
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_pairs))) +
    geom_bar(aes(y = n_sig_pairs), stat = &quot;identity&quot;, fill = &quot;grey80&quot;, color = &quot;black&quot;, width = 0.7) +
    geom_bar(aes(y = n_mbe_sig), stat = &quot;identity&quot;, fill = &quot;darkturquoise&quot;, width = 0.7) +
    geom_text(aes(y = n_mbe_sig, label = n_mbe_sig), vjust = -0.4, size = 3.5) +
    labs(
      x = &quot;Trait&quot;,
      y = paste0(&quot;# of FDR&lt;&quot;, FDRcutoff, &quot; (blue: MBE p&lt;&quot;, MBEcutoff, &quot;)&quot;),
      title = types[celltype]
    ) +
    theme_bw(base_size = 14) +
    theme(
      legend.position = &quot;none&quot;,
      axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
      axis.text.y = element_text(size = 10, color = &quot;black&quot;),
      plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
    )
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2, top = &quot;Number of pairs with MBE p &lt; 0.05, for the sig pairs&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-2.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-3-2.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ect-fdr-0.2" class="section level3">
<h3>ECT FDR &lt; 0.2</h3>
<pre class="r"><code>FDRcutoff &lt;- 0.2
MBEcutoff &lt;- 0.05

p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_summary &lt;- df %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_FDR_3suppSNP = sum(ECT_FDR_5suppSNP &lt; FDRcutoff, na.rm = TRUE)
    ) %&gt;%
    arrange(desc(n_sig_FDR_3suppSNP))
  
  df_summary$trait[df_summary$trait ==&quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  df_summary &lt;- df_summary %&gt;%
    mutate(trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait))
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_FDR_3suppSNP),
                       y = n_sig_FDR_3suppSNP,
                       fill = n_sig_FDR_3suppSNP)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7, color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;lightblue&quot;, high = &quot;steelblue&quot;) +
  labs(
    x = &quot;Trait&quot;,
    y = paste0(&quot;# of pairs at FDR &lt; &quot;, FDRcutoff),
    title = types[celltype]
  ) +
  theme_bw(base_size = 14) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
  )
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-1.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- list()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype, &quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric if it’s stored as character
  df &lt;- df %&gt;%
    mutate(mbe_pval_reverse_pair = suppressWarnings(as.numeric(mbe_pval_reverse_pair)))
  
  # Keep only FDR-significant pairs
  df_sig &lt;- df %&gt;%
    dplyr::filter(ECT_FDR_5suppSNP &lt; FDRcutoff)
  
    # Fix naming if needed
  df_sig$trait[df_sig$trait == &quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  # Summarize counts per trait
  df_summary &lt;- df_sig %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_pairs = n(),
      n_mbe_sig = sum(!is.na(mbe_pval_reverse_pair) &amp; mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    # Keep traits with at least one FDR-significant pair
    dplyr::filter(n_sig_pairs &gt;= 1) %&gt;%
    mutate(
      trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait)
    ) %&gt;%
    arrange(desc(n_sig_pairs))
  

  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_pairs))) +
    geom_bar(aes(y = n_sig_pairs), stat = &quot;identity&quot;, fill = &quot;grey80&quot;, color = &quot;black&quot;, width = 0.7) +
    geom_bar(aes(y = n_mbe_sig), stat = &quot;identity&quot;, fill = &quot;darkturquoise&quot;, width = 0.7) +
    geom_text(aes(y = n_mbe_sig, label = n_mbe_sig), vjust = -0.4, size = 3.5) +
    labs(
      x = &quot;Trait&quot;,
      y = paste0(&quot;# of FDR&lt;&quot;, FDRcutoff, &quot; (blue: MBE p&lt;&quot;, MBEcutoff, &quot;)&quot;),
      title = types[celltype]
    ) +
    theme_bw(base_size = 14) +
    theme(
      legend.position = &quot;none&quot;,
      axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
      axis.text.y = element_text(size = 10, color = &quot;black&quot;),
      plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
    )
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2, top = &quot;Number of pairs with MBE p &lt; 0.05, for the sig pairs&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-2.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-2.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-4-2.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-cell-type"
class="section level2">
<h2>Number of significant pairs at different cutoffs, for each cell
type</h2>
<pre class="r"><code>num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2 &amp; df$ECT_FDR_5suppSNP &gt;= 0.1,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))
  
}

sum &lt;- data.frame(celltype = types,
                  num_fdr01 = num_fdr01,
                  num_fdr02 = num_fdr02)


df &lt;- sum %&gt;%
  pivot_longer(cols = starts_with(&quot;num_fdr&quot;), names_to = &quot;event&quot;, values_to = &quot;total&quot;) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# Plot
p1 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(x = &quot;Cell types&quot;, y = &quot;Count&quot;) +
  ggtitle(&quot;# of pairs&quot;) +
  ylim(0, 60) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, 
                       celltype, &quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric safely (ignore &quot;0/1snp&quot; or NA)
  df$mbe_pval_reverse_pair &lt;- suppressWarnings(as.numeric(df$mbe_pval_reverse_pair))
  df$mbe_pval_reverse_pair[is.na(df$mbe_pval_reverse_pair)] &lt;- 0.5
  
  # --- FDR &lt; 0.1 ---
  df_fdr01 &lt;- df %&gt;%
    dplyr::filter(
      ECT_FDR_5suppSNP &lt; 0.1,
      !(mbe_pval_reverse_pair &lt; MBEcutoff)  # remove reverse-causal pairs
    ) %&gt;%
    dplyr::filter(!is.na(p_ECT))
  
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))
  
  # --- 0.1 &lt; FDR &lt; 0.2 ---
  df_fdr02 &lt;- df %&gt;%
    dplyr::filter(
      ECT_FDR_5suppSNP &lt; 0.2,
      ECT_FDR_5suppSNP &gt;= 0.1,
      !(mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    dplyr::filter(!is.na(p_ECT))
  
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))
}

# summary table
sum &lt;- data.frame(
  celltype = types,
  num_fdr01 = num_fdr01,
  num_fdr02 = num_fdr02
)



df &lt;- sum %&gt;%
  pivot_longer(
    cols = starts_with(&quot;num_fdr&quot;),
    names_to = &quot;event&quot;,
    values_to = &quot;total&quot;
  ) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;,
                                        &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# ---- Plot ----
p2 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(
    x = &quot;Cell types&quot;,
    y = &quot;Count (excluding MBE p &lt; 0.05)&quot;,
    title = &quot;# of pairs after removing reverse-causal signals&quot;
  ) +
  ylim(0, 60) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

all &lt;- grid.arrange(p1, p2, nrow = 1)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-5-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-5-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-5-1.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-5-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="traits-and-pathways-identified" class="section level2">
<h2>Traits and pathways identified</h2>
<div id="ibd-ebi-a-gcst003043" class="section level3">
<h3>IBD-ebi-a-GCST003043</h3>
<pre class="r"><code>trait &lt;- &quot;IBD-ebi-a-GCST003043&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  dplyr::mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-6-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>supp_snps_all &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/supp_snps_ECTFDR02.RDS&quot;)
supp_snps_trait &lt;- supp_snps_all[supp_snps_all$trait == trait,]

supp_snps_trait &lt;- supp_snps_trait[supp_snps_trait$factor %in% df_trait_fdr01$factor,]

DT::datatable(supp_snps_trait,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&quot;supporting snps for &quot;, trait)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-ad41175326dc68f35292" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ad41175326dc68f35292">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">supporting snps for IBD-ebi-a-GCST003043<\/caption>","data":[["604","605","606","607","608","609","610","611","612","613","614","615","616","617","618"],["thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte","thymocyte"],["IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043","IBD-ebi-a-GCST003043"],["04971_PC2","04971_PC2","04971_PC2","04971_PC2","04971_PC2","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1","04670_PC1"],["04971","04971","04971","04971","04971","04670","04670","04670","04670","04670","04670","04670","04670","04670","04670"],["Gastric acid secretion","Gastric acid secretion","Gastric acid secretion","Gastric acid secretion","Gastric acid secretion","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration","Leukocyte transendothelial migration"],["5e-05","5e-05","5e-05","5e-05","5e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05","8.3e-05"],["0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996","0.0996"],["5","5","5","5","5","10","10","10","10","10","10","10","10","10","10"],["rs1388585","rs2024092","rs2497318","rs35256947","rs7738430","rs1292053","rs13107612","rs1388585","rs259964","rs3776414","rs395157","rs648541","rs6740462","rs7097656","rs740287"],["12","19","10","2","6","17","4","12","20","5","5","11","2","10","7"],["40531691","1124031","94432000","231161026","31508836","57963537","102739980","40531691","57824309","10689562","38867732","114429934","65667272","82250831","107584780"],["0.001531","0.005837","0.003773","0.006124","7.166e-06","0.01326","0.0008946","0.001019","0.007458","0.001132","0.0116","0.001247","0.00909","0.001859","0.01426"],["0.1064045","0.1702472","0.1702472","0.1702472","0.000996074","0.198214","0.04333325","0.04333325","0.172777","0.04333325","0.198214","0.04333325","0.180501428571429","0.0516802","0.198214"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pwyID<\/th>\n      <th>pwyNAME<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>Num_supp_snp<\/th>\n      <th>SNP<\/th>\n      <th>chr<\/th>\n      <th>bp<\/th>\n      <th>pval_SNP_factor<\/th>\n      <th>fdr_SNP_factor<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-6-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-3">
Past versions of unnamed-chunk-6-3.png
</button>
</p>
<div id="fig-unnamed-chunk-6-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-6-3.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<table>
<colgroup>
<col width="7%" />
<col width="46%" />
<col width="46%" />
</colgroup>
<thead>
<tr class="header">
<th>Pathway</th>
<th>Role in Inflammatory Bowel Disease (IBD)</th>
<th>Role in T cell (CD8⁺)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Gastric acid secretion</strong></td>
<td><strong>Not a core IBD pathway.</strong> IBD mainly affects
ileum/colon; gastric acid secretion isn’t a canonical pathogenic
pathway, though gastric acidity can indirectly shape microbiota and
upper-GI environment.</td>
<td><strong>Not T-cell–intrinsic.</strong> This pathway is about
parietal cell function; CD8⁺ T cells are not using a “gastric acid
secretion” pathway. Any effect would be indirect via mucosal
environment, not a primary CD8⁺ signaling pathway.</td>
</tr>
<tr class="even">
<td><strong>Leukocyte transendothelial migration</strong></td>
<td><strong>Yes.</strong> Trafficking of leukocytes (including T cells,
monocytes, neutrophils) across gut endothelium is essential for
intestinal inflammation; adhesion molecules and migration pathways are
recognized contributors and therapeutic targets in IBD. (<a
href="https://en.wikipedia.org/wiki/Hyperprolactinaemia?utm_source=chatgpt.com"
title="Hyperprolactinaemia">PMC</a>)</td>
<td><strong>Yes.</strong> CD8⁺ effector/memory T cells must cross
endothelium to enter inflamed tissues; they use the generic leukocyte
transendothelial migration machinery (adhesion molecules, chemokines,
diapedesis). (<a
href="https://en.wikipedia.org/wiki/CD69?utm_source=chatgpt.com"
title="CD69">PubMed</a>)</td>
</tr>
<tr class="odd">
<td><strong>Cytokine–cytokine receptor interaction</strong></td>
<td><strong>Strongly yes.</strong> Dysregulated cytokine networks (TNF,
IL-6, IL-12/23, IL-17, etc.) and their receptors are central to IBD
pathogenesis and are major drug targets (anti-TNF, anti-IL-12/23, etc.).
(<a
href="https://ashpublications.org/blood/article/24/6/726/38119/Effect-of-Prolactin-on-Erythropoiesis-in-the-Mouse?utm_source=chatgpt.com"
title="Effect of Prolactin on Erythropoiesis in the Mouse | Blood">PMC</a>)</td>
<td><strong>Yes.</strong> Cytokines like IL-2, IL-7, IL-12, IL-15,
IL-18, IL-21 drive CD8⁺ T-cell activation, proliferation,
differentiation and survival via their receptors; cytokine–receptor
signaling is fundamental to CD8⁺ biology. (<a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC151869/?utm_source=chatgpt.com"
title="Prolactin modulates the naive B cell repertoire - PMC">Nature</a>)</td>
</tr>
<tr class="even">
<td><strong>Necroptosis</strong></td>
<td><strong>Yes.</strong> Necroptosis of intestinal epithelial and
immune cells contributes to barrier damage and sustained inflammation in
IBD and colitis; necroptosis regulators (RIPK3/MLKL, etc.) are emerging
therapeutic targets. (<a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3429555/?utm_source=chatgpt.com"
title="The ins and outs of human reticulocyte maturation">PMC</a>)</td>
<td><strong>Likely context-dependent role.</strong> Necroptosis can
regulate survival/death of immune cells; there is evidence it shapes
inflammatory responses, but a specific, well-defined CD8⁺
T-cell–intrinsic “necroptosis pathway” is less established than in
epithelium/macrophages.</td>
</tr>
<tr class="odd">
<td><strong>MAPK signaling pathway</strong></td>
<td><strong>Yes.</strong> MAPK cascades (ERK, JNK, p38) integrate stress
and cytokine signals in intestinal immune and epithelial cells,
promoting inflammatory gene expression and influencing cell survival and
proliferation in IBD. (<a
href="https://www.frontiersin.org/journals/immunology/articles/10.3389/fimmu.2022.1017115/full?utm_source=chatgpt.com"
title="Prolactin promotes proliferation of germinal center B cells ...">Nature</a>)</td>
<td><strong>Strongly yes.</strong> ERK/MAPK signaling downstream of TCR
is critical for CD8⁺ T-cell proliferation, survival, lineage decisions
and effector function. (<a
href="https://www.sciencedirect.com/science/article/pii/S2405844023041841?utm_source=chatgpt.com"
title="Interference of B lymphocyte tolerance by prolactin in ...">PMC</a>)</td>
</tr>
<tr class="even">
<td><strong>GnRH secretion</strong></td>
<td><strong>No clear direct role.</strong> GnRH (hypothalamic–pituitary
axis) can modulate immune function via systemic hormones, but “GnRH
secretion” is not a canonical KEGG pathway implicated in IBD
pathogenesis. Any links are indirect/endocrine rather than a core
disease pathway.</td>
<td><strong>Not a core CD8⁺ pathway.</strong> CD8⁺ T cells can be
influenced by endocrine milieu, but they do not rely on a specific “GnRH
secretion” signaling module; effects, if any, are systemic and
indirect.</td>
</tr>
<tr class="odd">
<td><strong>Thyroid hormone synthesis</strong></td>
<td><strong>No clear direct role.</strong> Thyroid disorders can
co-occur with autoimmune diseases, but thyroid hormone synthesis per se
is not a standard mechanistic pathway in IBD.</td>
<td><strong>Not T-cell–intrinsic.</strong> Thyroid hormones can modulate
immunity, but CD8⁺ T cells are not executing a “thyroid hormone
synthesis” pathway; any effect is via hormone signaling, not
synthesis.</td>
</tr>
<tr class="even">
<td><strong>Peroxisome</strong></td>
<td><strong>Yes (emerging).</strong> Peroxisome dysfunction in
intestinal epithelium and immune cells disturbs lipid metabolism and ROS
handling, compromises barrier integrity and host defense, and is
increasingly linked to intestinal inflammation and mucosal healing in
IBD. (<a
href="https://pubmed.ncbi.nlm.nih.gov/21781241/?utm_source=chatgpt.com"
title="Novel platelet activation receptor CLEC-2: from discovery ...">PMC</a>)</td>
<td><strong>Yes (metabolic/immune regulation).</strong> Peroxisomes
regulate fatty-acid oxidation, ROS and lipid signaling in immune cells,
and are now recognized as important metabolic regulators of T-cell
activation and immune synapse function. (<a
href="https://www.sciencedirect.com/science/article/pii/S0753332221012105?utm_source=chatgpt.com"
title="Exploring the antiplatelet activity of serotonin 5-HT 2A ...">PMC</a>)</td>
</tr>
</tbody>
</table>
<pre><code>                                                                      |</code></pre>
</div>
<div id="ret-ebi-a-gcst90002405" class="section level3">
<h3>RET-ebi-a-GCST90002405</h3>
<pre class="r"><code>trait &lt;- &quot;RET-ebi-a-GCST90002405&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/0c70bada0be6ace162a91b0485b3b8431ec9b363/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-2.png" target="_blank">0c70bad</a>
</td>
<td>
XSun
</td>
<td>
2025-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-7-2.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<table>
<colgroup>
<col width="8%" />
<col width="50%" />
<col width="40%" />
</colgroup>
<thead>
<tr class="header">
<th>Pathway</th>
<th>Role in <strong>Reticulocyte count</strong> (erythropoiesis / retic
maturation)</th>
<th>Cell type &amp; role</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Longevity regulating pathway – multiple
species</strong></td>
<td><strong>Indirect.</strong> This KEGG pathway bundles
mTOR/PI3K/FOXO/AMPK etc.; those components clearly regulate HSC aging
and erythropoiesis, but the “longevity” super-pathway itself isn’t a
canonical reticulocyte pathway. (<a
href="https://en.wikipedia.org/wiki/Hyperprolactinaemia?utm_source=chatgpt.com"
title="Hyperprolactinaemia">Wikipedia</a>)</td>
<td><strong>B cell (CD19+): Indirect.</strong> Same story: its mTOR/PI3K
arms are central to B-cell development and function, but “longevity
pathway” is just a meta-grouping. (<a
href="https://en.wikipedia.org/wiki/CD69?utm_source=chatgpt.com"
title="CD69">Wikipedia</a>)</td>
</tr>
<tr class="even">
<td><strong>Prolactin signaling pathway</strong></td>
<td><strong>Indirect but biologically plausible.</strong> Prolactin can
stimulate hematopoiesis and increase erythroid precursors / red cell
mass, but it’s not considered a core reticulocyte-count pathway in
humans. (<a
href="https://ashpublications.org/blood/article/24/6/726/38119/Effect-of-Prolactin-on-Erythropoiesis-in-the-Mouse?utm_source=chatgpt.com"
title="Effect of Prolactin on Erythropoiesis in the Mouse | Blood">ASH
Publications</a>)</td>
<td><strong>B cell (CD19+): Yes (immunomodulatory).</strong> B cells
express prolactin receptor; prolactin promotes B-cell survival, germinal
center B-cell proliferation and alters tolerance, especially in
autoimmunity. (<a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC151869/?utm_source=chatgpt.com"
title="Prolactin modulates the naive B cell repertoire - PMC">PMC</a>)</td>
</tr>
<tr class="odd">
<td><strong>Osteoclast differentiation</strong></td>
<td><strong>Indirect.</strong> Osteoclasts shape the bone marrow niche,
so extreme changes can influence hematopoiesis, but this pathway is not
a standard regulator of reticulocyte count.</td>
<td><strong>B cell (CD19+): Not primary.</strong> Mainly
myeloid/osteoclast focused; B cells are affected only indirectly via
bone/marrow environment.</td>
</tr>
<tr class="even">
<td><strong>Proteasome</strong></td>
<td><strong>Yes (core/housekeeping).</strong> Proteasome-mediated
protein degradation is essential for terminal erythroid differentiation
and reticulocyte maturation (clearance of unneeded proteins); inhibition
can impair maturation and alter reticulocyte output. (<a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3429555/?utm_source=chatgpt.com"
title="The ins and outs of human reticulocyte maturation">PMC</a>)</td>
<td><strong>B cell (CD19+): Yes (housekeeping).</strong> Proteasomes are
critical for antigen processing and plasma-cell function but act as
general machinery rather than a B-cell-specific signaling pathway.</td>
</tr>
<tr class="odd">
<td><strong>Natural killer cell mediated cytotoxicity</strong></td>
<td><strong>Not a primary pathway.</strong> NK effector mechanisms don’t
directly control erythropoiesis/reticulocyte count.</td>
<td><strong>B cell (CD19+): Indirect.</strong> NK–B crosstalk exists via
cytokines, but this KEGG pathway is NK-centric, not B-cell
intrinsic.</td>
</tr>
<tr class="even">
<td><strong>TNF signaling pathway</strong></td>
<td><strong>Yes (negative regulator).</strong> TNF-α and related
inflammatory signals inhibit erythropoiesis and contribute to anemia of
chronic disease, thereby lowering reticulocyte production. (<a
href="https://www.frontiersin.org/journals/immunology/articles/10.3389/fimmu.2022.1017115/full?utm_source=chatgpt.com"
title="Prolactin promotes proliferation of germinal center B cells ...">Frontiers</a>)</td>
<td><strong>B cell (CD19+): Yes (context-dependent).</strong> TNF
signaling affects B-cell survival, germinal center responses and
inflammatory activation, but it’s not uniquely B-cell specific.</td>
</tr>
<tr class="odd">
<td><strong>Hippo signaling pathway</strong></td>
<td><strong>Indirect / emerging.</strong> Hippo controls proliferation
and organ size; there is emerging work on Hippo pathway in hematopoietic
stem/progenitor regulation, but it is not a classic reticulocyte-count
pathway.</td>
<td><strong>Platelet: Indirect/possible.</strong> Some Hippo components
are expressed in megakaryocytes/platelets, but they’re not part of the
core textbook platelet-activation machinery.</td>
</tr>
<tr class="even">
<td><strong>Thermogenesis</strong></td>
<td><strong>Not a primary pathway.</strong> Brown‐fat thermogenesis can
change oxygen demand but there is no clear direct mechanistic link to
reticulocyte count.</td>
<td><strong>Platelet: Not primary.</strong> Platelets participate in
hemostasis, not thermogenesis. Any link would be systemic at best.</td>
</tr>
<tr class="odd">
<td><strong>Purine metabolism</strong></td>
<td><strong>Yes (core metabolic).</strong> Purine synthesis and salvage
are crucial for rapidly dividing erythroid precursors and for RBC
nucleotide balance; defects can cause anemia and impaired
erythropoiesis. (<a
href="https://www.sciencedirect.com/science/article/pii/S2405844023041841?utm_source=chatgpt.com"
title="Interference of B lymphocyte tolerance by prolactin in ...">ScienceDirect</a>)</td>
<td><strong>Platelet: Yes (housekeeping).</strong> Purines (ATP/ADP) are
central to platelet energy metabolism and ADP-mediated aggregation, but
again this is general metabolism rather than platelet-specific
signaling.</td>
</tr>
<tr class="even">
<td><strong>C-type lectin receptor signaling pathway</strong></td>
<td><strong>Not a primary pathway.</strong> C-type lectin receptors are
mostly on immune cells; they are not key in erythroid/reticulocyte
biology.</td>
<td><strong>Platelet: Yes (CLEC-2).</strong> Platelets highly express
the C-type lectin-like receptor CLEC-2, which delivers potent activation
and aggregation signals and contributes to thrombosis. (<a
href="https://pubmed.ncbi.nlm.nih.gov/21781241/?utm_source=chatgpt.com"
title="Novel platelet activation receptor CLEC-2: from discovery ...">PubMed</a>)</td>
</tr>
<tr class="odd">
<td><strong>Glutamatergic synapse</strong></td>
<td><strong>Not a primary pathway.</strong> This is a neuronal synapse
pathway; any glutamate transport in RBCs is minor and not known to
regulate reticulocyte count.</td>
<td><strong>Platelet: Not primary.</strong> Platelets are more clearly
linked to serotonin signaling; glutamatergic synapse signaling is not
central to their function.</td>
</tr>
<tr class="even">
<td><strong>Adrenergic signaling in cardiomyocytes</strong></td>
<td><strong>Indirect (systemic).</strong> β-adrenergic tone can
influence hemodynamics and oxygen demand, which may secondarily modulate
EPO levels, but this cardiomyocyte-focused pathway is not a canonical
reticulocyte-regulator.</td>
<td><strong>Platelet: Indirect/partial overlap.</strong> Platelets do
have adrenergic receptors, but the KEGG “adrenergic signaling in
cardiomyocytes” module is cardiac-specific; only some downstream
cAMP/PKA components overlap with platelet pathways.</td>
</tr>
<tr class="odd">
<td><strong>Serotonergic synapse</strong></td>
<td><strong>Not a primary pathway.</strong> Serotonin can have systemic
vascular effects, but there’s no clear direct link to reticulocyte
production.</td>
<td><strong>Platelet: Yes (amplifier).</strong> Platelets take up
serotonin and signal through 5-HT₂A receptors to amplify aggregation and
thrombosis. (<a
href="https://www.sciencedirect.com/science/article/pii/S0753332221012105?utm_source=chatgpt.com"
title="Exploring the antiplatelet activity of serotonin 5-HT 2A ...">ScienceDirect</a>)</td>
</tr>
<tr class="even">
<td><strong>Dopaminergic synapse</strong></td>
<td><strong>Not a primary pathway.</strong> Mostly neuronal; any effect
on erythropoiesis is indirect.</td>
<td><strong>Platelet: Indirect/minor.</strong> Platelets express some
dopamine receptors, but dopaminergic synapse signaling is not a core
platelet-activation pathway.</td>
</tr>
<tr class="odd">
<td><strong>GnRH signaling pathway</strong></td>
<td><strong>Indirect endocrine.</strong> GnRH primarily regulates
gonadotropins; any effect on erythropoiesis is via sex steroids, not
directly on reticulocytes.</td>
<td><strong>Platelet: Not primary.</strong> No strong evidence that
classical GnRH signaling is important in platelet biology.</td>
</tr>
<tr class="even">
<td><strong>RNA degradation</strong></td>
<td><strong>Yes (housekeeping; relevant to maturation).</strong>
Reticulocytes lose their residual RNA as they mature to erythrocytes;
RNA degradation machinery is part of this maturation process, though
it’s a general cellular function rather than an erythroid-specific
pathway. (<a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3429555/?utm_source=chatgpt.com"
title="The ins and outs of human reticulocyte maturation">PMC</a>)</td>
<td><strong>Platelet: Yes (generic).</strong> Platelets contain residual
RNA and rely on RNA turnover for post-transcriptional regulation, but
again this is general, not platelet-specific.</td>
</tr>
<tr class="odd">
<td><strong>cAMP signaling pathway</strong></td>
<td><strong>Yes (functional, not so much “count”).</strong> cAMP
signaling modulates red-cell deformability and ATP release; it
influences RBC/reticulocyte function more than steady-state reticulocyte
number. (<a
href="https://www.sciencedirect.com/science/article/pii/S0021925820483656?utm_source=chatgpt.com"
title="Hydroxycarbamide Decreases Sickle Reticulocyte ...">ScienceDirect</a>)</td>
<td><strong>Platelet: Yes (core negative regulator).</strong> Increased
cAMP (e.g. via prostacyclin/NO pathways) is a key inhibitory signal that
limits platelet activation and aggregation.</td>
</tr>
<tr class="even">
<td><strong>Cholinergic synapse</strong></td>
<td><strong>Not a primary pathway.</strong> Neuronal, with no strong
evidence for direct reticulocyte control.</td>
<td><strong>Platelet: Not primary.</strong> Some cholinergic components
may be present, but not central to platelet physiology.</td>
</tr>
<tr class="odd">
<td><strong>Cellular senescence</strong></td>
<td><strong>Indirect but important (upstream).</strong> Senescence of
hematopoietic stem/progenitor cells contributes to age-related decline
and skewing of hematopoiesis, which can alter RBC/reticulocyte output,
but this is at the stem-cell level rather than a specific reticulocyte
pathway. (<a
href="https://pubmed.ncbi.nlm.nih.gov/35941268/?utm_source=chatgpt.com"
title="Mechanisms involved in hematopoietic stem cell aging">PubMed</a>)</td>
<td><strong>Platelet: Indirect.</strong> HSC/megakaryocyte aging and
niche senescence change platelet counts and reactivity, but not via a
platelet-specific “senescence” signaling module.</td>
</tr>
<tr class="even">
<td><strong>Endocytosis</strong></td>
<td><strong>Yes (core to maturation).</strong> Reticulocyte maturation
heavily depends on endocytosis and exocytosis for plasma-membrane and
protein remodeling; this is a well-described mechanism of reticulocyte →
erythrocyte maturation. (<a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3429555/?utm_source=chatgpt.com"
title="The ins and outs of human reticulocyte maturation">PMC</a>)</td>
<td><strong>Platelet: Yes (generic).</strong> Endocytosis is involved in
receptor trafficking and granule handling, but as general vesicle
biology rather than a named platelet-signaling pathway.</td>
</tr>
<tr class="odd">
<td><strong>mTOR signaling pathway</strong></td>
<td><strong>Yes (core).</strong> mTORC1/2 signaling is critical for
hematopoietic lineage development and erythropoiesis; perturbation of
mTOR alters erythroid maturation and red-cell output. (<a
href="https://en.wikipedia.org/wiki/Hyperprolactinaemia?utm_source=chatgpt.com"
title="Hyperprolactinaemia">Wikipedia</a>)</td>
<td><strong>Platelet: Yes (core).</strong> Platelets use an mTOR–S6K1
axis to regulate Rac1 activation, spreading, aggregation and thromboxane
generation; mTOR inhibition reduces platelet activation. (<a
href="https://acrjournals.onlinelibrary.wiley.com/doi/abs/10.1002/art.24500?utm_source=chatgpt.com"
title="Prolactin alters the mechanisms of B cell tolerance induction">Acr
Journals</a>)</td>
</tr>
<tr class="even">
<td><strong>PI3K–Akt signaling pathway</strong></td>
<td><strong>Yes (core).</strong> PI3K/Akt is a central growth-factor
pathway in hematopoiesis and cooperates with mTOR in erythroid lineage
development and EPO-responsive progenitors. (<a
href="https://en.wikipedia.org/wiki/Hyperprolactinaemia?utm_source=chatgpt.com"
title="Hyperprolactinaemia">Wikipedia</a>)</td>
<td><strong>Platelet: Yes (core).</strong> PI3K/Akt signaling is
critical for platelet activation, integrin signaling and granule
secretion. (<a
href="https://link.springer.com/article/10.1007/s11102-024-01493-x?utm_source=chatgpt.com"
title="Exploring sex-specific hematological changes and their ...">SpringerLink</a>)</td>
</tr>
<tr class="odd">
<td><strong>Ferroptosis</strong></td>
<td><strong>Yes (emerging but direct).</strong> Ferroptosis
(iron-dependent lipid peroxidation) can damage erythroid cells and is
implicated in anemia and ineffective erythropoiesis (e.g. β-thalassemia,
MDS). (<a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8577445/?utm_source=chatgpt.com"
title="Embryonal erythropoiesis and aging exploit ferroptosis - PMC">PMC</a>)</td>
<td><strong>Platelet: Not a primary pathway.</strong> Ferroptosis
research is mainly in nucleated cells and RBCs; platelets are anucleate
and not a main focus of ferroptosis literature.</td>
</tr>
<tr class="even">
<td><strong>Vasopressin-regulated water reabsorption</strong></td>
<td><strong>Not a primary pathway.</strong> Vasopressin affects plasma
volume and osmolality; it may change hemoglobin concentration
measurements but not reticulocyte production itself.</td>
<td><strong>T cell (CD8+): Indirect at most.</strong> Vasopressin can
have immunomodulatory effects, but CD8⁺ T-cell activation is dominated
by TCR/cytokine pathways, not this renal signaling module.</td>
</tr>
<tr class="odd">
<td><strong>Aldosterone synthesis and secretion</strong></td>
<td><strong>Not a primary pathway.</strong> Aldosterone mainly regulates
salt/water balance and blood pressure; any effect on erythropoiesis is
very indirect.</td>
<td><strong>T cell (CD8+): Indirect.</strong> Mineralocorticoid
signaling can modulate inflammation in some contexts, but not a core
T-cell activation pathway.</td>
</tr>
<tr class="even">
<td><strong>Adherens junction</strong></td>
<td><strong>Indirect.</strong> Cell–cell adhesion in bone-marrow and
spleen niches matters for hematopoiesis, but “adherens junction” as a
KEGG module isn’t a specific reticulocyte regulator.</td>
<td><strong>T cell (CD8+): Indirect/partial.</strong> Some adhesion
molecules also contribute to immunological synapse formation, but the
canonical CD8⁺ pathway is better captured by TCR and integrin signaling
modules.</td>
</tr>
<tr class="odd">
<td><strong>Valine, leucine and isoleucine degradation</strong></td>
<td><strong>Indirect metabolic.</strong> BCAA metabolism supports
cellular energy and can feed into mTOR signaling, but there’s no strong
evidence this pathway specifically controls reticulocyte count.</td>
<td><strong>T cell (CD8+): Indirect but potentially relevant.</strong>
BCAA metabolism contributes to T-cell activation and growth via
nutrient-sensing pathways, but this KEGG degradation pathway is not the
main way we describe CD8⁺ T-cell signaling.</td>
</tr>
<tr class="even">
<td><strong>MAPK signaling pathway</strong></td>
<td><strong>Yes (core signaling).</strong> MAPK cascades are part of
EPO-receptor and growth-factor signaling controlling erythroid
proliferation and survival; they are important but usually discussed
together with JAK–STAT and PI3K/Akt. (<a
href="https://journals.sagepub.com/doi/10.1191/096120301717164930?utm_source=chatgpt.com"
title="Effects of prolactin on hematopoiesis">SAGE Journals</a>)</td>
<td><strong>T cell (CD8+): Yes (core).</strong> TCR engagement activates
Ras–MAPK (ERK, JNK, p38), which is crucial for CD8⁺ T-cell activation,
survival and lineage decisions. (<a
href="https://pubmed.ncbi.nlm.nih.gov/10721006/?utm_source=chatgpt.com"
title="Association between elevated prolactin levels and ...">PubMed</a>)</td>
</tr>
</tbody>
</table>
</div>
<div id="mch-ebi-a-gcst90002390" class="section level3">
<h3>MCH-ebi-a-GCST90002390</h3>
<pre class="r"><code>trait &lt;- &quot;MCH-ebi-a-GCST90002390&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-1.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-2.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-2.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/c81644744954e1781983b8656df97d40ea6925f1/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-8-2.png" target="_blank">c816447</a>
</td>
<td>
XSun
</td>
<td>
2025-11-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<table>
<colgroup>
<col width="19%" />
<col width="41%" />
<col width="38%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Pathway</strong></th>
<th><strong>Role in Mean Corpuscular Hemoglobin (MCH)</strong></th>
<th><strong>Role in Cell Type</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Ribosome</strong></td>
<td>✔ <strong>Yes. Strong role.</strong> Ribosome pathways determine
global protein synthesis, including hemoglobin production; defects
affect MCH.</td>
<td>✔ <strong>Yes (B cell CD19+).</strong> Ribosome activity is
essential for B-cell activation, antibody synthesis, proliferation.</td>
</tr>
<tr class="even">
<td><strong>Long-term depression (LTD)</strong></td>
<td>✖ <strong>No direct role.</strong> LTD is a neuronal synaptic
plasticity pathway; not relevant to erythropoiesis or hemoglobin.</td>
<td>✖ <strong>No (B cell CD19+).</strong> LTD is neuron-specific; does
not function in B-cells.</td>
</tr>
<tr class="odd">
<td><strong>Insulin signaling pathway</strong></td>
<td>✔ <strong>Indirect role.</strong> Insulin affects iron metabolism,
erythropoiesis, and red-cell indices (MCH/MCHC).</td>
<td>✔ <strong>Yes (CD15+ leukocyte).</strong> Insulin signaling
modulates neutrophil metabolism, survival, and inflammatory
response.</td>
</tr>
<tr class="even">
<td><strong>Viral protein interaction with cytokine &amp; cytokine
receptor</strong></td>
<td>✔ <strong>Indirect role.</strong> Anything modulating cytokines can
indirectly alter erythropoiesis during inflammation, but not a core MCH
pathway.</td>
<td>✔ <strong>Yes (Platelet).</strong> Platelets express cytokine
receptors (e.g., IL-6R, TNFR) and respond strongly during viral
infections.</td>
</tr>
</tbody>
</table>
</div>
<div id="mpv-ebi-a-gcst90002395" class="section level3">
<h3>MPV-ebi-a-GCST90002395</h3>
<pre class="r"><code>trait &lt;- &quot;MPV-ebi-a-GCST90002395&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-9-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-9-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-9-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-9-2.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<table>
<colgroup>
<col width="0%" />
<col width="14%" />
<col width="40%" />
<col width="5%" />
<col width="39%" />
</colgroup>
<thead>
<tr class="header">
<th>#</th>
<th>Pathway</th>
<th>Role in Mean Platelet Volume (MPV)</th>
<th>Cell type</th>
<th>Role in that cell type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td><strong>IL-17 signaling pathway</strong></td>
<td><strong>Indirect</strong> – IL-17 is a pro-inflammatory cytokine;
inflammation can affect megakaryopoiesis and platelet indices including
MPV.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>Indirect</strong> – IL-17 shapes inflammatory milieu and
B-cell responses but is not a core B-cell intrinsic pathway.</td>
</tr>
<tr class="even">
<td>2</td>
<td><strong>Hematopoietic cell lineage</strong></td>
<td><strong>Yes</strong> – directly involved in differentiation of
hematopoietic progenitors into megakaryocytes and platelets, thus
influencing MPV.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>Yes</strong> – governs lineage commitment and development of
B-cell precursors.</td>
</tr>
<tr class="odd">
<td>3</td>
<td><strong>Neurotrophin signaling pathway</strong></td>
<td><strong>No clear role</strong> – primarily neuronal; no established
direct regulation of platelet size.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>No clear role</strong> – mainly CNS; only minor or
speculative immune effects.</td>
</tr>
<tr class="even">
<td>4</td>
<td><strong>Rap1 signaling pathway</strong></td>
<td><strong>Indirect</strong> – Rap1 is crucial for integrin activation
in platelets; primarily affects function, but may secondarily relate to
platelet production/turnover.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>Yes</strong> – regulates adhesion, migration, and
immune-synapse–like interactions in lymphocytes.</td>
</tr>
<tr class="odd">
<td>5</td>
<td><strong>Th17 cell differentiation</strong></td>
<td><strong>Indirect</strong> – Th17-driven inflammation can modulate
platelet production and MPV as an inflammatory marker.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>Indirect</strong> – Th17 cells crosstalk with B cells (e.g.,
via IL-17, IL-21), but pathway is T-cell–centric.</td>
</tr>
<tr class="even">
<td>6</td>
<td><strong>MAPK signaling pathway</strong></td>
<td><strong>Yes/Indirect</strong> – MAPK cascades operate in
megakaryocytes and can influence proliferation, differentiation, and
platelet parameters.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>Yes</strong> – key pathway in B-cell activation,
proliferation, and differentiation.</td>
</tr>
<tr class="odd">
<td>7</td>
<td><strong>Toll-like receptor (TLR) signaling pathway</strong></td>
<td><strong>Indirect</strong> – platelet and megakaryocyte TLRs link
inflammation and thrombopoiesis, with MPV often increased in
inflammatory states.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>Yes/Indirect</strong> – B cells express some TLRs; TLR
signaling augments activation and antibody responses.</td>
</tr>
<tr class="even">
<td>8</td>
<td><strong>Dopaminergic synapse</strong></td>
<td><strong>No clear role</strong> – classic pathway is neuronal; no
established direct effect on MPV.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>No clear role</strong> – dopamine can modulate immunity, but
“dopaminergic synapse” per se is not a core B-cell pathway.</td>
</tr>
<tr class="odd">
<td>9</td>
<td><strong>Thyroid hormone synthesis</strong></td>
<td><strong>Indirect</strong> – thyroid status (hypo/hyperthyroidism) is
associated with altered platelet counts/MPV via systemic endocrine
effects.</td>
<td><strong>B cell (CD19+)</strong></td>
<td><strong>Indirect (systemic)</strong> – thyroid hormones affect
immune system broadly, but not specifically B-cell signaling.</td>
</tr>
<tr class="even">
<td>10</td>
<td><strong>Hippo signaling pathway</strong></td>
<td><strong>Indirect</strong> – Hippo controls cell growth and organ
size; possible role in megakaryocyte biology but not a classic MPV
pathway.</td>
<td><strong>Monocyte (CD14+)</strong></td>
<td><strong>Yes/Indirect</strong> – Hippo components modulate
monocyte/macrophage activation and inflammatory responses.</td>
</tr>
<tr class="odd">
<td>11</td>
<td><strong>Longevity regulating pathway</strong></td>
<td><strong>Indirect</strong> – includes nutrient-sensing and
stress-response pathways (e.g., AMPK/FOXO) that may influence
hematopoietic stem cells and platelet production.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>Yes/Indirect</strong> – impacts T-cell survival, metabolism,
and memory formation.</td>
</tr>
<tr class="even">
<td>12</td>
<td><strong>PI3K-Akt signaling pathway</strong></td>
<td><strong>Yes/Indirect</strong> – central in megakaryocyte
survival/proliferation and platelet activation; can affect platelet mass
and MPV.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>Yes</strong> – major pathway for activation, proliferation,
survival, and effector function of CD8+ T cells.</td>
</tr>
<tr class="odd">
<td>13</td>
<td><strong>Long-term depression (LTD)</strong></td>
<td><strong>No clear role</strong> – synaptic plasticity pathway in
neurons; not linked to platelet indices.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>No clear role</strong> – no established direct T-cell
function.</td>
</tr>
<tr class="even">
<td>14</td>
<td><strong>Antigen processing and presentation</strong></td>
<td><strong>No direct role</strong> – primarily an APC process; does not
directly regulate MPV.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>Indirect (critical for activation)</strong> – T cells rely
on this pathway in APCs for antigen presentation, but it is not a T-cell
intrinsic signaling pathway.</td>
</tr>
<tr class="odd">
<td>15</td>
<td><strong>Growth hormone (GH) synthesis, secretion and
action</strong></td>
<td><strong>Indirect</strong> – GH/IGF axis can modulate hematopoiesis
and platelet counts/volume.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>Indirect</strong> – GH receptors on T cells influence
growth, survival, and immune modulation.</td>
</tr>
<tr class="even">
<td>16</td>
<td><strong>GABAergic synapse</strong></td>
<td><strong>No clear role</strong> – classic neurotransmission pathway;
not known to shape MPV directly.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>No clear role</strong> – GABA can affect immune cells, but
the canonical “GABAergic synapse” pathway is not core T-cell
signaling.</td>
</tr>
<tr class="odd">
<td>17</td>
<td><strong>Ribosome biogenesis in eukaryotes</strong></td>
<td><strong>Yes</strong> – ribosome production is essential for protein
synthesis in megakaryocytes and platelet formation, influencing platelet
mass/size.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>Yes</strong> – T-cell activation requires upregulated
ribosome biogenesis for clonal expansion and effector function.</td>
</tr>
<tr class="even">
<td>18</td>
<td><strong>Vasopressin-regulated water reabsorption</strong></td>
<td><strong>No direct role</strong> – regulates renal water handling;
any effect on MPV would be via hemoconcentration/dilution rather than
biology of platelets.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>No clear role</strong> – not a T-cell signaling
pathway.</td>
</tr>
<tr class="odd">
<td>19</td>
<td><strong>GnRH secretion</strong></td>
<td><strong>No clear role</strong> – reproductive axis; no established
direct regulation of MPV.</td>
<td><strong>T cell (CD8+)</strong></td>
<td><strong>No clear role</strong> – GnRH can modulate immunity
hormonally, but the dedicated “GnRH secretion” pathway is not T-cell
intrinsic.</td>
</tr>
</tbody>
</table>
</div>
<div id="t1d-ebi-a-gcst90014023" class="section level3">
<h3>T1D-ebi-a-GCST90014023</h3>
<pre class="r"><code>trait &lt;- &quot;T1D-ebi-a-GCST90014023&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-10-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-10-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-10-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-10-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-10-2.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-10-2.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<table>
<colgroup>
<col width="7%" />
<col width="50%" />
<col width="42%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Pathway</strong></th>
<th><strong>Role in Type 1 Diabetes</strong></th>
<th><strong>Role in B cell (CD19+)</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Longevity regulating pathway</strong></td>
<td><strong>Indirect.</strong> Includes AMPK, FOXO, mTOR, insulin/IGF
signaling — all relevant to metabolic regulation and immune aging. These
pathways influence autoimmunity risk and T1D susceptibility but are
<strong>not a primary T1D pathway</strong>.</td>
<td><strong>Yes/Indirect.</strong> B-cell activation, survival, and
differentiation depend on FOXO, AKT, AMPK, and mTOR pathways included in
this module. Influences antibody production and immune homeostasis.</td>
</tr>
<tr class="even">
<td><strong>Propanoate metabolism</strong></td>
<td><strong>No clear role.</strong> Propionate is a short-chain fatty
acid linked mainly to gut microbiome and metabolic traits. Some evidence
suggests SCFAs modulate autoimmunity, but <strong>not a core T1D
pathway</strong>.</td>
<td><strong>Indirect.</strong> SCFAs (propionate/propanoate) may
influence B-cell function via metabolic and epigenetic effects, but
<strong>not part of canonical B-cell signaling</strong>.</td>
</tr>
<tr class="odd">
<td><strong>Endocytosis</strong></td>
<td><strong>Yes/Indirect.</strong> Antigen uptake and processing by APCs
and β-cell antigen presentation involve endocytosis. Also plays a role
in immune activation and autoantigen exposure in T1D.</td>
<td><strong>Yes.</strong> B cells rely heavily on endocytosis for
antigen uptake through the B-cell receptor (BCR), internalization,
processing, and presentation to T cells. Core B-cell function.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="plots-for-all-pairs-at-ect-fdr-0.1" class="section level2">
<h2>Plots for all pairs at ECT FDR &lt; 0.1</h2>
<div id="plots" class="section level3">
<h3>Plots</h3>
<pre class="r"><code># candidate_pairs &lt;- data.frame(celltype = c(&quot;B_cell&quot;,&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;),
#                               trait = c(&quot;Psoriasis-ebi-a-GCST90038681&quot;,&quot;RA-ukb-a-105&quot;,&quot;SLE-ebi-a-GCST90018917&quot;,&quot;SLE-ebi-a-GCST90018917&quot;,
#                                         &quot;SLE-ebi-a-GCST90018917&quot;,&quot;IBD-ebi-a-GCST003043&quot;,&quot;IBD-ebi-a-GCST003043&quot;),
#                               factor = c(&quot;04062_PC2&quot;,&quot;04071_PC4&quot;,&quot;04120_PC1&quot;,&quot;04015_PC2&quot;,&quot;04022_PC4&quot;,&quot;00230_PC2&quot;,&quot;04924_PC1&quot;)
#                               )

fdr_cutoff &lt;- 0.2
df_all_fdr01$celltype &lt;- celltypes[df_all_fdr01$celltype]

for (i in 1:nrow(df_all_fdr01)){
  
  celltype &lt;- df_all_fdr01$celltype[i]
  trait &lt;- df_all_fdr01$trait[i]
  factor &lt;- df_all_fdr01$factor[i]
  
  file_harmo &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;,celltype, &quot;-&quot;, trait, &quot;_harmo.RDS&quot;)
  dt_harmo &lt;- readRDS(file_harmo)
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EUR_SLEremoved_noMHCadded.RDS&quot;))
  
  dat &lt;- dt_harmo[[factor]]
  snp_select &lt;- dat[dat$FDR.exposure &lt; fdr_cutoff,]
  snp_select$chr &lt;- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr &lt;- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]
  
  pathwayname &lt;- df$pathwayName[df$factor == factor &amp; df$trait ==trait]
  
  eff &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\nECT p-value =&quot;, ect_p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)
  
  
  #snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff &amp; dat$steiger_dir ==T,]
  snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff,]
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  coef &lt;- summary(fit)
  p &lt;- round(coef$coefficients[,4], digits = 4)
  #ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]
  
  eff_rev &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome)) +
    geom_point(size = 4, color =&quot;steelblue&quot;) + 
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\n p-value =&quot;, p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)
  

  
  file_gene_loading &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/gene_loading/&quot;,celltype, &quot;_&quot;, factor, &quot;-&quot;, trait, &quot;.RDS&quot;)
  gene_loadings_pc &lt;- readRDS(file_gene_loading)
  
  gene_loadings_pc_abs &lt;- abs(gene_loadings_pc)
  gene_loadings_pc_abs_sort &lt;- sort(gene_loadings_pc_abs,decreasing = T)
  topgenes &lt;- names(gene_loadings_pc_abs_sort)[1:10]
  
  gene_loadings_pc &lt;- as.data.frame(cbind(names(gene_loadings_pc),gene_loadings_pc))
  names(gene_loadings_pc) &lt;- c(&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)
  
  # Extract gene information based on the gene symbol
  gene_info &lt;- genes(EnsDb.Hsapiens.v75, filter = ~ gene_name ==  gene_loadings_pc$gene_name)
  gene_info &lt;- as.data.frame(gene_info)
  gene_info &lt;- gene_info[!duplicated(gene_info$gene_name),]
  
  gene_info &lt;- merge(gene_info,gene_loadings_pc, by=&quot;gene_name&quot;)
  gene_info &lt;- gene_info[gene_info$seqnames %in%c(1:22),]
  
  loading_plot &lt;- gene_info[,c(&quot;seqnames&quot;,&quot;start&quot;,&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)]
  colnames(loading_plot) &lt;- c(&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;gene_name&quot;,&quot;loadings&quot;)
  loading_plot$loadings &lt;- round(as.numeric(loading_plot$loadings),digits = 4)
  
  don &lt;- loading_plot %&gt;%
    
    # Compute chromosome size
    group_by(chromosome_name) %&gt;%
    summarise(chr_len=max(start_position)) %&gt;%
    
    # Calculate cumulative position of each chromosome
    mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %&gt;%
    dplyr::select(-chr_len) %&gt;%
    
    # Add this info to the initial dataset
    left_join(loading_plot, ., by=c(&quot;chromosome_name&quot;=&quot;chromosome_name&quot;)) %&gt;%
    
    # Add a cumulative position of each SNP
    arrange(chromosome_name, start_position) %&gt;%
    mutate( start_positioncum=start_position+tot) %&gt;%
    
    # Add highlight and annotation information
    #mutate( is_annotate=ifelse(gene_name %in% loading_plot_top10$gene_name, &quot;yes&quot;, &quot;no&quot;))
    mutate( is_annotate=ifelse(gene_name %in% topgenes, &quot;yes&quot;, &quot;no&quot;))
  # Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
  axisdf = don %&gt;% group_by(chromosome_name) %&gt;% summarize(center=( max(start_positioncum) + min(start_positioncum) ) / 2 )
  
  manhplot_gene &lt;- ggplot(don, aes(x=start_positioncum, y=loadings)) +
    
    # Show all points
    #rasterise(geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3),dpi=80 )+
    geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3) +
    scale_color_manual(values = rep(c(&quot;red&quot;, &quot;steelblue&quot;), 22)) +
    
    # custom X axis:
    scale_x_continuous( label = as.integer(axisdf$chromosome_name), breaks= axisdf$center ) +
    #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
    
    #ylab(expression(paste(&quot;-log&quot;[10],&quot;(p-value)&quot;))) +
    #ylim(0,max(-log10(don$loadings)) + 2) +
    #ylim(min(as.numeric(don$loadings)),max(as.numeric(don$loadings))) +
    #ylab(&quot;&quot;) +
    ylab(&quot;Gene Loadings&quot;) +
    xlab(&quot;Chromosome&quot;) +
    
    # Custom the theme:
    theme_bw() +
    #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
    theme(
      legend.position=&quot;none&quot;,
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank(),
      axis.title.x = element_text(size = 18),
      axis.text.x = element_text(size = 16, color = &quot;black&quot;),
      axis.title.y = element_text(size = 18),
      axis.text.y = element_text(size = 16, color = &quot;black&quot;),
      text= element_text(family=&quot;Times&quot;),
      #axis.text.y=element_blank(),
      #axis.ticks.y=element_blank()
    ) +
    geom_label_repel( data=subset(don, is_annotate==&quot;yes&quot;), aes(label=gene_name), size=4)
  
    
    cat(paste0(celltype,&quot;-&quot;, pathwayname, &quot;-&quot;, trait ))
  
  
    all &lt;- grid.arrange(eff, eff_rev, nrow = 1)
    
    print(manhplot_gene)
    print(paste0(&quot;Top 10 genes with largest abs loading: &quot;, paste0(topgenes, collapse = &quot;,&quot;)))
  
}</code></pre>
<pre><code>CD15_positive_leukocyte-Insulin signaling pathway-MCH-ebi-a-GCST90002390</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-2.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-2.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: IKBKB,TSC1,CBL,PHKA2,SOCS4,PHKB,MAPK3,RPS6,RAF1,INPPL1&quot;
platelet-NF-kappa B signaling pathway-HB-ebi-a-GCST90002384</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-3">
Past versions of unnamed-chunk-11-3.png
</button>
</p>
<div id="fig-unnamed-chunk-11-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-3.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-4">
Past versions of unnamed-chunk-11-4.png
</button>
</p>
<div id="fig-unnamed-chunk-11-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-4.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: LTB,LY96,TNFRSF1A,PLCG2,CD14,NFKBIA,NFKB1,PTGS2,CFLAR,PARP1&quot;
platelet-Hippo signaling pathway-RET-ebi-a-GCST90002405</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-5">
Past versions of unnamed-chunk-11-5.png
</button>
</p>
<div id="fig-unnamed-chunk-11-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-5.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-6">
Past versions of unnamed-chunk-11-6.png
</button>
</p>
<div id="fig-unnamed-chunk-11-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-6.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: ACTB,BIRC2,CTNNB1,FZD7,TEAD4,FZD1,PPP1CA,PPP2R2D,PPP2R2B,NF2&quot;
platelet-Thermogenesis-RET-ebi-a-GCST90002405</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-7">
Past versions of unnamed-chunk-11-7.png
</button>
</p>
<div id="fig-unnamed-chunk-11-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-7.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-8">
Past versions of unnamed-chunk-11-8.png
</button>
</p>
<div id="fig-unnamed-chunk-11-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-8.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: ACTG1,PRKAA1,GRB2,COX4I1,GNAS,KDM3B,PRKACB,SLC25A20,MGLL,NDUFB9&quot;
platelet-Purine metabolism-RET-ebi-a-GCST90002405</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-9">
Past versions of unnamed-chunk-11-9.png
</button>
</p>
<div id="fig-unnamed-chunk-11-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-9.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-10">
Past versions of unnamed-chunk-11-10.png
</button>
</p>
<div id="fig-unnamed-chunk-11-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-10.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: NT5C2,AK3,NT5C,PDE4D,PDE5A,ADCY3,DGUOK,NME1-NME2,RRM2,PAPSS1&quot;
B_cell-Chemokine signaling pathway-Psoriasis-ebi-a-GCST90038681</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-11">
Past versions of unnamed-chunk-11-11.png
</button>
</p>
<div id="fig-unnamed-chunk-11-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-11.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-12">
Past versions of unnamed-chunk-11-12.png
</button>
</p>
<div id="fig-unnamed-chunk-11-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-12.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3R2,RHOA,PTK2,PRKCD,DOCK2,NFKB1,ELMO1,IKBKG,VAV2,CDC42&quot;
thymocyte-Gastric acid secretion-IBD-ebi-a-GCST003043</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-13">
Past versions of unnamed-chunk-11-13.png
</button>
</p>
<div id="fig-unnamed-chunk-11-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-13.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-14">
Past versions of unnamed-chunk-11-14.png
</button>
</p>
<div id="fig-unnamed-chunk-11-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-14.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PLCB1,CALM2,ATP1B3,KCNJ10,GNAI3,ITPR1,PRKCA,CALM3,ATP1A1,ACTB&quot;
thymocyte-Leukocyte transendothelial migration-IBD-ebi-a-GCST003043</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-15">
Past versions of unnamed-chunk-11-15.png
</button>
</p>
<div id="fig-unnamed-chunk-11-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-15.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-16">
Past versions of unnamed-chunk-11-16.png
</button>
</p>
<div id="fig-unnamed-chunk-11-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-11-16.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: CTNNA1,ACTB,PIK3R2,ITGAL,ACTN4,ITGB2,RAC2,MSN,MAPK13,SIPA1&quot;</code></pre>
</div>
<div id="gene-function-on-the-cell-types-nots-from-chatgpt"
class="section level3">
<h3>Gene function on the cell types , nots from chatgpt</h3>
<ul>
<li>B cell (CD19+), Psoriasis-ebi-a-GCST90038681, 04062_PC2, Chemokine
signaling pathway</li>
</ul>
<table>
<colgroup>
<col width="4%" />
<col width="67%" />
<col width="27%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Gene</th>
<th align="center">Evidence for B-cell activation involvement</th>
<th align="center">Notes / uncertainties</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">IKBKB(IKKβ)</td>
<td align="center">There is strong evidence: A nonsense mutation in
IKBKB caused absence of IKKβ and a lack of T- and B-cell activation
through their antigen receptors. (PMC) Also, canonical NF-κB pathway
downstream of BCR or CD40 in B cells involves IKK complex. (JACI
Online)</td>
<td align="center">This is a good “hit” for B-cell activation.</td>
</tr>
<tr class="even">
<td align="center">RAF1 (C-RAF)</td>
<td align="center">Evidence that RAF family (including RAF-1) are
involved in B cell activation/differentiation: e.g., a thesis shows that
B-cell specific knockout of B-Raf and Raf-1 impacted B cell activation,
expansion/plasma cell differentiation. (LMU Munich Theses) However, the
study indicates that in certain mature B-cells, ERK phosphorylation
could happen independent of Raf-1. (Science)</td>
<td align="center">So RAF1 is involved, but perhaps not strictly
essential in all contexts. It likely participates in the
BCR→Ras→Raf→MEK→ERK axis in B cells.</td>
</tr>
<tr class="odd">
<td align="center">MAPK3(ERK1)</td>
<td align="center">More general MAPK/ERK signalling is known to be
important in B cell activation: e.g., BCR-induced cell spreading is
inhibited by ERK inhibitors. (MicroPublication) Because MAPK3 is a core
ERK1, it likely participates downstream of B-cell activation
signals.</td>
<td align="center">Good candidate to include.</td>
</tr>
<tr class="even">
<td align="center">INPPL1(SHIP2)</td>
<td align="center">Some relevant evidence: INPPL1 expression is
annotated for “B-cell function” in a cell‐type summary. (Human Protein
Atlas) Also SHIP family (primarily SHIP1) are established regulators of
BCR signalling (often negative regulators) via PI3K signalling in B
cells. Though direct literature for SHIP2 (INPPL1) in B cell activation
is sparse.</td>
<td align="center">Likely involved (perhaps via regulatory/inhibitory
signalling) but evidence is modest.</td>
</tr>
<tr class="odd">
<td align="center">RPS6(Ribosomal protein S6)</td>
<td align="center">There is evidence for involvement in translation
regulation in immune cells, but I did not identify a direct paper
linking RPS6 to B-cell activation signalling specifically.</td>
<td align="center">Probably a very indirect player (translation/protein
synthesis) rather than direct activation signalling; you may decide to
exclude or mark as secondary.</td>
</tr>
</tbody>
</table>
<ul>
<li>Platelet, HB-ebi-a-GCST90002384, 04064_PC1, NF-kappa B signaling
pathway</li>
</ul>
<table>
<colgroup>
<col width="5%" />
<col width="68%" />
<col width="26%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Gene</th>
<th align="center">Evidence for involvement in platelet activation</th>
<th align="center">Notes / uncertainties</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">PLCG2(Phospholipase C γ2)</td>
<td align="center">Strong evidence: The literature shows that platelet
activation (especially via collagen receptor GP VI or integrins)
involves activation of PLCγ2. (PubMed) For example: “Agonist‐induced
platelet activation … involves different signalling pathways leading to
the activation of phospholipase C … γ2.” (PubMed)</td>
<td align="center">This is one of the clearest “hits” on your list for
platelet activation.</td>
</tr>
<tr class="even">
<td align="center">NFKB1 (NF-κB p50 subunit)</td>
<td align="center">Yes: There is evidence that NF-κB family members in
platelets (despite them being anuclear) are engaged in platelet
activation/priming and aggregation. (PMC) For example: “NF-κB inhibitors
prevent platelet activation following stimulation with several agonists,
suggesting that NF‐κB in platelets functions rather non-genomically.”
(PMC)</td>
<td align="center">Good candidate. Although the mechanistic depth may
vary, the role is plausible.</td>
</tr>
<tr class="odd">
<td align="center">NFKBIA (IκBα)</td>
<td align="center">Yes: Since IκBα is the inhibitor of NF-κB, and NF-κB
signalling is present and functional in platelets (see above), there is
support for involvement of the pathway including IκBα in platelet
activation. (PubMed)</td>
<td align="center">While the direct role of IκBα specifically in
platelet activation may be less studied, given the pathway it is likely
involved.</td>
</tr>
<tr class="even">
<td align="center">TNFRSF1A(TNF-receptor 1)</td>
<td align="center">Moderate evidence: There is study evidence that
platelets express TNF receptors (including TNFR1) and that TNF-α can
trigger platelet activation/aggregation. (PubMed) For example: “TNF-α
behaves as a trigger of platelet activation through stimulation of the
arachidonic acid pathway.” (PubMed)</td>
<td align="center">Good candidate for inclusion as a regulator/modulator
of platelet activation (rather than core immediate signalling).</td>
</tr>
</tbody>
</table>
<ul>
<li>T cell (CD8+), IBD-ebi-a-GCST003043, 04670_PC1, Leukocyte
transendothelial migration</li>
</ul>
<table>
<colgroup>
<col width="5%" />
<col width="68%" />
<col width="26%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Gene</th>
<th align="center">Evidence for involvement in platelet activation</th>
<th align="center">Notes / uncertainties</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">PLCG2(Phospholipase C γ2)</td>
<td align="center">Strong evidence: The literature shows that platelet
activation (especially via collagen receptor GP VI or integrins)
involves activation of PLCγ2. (PubMed) For example: “Agonist‐induced
platelet activation … involves different signalling pathways leading to
the activation of phospholipase C … γ2.” (PubMed)</td>
<td align="center">This is one of the clearest “hits” on your list for
platelet activation.</td>
</tr>
<tr class="even">
<td align="center">NFKB1 (NF-κB p50 subunit)</td>
<td align="center">Yes: There is evidence that NF-κB family members in
platelets (despite them being anuclear) are engaged in platelet
activation/priming and aggregation. (PMC) For example: “NF-κB inhibitors
prevent platelet activation following stimulation with several agonists,
suggesting that NF‐κB in platelets functions rather non-genomically.”
(PMC)</td>
<td align="center">Good candidate. Although the mechanistic depth may
vary, the role is plausible.</td>
</tr>
<tr class="odd">
<td align="center">NFKBIA (IκBα)</td>
<td align="center">Yes: Since IκBα is the inhibitor of NF-κB, and NF-κB
signalling is present and functional in platelets (see above), there is
support for involvement of the pathway including IκBα in platelet
activation. (PubMed)</td>
<td align="center">While the direct role of IκBα specifically in
platelet activation may be less studied, given the pathway it is likely
involved.</td>
</tr>
<tr class="even">
<td align="center">TNFRSF1A(TNF-receptor 1)</td>
<td align="center">Moderate evidence: There is study evidence that
platelets express TNF receptors (including TNFR1) and that TNF-α can
trigger platelet activation/aggregation. (PubMed) For example: “TNF-α
behaves as a trigger of platelet activation through stimulation of the
arachidonic acid pathway.” (PubMed)</td>
<td align="center">Good candidate for inclusion as a regulator/modulator
of platelet activation (rather than core immediate signalling).</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="eas" class="section level1">
<h1>EAS</h1>
<div id="detailed-results-for-all-pairs-at-ect-fdr-0.2-1"
class="section level2">
<h2>Detailed results for all pairs at ECT FDR &lt; 0.2</h2>
<pre class="r"><code>df_all_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  df &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2,]
  df_all_fdr02 &lt;- rbind(df_all_fdr02,df_fdr02)
  
}


df_all_fdr02$celltype &lt;- types[df_all_fdr02$celltype]
df_all_fdr02 &lt;- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP),]

df_all_fdr01 &lt;- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP &lt; 0.1,]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Pairs with ECT FDR &lt;= 0.2 &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-66aaeb19fe992dfb51c2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-66aaeb19fe992dfb51c2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Pairs with ECT FDR &lt;= 0.2 <\/caption>","data":[["31","42","5","61","7","10","11","12","14","15","2","17","18","19","20","3","4","23","41","25","6","27","28","30","32","34","38","43","44","45","46","50","9"],["Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Monocyte (CD14+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","B cell (CD19+)","B cell (CD19+)","Granulocyte (CD15+)","Monocyte (CD14+)","Granulocyte (CD15+)","B cell (CD19+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","Granulocyte (CD15+)","B cell (CD19+)"],["height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","ATD-bbj-a-90","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","BMI-ebi-a-GCST90018727","MCV-ebi-a-GCST90018746","height-ebi-a-GCST90018739","BMI-ebi-a-GCST90018727","ATD-bbj-a-90","height-ebi-a-GCST90018739","MCV-ebi-a-GCST90018746","MCV-ebi-a-GCST90018746","height-ebi-a-GCST90018739","MCV-ebi-a-GCST90018746","height-ebi-a-GCST90018739","MCV-ebi-a-GCST90018746","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","height-ebi-a-GCST90018739","ATD-bbj-a-90","RA-bbj-a-151","height-ebi-a-GCST90018739","BMI-ebi-a-GCST90018727","ATD-bbj-a-90","WBC-ebi-a-GCST90018758"],["05167_PC3","05161_PC3","04935_PC3","05200_PC3","05165_PC3","03015_PC3","04972_PC4","04726_PC4","04550_PC2","04360_PC3","04931_PC1","05163_PC3","05016_PC5","04070_PC3","04360_PC3","04611_PC3","04810_PC4","00513_PC3","05200_PC1","05131_PC3","04921_PC4","04630_PC2","04150_PC3","04261_PC3","04730_PC4","04110_PC3","05226_PC3","04910_PC3","04750_PC3","04151_PC4","04151_PC4","04150_PC3","00270_PC2"],["05167","05161","04935","05200","05165","03015","04972","04726","04550","04360","04931","05163","05016","04070","04360","04611","04810","00513","05200","05131","04921","04630","04150","04261","04730","04110","05226","04910","04750","04151","04151","04150","00270"],["Kaposi sarcoma-associated herpesvirus infection","Hepatitis B","Growth hormone synthesis, secretion and action","Pathways in cancer","Human papillomavirus infection","mRNA surveillance pathway","Pancreatic secretion","Serotonergic synapse","Signaling pathways regulating pluripotency of stem cells","Axon guidance","Insulin resistance","Human cytomegalovirus infection","Huntington disease","Phosphatidylinositol signaling system","Axon guidance","Platelet activation","Regulation of actin cytoskeleton","Various types of N-glycan biosynthesis","Pathways in cancer","Shigellosis","Oxytocin signaling pathway","JAK-STAT signaling pathway","mTOR signaling pathway","Adrenergic signaling in cardiomyocytes","Long-term depression","Cell cycle","Gastric cancer","Insulin signaling pathway","Inflammatory mediator regulation of TRP channels","PI3K-Akt signaling pathway","PI3K-Akt signaling pathway","mTOR signaling pathway","Cysteine and methionine metabolism"],[14,6,8,7,12,5,6,6,11,5,7,7,5,5,6,8,5,11,5,11,6,7,6,5,8,9,9,5,5,12,10,5,9],[1e-05,1.3e-05,1.4e-05,1.5e-05,0.000215,0.000337,0.000369,0.00039,0.000672,0.000689,3.1e-05,0.0010999,0.0011999,0.0012999,0.0012999,5.6e-05,0.000107,0.0020998,0.000106,0.0025997,0.000285,0.0037996,0.0038996,0.0043996,0.0044996,0.0049995,0.0063994,0.0077992,0.0077992,0.0078992,0.0081992,0.0089991,0.000762],[0.00222,0.00222,0.00222,0.00222,0.025456,0.02886,0.02886,0.02886,0.0407888,0.0407888,0.049445,0.0549672,0.0549672,0.0549672,0.0549672,0.056068,0.056068,0.0828721,0.084535,0.0961889,0.09956,0.1282535,0.1282535,0.1331882,0.1331882,0.1409383,0.172202,0.1866895,0.1866895,0.1866895,0.1866895,0.1973136,0.199644],[186,162,119,531,330,91,102,115,140,181,108,225,264,99,181,124,213,39,531,236,153,162,153,149,60,124,149,137,100,354,354,153,49],[119,105,65,255,174,64,35,39,51,74,69,136,146,63,74,71,120,24,290,150,79,73,81,70,28,69,60,87,47,156,156,81,32],["disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","bloodcell_EAS_new","disease_EAS","disease_EAS","disease_EAS","disease_EAS","bloodcell_EAS_new","bloodcell_EAS_new","disease_EAS","bloodcell_EAS_new","disease_EAS","bloodcell_EAS_new","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","disease_EAS","bloodcell_EAS_new"],["-0.335320189440536","0.620471254657609","0.774097800142423","-0.924999067074685","-0.628082585903604","0/1snp","-0.0331521299069109","0.0320183424815683","0.012336281274246","-0.523110577590398","-0.0780203996195739","0.382195178073134","0.568640271452703","0/1snp","-0.312450227158299","-0.920969134117914","0.320301620905627","0.302813271864685","0.0289598358094416","-0.140027545018815","-0.77540583139736","-0.0676601639868393","0.86014429611751","-0.579958541338593","0.421358059436136","0.828820548905881","-0.763514895581253","0/1snp","0.0504123419360684","-0.395194415278412","0.380136725276964","0/1snp","0.431538751405459"],["0.696464692880394","0.430496336025231","0.359173780334774","0.296237509071482","0.474956008235431","0/1snp","0.965903671216125","0.968750524709777","0.987883789277398","0.548349402716188","0.784298192893093","0.661729006909189","0.675589994882587","0/1snp","0.700367394570114","0.107689727067654","0.479354160984548","0.672074254887795","0.899252620378613","0.88745417970635","0.0783774426584705","0.933647714663954","0.312849596924828","0.585654893451985","0.523797160148899","0.298773729511107","0.367999370227373","0/1snp","0.595598480652005","0.60041055366391","0.807294644487019","0/1snp","0.642968949046111"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n      <th>trait_category<\/th>\n      <th>mbe_reverse_pair<\/th>\n      <th>mbe_pval_reverse_pair<\/th>\n      <th>acat_p<\/th>\n      <th>acat_p_MHCremoved<\/th>\n      <th>acat_p_correct_pairnull<\/th>\n      <th>acat_p_correct_noMHC_pairnull<\/th>\n      <th>acat_p_correct_traitnull<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[6,7,8,9,10,14,15,16,17,18]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-trait-1"
class="section level2">
<h2>Number of significant pairs at different cutoffs, for each
trait</h2>
<div id="ect-fdr-0.1-1" class="section level3">
<h3>ECT FDR &lt; 0.1</h3>
<pre class="r"><code>FDRcutoff &lt;- 0.1
MBEcutoff &lt;- 0.05

p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_summary &lt;- df %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_FDR_3suppSNP = sum(ECT_FDR_5suppSNP &lt; FDRcutoff, na.rm = TRUE)
    ) %&gt;%
    arrange(desc(n_sig_FDR_3suppSNP))
  
  df_summary$trait[df_summary$trait ==&quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  df_summary &lt;- df_summary %&gt;%
    mutate(trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait))
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_FDR_3suppSNP),
                       y = n_sig_FDR_3suppSNP,
                       fill = n_sig_FDR_3suppSNP)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7, color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;lightblue&quot;, high = &quot;steelblue&quot;) +
  labs(
    x = &quot;Trait&quot;,
    y = paste0(&quot;# of pairs at FDR &lt; &quot;, FDRcutoff),
    title = types[celltype]
  ) +
  theme_bw(base_size = 14) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
  )
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- list()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype, &quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric if it’s stored as character
  df &lt;- df %&gt;%
    mutate(mbe_pval_reverse_pair = suppressWarnings(as.numeric(mbe_pval_reverse_pair)))
  
  # Keep only FDR-significant pairs
  df_sig &lt;- df %&gt;%
    dplyr::filter(ECT_FDR_5suppSNP &lt; FDRcutoff)
  
  # Fix naming if needed
  df_sig$trait[df_sig$trait == &quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  # Summarize counts per trait
  df_summary &lt;- df_sig %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_pairs = n(),
      n_mbe_sig = sum(!is.na(mbe_pval_reverse_pair) &amp; mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    # Keep traits with at least one FDR-significant pair
    dplyr::filter(n_sig_pairs &gt;= 1) %&gt;%
    mutate(
      trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait)
    ) %&gt;%
    arrange(desc(n_sig_pairs))
  
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_pairs))) +
    geom_bar(aes(y = n_sig_pairs), stat = &quot;identity&quot;, fill = &quot;grey80&quot;, color = &quot;black&quot;, width = 0.7) +
    geom_bar(aes(y = n_mbe_sig), stat = &quot;identity&quot;, fill = &quot;darkturquoise&quot;, width = 0.7) +
    geom_text(aes(y = n_mbe_sig, label = n_mbe_sig), vjust = -0.4, size = 3.5) +
    labs(
      x = &quot;Trait&quot;,
      y = paste0(&quot;# of FDR&lt;&quot;, FDRcutoff, &quot; (blue: MBE p&lt;&quot;, MBEcutoff, &quot;)&quot;),
      title = types[celltype]
    ) +
    theme_bw(base_size = 14) +
    theme(
      legend.position = &quot;none&quot;,
      axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
      axis.text.y = element_text(size = 10, color = &quot;black&quot;),
      plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
    )
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2, top = &quot;Number of pairs with MBE p &lt; 0.05, for the sig pairs&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-2">
Past versions of unnamed-chunk-13-2.png
</button>
</p>
<div id="fig-unnamed-chunk-13-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-2.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-13-2.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ect-fdr-0.2-1" class="section level3">
<h3>ECT FDR &lt; 0.2</h3>
<pre class="r"><code>FDRcutoff &lt;- 0.2
MBEcutoff &lt;- 0.05

p &lt;- list()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_summary &lt;- df %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_FDR_3suppSNP = sum(ECT_FDR_5suppSNP &lt; FDRcutoff, na.rm = TRUE)
    ) %&gt;%
    arrange(desc(n_sig_FDR_3suppSNP))
  
  df_summary$trait[df_summary$trait ==&quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  df_summary &lt;- df_summary %&gt;%
    mutate(trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait))
  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_FDR_3suppSNP),
                       y = n_sig_FDR_3suppSNP,
                       fill = n_sig_FDR_3suppSNP)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7, color = &quot;black&quot;) +
  scale_fill_gradient(low = &quot;lightblue&quot;, high = &quot;steelblue&quot;) +
  labs(
    x = &quot;Trait&quot;,
    y = paste0(&quot;# of pairs at FDR &lt; &quot;, FDRcutoff),
    title = types[celltype]
  ) +
  theme_bw(base_size = 14) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
  )
  
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- list()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype, &quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric if it’s stored as character
  df &lt;- df %&gt;%
    mutate(mbe_pval_reverse_pair = suppressWarnings(as.numeric(mbe_pval_reverse_pair)))
  
  # Keep only FDR-significant pairs
  df_sig &lt;- df %&gt;%
    dplyr::filter(ECT_FDR_5suppSNP &lt; FDRcutoff)
  
    # Fix naming if needed
  df_sig$trait[df_sig$trait == &quot;asthma_pmid36778051&quot;] &lt;- &quot;asthma-pmid36778051&quot;
  
  # Summarize counts per trait
  df_summary &lt;- df_sig %&gt;%
    group_by(trait) %&gt;%
    summarise(
      n_sig_pairs = n(),
      n_mbe_sig = sum(!is.na(mbe_pval_reverse_pair) &amp; mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    # Keep traits with at least one FDR-significant pair
    dplyr::filter(n_sig_pairs &gt;= 1) %&gt;%
    mutate(
      trait_simple = sub(&quot;-.*&quot;, &quot;&quot;, trait)
    ) %&gt;%
    arrange(desc(n_sig_pairs))
  

  
  p[[celltype]] &lt;- ggplot(df_summary, aes(x = reorder(trait_simple, -n_sig_pairs))) +
    geom_bar(aes(y = n_sig_pairs), stat = &quot;identity&quot;, fill = &quot;grey80&quot;, color = &quot;black&quot;, width = 0.7) +
    geom_bar(aes(y = n_mbe_sig), stat = &quot;identity&quot;, fill = &quot;darkturquoise&quot;, width = 0.7) +
    geom_text(aes(y = n_mbe_sig, label = n_mbe_sig), vjust = -0.4, size = 3.5) +
    labs(
      x = &quot;Trait&quot;,
      y = paste0(&quot;# of FDR&lt;&quot;, FDRcutoff, &quot; (blue: MBE p&lt;&quot;, MBEcutoff, &quot;)&quot;),
      title = types[celltype]
    ) +
    theme_bw(base_size = 14) +
    theme(
      legend.position = &quot;none&quot;,
      axis.text.x = element_text(angle = 90, hjust = 0, size = 10, color = &quot;black&quot;),
      axis.text.y = element_text(size = 10, color = &quot;black&quot;),
      plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5)
    )
}

all &lt;- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2, top = &quot;Number of pairs with MBE p &lt; 0.05, for the sig pairs&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-2.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-2.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-14-2.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="number-of-significant-pairs-at-different-cutoffs-for-each-cell-type-1"
class="section level2">
<h2>Number of significant pairs at different cutoffs, for each cell
type</h2>
<pre class="r"><code>num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.1,]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))
  
  df_fdr02 &lt;- df[df$ECT_FDR_5suppSNP &lt; 0.2 &amp; df$ECT_FDR_5suppSNP &gt;= 0.1,]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))
  
}

sum &lt;- data.frame(celltype = types,
                  num_fdr01 = num_fdr01,
                  num_fdr02 = num_fdr02)


df &lt;- sum %&gt;%
  pivot_longer(cols = starts_with(&quot;num_fdr&quot;), names_to = &quot;event&quot;, values_to = &quot;total&quot;) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# Plot
p1 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(x = &quot;Cell types&quot;, y = &quot;Count&quot;) +
  ggtitle(&quot;# of pairs&quot;) +
  ylim(0, 60) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

num_fdr01 &lt;- c()
num_fdr02 &lt;- c()
for (celltype in celltypes) {
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, 
                       celltype, &quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  # Convert mbe_pval_reverse_pair to numeric safely (ignore &quot;0/1snp&quot; or NA)
  df$mbe_pval_reverse_pair &lt;- suppressWarnings(as.numeric(df$mbe_pval_reverse_pair))
  df$mbe_pval_reverse_pair[is.na(df$mbe_pval_reverse_pair)] &lt;- 0.5
  
  # --- FDR &lt; 0.1 ---
  df_fdr01 &lt;- df %&gt;%
    dplyr::filter(
      ECT_FDR_5suppSNP &lt; 0.1,
      !(mbe_pval_reverse_pair &lt; MBEcutoff)  # remove reverse-causal pairs
    ) %&gt;%
    dplyr::filter(!is.na(p_ECT))
  
  num_fdr01 &lt;- c(num_fdr01, nrow(df_fdr01))
  
  # --- 0.1 &lt; FDR &lt; 0.2 ---
  df_fdr02 &lt;- df %&gt;%
    dplyr::filter(
      ECT_FDR_5suppSNP &lt; 0.2,
      ECT_FDR_5suppSNP &gt;= 0.1,
      !(mbe_pval_reverse_pair &lt; MBEcutoff)
    ) %&gt;%
    dplyr::filter(!is.na(p_ECT))
  
  num_fdr02 &lt;- c(num_fdr02, nrow(df_fdr02))
}

# summary table
sum &lt;- data.frame(
  celltype = types,
  num_fdr01 = num_fdr01,
  num_fdr02 = num_fdr02
)



df &lt;- sum %&gt;%
  pivot_longer(
    cols = starts_with(&quot;num_fdr&quot;),
    names_to = &quot;event&quot;,
    values_to = &quot;total&quot;
  ) %&gt;%
  mutate(
    event = recode(event,
                   &quot;num_fdr01&quot; = &quot;FDR &lt; 0.1&quot;,
                   &quot;num_fdr02&quot; = &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  )

df$event &lt;- factor(df$event, levels = c(&quot;FDR &lt; 0.1&quot;,
                                        &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# ---- Plot ----
p2 &lt;- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = &quot;identity&quot;, position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = &quot;group&quot;,
    values = c(&quot;#F8766D&quot;, &quot;darkturquoise&quot;),
    labels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;)
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = &quot;black&quot;,
    size = 3,
    vjust = -0.3
  ) +
  labs(
    x = &quot;Cell types&quot;,
    y = &quot;Count (excluding MBE p &lt; 0.05)&quot;,
    title = &quot;# of pairs after removing reverse-causal signals&quot;
  ) +
  ylim(0, 60) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = &quot;black&quot;),
    text = element_text(family = &quot;Times&quot;)
  )

all &lt;- grid.arrange(p1, p2, nrow = 1)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/6f0c8705006fa3ea4e69d3218cc656c67dcd2ab7/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-15-1.png" target="_blank">6f0c870</a>
</td>
<td>
XSun
</td>
<td>
2025-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="traits-and-pathways-identified-1" class="section level2">
<h2>Traits and pathways identified</h2>
<div id="height-ebi-a-gcst90018739" class="section level3">
<h3>height-ebi-a-GCST90018739</h3>
<pre class="r"><code>trait &lt;- &quot;height-ebi-a-GCST90018739&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  dplyr::mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-16-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-16-1.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-16-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>supp_snps_all &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/supp_snps_ECTFDR02.RDS&quot;)
supp_snps_trait &lt;- supp_snps_all[supp_snps_all$trait == trait,]

supp_snps_trait &lt;- supp_snps_trait[supp_snps_trait$factor %in% df_trait_fdr01$factor,]

DT::datatable(supp_snps_trait,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,paste0(&quot;supporting snps for &quot;, trait)),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-1814506231684700bf53" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1814506231684700bf53">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">supporting snps for height-ebi-a-GCST90018739<\/caption>","data":[[],[],[],[],[],[],[],[],[],[],[],[],[],[]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>trait<\/th>\n      <th>factor<\/th>\n      <th>pwyID<\/th>\n      <th>pwyNAME<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>Num_supp_snp<\/th>\n      <th>SNP<\/th>\n      <th>chr<\/th>\n      <th>bp<\/th>\n      <th>pval_SNP_factor<\/th>\n      <th>fdr_SNP_factor<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-16-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-3">
Past versions of unnamed-chunk-16-3.png
</button>
</p>
<div id="fig-unnamed-chunk-16-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/dcc84263a7f25929dbc8f217a0cab74fa2fc21dd/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-16-3.png" target="_blank">dcc8426</a>
</td>
<td>
XSun
</td>
<td>
2025-11-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-16-3.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mcv-ebi-a-gcst90018746" class="section level3">
<h3>MCV-ebi-a-GCST90018746</h3>
<pre class="r"><code>trait &lt;- &quot;MCV-ebi-a-GCST90018746&quot;

df_trait_fdr01 &lt;- c()
df_trait_fdr02 &lt;- c()
for(celltype in celltypes){
  
  df &lt;- readRDS(paste0(folder_ect_summary, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  df_fdr01 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.1, ]
  df_fdr01 &lt;- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 &lt;- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 &lt;- df[df$trait == trait &amp; df$ECT_FDR_5suppSNP&lt; 0.2, ]
  df_fdr02 &lt;- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 &lt;- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr01$celltype &lt;- types[df_trait_fdr01$celltype]

# Prepare data
selected &lt;- df_trait_fdr01 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.1&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-17-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-17-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_trait_fdr02$celltype &lt;- types[df_trait_fdr02$celltype]

# Prepare data
selected &lt;- df_trait_fdr02 %&gt;%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName,&quot; &quot;,factor)
  )

# Plot
p &lt;- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = &quot;white&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = &quot;black&quot;,
    size = 4.5,
    family = &quot;Times&quot;
  ) +
  labs(
    x = expression(paste(&quot;-log&quot;[10], &quot;(ECT p-value)&quot;)),
    y = NULL,
    title = paste0(unique(selected$trait),&quot; ECT FDR &lt; 0.2&quot;)
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = &quot;Times&quot;),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 8)

p</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-17-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-2">
Past versions of unnamed-chunk-17-2.png
</button>
</p>
<div id="fig-unnamed-chunk-17-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-17-2.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="plots-for-all-pairs-at-ect-fdr-0.1-1" class="section level2">
<h2>Plots for all pairs at ECT FDR &lt; 0.1</h2>
<div id="plots-1" class="section level3">
<h3>Plots</h3>
<pre class="r"><code># candidate_pairs &lt;- data.frame(celltype = c(&quot;B_cell&quot;,&quot;B_cell&quot;,&quot;CD14_positive_monocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;,&quot;thymocyte&quot;),
#                               trait = c(&quot;Psoriasis-ebi-a-GCST90038681&quot;,&quot;RA-ukb-a-105&quot;,&quot;SLE-ebi-a-GCST90018917&quot;,&quot;SLE-ebi-a-GCST90018917&quot;,
#                                         &quot;SLE-ebi-a-GCST90018917&quot;,&quot;IBD-ebi-a-GCST003043&quot;,&quot;IBD-ebi-a-GCST003043&quot;),
#                               factor = c(&quot;04062_PC2&quot;,&quot;04071_PC4&quot;,&quot;04120_PC1&quot;,&quot;04015_PC2&quot;,&quot;04022_PC4&quot;,&quot;00230_PC2&quot;,&quot;04924_PC1&quot;)
#                               )

fdr_cutoff &lt;- 0.2
df_all_fdr01$celltype &lt;- celltypes[df_all_fdr01$celltype]

for (i in 1:nrow(df_all_fdr01)){
  
  celltype &lt;- df_all_fdr01$celltype[i]
  trait &lt;- df_all_fdr01$trait[i]
  factor &lt;- df_all_fdr01$factor[i]
  
  file_harmo &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/&quot;,celltype, &quot;-&quot;, trait, &quot;_harmo.RDS&quot;)
  dt_harmo &lt;- readRDS(file_harmo)
  
  df &lt;- readRDS(paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/&quot;, celltype,&quot;_ECT_summary_EAS_SLEremoved.RDS&quot;))
  
  dat &lt;- dt_harmo[[factor]]
  snp_select &lt;- dat[dat$FDR.exposure &lt; fdr_cutoff,]
  snp_select$chr &lt;- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr &lt;- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]
  
  pathwayname &lt;- df$pathwayName[df$factor == factor &amp; df$trait ==trait]
  
  eff &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\nECT p-value =&quot;, ect_p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)
  
  
  #snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff &amp; dat$steiger_dir ==T,]
  snp_select &lt;- dat[dat$FDR.exposure &gt; fdr_cutoff,]
  
  fit &lt;- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  coef &lt;- summary(fit)
  p &lt;- round(coef$coefficients[,4], digits = 4)
  #ect_p &lt;- df$p_ECT[df$factor == factor &amp; df$trait ==trait]
  
  eff_rev &lt;- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome)) +
    geom_point(size = 4, color =&quot;steelblue&quot;) + 
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = &quot;black&quot;),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = &quot;black&quot;),
          text= element_text(family=&quot;Times&quot;),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = &quot;grey&quot;, size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste(&quot;R^2 = &quot;, signif(summary(fit)$r.squared, 3), 
                             &quot;\n p-value =&quot;, p),
               color = &quot;black&quot;, family = &quot;Times&quot;, size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;) +
    geom_hline(yintercept = 0, color = &quot;grey&quot;, linetype = &quot;dashed&quot;)
  

  
  file_gene_loading &lt;- paste0(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/gene_loading/&quot;,celltype, &quot;_&quot;, factor, &quot;-&quot;, trait, &quot;.RDS&quot;)
  gene_loadings_pc &lt;- readRDS(file_gene_loading)
  
  gene_loadings_pc_abs &lt;- abs(gene_loadings_pc)
  gene_loadings_pc_abs_sort &lt;- sort(gene_loadings_pc_abs,decreasing = T)
  topgenes &lt;- names(gene_loadings_pc_abs_sort)[1:10]
  
  gene_loadings_pc &lt;- as.data.frame(cbind(names(gene_loadings_pc),gene_loadings_pc))
  names(gene_loadings_pc) &lt;- c(&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)
  
  # Extract gene information based on the gene symbol
  gene_info &lt;- genes(EnsDb.Hsapiens.v75, filter = ~ gene_name ==  gene_loadings_pc$gene_name)
  gene_info &lt;- as.data.frame(gene_info)
  gene_info &lt;- gene_info[!duplicated(gene_info$gene_name),]
  
  gene_info &lt;- merge(gene_info,gene_loadings_pc, by=&quot;gene_name&quot;)
  gene_info &lt;- gene_info[gene_info$seqnames %in%c(1:22),]
  
  loading_plot &lt;- gene_info[,c(&quot;seqnames&quot;,&quot;start&quot;,&quot;gene_name&quot;,&quot;gene_loadings_pc&quot;)]
  colnames(loading_plot) &lt;- c(&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;gene_name&quot;,&quot;loadings&quot;)
  loading_plot$loadings &lt;- round(as.numeric(loading_plot$loadings),digits = 4)
  
  don &lt;- loading_plot %&gt;%
    
    # Compute chromosome size
    group_by(chromosome_name) %&gt;%
    summarise(chr_len=max(start_position)) %&gt;%
    
    # Calculate cumulative position of each chromosome
    mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %&gt;%
    dplyr::select(-chr_len) %&gt;%
    
    # Add this info to the initial dataset
    left_join(loading_plot, ., by=c(&quot;chromosome_name&quot;=&quot;chromosome_name&quot;)) %&gt;%
    
    # Add a cumulative position of each SNP
    arrange(chromosome_name, start_position) %&gt;%
    mutate( start_positioncum=start_position+tot) %&gt;%
    
    # Add highlight and annotation information
    #mutate( is_annotate=ifelse(gene_name %in% loading_plot_top10$gene_name, &quot;yes&quot;, &quot;no&quot;))
    mutate( is_annotate=ifelse(gene_name %in% topgenes, &quot;yes&quot;, &quot;no&quot;))
  # Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
  axisdf = don %&gt;% group_by(chromosome_name) %&gt;% summarize(center=( max(start_positioncum) + min(start_positioncum) ) / 2 )
  
  manhplot_gene &lt;- ggplot(don, aes(x=start_positioncum, y=loadings)) +
    
    # Show all points
    #rasterise(geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3),dpi=80 )+
    geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3) +
    scale_color_manual(values = rep(c(&quot;red&quot;, &quot;steelblue&quot;), 22)) +
    
    # custom X axis:
    scale_x_continuous( label = as.integer(axisdf$chromosome_name), breaks= axisdf$center ) +
    #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
    
    #ylab(expression(paste(&quot;-log&quot;[10],&quot;(p-value)&quot;))) +
    #ylim(0,max(-log10(don$loadings)) + 2) +
    #ylim(min(as.numeric(don$loadings)),max(as.numeric(don$loadings))) +
    #ylab(&quot;&quot;) +
    ylab(&quot;Gene Loadings&quot;) +
    xlab(&quot;Chromosome&quot;) +
    
    # Custom the theme:
    theme_bw() +
    #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
    theme(
      legend.position=&quot;none&quot;,
      panel.border = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank(),
      axis.title.x = element_text(size = 18),
      axis.text.x = element_text(size = 16, color = &quot;black&quot;),
      axis.title.y = element_text(size = 18),
      axis.text.y = element_text(size = 16, color = &quot;black&quot;),
      text= element_text(family=&quot;Times&quot;),
      #axis.text.y=element_blank(),
      #axis.ticks.y=element_blank()
    ) +
    geom_label_repel( data=subset(don, is_annotate==&quot;yes&quot;), aes(label=gene_name), size=4)
  
    
    cat(paste0(celltype,&quot;-&quot;, pathwayname, &quot;-&quot;, trait ))
  
  
    all &lt;- grid.arrange(eff, eff_rev, nrow = 1)
    
    print(manhplot_gene)
    print(paste0(&quot;Top 10 genes with largest abs loading: &quot;, paste0(topgenes, collapse = &quot;,&quot;)))
  
}</code></pre>
<pre><code>CD15_positive_leukocyte-Kaposi sarcoma-associated herpesvirus infection-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-1.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-2">
Past versions of unnamed-chunk-18-2.png
</button>
</p>
<div id="fig-unnamed-chunk-18-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-2.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,GNG2,JAK1,RB1,JAK2,MAP2K4,RCAN1,MAP2K1,GNG10,IFNAR2&quot;
CD15_positive_leukocyte-Hepatitis B-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-3">
Past versions of unnamed-chunk-18-3.png
</button>
</p>
<div id="fig-unnamed-chunk-18-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-3.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-4">
Past versions of unnamed-chunk-18-4.png
</button>
</p>
<div id="fig-unnamed-chunk-18-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-4.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,TRAF6,JAK2,MAP2K4,JAK1,RB1,MAP2K1,ARAF,TYK2,NRAS&quot;
CD15_positive_leukocyte-Growth hormone synthesis, secretion and action-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-5">
Past versions of unnamed-chunk-18-5.png
</button>
</p>
<div id="fig-unnamed-chunk-18-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-5.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-6">
Past versions of unnamed-chunk-18-6.png
</button>
</p>
<div id="fig-unnamed-chunk-18-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-6.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,JAK2,GSK3B,AKT1,MAP2K4,GNAQ,RAF1,EP300,MAPK3,ITPR2&quot;
CD15_positive_leukocyte-Pathways in cancer-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-7">
Past versions of unnamed-chunk-18-7.png
</button>
</p>
<div id="fig-unnamed-chunk-18-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-7.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-8">
Past versions of unnamed-chunk-18-8.png
</button>
</p>
<div id="fig-unnamed-chunk-18-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-8.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,GNG2,TRAF6,JAK1,JAK2,RB1,GNAQ,MAP2K1,HDAC2,IFNAR2&quot;
CD15_positive_leukocyte-Human papillomavirus infection-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-9">
Past versions of unnamed-chunk-18-9.png
</button>
</p>
<div id="fig-unnamed-chunk-18-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-9.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-10">
Past versions of unnamed-chunk-18-10.png
</button>
</p>
<div id="fig-unnamed-chunk-18-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-10.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,PPP2R2D,JAK1,RB1,PPP2CA,MAP2K1,HDAC2,CDC42,NRAS,TBPL1&quot;
CD15_positive_leukocyte-mRNA surveillance pathway-ATD-bbj-a-90</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-11">
Past versions of unnamed-chunk-18-11.png
</button>
</p>
<div id="fig-unnamed-chunk-18-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-11.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-12">
Past versions of unnamed-chunk-18-12.png
</button>
</p>
<div id="fig-unnamed-chunk-18-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-12.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PABPC1L,PNN,SMG1,PELO,WDR33,NXF1,UPF2,PABPN1,CPSF4,CPSF2&quot;
CD15_positive_leukocyte-Pancreatic secretion-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-13">
Past versions of unnamed-chunk-18-13.png
</button>
</p>
<div id="fig-unnamed-chunk-18-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-13.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-14">
Past versions of unnamed-chunk-18-14.png
</button>
</p>
<div id="fig-unnamed-chunk-18-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-14.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PLCB2,RAB11A,ITPR1,RAP1A,GNAQ,ITPR2,TPCN2,AMY1C,PLCB1,KCNQ1&quot;
CD15_positive_leukocyte-Serotonergic synapse-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-15">
Past versions of unnamed-chunk-18-15.png
</button>
</p>
<div id="fig-unnamed-chunk-18-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-15.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-16">
Past versions of unnamed-chunk-18-16.png
</button>
</p>
<div id="fig-unnamed-chunk-18-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-16.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PLA2G4B,MAPK3,GNAI2,PTGS1,ITPR2,ITPR1,GNG5,GNB2,GNB1,MAPK1&quot;
CD15_positive_leukocyte-Signaling pathways regulating pluripotency of stem cells-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-17">
Past versions of unnamed-chunk-18-17.png
</button>
</p>
<div id="fig-unnamed-chunk-18-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-17.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-18.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-18">
Past versions of unnamed-chunk-18-18.png
</button>
</p>
<div id="fig-unnamed-chunk-18-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-18.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,ACVR1,JAK2,JAK1,NRAS,MAP2K1,GSK3B,CTNNB1,AKT1,RIF1&quot;
CD15_positive_leukocyte-Axon guidance-BMI-ebi-a-GCST90018727</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-19.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-19">
Past versions of unnamed-chunk-18-19.png
</button>
</p>
<div id="fig-unnamed-chunk-18-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-19.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-20.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-20">
Past versions of unnamed-chunk-18-20.png
</button>
</p>
<div id="fig-unnamed-chunk-18-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-20.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,NRAS,RAF1,EPHB1,GSK3B,MAPK3,CDC42,PIK3R1,LIMK2,MAPK1&quot;
CD14_positive_monocyte-Insulin resistance-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-21.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-21">
Past versions of unnamed-chunk-18-21.png
</button>
</p>
<div id="fig-unnamed-chunk-18-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-21.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-22.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-22">
Past versions of unnamed-chunk-18-22.png
</button>
</p>
<div id="fig-unnamed-chunk-18-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-22.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PPP1CA,NR1H2,PPP1CC,RPS6KB1,G6PC3,PCK2,RELA,STAT3,PIK3CA,PTEN&quot;
CD15_positive_leukocyte-Human cytomegalovirus infection-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-23.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-23">
Past versions of unnamed-chunk-18-23.png
</button>
</p>
<div id="fig-unnamed-chunk-18-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-23.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-24.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-24">
Past versions of unnamed-chunk-18-24.png
</button>
</p>
<div id="fig-unnamed-chunk-18-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-24.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,GNG2,JAK1,GNAQ,RB1,RPS6KB2,GNA13,GNG10,MAP2K1,GNG5&quot;
CD15_positive_leukocyte-Huntington disease-BMI-ebi-a-GCST90018727</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-25.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-25">
Past versions of unnamed-chunk-18-25.png
</button>
</p>
<div id="fig-unnamed-chunk-18-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-25.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-26.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-26">
Past versions of unnamed-chunk-18-26.png
</button>
</p>
<div id="fig-unnamed-chunk-18-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-26.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: DNAH1,NDUFB3,ULK1,TBP,NDUFS7,NDUFS3,TUBA1C,CASP9,CYC1,NRBF2&quot;
CD15_positive_leukocyte-Phosphatidylinositol signaling system-ATD-bbj-a-90</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-27.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-27">
Past versions of unnamed-chunk-18-27.png
</button>
</p>
<div id="fig-unnamed-chunk-18-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-27.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-28.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-28">
Past versions of unnamed-chunk-18-28.png
</button>
</p>
<div id="fig-unnamed-chunk-18-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-28.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,IPMK,MTM1,ITPR2,PI4KB,IP6K2,PIK3R1,CALM2,MTMR4,PRKCB&quot;
CD15_positive_leukocyte-Axon guidance-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-29.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-29">
Past versions of unnamed-chunk-18-29.png
</button>
</p>
<div id="fig-unnamed-chunk-18-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-29.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-30.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-30">
Past versions of unnamed-chunk-18-30.png
</button>
</p>
<div id="fig-unnamed-chunk-18-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-30.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,NRAS,RAF1,EPHB1,GSK3B,MAPK3,CDC42,PIK3R1,LIMK2,MAPK1&quot;
B_cell-Platelet activation-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-31.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-31">
Past versions of unnamed-chunk-18-31.png
</button>
</p>
<div id="fig-unnamed-chunk-18-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-31.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-32.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-32">
Past versions of unnamed-chunk-18-32.png
</button>
</p>
<div id="fig-unnamed-chunk-18-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-32.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: VWF,PTGS1,GP9,TLN1,MYLK,GP1BA,ITGB3,GP6,P2RY12,ADCY3&quot;
B_cell-Regulation of actin cytoskeleton-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-33.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-33">
Past versions of unnamed-chunk-18-33.png
</button>
</p>
<div id="fig-unnamed-chunk-18-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-33.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-34.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-34">
Past versions of unnamed-chunk-18-34.png
</button>
</p>
<div id="fig-unnamed-chunk-18-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-34.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: ITGB5,ACTN1,MYLK,MAPK3,ITGB3,NRAS,NCKAP1L,PIP4K2A,ITGA2B,MYL9&quot;
CD15_positive_leukocyte-Various types of N-glycan biosynthesis-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-35.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-35">
Past versions of unnamed-chunk-18-35.png
</button>
</p>
<div id="fig-unnamed-chunk-18-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-35.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-36.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-36">
Past versions of unnamed-chunk-18-36.png
</button>
</p>
<div id="fig-unnamed-chunk-18-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-36.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: STT3B,MAN2A2,B4GALT2,STT3A,DDOST,MGAT2,MGAT1,MGAT4B,ALG12,MGAT4A&quot;
CD14_positive_monocyte-Pathways in cancer-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-37.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-37">
Past versions of unnamed-chunk-18-37.png
</button>
</p>
<div id="fig-unnamed-chunk-18-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-37.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-38.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-38">
Past versions of unnamed-chunk-18-38.png
</button>
</p>
<div id="fig-unnamed-chunk-18-38" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-38.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: FH,HDAC2,CDKN1B,CTBP1,CSF3R,RASSF5,ARHGEF1,NFKB2,BAX,RELA&quot;
CD15_positive_leukocyte-Shigellosis-height-ebi-a-GCST90018739</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-39.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-39">
Past versions of unnamed-chunk-18-39.png
</button>
</p>
<div id="fig-unnamed-chunk-18-39" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-39.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-40.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-40">
Past versions of unnamed-chunk-18-40.png
</button>
</p>
<div id="fig-unnamed-chunk-18-40" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-40.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: PIK3CB,UBE2D2,BCL10,UBE2N,TRAF6,FBXW11,UBE2V2,RBCK1,CDC42,MAPK9&quot;
B_cell-Oxytocin signaling pathway-MCV-ebi-a-GCST90018746</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-41.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-41">
Past versions of unnamed-chunk-18-41.png
</button>
</p>
<div id="fig-unnamed-chunk-18-41" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-41.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-42.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-42">
Past versions of unnamed-chunk-18-42.png
</button>
</p>
<div id="fig-unnamed-chunk-18-42" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/31933eca2daebe7a0f2689257b715407fd00d0e3/docs/figure/revision_summarizing_ECT_setting_SLEremoved.Rmd/unnamed-chunk-18-42.png" target="_blank">31933ec</a>
</td>
<td>
XSun
</td>
<td>
2025-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Top 10 genes with largest abs loading: MAPK3,CALM3,CDKN1A,MAP2K2,MYL6B,CAMK2B,CALML4,PIK3CG,RHOA,PLA2G4B&quot;</code></pre>
<!-- ### Gene function on the cell types , nots from chatgpt  -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggrastr_1.0.1             ggrepel_0.9.1            
 [3] EnsDb.Hsapiens.v75_2.99.0 ensembldb_2.20.2         
 [5] AnnotationFilter_1.20.0   GenomicFeatures_1.48.3   
 [7] AnnotationDbi_1.58.0      Biobase_2.56.0           
 [9] GenomicRanges_1.48.0      GenomeInfoDb_1.39.9      
[11] IRanges_2.30.0            S4Vectors_0.34.0         
[13] BiocGenerics_0.42.0       tidyr_1.3.0              
[15] forcats_0.5.1             gridExtra_2.3            
[17] ggplot2_4.0.0             dplyr_1.1.4              

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0            rjson_0.2.21               
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] XVector_0.36.0              fs_1.5.2                   
  [7] dichromat_2.0-0.1           rstudioapi_0.13            
  [9] farver_2.1.0                DT_0.22                    
 [11] bit64_4.0.5                 fansi_1.0.3                
 [13] xml2_1.3.3                  codetools_0.2-18           
 [15] cachem_1.0.6                knitr_1.39                 
 [17] jsonlite_1.8.0              workflowr_1.7.0            
 [19] Rsamtools_2.12.0            dbplyr_2.1.1               
 [21] png_0.1-7                   compiler_4.2.0             
 [23] httr_1.4.3                  assertthat_0.2.1           
 [25] Matrix_1.5-3                fastmap_1.1.0              
 [27] lazyeval_0.2.2              cli_3.6.1                  
 [29] later_1.3.0                 htmltools_0.5.2            
 [31] prettyunits_1.1.1           tools_4.2.0                
 [33] gtable_0.3.6                glue_1.6.2                 
 [35] GenomeInfoDbData_1.2.8      rappdirs_0.3.3             
 [37] Rcpp_1.0.12                 jquerylib_0.1.4            
 [39] vctrs_0.6.5                 Biostrings_2.64.0          
 [41] rtracklayer_1.56.0          crosstalk_1.2.0            
 [43] xfun_0.41                   stringr_1.5.1              
 [45] lifecycle_1.0.4             restfulr_0.0.14            
 [47] XML_3.99-0.14               zlibbioc_1.42.0            
 [49] scales_1.4.0                hms_1.1.1                  
 [51] promises_1.2.0.1            MatrixGenerics_1.8.0       
 [53] ProtGenerics_1.28.0         parallel_4.2.0             
 [55] SummarizedExperiment_1.26.1 RColorBrewer_1.1-3         
 [57] yaml_2.3.5                  curl_4.3.2                 
 [59] memoise_2.0.1               sass_0.4.1                 
 [61] biomaRt_2.54.1              stringi_1.7.6              
 [63] RSQLite_2.3.1               highr_0.9                  
 [65] BiocIO_1.6.0                filelock_1.0.2             
 [67] BiocParallel_1.30.3         rlang_1.1.2                
 [69] pkgconfig_2.0.3             bitops_1.0-7               
 [71] matrixStats_0.62.0          evaluate_0.15              
 [73] lattice_0.20-45             purrr_1.0.2                
 [75] labeling_0.4.2              GenomicAlignments_1.32.0   
 [77] htmlwidgets_1.5.4           bit_4.0.4                  
 [79] tidyselect_1.2.0            magrittr_2.0.3             
 [81] R6_2.5.1                    generics_0.1.2             
 [83] DelayedArray_0.22.0         DBI_1.2.2                  
 [85] pillar_1.9.0                whisker_0.4                
 [87] withr_2.5.0                 KEGGREST_1.36.3            
 [89] RCurl_1.98-1.7              tibble_3.2.1               
 [91] crayon_1.5.1                utf8_1.2.2                 
 [93] BiocFileCache_2.4.0         rmarkdown_2.25             
 [95] progress_1.2.2              grid_4.2.0                 
 [97] blob_1.2.3                  git2r_0.30.1               
 [99] S7_0.2.0                    digest_0.6.29              
[101] httpuv_1.6.5                beeswarm_0.4.0             
[103] vipor_0.4.5                 bslib_0.3.1                </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
