<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-11-25" />

<title>Summarizing ECT results – ≥ 5 supporting SNPs &amp; ≥ 20 genes, GTEx</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">factor_analysis_new</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xsun1229/factor_analysis_new">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Summarizing ECT results – ≥ 5 supporting
SNPs &amp; ≥ 20 genes, GTEx</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-11-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-12-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>factor_analysis_new/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree8189e6f8a25542d8e53360d82cf082b1520d8b46targetblank8189e6fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/factor_analysis_new/tree/8189e6f8a25542d8e53360d82cf082b1520d8b46" target="_blank">8189e6f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229factoranalysisnewtree8189e6f8a25542d8e53360d82cf082b1520d8b46targetblank8189e6fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/factor_analysis_new/tree/8189e6f8a25542d8e53360d82cf082b1520d8b46" target="_blank">8189e6f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/revision_summarizing_ECT_setting_SLEremoved_gtex.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/revision_summarizing_ECT_setting_SLEremoved_gtex.Rmd</code>)
and HTML
(<code>docs/revision_summarizing_ECT_setting_SLEremoved_gtex.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/449bd0104148e8e21c4e4d42c2833f17e6764f39/analysis/revision_summarizing_ECT_setting_SLEremoved_gtex.Rmd" target="_blank">449bd01</a>
</td>
<td>
XSun
</td>
<td>
2025-11-25
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/factor_analysis_new/449bd0104148e8e21c4e4d42c2833f17e6764f39/docs/revision_summarizing_ECT_setting_SLEremoved_gtex.html" target="_blank">449bd01</a>
</td>
<td>
XSun
</td>
<td>
2025-11-25
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>We used data from GTEx whole blood tissue, it contains 670 samples
for the gene expression data.</p>
<p>We applied cutoffs of + - <code>supporting SNP ≥ 5</code> and -
<code>gene number in pathway ≥ 20</code></p>
<p>to filter the pairs before performing FDR control. All results
presented below are based on this setting.</p>
<pre class="r"><code>library(dplyr)
library(ggplot2)
library(gridExtra)

celltypes &lt;- c(&quot;Whole_Blood&quot;)
types &lt;- c(&quot;Whole Blood&quot;)
names(celltypes) &lt;- types
names(types) &lt;- celltypes

folder_ect_summary &lt;- &quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_gtex/ECT_summary/&quot;</code></pre>
<div id="summary-for-the-ect-results" class="section level2">
<h2>Summary for the ECT results</h2>
<pre class="r"><code>df &lt;- readRDS(&quot;/project/xinhe/xsun/pathway_factor/analysis/1.ECT_gtex/ECT_summary/Whole_Blood_ECT_summary_poolalltraits_selected.RDS&quot;)


df_all &lt;- df[complete.cases(df$ECT_FDR_5suppSNP),]
df_all_fdr02 &lt;- df_all[df_all$ECT_FDR_5suppSNP &lt; 0.2 &amp; df_all$ECT_FDR_5suppSNP &gt;0.1,]
df_all_fdr01 &lt;- df_all[df_all$ECT_FDR_5suppSNP &lt; 0.1,]


df_count &lt;- data.frame(
  cutoff = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;),
  count  = c(nrow(df_all_fdr01), nrow(df_all_fdr02))
)

df_count$cutoff &lt;- factor(df_count$cutoff, 
                          levels = c(&quot;FDR &lt; 0.1&quot;, &quot;0.1 &lt; FDR &lt; 0.2&quot;))

# bar plot
ggplot(df_count, aes(x = cutoff, y = count, fill = cutoff)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.6, color = &quot;white&quot;) +
  scale_fill_manual(values = c(
    &quot;FDR &lt; 0.1&quot; = &quot;steelblue&quot;,
    &quot;0.1 &lt; FDR &lt; 0.2&quot; = &quot;steelblue&quot;
  )) +
  labs(x = &quot;Cutoff&quot;, y = &quot;Number of pairs&quot;, 
       title = &quot;Number of Factor–Trait Pairs Passing FDR Cutoffs&quot;) +
  theme_minimal(base_size = 14) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/revision_summarizing_ECT_setting_SLEremoved_gtex.Rmd/unnamed-chunk-2-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/xsun1229/factor_analysis_new/blob/449bd0104148e8e21c4e4d42c2833f17e6764f39/docs/figure/revision_summarizing_ECT_setting_SLEremoved_gtex.Rmd/unnamed-chunk-2-1.png" target="_blank">449bd01</a>
</td>
<td>
XSun
</td>
<td>
2025-11-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="detailed-results-for-all-pairs-at-ect-fdr-0.1"
class="section level2">
<h2>Detailed results for all pairs at ECT FDR &lt; 0.1</h2>
<pre class="r"><code>DT::datatable(df_all_fdr01,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Pairs with ECT FDR &lt;= 0.1 &#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-0c923cdc0df2ab45b6f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c923cdc0df2ab45b6f5">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Pairs with ECT FDR &lt;= 0.1 <\/caption>","data":[["2336","3306","14279","3553","3555","2144","2466","3155","3192","7126","14409","17097","17256","3036","14039","2604","3211","6445","7134","14085"],["Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood","Whole Blood"],["04110_PC1","04360_PC1","04114_PC5","04611_PC1","04071_PC1","04914_PC1","00620_PC1","04014_PC1","04670_PC5","00514_PC3","04621_PC4","04923_PC2","00970_PC3","04072_PC1","03020_PC2","03015_PC2","04015_PC1","04114_PC3","04120_PC4","04217_PC4"],["04110","04360","04114","04611","04071","04914","00620","04014","04670","00514","04621","04923","00970","04072","03020","03015","04015","04114","04120","04217"],["Cell cycle","Axon guidance","Oocyte meiosis","Platelet activation","Sphingolipid signaling pathway","Progesterone-mediated oocyte maturation","Pyruvate metabolism","Ras signaling pathway","Leukocyte transendothelial migration","Other types of O-glycan biosynthesis","NOD-like receptor signaling pathway","Regulation of lipolysis in adipocytes","Aminoacyl-tRNA biosynthesis","Phospholipase D signaling pathway","RNA polymerase","mRNA surveillance pathway","Rap1 signaling pathway","Oocyte meiosis","Ubiquitin mediated proteolysis","Necroptosis"],["HB-ebi-a-GCST90002384","HCT-ebi-a-GCST90002383","RBC-ebi-a-GCST90002403","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","HB-ebi-a-GCST90002384","HB-ebi-a-GCST90002384","HCT-ebi-a-GCST90002383","HCT-ebi-a-GCST90002383","LDL-ukb-d-30780_irnt","RBC-ebi-a-GCST90002403","UC-ebi-a-GCST003045","UC-ebi-a-GCST003045","HCT-ebi-a-GCST90002383","RBC-ebi-a-GCST90002403","HB-ebi-a-GCST90002384","HCT-ebi-a-GCST90002383","IBD-ebi-a-GCST003043","LDL-ukb-d-30780_irnt","RBC-ebi-a-GCST90002403"],["Hemoglobin ","Hematocrit","Red blood cell count ","Hematocrit","Hematocrit","Hemoglobin ","Hemoglobin ","Hematocrit","Hematocrit","LDL cholesterol ","Red blood cell count ","Ulcerative colitis ","Ulcerative colitis ","Hematocrit","Red blood cell count ","Hemoglobin ","Hematocrit","Inflammatory bowel disease ","LDL cholesterol ","Red blood cell count "],[16,8,5,5,9,16,9,5,15,6,7,8,5,8,28,24,7,5,8,14],[2e-06,1.1e-05,2.6e-05,4.1e-05,6.5e-05,0.0002,0.000141,0.000206,0.000105,0.000215,0.000184,0.000203,0.000125,0.00024,0.000256,0.000402,0.000406,0.0004,0.000382,0.000417],[0.009614,0.0264385,0.0416606666666667,0.04927175,0.062491,0.0795003846153846,0.0795003846153846,0.0795003846153846,0.0795003846153846,0.0795003846153846,0.0795003846153846,0.0795003846153846,0.0795003846153846,0.0820394666666667,0.0820394666666667,0.0954532857142857,0.0954532857142857,0.0954532857142857,0.0954532857142857,0.0954532857142857],[124,181,128,124,119,99,40,232,112,46,181,55,66,148,31,91,210,128,136,162],[118,140,108,110,111,81,36,177,93,35,146,42,44,119,31,74,175,108,124,123]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>factor<\/th>\n      <th>pathwayID<\/th>\n      <th>pathwayName<\/th>\n      <th>trait_id<\/th>\n      <th>Trait<\/th>\n      <th>num_supp_SNP<\/th>\n      <th>p_ECT<\/th>\n      <th>ECT_FDR_5suppSNP<\/th>\n      <th>n_genes_inpathway<\/th>\n      <th>n_genes_expressed<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[7,8,9,10,11]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Highlight:</p>
<ul>
<li>Ubiquitin-mediated proteolysis and LDL cholesterol</li>
</ul>
<p>ref: <a
href="https://www.sciencedirect.com/science/article/abs/pii/S0962892423001927"
class="uri">https://www.sciencedirect.com/science/article/abs/pii/S0962892423001927</a></p>
<ul>
<li>Other O-glycan biosynthesis and LDL cholesterol</li>
</ul>
<p>ref: <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC4067166/"
class="uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC4067166/</a></p>
<p>The non-mucin O-glycans in this pathway include:</p>
<ul>
<li>O-fucose glycosylation (e.g., NOTCH signaling)</li>
<li>O-mannose glycosylation (α-dystroglycan)</li>
<li>O-GlcNAc modification</li>
<li>O-linked glycans on LDLR and other lipoprotein receptors</li>
</ul>
<p>These modifications directly affect:</p>
<ul>
<li>LDLR stability</li>
<li>LDLR recycling</li>
<li>LDL particle clearance</li>
<li>Apolipoprotein glycosylation (ApoE, ApoC-III)</li>
</ul>
<p>Changes in these O-glycans alter plasma LDL levels and cholesterol
homeostasis.</p>
<table>
<colgroup>
<col width="19%" />
<col width="9%" />
<col width="10%" />
<col width="15%" />
<col width="45%" />
</colgroup>
<thead>
<tr class="header">
<th>Pathway</th>
<th>Trait</th>
<th>Evidence Level</th>
<th>Mechanism Type</th>
<th>Short Rationale</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Cell cycle</td>
<td>Hemoglobin</td>
<td><strong>Strong</strong></td>
<td>Cell-intrinsic</td>
<td>Erythroid progenitors require cell-cycle exit for globin expression;
disruptions alter Hb.</td>
</tr>
<tr class="even">
<td>Axon guidance</td>
<td>Hematocrit</td>
<td><strong>Indirect / Weak</strong></td>
<td>Vascular/hematopoietic</td>
<td>SLIT/ROBO, Eph/ephrin affect HSPC niches but not steady Hct
directly.</td>
</tr>
<tr class="odd">
<td>Oocyte meiosis</td>
<td>RBC count</td>
<td><strong>None</strong></td>
<td>None</td>
<td>No erythroid relevance; only generic cell-cycle overlap.</td>
</tr>
<tr class="even">
<td>Platelet activation</td>
<td>Hematocrit</td>
<td><strong>Indirect</strong></td>
<td>Hemodynamic/platelet physiology</td>
<td>Hct modulates platelet activation; platelet pathways don’t determine
RBC mass.</td>
</tr>
<tr class="odd">
<td>Sphingolipid signaling</td>
<td>Hematocrit</td>
<td><strong>Moderate</strong></td>
<td>Cell-intrinsic + Inflammatory</td>
<td>S1P/ceramide regulate erythroid survival &amp; inflammatory
anemia.</td>
</tr>
<tr class="even">
<td>Progesterone-mediated oocyte maturation</td>
<td>Hemoglobin</td>
<td><strong>Weak/Indirect</strong></td>
<td>Endocrine systemic</td>
<td>Progesterone alters plasma volume/iron loss, not via this KEGG
pathway.</td>
</tr>
<tr class="odd">
<td>Pyruvate metabolism</td>
<td>Hemoglobin</td>
<td><strong>Strong</strong></td>
<td>Cell-intrinsic (RBC glycolysis)</td>
<td>RBCs require glycolysis; pyruvate kinase defects cause low Hb.</td>
</tr>
<tr class="even">
<td>Ras signaling</td>
<td>Hematocrit</td>
<td><strong>Strong</strong></td>
<td>Cell-intrinsic</td>
<td>Key EPO/SCF pathway in erythropoiesis; modulates red-cell
output.</td>
</tr>
<tr class="odd">
<td>Leukocyte transendothelial migration</td>
<td>Hematocrit</td>
<td><strong>Indirect</strong></td>
<td>Inflammation</td>
<td>LTM influences vascular inflammation; Hct affects flow, not vice
versa.</td>
</tr>
<tr class="even">
<td>Other O-glycan biosynthesis</td>
<td>LDL cholesterol</td>
<td><strong>Moderate / Strong</strong></td>
<td>Metabolic</td>
<td>Glycosylation alters LDLR function and apo-protein clearance.</td>
</tr>
<tr class="odd">
<td>NOD-like receptor signaling</td>
<td>RBC count</td>
<td><strong>Strong</strong></td>
<td>Immune/inflammatory</td>
<td>NLRP3 inflammasome drives ineffective erythropoiesis and
inflammatory anemia.</td>
</tr>
<tr class="even">
<td>Regulation of lipolysis in adipocytes</td>
<td>Ulcerative colitis</td>
<td><strong>Indirect</strong></td>
<td>Metabolic systemic</td>
<td>Lipolytic pathways altered in UC; systemic metabolic consequence not
primary cause.</td>
</tr>
<tr class="odd">
<td>Aminoacyl-tRNA biosynthesis</td>
<td>Ulcerative colitis</td>
<td><strong>Indirect</strong></td>
<td>Metabolic/cellular stress</td>
<td>Commonly enriched signature in UC omics; reflects translational
stress.</td>
</tr>
<tr class="even">
<td>Phospholipase D signaling</td>
<td>Hematocrit</td>
<td><strong>Weak / Indirect</strong></td>
<td>Signaling</td>
<td>Broad signaling role; no direct erythropoietic mechanism.</td>
</tr>
<tr class="odd">
<td>RNA polymerase</td>
<td>RBC count</td>
<td><strong>Strong</strong></td>
<td>Cell-intrinsic</td>
<td>Essential for erythroid transcription and globin gene
expression.</td>
</tr>
<tr class="even">
<td>mRNA surveillance</td>
<td>Hemoglobin</td>
<td><strong>Strong</strong></td>
<td>Cell-intrinsic</td>
<td>NMD regulates globin transcripts; central to thalassemia/β-globin
expression.</td>
</tr>
<tr class="odd">
<td>Rap1 signaling</td>
<td>Hematocrit</td>
<td><strong>Indirect</strong></td>
<td>Vascular + RBC adhesion</td>
<td>Influences integrin signaling and RBC adhesion; not a core Hct
regulator.</td>
</tr>
<tr class="even">
<td>Oocyte meiosis</td>
<td>IBD</td>
<td><strong>None</strong></td>
<td>None</td>
<td>No relevance; pathway is oocyte-specific.</td>
</tr>
<tr class="odd">
<td>Ubiquitin-mediated proteolysis</td>
<td>LDL cholesterol</td>
<td><strong>Strong</strong></td>
<td>Metabolic</td>
<td>IDOL–LDLR axis controls LDL receptor degradation → LDL levels.</td>
</tr>
<tr class="even">
<td>Necroptosis</td>
<td>RBC count</td>
<td><strong>Moderate / Strong</strong></td>
<td>Cell-intrinsic</td>
<td>RBCs undergo regulated necroptosis; contributes to hemolysis and low
RBC count.</td>
</tr>
</tbody>
</table>
</div>
<div id="examing-some-pairs" class="section level2">
<h2>Examing some pairs</h2>
<div id="oocyte-meiosis-red-blood-cell-count" class="section level3">
<h3>Oocyte meiosis &gt; Red blood cell count</h3>
<p>We extract the top 10 genes with highest loadings on the factor, and
then examine their common functions (by ChatGPT)</p>
<table>
<colgroup>
<col width="5%" />
<col width="19%" />
<col width="14%" />
<col width="59%" />
</colgroup>
<thead>
<tr class="header">
<th>Gene</th>
<th>Main Function</th>
<th>RBC Role Through This Function?</th>
<th>Explanation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>ADCY5</strong></td>
<td>cAMP production, GPCR signaling</td>
<td><strong>Possibly Yes</strong></td>
<td>cAMP signaling modulates <strong>EPO receptor signaling</strong> and
erythroid proliferation. Not RBC-specific, but pathway is relevant.</td>
</tr>
<tr class="even">
<td><strong>RBX1</strong></td>
<td>E3 ubiquitin ligase component</td>
<td><strong>Possible</strong></td>
<td>Ubiquitin–proteasome system controls erythroblast cell-cycle and
apoptosis, but RBX1 is <strong>general</strong>, not
erythroid-specific.</td>
</tr>
<tr class="odd">
<td><strong>PTTG2</strong></td>
<td>Cell cycle / chromatid separation</td>
<td><strong>Possible (weak)</strong></td>
<td>Erythroblasts divide rapidly, so cell-cycle genes matter, but PTTG2
is <strong>not known</strong> to have an erythroid-specific role.</td>
</tr>
<tr class="even">
<td><strong>CCNE2</strong></td>
<td>Cyclin E2, G1/S transition</td>
<td><strong>Possible</strong></td>
<td>G1/S regulators are essential during early erythroid expansion, but
CCNE2 is <strong>general</strong>, not RBC-specific.</td>
</tr>
<tr class="odd">
<td><strong>CDC16</strong></td>
<td>APC/C subunit (mitosis control)</td>
<td><strong>Possible (weak)</strong></td>
<td>APC/C activity is required in all dividing cells; erythroblasts also
rely on it, but CDC16 itself is <strong>not RBC-enriched</strong>.</td>
</tr>
<tr class="even">
<td><strong>ANAPC13</strong></td>
<td>APC/C assembly</td>
<td><strong>Possible (weak)</strong></td>
<td>Same as CDC16—mitotic machinery required, but not specific to
erythropoiesis.</td>
</tr>
<tr class="odd">
<td><strong>CAMK2G</strong></td>
<td>Ca²⁺-dependent kinase</td>
<td><strong>Unlikely</strong></td>
<td>Ca²⁺ signaling is important in many cells; erythropoiesis mainly
depends on CaMKIIβ, not CAMK2G specifically.</td>
</tr>
<tr class="even">
<td><strong>MAPK13</strong></td>
<td>p38 MAPK signaling (stress, inflammation)</td>
<td><strong>Possible</strong></td>
<td>p38 MAPK pathway regulates erythroid differentiation; MAPK13 is
<strong>p38δ</strong>, which is less studied but may have minor
roles.</td>
</tr>
<tr class="odd">
<td><strong>SPDYA</strong></td>
<td>Activates CDK1/2</td>
<td><strong>Possible</strong></td>
<td>CDKs are required for erythroblast cycling, but SPDYA is not a known
erythroid regulator.</td>
</tr>
<tr class="even">
<td><strong>IGF1</strong></td>
<td>Growth factor</td>
<td><strong>Strong Yes</strong></td>
<td>IGF1 <strong>directly stimulates erythroid progenitor proliferation
and survival</strong> (BFU-E, CFU-E). Clear RBC biology.</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gridExtra_2.3 ggplot2_4.0.0 dplyr_1.1.4  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12        highr_0.9          RColorBrewer_1.1-3 pillar_1.9.0      
 [5] compiler_4.2.0     bslib_0.3.1        later_1.3.0        jquerylib_0.1.4   
 [9] git2r_0.30.1       workflowr_1.7.0    tools_4.2.0        digest_0.6.29     
[13] gtable_0.3.6       jsonlite_1.8.0     evaluate_0.15      lifecycle_1.0.4   
[17] tibble_3.2.1       pkgconfig_2.0.3    rlang_1.1.2        cli_3.6.1         
[21] rstudioapi_0.13    crosstalk_1.2.0    yaml_2.3.5         xfun_0.41         
[25] fastmap_1.1.0      withr_2.5.0        stringr_1.5.1      knitr_1.39        
[29] htmlwidgets_1.5.4  generics_0.1.2     fs_1.5.2           vctrs_0.6.5       
[33] sass_0.4.1         DT_0.22            grid_4.2.0         rprojroot_2.0.3   
[37] tidyselect_1.2.0   glue_1.6.2         R6_2.5.1           fansi_1.0.3       
[41] rmarkdown_2.25     farver_2.1.0       magrittr_2.0.3     whisker_0.4       
[45] scales_1.4.0       promises_1.2.0.1   htmltools_0.5.2    dichromat_2.0-0.1 
[49] httpuv_1.6.5       labeling_0.4.2     S7_0.2.0           utf8_1.2.2        
[53] stringi_1.7.6     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
