---
title: "Heritability of PCs"
author: "XSun"
date: "2022-10-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r echo=FALSE, message=FALSE, warning=FALSE}
suppressMessages(library(ggplot2))
suppressMessages(library(gridExtra))
```


## Introduction

We computed the heritability of PCs with gcta.

## Results (Computed with whole genome variants, GCTA)

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=12}
celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
path <- c("4.B_cell_kegg","6.CD14_positive_monocyte_kegg","8.CD15_positive_leukocyte_kegg","16.platelet_kegg","10.T_cell_kegg","14.thymocyte_kegg")
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")


p1 <- list()
p2 <- list()
for (i in 1:length(celltypes)){
  
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/3.heritability/")
  file_hsq <- paste0(folder_celltype,"hsq_processed_co.txt")
  
  H <- read.table(file_hsq, header = T)
  sum_sorted <- H[order(as.numeric(H$hsq)),]
  sum_sorted$order <- seq(1,nrow(H),by=1)
  
  p1[[i]] <- ggplot(sum_sorted, aes(x=order, y=hsq)) + theme_bw(base_line_size =0.3) +
    geom_errorbar(aes(ymin=hsq-SE, ymax=hsq+SE), width=0.1, color = "grey",alpha=0.6) +
    geom_point(size = 0.8, color = "steelblue") + 
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + 
    xlab("PCs sorted by heritability") + ylab("Heritability \u00B1 SE") + 
    ylim(-2,3)  + 
    theme(axis.title.x = element_text(size = 14),
          axis.text.x = element_text(size = 12, color = "black"),
          axis.title.y = element_text(size = 14),
          axis.text.y = element_text(size = 12, color = "black"),
          text= element_text(family="Times New Roman")) 
  
  plotdata <- data.frame(observed = -log10(sort(H$pval)),
                         expected = -log10(ppoints(nrow(H))))
  
  
  p2[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5, color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
  
  
  
}


all_interval <- grid.arrange(p1[[1]],p1[[2]],p1[[3]],p1[[4]],p1[[5]],p1[[6]], nrow = 2)

all_qq <- grid.arrange(p2[[1]],p2[[2]],p2[[3]],p2[[4]],p2[[5]],p2[[6]], nrow = 2)

```

## Results (Computed with whole genome variants, GEMMA, computed using REML model)

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=12}
celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
path <- c("4.B_cell_kegg","6.CD14_positive_monocyte_kegg","8.CD15_positive_leukocyte_kegg","16.platelet_kegg","10.T_cell_kegg","14.thymocyte_kegg")
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")


p1 <- list()
p2 <- list()
for (i in 1:length(celltypes)){
  
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/4.heritability_gemma/")
  file_hsq <- paste0(folder_celltype,"results/hsq_se_wg.rdata")
  
  load(file_hsq)
  
  sum_sorted <- df[order(as.numeric(df$pve_all_reml)),]
  sum_sorted$order <- seq(1,nrow(sum_sorted),by=1)
  
  p1[[i]] <- ggplot(sum_sorted, aes(x=order, y=pve_all_reml)) + theme_bw(base_line_size =0.3) +
    geom_errorbar(aes(ymin=pve_all_reml-se_all_reml, ymax=pve_all_reml+se_all_reml), width=0.1, color = "grey",alpha=0.6) +
    geom_point(size = 0.8, color = "steelblue") + 
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + 
    xlab("PCs sorted by heritability") + ylab("Heritability \u00B1 SE") + 
    ylim(-2,3)  + 
    theme(axis.title.x = element_text(size = 14),
          axis.text.x = element_text(size = 12, color = "black"),
          axis.title.y = element_text(size = 14),
          axis.text.y = element_text(size = 12, color = "black"),
          text= element_text(family="Times New Roman")) 
  
  df$pval_reml[is.na(df$pval_reml)] <- 1
  
  plotdata <- data.frame(observed = -log10(sort(as.numeric(as.character(df$pval_reml)))),
                         expected = -log10(ppoints(nrow(df))))
  
  
  p2[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5, color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times New Roman")) 
  
  
  
  
}


all_interval <- grid.arrange(p1[[1]],p1[[2]],p1[[3]],p1[[4]],p1[[5]],p1[[6]], nrow = 2)

all_qq <- grid.arrange(p2[[1]],p2[[2]],p2[[3]],p2[[4]],p2[[5]],p2[[6]], nrow = 2)

```

## Summary table

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE, fig.width=5,fig.height=5}
celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
path <- c("4.B_cell_kegg","6.CD14_positive_monocyte_kegg","8.CD15_positive_leukocyte_kegg","16.platelet_kegg","10.T_cell_kegg","14.thymocyte_kegg")
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")


i <- 1
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/4.heritability_gemma/")
  file_hsq <- paste0(folder_celltype,"results/hsq_se_wg.rdata")
  
  load(file_hsq)
  df$factor <- rownames(df)
  df$pval_reml[is.na(df$pval_reml)] <- 1
  df <- df[,4:7]
  colnames(df) <- c("hsq_gemma","se_gemma","pval_gemma","factor")
  
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/3.heritability/")
  file_hsq <- paste0(folder_celltype,"hsq_processed_co.txt")
  
  H <- read.table(file_hsq, header = T)
  H <- data.frame(cbind(H$factors,H$hsq,H$SE,H$pval))
  colnames(H) <- c("factor","hsq_gcta","se_gcta","pval_gcta")
  
  df <- merge(df,H, by="factor")
  print(type[[i]])
  #print(htmltools::tagList(DT::datatable(df,options = list(pageLength =5))))
 DT::datatable(df, options = list(pageLength =5))

i <- 2
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/4.heritability_gemma/")
  file_hsq <- paste0(folder_celltype,"results/hsq_se_wg.rdata")
  
  load(file_hsq)
  df$factor <- rownames(df)
  df$pval_reml[is.na(df$pval_reml)] <- 1
  df <- df[,4:7]
  colnames(df) <- c("hsq_gemma","se_gemma","pval_gemma","factor")
  
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/3.heritability/")
  file_hsq <- paste0(folder_celltype,"hsq_processed_co.txt")
  
  H <- read.table(file_hsq, header = T)
  H <- data.frame(cbind(H$factors,H$hsq,H$SE,H$pval))
  colnames(H) <- c("factor","hsq_gcta","se_gcta","pval_gcta")
  
  df <- merge(df,H, by="factor")
  print(type[[i]])
  DT::datatable(df, options = list(pageLength =5))

i <- 3
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/4.heritability_gemma/")
  file_hsq <- paste0(folder_celltype,"results/hsq_se_wg.rdata")
  
  load(file_hsq)
  df$factor <- rownames(df)
  df$pval_reml[is.na(df$pval_reml)] <- 1
  df <- df[,4:7]
  colnames(df) <- c("hsq_gemma","se_gemma","pval_gemma","factor")
  
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/3.heritability/")
  file_hsq <- paste0(folder_celltype,"hsq_processed_co.txt")
  
  H <- read.table(file_hsq, header = T)
  H <- data.frame(cbind(H$factors,H$hsq,H$SE,H$pval))
  colnames(H) <- c("factor","hsq_gcta","se_gcta","pval_gcta")
  
  df <- merge(df,H, by="factor")
  print(type[[i]])
  DT::datatable(df, options = list(pageLength =5))
  
  
  
  i <- 4
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/4.heritability_gemma/")
  file_hsq <- paste0(folder_celltype,"results/hsq_se_wg.rdata")
  
  load(file_hsq)
  df$factor <- rownames(df)
  df$pval_reml[is.na(df$pval_reml)] <- 1
  df <- df[,4:7]
  colnames(df) <- c("hsq_gemma","se_gemma","pval_gemma","factor")
  
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/3.heritability/")
  file_hsq <- paste0(folder_celltype,"hsq_processed_co.txt")
  
  H <- read.table(file_hsq, header = T)
  H <- data.frame(cbind(H$factors,H$hsq,H$SE,H$pval))
  colnames(H) <- c("factor","hsq_gcta","se_gcta","pval_gcta")
  
  df <- merge(df,H, by="factor")
  print(type[[i]])
  DT::datatable(df, options = list(pageLength =5))
  
  df <- df[complete.cases(df$hsq_gemma),]
  plot(df$hsq_gemma,df$hsq_gcta)
  abline(a=0,b=1,col="red")
  
  i <- 5
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/4.heritability_gemma/")
  file_hsq <- paste0(folder_celltype,"results/hsq_se_wg.rdata")
  
  load(file_hsq)
  df$factor <- rownames(df)
  df$pval_reml[is.na(df$pval_reml)] <- 1
  df <- df[,4:7]
  colnames(df) <- c("hsq_gemma","se_gemma","pval_gemma","factor")
  
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/3.heritability/")
  file_hsq <- paste0(folder_celltype,"hsq_processed_co.txt")
  
  H <- read.table(file_hsq, header = T)
  H <- data.frame(cbind(H$factors,H$hsq,H$SE,H$pval))
  colnames(H) <- c("factor","hsq_gcta","se_gcta","pval_gcta")
  
  df <- merge(df,H, by="factor")
  print(type[[i]])
  DT::datatable(df, options = list(pageLength =5))
  
  
  i <- 6
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/4.heritability_gemma/")
  file_hsq <- paste0(folder_celltype,"results/hsq_se_wg.rdata")
  
  load(file_hsq)
  df$factor <- rownames(df)
  df$pval_reml[is.na(df$pval_reml)] <- 1
  df <- df[,4:7]
  colnames(df) <- c("hsq_gemma","se_gemma","pval_gemma","factor")
  
  folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/3.heritability/")
  file_hsq <- paste0(folder_celltype,"hsq_processed_co.txt")
  
  H <- read.table(file_hsq, header = T)
  H <- data.frame(cbind(H$factors,H$hsq,H$SE,H$pval))
  colnames(H) <- c("factor","hsq_gcta","se_gcta","pval_gcta")
  
  df <- merge(df,H, by="factor")
  print(type[[i]])
  DT::datatable(df, options = list(pageLength =5))
```

<!-- ## Results (Computed with independent GWAS variants, GEMMA) -->

<!-- ### Summary for the significant pairs (ECT FDR < 0.2) -->

<!-- The table below contains the heritability computed by GCTA and GEMMA -->

<!-- The "pve" columns contain the heritability computed from independent GWAS variants  (GEMMA, LMM), the "gcta" columns contain the heritability computed from whole genome variants(GCTA). -->

<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=12} -->
<!-- load("/project2/xinhe/xsun/CEDAR/global_analysis_FDR_corrected/results/ECT_summary_pveadded.rdata") -->
<!-- DT::datatable(sum_sorted, options = list(pageLength =10)) -->
<!-- ``` -->

<!-- ### Heritability plot (Computed with independent GWAS variants, GEMMA) -->

<!-- We computed the heritability for each factor - trait pair.  -->

<!-- So, the number of dots in each plot varies from 14,000 to 22,000. -->

<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE,fig.height=5, fig.width=12} -->
<!-- celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte") -->
<!-- path <- c("4.B_cell_kegg","6.CD14_positive_monocyte_kegg","8.CD15_positive_leukocyte_kegg","16.platelet_kegg","10.T_cell_kegg","14.thymocyte_kegg") -->
<!-- type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)") -->


<!-- p1 <- list() -->
<!-- p2 <- list() -->
<!-- for (i in 1:length(celltypes)){ -->

<!--   folder_celltype <- paste0("/project2/xinhe/xsun/CEDAR/",path[i],"/9.pcs_40genes_rerun/4.heritability_gemma/") -->
<!--   file_hsq <- paste0(folder_celltype,"results/hsq_se.rdata") -->

<!--   load(file_hsq) -->

<!--   hsq_alltraits <- c() -->
<!--   se_alltraits <- c() -->
<!--   for (j in 1:length(hsq_all)){ -->

<!--     hsq_alltraits <- c(hsq_alltraits,hsq_all[[j]]) -->
<!--     se_alltraits <- c(se_alltraits,hsq_se_all[[j]]) -->
<!--   } -->

<!--   ######### -->
<!--   sum <- as.data.frame(cbind(hsq_alltraits,se_alltraits)) -->
<!--   sum_sorted <- sum[order(as.numeric(sum$hsq_alltraits)),] -->
<!--   sum_sorted$order <- seq(1,nrow(sum_sorted),by=1) -->
<!--   colnames(sum_sorted) <- c("hsq","SE","order") -->


<!--   p1[[i]] <- ggplot(sum_sorted, aes(x=order, y=hsq)) + theme_bw(base_line_size =0.3) + -->
<!--     geom_errorbar(aes(ymin=hsq-SE, ymax=hsq+SE), width=0.1, color = "grey",alpha=0.6) + -->
<!--     geom_point(size = 0.8, color = "steelblue") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     xlab("PCs sorted by heritability") + ylab("Heritability \u00B1 SE") + -->
<!--     ylim(-1,1)  + -->
<!--     theme(axis.title.x = element_text(size = 14), -->
<!--           axis.text.x = element_text(size = 12, color = "black"), -->
<!--           axis.title.y = element_text(size = 14), -->
<!--           axis.text.y = element_text(size = 12, color = "black"), -->
<!--           text= element_text(family="Times New Roman")) -->

<!--   zscore <- sum_sorted$hsq/sum_sorted$SE -->
<!--   pval <- 2*pnorm(-abs(zscore)) -->
<!--   sum_sorted$hsq_pval <- pval -->

<!--   plotdata <- data.frame(observed = -log10(sort(as.numeric(sum_sorted$hsq_pval))), -->
<!--                          expected = -log10(ppoints(nrow(sum_sorted)))) -->


<!--   p2[[i]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) + -->
<!--     geom_point(aes(expected, observed), shape = 1, size = 1.5, color = "steelblue") + -->
<!--     geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") + -->
<!--     ggtitle(type[i]) + theme(plot.title = element_text(hjust = 0.5)) + -->
<!--     labs(x = expression(paste("Expected -log"[10],"(p-value)")), -->
<!--          y = expression(paste("Observed -log"[10],"(p-value)"))) + -->

<!--     theme(axis.title.x = element_text(size = 10), -->
<!--           axis.text.x = element_text(size = 8, color = "black"), -->
<!--           axis.title.y = element_text(size = 10), -->
<!--           axis.text.y = element_text(size = 8, color = "black"), -->
<!--           text= element_text(family="Times New Roman")) -->




<!-- } -->


<!-- all_interval <- grid.arrange(p1[[1]],p1[[2]],p1[[3]],p1[[4]],p1[[5]],p1[[6]], nrow = 2) -->
<!-- all_qq <- grid.arrange(p2[[1]],p2[[2]],p2[[3]],p2[[4]],p2[[5]],p2[[6]], nrow = 2) -->
<!-- ``` -->