---
title: "ECT result analyzing"
author: "XSun"
date: "2024-02-22"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

## Functions used

```{r message=FALSE}
library(tibble)
library(dplyr)
library(ggplot2)
library(grid)
library(tidyverse)
library(ComplexHeatmap)
library(circlize)
library(gridExtra)

traits <- c("eo", "gran", "hct", "hgb", "lymph", "mch", "mchc", "mcv", 
            "mpv", "mono", "myeloid_wbc", "neut", "plt", "pdw", "pct", 
            "rbc", "rdw", "ret", "wbc", "baso", "T1D", "ukb.allasthma", 
            "EUR.IBD", "EUR.CD", "EUR.UC", "allergy")
celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")

order <- c("plt","mpv","pdw","pct",
           "rbc","mcv","hct","mch","mchc","hgb","rdw","ret",
           "mono","wbc","eo","neut","gran","myeloid_wbc","baso",
           "lymph",
           "EUR.UC","EUR.CD","EUR.IBD","T1D","ukb.allasthma","allergy")

plot_selected_pwys <- function(trait, summary_df) {
  
  # Subset summary_df for the given trait
  selected <- summary_df[summary_df$trait == trait, ]
  
  # Handle duplicated pathway names by appending a space
  index <- which(duplicated(selected$pwy_name))
  selected$pwy_name[index] <- paste0(selected$pwy_name[index], " ")
  
  # Transform p-values and order by -log10(pval)
  selected$lgectpval <- -log10(selected$pval_ECT)
  selected <- selected[order(selected$lgectpval, decreasing = TRUE), ]
  
  # Append note to the last num_snp_pass_cutoff value
  selected$num_snp_pass_cutoff[nrow(selected)] <- paste0(selected$num_snp_pass_cutoff[nrow(selected)], " (# of supporting variants)")
  
  # Set pathway names as a factor with levels in the desired order
  selected$pwy_name <- factor(selected$pwy_name, levels = rev(unique(selected$pwy_name)))
  
  # Create the plot
  p <- ggplot(selected) +
    theme_bw(base_line_size = 0.3) +
    geom_col(aes(x = lgectpval, y = pwy_name, fill = celltypes), width = 0.6) +
    geom_text(aes(0, y = pwy_name, label = pwy_name), hjust = 0, nudge_x = 0.1, colour = "white", size = 3, family = "Times") + 
    geom_text(aes(lgectpval, y = pwy_name, label = num_snp_pass_cutoff), hjust = 0, nudge_x = 0.01, colour = "black", size = 3, family = "Times") +
    labs(x = expression(paste("-log"[10], "(ECT p-value)")), y = "Pathways") +
    theme(axis.text.y = element_blank(), text = element_text(family = "Times")) +
    xlim(0, ceiling(max(selected$lgectpval))) +
    ggtitle(trait)
  
  return(p)
}


plot_selected_traits <- function(pwy, summary_df) {
  
  # Subset summary_df for the given trait
  selected <- summary_df[summary_df$pwy_name == pwy, ]
  
  # Handle duplicated trait names by appending a space
  index <- which(duplicated(selected$trait))
  selected$trait[index] <- paste0(selected$trait[index], " ")
  
  # Transform p-values and order by -log10(pval)
  selected$lgectpval <- -log10(selected$pval_ECT)
  selected <- selected[order(selected$lgectpval, decreasing = TRUE), ]
  
  # Append note to the last num_snp_pass_cutoff value
  selected$num_snp_pass_cutoff[nrow(selected)] <- paste0(selected$num_snp_pass_cutoff[nrow(selected)], " (# of supporting variants)")
  
  # Set pathway names as a factor with levels in the desired order
  selected$trait <- factor(selected$trait, levels = rev(unique(selected$trait)))
  
  # Create the plot
  p <- ggplot(selected) +
    theme_bw(base_line_size = 0.3) +
    geom_col(aes(x = lgectpval, y = trait, fill = celltypes), width = 0.6) +
    geom_text(aes(0, y = trait, label = trait), hjust = 0, nudge_x = 0.1, colour = "white", size = 3, family = "Times") + 
    geom_text(aes(lgectpval, y = trait, label = num_snp_pass_cutoff), hjust = 0, nudge_x = 0.01, colour = "black", size = 3, family = "Times") +
    labs(x = expression(paste("-log"[10], "(ECT p-value)")), y = "Traits") +
    theme(axis.text.y = element_blank(), text = element_text(family = "Times")) +
    xlim(0, ceiling(max(selected$lgectpval))) +
    ggtitle(pwy)
  
  return(p)
}

generate_heatmap <- function(df_all) {
  # Count the number of occurrences for each trait-pwy_name pair
  count_data <- df_all %>%
    dplyr::group_by(trait, pwy_name) %>%
    dplyr::summarise(n = n(), .groups = 'drop')
  
  # Create a matrix for the heatmap
  heatmap_data <- count_data %>%
    tidyr::pivot_wider(names_from = pwy_name, values_from = n, values_fill = list(n = 0)) %>%
    column_to_rownames(var = "trait") %>%
    as.matrix()
  
  # Determine the range of integer values
  value_range <- range(heatmap_data, na.rm = TRUE)
  value_range <- c(floor(value_range[1]), ceiling(value_range[2]))  # Ensure they are integers
  
  # Create a sequence of integers within this range
  int_seq <- seq(value_range[1], value_range[2], by = 1)
  colors <- colorRampPalette(c("white", "red"))(length(int_seq))
  
  # Create a named vector where each integer is associated with a specific color
  color_mapping <- setNames(colors, int_seq)
  
  # Generate the heatmap with the specified color mapping
  ComplexHeatmap::Heatmap(heatmap_data, name = "count",
                          column_title = "Pathway Name", row_title = "Trait",
                          col = color_mapping,  # Use the discrete color mapping
                          border = FALSE, 
                          show_row_dend = FALSE, 
                          show_column_dend = FALSE)
  
  
  
}

generate_heatmap_filter <- function(df_all, min_occurrence = 1) {
  # Count the number of occurrences for each trait-pathway pair
  count_data <- df_all %>%
    dplyr::group_by(trait, pwy_name) %>%
    dplyr::summarise(n = n(), .groups = 'drop')
  
  # Create a matrix for the heatmap
  heatmap_data <- count_data %>%
    tidyr::pivot_wider(names_from = pwy_name, values_from = n, values_fill = list(n = 0)) %>%
    column_to_rownames(var = "trait") %>%
    as.matrix()
  
  # Filter rows and columns based on the minimum number of occurrences
  heatmap_data <- heatmap_data[rowSums(heatmap_data) > min_occurrence, ]
  heatmap_data <- heatmap_data[, colSums(heatmap_data) > min_occurrence]
  
  # Determine the range of integer values
  value_range <- range(heatmap_data, na.rm = TRUE)
  value_range <- c(floor(value_range[1]), ceiling(value_range[2]))  # Ensure they are integers
  
  # Create a sequence of integers within this range
  int_seq <- seq(value_range[1], value_range[2], by = 1)
  colors <- colorRampPalette(c("white", "red"))(length(int_seq))
  
  # Create a named vector where each integer is associated with a specific color
  color_mapping <- setNames(colors, int_seq)
  
  # Generate the heatmap with the specified color mapping
  ComplexHeatmap::Heatmap(heatmap_data, name = "count",
                          column_title = "Pathway Name", row_title = "Trait",
                          col = color_mapping,  # Use the discrete color mapping
                          border = FALSE, 
                          show_row_dend = FALSE, 
                          show_column_dend = FALSE)
}

generate_heatmap_filter_reorder <- function(df_all, min_occurrence = 1, order) {
  # Define the specific order for rows

  
  # Count the number of occurrences for each trait-pathway pair
  count_data <- df_all %>%
    dplyr::group_by(trait, pwy_name) %>%
    dplyr::summarise(n = n(), .groups = 'drop')
  
  # Create a matrix for the heatmap
  heatmap_data <- count_data %>%
    tidyr::pivot_wider(names_from = pwy_name, values_from = n, values_fill = list(n = 0)) %>%
    column_to_rownames(var = "trait") %>%
    as.matrix()
  
  # Filter rows and columns based on the minimum number of occurrences
  heatmap_data <- heatmap_data[rowSums(heatmap_data) > min_occurrence, ]
  heatmap_data <- heatmap_data[, colSums(heatmap_data) > min_occurrence]
  
  
  # Reorder rows according to the specified order. 
  # Make sure all traits in the specified 'order' are present in 'heatmap_data' before reordering
  existing_traits = intersect(order, rownames(heatmap_data)) # Traits present in heatmap_data
  heatmap_data <- heatmap_data[existing_traits, ] # Reorder based on these existing traits
  
  # Ensure that the heatmap data contains only the rows present in the specified 'order'
  ordered_heatmap_data <- heatmap_data[match(order, rownames(heatmap_data)), ]
  
  # Determine the range of integer values
  value_range <- range(ordered_heatmap_data, na.rm = TRUE)
  value_range <- c(floor(value_range[1]), ceiling(value_range[2]))  # Ensure they are integers
  
  # Create a sequence of integers within this range
  int_seq <- seq(value_range[1], value_range[2], by = 1)
  colors <- colorRampPalette(c("white", "red"))(length(int_seq))
  
  # Create a named vector where each integer is associated with a specific color
  color_mapping <- setNames(colors, int_seq)
  
  # Generate the heatmap with the specified color mapping
  # Now, create the heatmap with the ordered data
  ordered_heatmap_data <- ordered_heatmap_data[complete.cases(ordered_heatmap_data),]
  # ComplexHeatmap::Heatmap(ordered_heatmap_data, name = "count",
  #                         column_title = "Pathway Name", row_title = "Trait",
  #                         col = color_mapping,  # Use the discrete color mapping
  #                         border = FALSE, 
  #                         show_row_dend = FALSE, 
  #                         show_column_dend = FALSE,
  #                         cluster_rows = FALSE, # Do not cluster rows
  #                         cluster_columns = FALSE) # Do not cluster columns
  
  ComplexHeatmap::Heatmap(ordered_heatmap_data, name = "count",
                          column_title = "Pathway Name", row_title = "Trait",
                          col = color_mapping,  # Use the discrete color mapping
                          border = FALSE, 
                          show_row_dend = FALSE, 
                          show_column_dend = FALSE,
                          cluster_rows = FALSE,  # Do not cluster rows
                          cluster_columns = FALSE,  # Do not cluster columns
                          column_names_rot = 60,
                          column_names_max_height = unit(20, "cm"))  

}

```

# Heat map for Pathways & traits identified at ECT FDR < 0.2/0.1

The x-axis of the heat map is the pathways identified at different cutoffs (ECT FDR < 0.2/0.1), the y-axis is the traits. At ECT FDR <0.2, only pathways and traits occurred more than one time are shown. At ECT FDR <0.1, all pathway and trait pairs are shown.

## ECT FDR < 0.2

```{r fig.width=15, fig.height=15}
load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT/results/ECT_summary_mbeadded_ectfdr02_allcelltypes.rdata")

print("min occurence for each pathway / trait = 2")

generate_heatmap_filter_reorder(df_all = df_all,min_occurrence = 1, order = order)

```


## ECT FDR < 0.1

```{r }
load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT/results/ECT_summary_mbeadded_ectfdr02_allcelltypes.rdata")

df_all <- df_all[df_all$fdr_ECT < 0.1,]

print("min occurence for each pathway / trait = 1")

generate_heatmap(df_all = df_all)

```





# Heat map for the grouped traits and cell types at ECT FDR < 0.2

We group out traits into 4 groups:

```{r echo=T, eval=T}
innate <- c("eo","gran","neut","mono","myeloid_wbc","wbc","baso")
noninnate <- c("hct","hgb","mch","mchc","mcv","mpv","plt","pdw","pct","rbc","rdw","ret")
lymphcytes <- c("lymph")
diseases <- c("T1D","ukb.allasthma","EUR.IBD","EUR.CD","EUR.UC","allergy")
```

Then we count the number of pairs identified for each group - cell type.

```{r eval=T}
load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT/results/ECT_summary_mbeadded_ectfdr02_allcelltypes.rdata")
df_all$grouped_trait <- case_when(
  df_all$trait %in% innate ~ "Innate",
  df_all$trait %in% noninnate ~ "Noninnate",
  df_all$trait %in% lymphcytes ~ "Lymphocytes",
  df_all$trait %in% diseases ~ "Diseases",
  TRUE ~ as.character(df_all$trait) # Just in case there are traits not categorized
)

aggregated_data <- df_all %>%
  group_by(celltypes, grouped_trait) %>%
  summarise(count = n(), .groups = 'drop')


heatmap_data <- aggregated_data %>%
  pivot_wider(names_from = grouped_trait, values_from = count, values_fill = list(count = 0))

heatmap_matrix <- as.matrix(heatmap_data[,-1]) # Exclude the first column which contains the cell types

rownames(heatmap_matrix) <- heatmap_data$celltypes

Heatmap(heatmap_matrix,
        name = "Count", # Names the color legend
        col = colorRamp2(c(min(heatmap_matrix), mean(heatmap_matrix), max(heatmap_matrix)), c("blue", "white", "red")),
        column_title = "Grouped Traits", # Title for the column
        row_title = "Cell Types", # Title for the row
        column_names_side = "top",
        row_names_side = "left",
        show_row_names = TRUE,
        show_column_names = TRUE,
        show_row_dend = FALSE, 
        show_column_dend = FALSE)
```



# Histogram for the number of pathways identified across traits at ECT FDR < 0.2/0.1

## ECT FDR < 0.2

```{r}
load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT/results/ECT_summary_mbeadded_ectfdr02_allcelltypes.rdata")

num_pwy_pertrait <- lapply(traits,function(trait) {
       sum(!duplicated(df_all$factor[df_all$trait == trait]))
  })

num_pwy_pertrait <- unlist(num_pwy_pertrait)
names(num_pwy_pertrait) <- traits

data_long <- enframe(num_pwy_pertrait, name = "trait", value = "count")

# Plot
ggplot(data_long, aes(x = trait, y = count)) +
  geom_bar(stat = "identity", fill = "skyblue") +
  theme_minimal() +
  labs(title = "Number of Pathways per Trait at FDR < 0.2",
       x = "Trait",
       y = "Number of Pathways") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

plot_selected_pwys(trait = "baso",summary_df = df_all)
plot_selected_pwys(trait = "ret",summary_df = df_all)
plot_selected_pwys(trait = "lymph",summary_df = df_all)


```

## ECT FDR < 0.1

```{r}
load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT/results/ECT_summary_mbeadded_ectfdr02_allcelltypes.rdata")

df_all <- df_all[df_all$fdr_ECT < 0.1,]

num_pwy_pertrait <- lapply(traits,function(trait) {
       sum(!duplicated(df_all$factor[df_all$trait == trait]))
  })

num_pwy_pertrait <- unlist(num_pwy_pertrait)
names(num_pwy_pertrait) <- traits

data_long <- enframe(num_pwy_pertrait, name = "trait", value = "count")

# Plot
ggplot(data_long, aes(x = trait, y = count)) +
  geom_bar(stat = "identity", fill = "skyblue") +
  theme_minimal() +
  labs(title = "Number of Pathways per Trait at FDR < 0.1",
       x = "Trait",
       y = "Number of Pathways") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

plot_selected_pwys(trait = "ret",summary_df = df_all)
plot_selected_pwys(trait = "pct",summary_df = df_all)
plot_selected_pwys(trait = "lymph",summary_df = df_all)


```








# If the highlighted pairs are driven by the same supporting variants (baso, ret and lymph)

```{r message=FALSE, fig.width=10, fig.height=10}

load("/project/xinhe/xsun/pathway_factor/analysis/3.results_analysis/results/supp_snp_pwy_hist_heatmap.rdata")

print("baso: the total pathways identified (at ECT FDR <0.2) is: 18")
print(p_all[[1]])
print(p_heatmap_all[[1]])

print("ret: the total pathways identified (at ECT FDR <0.2) is: 18")
print(p_all[[2]])
print(p_heatmap_all[[2]])

print("lymph: the total pathways identified (at ECT FDR <0.2) is: 10")
print(p_all[[3]])
print(p_heatmap_all[[3]])

```





# MAGMA p-values for the pathways genes

## Top 10 genes with the largest absolute values

```{r, fig.height=20, fig.width=20}
load("/project/xinhe/xsun/pathway_factor/analysis/3.results_analysis/results/magma_p_top_genes.rdata")
p <- list()
for (i in 1:length(magma_added)) {
  
  magma_p <- magma_added[[i]]$magma_p
  
  data_frame <- data.frame(
    Observed = sort(magma_p),
    Theoretical = qunif(ppoints(length(magma_p)))
  )
  
  # Generate the Q-Q plot
  p[[i]] <- ggplot(data_frame, aes(x = Theoretical, y = Observed)) +
    geom_point() +  # Plot the points
    geom_abline(intercept = 0, slope = 1, color = "red", linetype = "dashed") +  # Add a reference line
    xlab("Expected P-values") +  # Label the x-axis
    ylab("Observed P-Values") +  # Label the y-axis
    ggtitle(names(magma_added)[i])
  
}
do.call("grid.arrange", c(p, ncol = 5,nrow=5))


```




## All genes in the pathways

```{r, fig.height=20, fig.width=20}
load("/project/xinhe/xsun/pathway_factor/analysis/3.results_analysis/results/magma_p_all_genes.rdata")
p <- list()
for (i in 1:length(magma_added)) {
  
  magma_p <- magma_added[[i]]$magma_p
  
  data_frame <- data.frame(
    Observed = sort(magma_p),
    Theoretical = qunif(ppoints(length(magma_p)))
  )
  
  # Generate the Q-Q plot
  p[[i]] <- ggplot(data_frame, aes(x = Theoretical, y = Observed)) +
    geom_point() +  # Plot the points
    geom_abline(intercept = 0, slope = 1, color = "red", linetype = "dashed") +  # Add a reference line
    xlab("Expected P-values") +  # Label the x-axis
    ylab("Observed P-Values") +  # Label the y-axis
    ggtitle(names(magma_added)[i])
  
}
do.call("grid.arrange", c(p, ncol = 5,nrow=5))


```





# Histogram for pathways involved in multiple traits at ECT FDR < 0.2/0.1

## ECT FDR < 0.2

```{r}

load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT/results/ECT_summary_mbeadded_ectfdr02_allcelltypes.rdata")

pwys <- df_all$pwy_code[!duplicated(df_all$pwy_code)]

num_trait_perpwy <- lapply(pwys,function(pwy) {
       sum(!duplicated(df_all$trait[df_all$pwy_code == pwy]))
  })

num_trait_perpwy <- unlist(num_trait_perpwy)
names(num_trait_perpwy) <- df_all$pwy_name[match(pwys, df_all$pwy_code)]

num_trait_perpwy <- num_trait_perpwy[num_trait_perpwy>1]

data_long <- enframe(num_trait_perpwy, name = "pwy", value = "count")

# Plot
ggplot(data_long, aes(x = pwy, y = count)) +
  geom_bar(stat = "identity", fill = "skyblue") +
  theme_minimal() +
  labs(title = "Number of Traits per pwy at FDR < 0.1",
       x = "pwy",
       y = "Number of Traits") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))





plot_selected_traits("Cellular senescence",df_all)
plot_selected_traits("cAMP signaling pathway",df_all)





```





## ECT FDR < 0.1

```{r}

load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT/results/ECT_summary_mbeadded_ectfdr02_allcelltypes.rdata")

df_all <- df_all[df_all$fdr_ECT < 0.1,]

pwys <- df_all$pwy_code[!duplicated(df_all$pwy_code)]

num_trait_perpwy <- lapply(pwys,function(pwy) {
       sum(!duplicated(df_all$trait[df_all$pwy_code == pwy]))
  })

num_trait_perpwy <- unlist(num_trait_perpwy)
names(num_trait_perpwy) <- df_all$pwy_name[match(pwys, df_all$pwy_code)]

num_trait_perpwy <- num_trait_perpwy[num_trait_perpwy>1]

data_long <- enframe(num_trait_perpwy, name = "pwy", value = "count")

# Plot
ggplot(data_long, aes(x = pwy, y = count)) +
  geom_bar(stat = "identity", fill = "skyblue") +
  theme_minimal() +
  labs(title = "Number of Traits per pwy at FDR < 0.1",
       x = "pwy",
       y = "Number of Traits") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

plot_selected_traits("Small cell lung cancer",df_all)
```





# Cis genes of supporting variants -- for pairs with ECT FDR < 0.1


```{r}
load("/project/xinhe/xsun/pathway_factor/analysis/3.results_analysis/results/nearby_gene_added.rdata")

df <- df_all[,c("celltypes","factor","trait","pwy_name","fdr_ECT","num_snp_pass_cutoff","gene_near_supp_snps")]
colnames(df) <- c("celltypes","factor","trait","pwy_name","fdr_ECT","#of supporting SNPs","gene near supporting SNPs")
DT::datatable(df)
```


[Locus Zoom plots](https://uchicago.box.com/s/8f349903pupvsubmjvbs6daawrjjla1x)


# If the supporting SNPs are trans-eqtls reported in eQTLGen

Using top 10 genes with largest abs(gene loadings) and the supporting SNPs, we have 7960 gene ~ SNP pairs. 17 of them were reported by eQTLGen

```{r}

load("/project/xinhe/xsun/pathway_factor/analysis/3.results_analysis/results/trans_pairs.rdata")
DT::datatable(pairs_all[pairs_all$trans,])
```



