---
title: "The number of pairs at ECT FDR < 0.1 & 0.2, original ECT strategy"
author: "XSun"
date: "2026-01-05"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---


```{r warning=FALSE, message=FALSE}

library(ggplot2)
library(tidyr)
library(gridExtra)

celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")
names(type) <- celltypes
names(celltypes) <- type

```


```{r warning=FALSE, message=FALSE}

library(ggplot2)
library(tidyr)

celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")
names(type) <- celltypes
names(celltypes) <- type

source("/project/xinhe/xsun/pathway_factor/data_v2/traits_finalselection_SLEremoved.R")
source("/project/xinhe/xsun/r_functions/qqplot_sigle.R")

```


**ECT strategy: When generating “random” GWAS effects of a SNP, we sample from GWAS SNPs with comparable AF with that SNP.**

```{r warning=FALSE, message=FALSE}


df_all_celltype_real <- c()
for(celltype in celltypes){
  
  df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final_FDR02/", celltype,"_ECT_summary_EUR_SLEremoved.RDS"))
  df <- df[df$trait %in% traits_EUR,]
  
  df_all_celltype_real <- rbind(df_all_celltype_real, df)
  
 
}

df_all_celltype_real_5snps <- df_all_celltype_real[df_all_celltype_real$num_supp_SNP >=5, ]
num_candidate_pair_real <- nrow(df_all_celltype_real_5snps)




num_candidate_pair_perm <- c()
df_candidate_pair_perm_all <- c()
for (perm in c(1:10)){
  
  df_all_celltype_perm <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm", perm, "/", celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$trait_id %in% traits_EUR,]
    
    df_all_celltype_perm <- rbind(df_all_celltype_perm, df)
  }
  
  df_all_celltype_perm_5snps <- df_all_celltype_perm[df_all_celltype_perm$num_supp_SNP >=5, ]
  df_candidate_pair_perm_all <- rbind(df_candidate_pair_perm_all,df_all_celltype_perm_5snps)
  num_candidate_pair_perm <- c(num_candidate_pair_perm, nrow(df_all_celltype_perm_5snps))
  
}


```

# QQ plots for the ECT pvalues

```{r warning=FALSE, message=FALSE, fig.width=6, fig.height=3}

qq_real <- qqplot(df_all_celltype_real_5snps$p_ECT, title = "ECT p (real data)")

qq_random <- qqplot(df_candidate_pair_perm_all$p_ECT, title = "ECT p (permutation data, 10 permutation pooled)")

all <- grid.arrange(qq_real,qq_random, nrow = 1)
```

```{r warning=FALSE, message=FALSE, fig.width=12, fig.height=9}

p <- list()
for (perm in c(1:10)){
  
  df_all_celltype_perm <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm", perm, "/", celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$trait_id %in% traits_EUR,]
    
    df_all_celltype_perm <- rbind(df_all_celltype_perm, df)
  }
  
  df_all_celltype_perm_5snps <- df_all_celltype_perm[df_all_celltype_perm$num_supp_SNP >=5, ]
  
  p[[perm]] <- qqplot(df_all_celltype_perm_5snps$p_ECT, title = paste0("Permutation ", perm))
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]],p[[7]],p[[8]],p[[9]],p[[10]], nrow = 3)

```


# The number of pairs at different cutoffs for each permutation and real data

We permuted the genotype and expression data in CEDAR by shuffling sample labels 10 times and performed the entire pipeline on each permuted dataset.

We then examined, across the 10 permutations, the number of pairs passing different cutoffs.



## Cutoff: ECT p < 0.05

```{r warning=FALSE, message=FALSE}

cutoff <- 0.05

## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final_FDR02/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_EUR_SLEremoved.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])


expected_real <- num_candidate_pair_real * cutoff
expected_perm <- num_candidate_pair_perm * cutoff

df_expected_perm <- data.frame(
  Permutation = 1:length(expected_perm),
  expected = expected_perm
)

p <- ggplot(df_long, aes(x = Permutation, y = value)) +
  
  ## observed permutation (solid red)
  geom_line(color = "#d62728", linewidth = 1) +
  geom_point(color = "#d62728", size = 2) +
  
  ## observed real-data reference
  geom_hline(
    yintercept = num_passcutoff_real,
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0(
      "Observed (real): ECT p < ", cutoff, 
      " = ", num_passcutoff_real
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected real data (dashed blue)
  geom_hline(
    yintercept = expected_real,
    linetype = "dashed",
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = expected_real,
    label = paste0(
      "Expected (real): ", num_candidate_pair_real, " * ", cutoff, 
      " = ", round(expected_real, 2)
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected permutation data (dashed red)
  geom_line(
    data = df_expected_perm,
    aes(x = Permutation, y = expected),
    inherit.aes = FALSE,
    linetype = "dashed",
    linewidth = 1,
    color = "#d62728"
  ) +
  
  ggtitle(
    paste0(
      "ECT p < ", cutoff, "\n",
      "dashed: expected, solid: observed\n",
      "blue: real, red: permutation"
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs"
  ) +
  
  scale_x_continuous(breaks = 1:10) +
  
  theme_bw(base_size = 12)


p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```




## Cutoff: ECT p < 0.01

```{r warning=FALSE, message=FALSE}

cutoff <- 0.01


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final_FDR02/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_EUR_SLEremoved.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])

expected_real <- num_candidate_pair_real * cutoff
expected_perm <- num_candidate_pair_perm * cutoff

df_expected_perm <- data.frame(
  Permutation = 1:length(expected_perm),
  expected = expected_perm
)

p <- ggplot(df_long, aes(x = Permutation, y = value)) +
  
  ## observed permutation (solid red)
  geom_line(color = "#d62728", linewidth = 1) +
  geom_point(color = "#d62728", size = 2) +
  
  ## observed real-data reference
  geom_hline(
    yintercept = num_passcutoff_real,
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0(
      "Observed (real): ECT p < ", cutoff, 
      " = ", num_passcutoff_real
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected real data (dashed blue)
  geom_hline(
    yintercept = expected_real,
    linetype = "dashed",
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = expected_real,
    label = paste0(
      "Expected (real): ", num_candidate_pair_real, " * ", cutoff, 
      " = ", round(expected_real, 2)
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected permutation data (dashed red)
  geom_line(
    data = df_expected_perm,
    aes(x = Permutation, y = expected),
    inherit.aes = FALSE,
    linetype = "dashed",
    linewidth = 1,
    color = "#d62728"
  ) +
  
  ggtitle(
    paste0(
      "ECT p < ", cutoff, "\n",
      "dashed: expected, solid: observed\n",
      "blue: real, red: permutation"
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs"
  ) +
  
  scale_x_continuous(breaks = 1:10) +
  
  theme_bw(base_size = 12)

p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```


## Cutoff: ECT p < 0.001

```{r warning=FALSE, message=FALSE}

cutoff <- 0.001


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final_FDR02/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_EUR_SLEremoved.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])

expected_real <- num_candidate_pair_real * cutoff
expected_perm <- num_candidate_pair_perm * cutoff

df_expected_perm <- data.frame(
  Permutation = 1:length(expected_perm),
  expected = expected_perm
)

p <- ggplot(df_long, aes(x = Permutation, y = value)) +
  
  ## observed permutation (solid red)
  geom_line(color = "#d62728", linewidth = 1) +
  geom_point(color = "#d62728", size = 2) +
  
  ## observed real-data reference
  geom_hline(
    yintercept = num_passcutoff_real,
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0(
      "Observed (real): ECT p < ", cutoff, 
      " = ", num_passcutoff_real
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected real data (dashed blue)
  geom_hline(
    yintercept = expected_real,
    linetype = "dashed",
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = expected_real,
    label = paste0(
      "Expected (real): ", num_candidate_pair_real, " * ", cutoff, 
      " = ", round(expected_real, 2)
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected permutation data (dashed red)
  geom_line(
    data = df_expected_perm,
    aes(x = Permutation, y = expected),
    inherit.aes = FALSE,
    linetype = "dashed",
    linewidth = 1,
    color = "#d62728"
  ) +
  
  ggtitle(
    paste0(
      "ECT p < ", cutoff, "\n",
      "dashed: expected, solid: observed\n",
      "blue: real, red: permutation"
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs"
  ) +
  
  scale_x_continuous(breaks = 1:10) +
  
  theme_bw(base_size = 12)
p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```

## Cutoff: ECT p < 0.0001

```{r warning=FALSE, message=FALSE}

cutoff <- 0.0001


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final_FDR02/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_EUR_SLEremoved.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])

expected_real <- num_candidate_pair_real * cutoff
expected_perm <- num_candidate_pair_perm * cutoff

df_expected_perm <- data.frame(
  Permutation = 1:length(expected_perm),
  expected = expected_perm
)

p <- ggplot(df_long, aes(x = Permutation, y = value)) +
  
  ## observed permutation (solid red)
  geom_line(color = "#d62728", linewidth = 1) +
  geom_point(color = "#d62728", size = 2) +
  
  ## observed real-data reference
  geom_hline(
    yintercept = num_passcutoff_real,
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0(
      "Observed (real): ECT p < ", cutoff, 
      " = ", num_passcutoff_real
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected real data (dashed blue)
  geom_hline(
    yintercept = expected_real,
    linetype = "dashed",
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = expected_real,
    label = paste0(
      "Expected (real): ", num_candidate_pair_real, " * ", cutoff, 
      " = ", round(expected_real, 2)
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected permutation data (dashed red)
  geom_line(
    data = df_expected_perm,
    aes(x = Permutation, y = expected),
    inherit.aes = FALSE,
    linetype = "dashed",
    linewidth = 1,
    color = "#d62728"
  ) +
  
  ggtitle(
    paste0(
      "ECT p < ", cutoff, "\n",
      "dashed: expected, solid: observed\n",
      "blue: real, red: permutation"
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs"
  ) +
  
  scale_x_continuous(breaks = 1:10) +
  
  theme_bw(base_size = 12)

p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```


## Cutoff: ECT p < 0.00001

```{r warning=FALSE, message=FALSE}

cutoff <- 0.00001


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final_FDR02/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_EUR_SLEremoved.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])

expected_real <- num_candidate_pair_real * cutoff
expected_perm <- num_candidate_pair_perm * cutoff

df_expected_perm <- data.frame(
  Permutation = 1:length(expected_perm),
  expected = expected_perm
)

p <- ggplot(df_long, aes(x = Permutation, y = value)) +
  
  ## observed permutation (solid red)
  geom_line(color = "#d62728", linewidth = 1) +
  geom_point(color = "#d62728", size = 2) +
  
  ## observed real-data reference
  geom_hline(
    yintercept = num_passcutoff_real,
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0(
      "Observed (real): ECT p < ", cutoff, 
      " = ", num_passcutoff_real
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected real data (dashed blue)
  geom_hline(
    yintercept = expected_real,
    linetype = "dashed",
    color = "#1f77b4"
  ) +
  
  annotate(
    "text",
    x = Inf, y = expected_real,
    label = paste0(
      "Expected (real): ", num_candidate_pair_real, " * ", cutoff, 
      " = ", round(expected_real, 2)
    ),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  ## expected permutation data (dashed red)
  geom_line(
    data = df_expected_perm,
    aes(x = Permutation, y = expected),
    inherit.aes = FALSE,
    linetype = "dashed",
    linewidth = 1,
    color = "#d62728"
  ) +
  
  ggtitle(
    paste0(
      "ECT p < ", cutoff, "\n",
      "dashed: expected, solid: observed\n",
      "blue: real, red: permutation"
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs"
  ) +
  
  scale_x_continuous(breaks = 1:10) +
  
  theme_bw(base_size = 12)

p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```



# Examing the permutation with large number of pairs

## Permutation 2

```{r warning=FALSE, message=FALSE}

permu <- 2

folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm",permu,"/")

df_all_fdr02 <- c()
for(celltype in celltypes){
  
  df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df <- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 <- df[df$ECT_FDR_5suppSNP < 0.2,]
  df_all_fdr02 <- rbind(df_all_fdr02,df_fdr02)
  
}

rownames(df_all_fdr02) <- NULL

df_all_fdr02 <- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP < 0.2,]
df_all_fdr02 <- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP, decreasing = F),]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Pairs with ECT FDR < 0.2 for permutation ', permu)),options = list(pageLength = 10) )
```


## Permutation 7

```{r warning=FALSE, message=FALSE}

permu <- 7

folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm",permu,"/")

df_all_fdr02 <- c()
for(celltype in celltypes){
  
  df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df <- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 <- df[df$ECT_FDR_5suppSNP < 0.2,]
  df_all_fdr02 <- rbind(df_all_fdr02,df_fdr02)
  
}

rownames(df_all_fdr02) <- NULL

df_all_fdr02 <- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP < 0.2,]
df_all_fdr02 <- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP, decreasing = F),]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Pairs with ECT FDR < 0.2 for permutation ', permu)),options = list(pageLength = 10) )
```


## Permutation 8

```{r warning=FALSE, message=FALSE}

permu <- 8

folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary/perm",permu,"/")

df_all_fdr02 <- c()
for(celltype in celltypes){
  
  df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df <- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 <- df[df$ECT_FDR_5suppSNP < 0.2,]
  df_all_fdr02 <- rbind(df_all_fdr02,df_fdr02)
  
}

rownames(df_all_fdr02) <- NULL

df_all_fdr02 <- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP < 0.2,]
df_all_fdr02 <- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP, decreasing = F),]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Pairs with ECT FDR < 0.2 for permutation ', permu)),options = list(pageLength = 10) )
```


### effect size plots for the top pairs (ECT FDR < 0.1)


```{r warning=FALSE, message=FALSE, fig.height=3, fig.width=4}

df_all_fdr01 <- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP < 0.1,]
df_all_fdr01$celltype <- celltypes[df_all_fdr01$celltype]

fdr_cutoff <- 0.2

for (i in 1:nrow(df_all_fdr01)){
  
  celltype <- df_all_fdr01$celltype[i]
  trait <- df_all_fdr01$trait_id[i]
  factor <- df_all_fdr01$factor[i]
  
  file_harmo <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/assoc_harmonize/perm",permu,"/",celltype, "-", trait, "_harmo.RDS")
  dt_harmo <- readRDS(file_harmo)
  
  dat <- dt_harmo[[factor]]
  snp_select <- dat[dat$FDR.exposure < fdr_cutoff,]
  snp_select$chr <- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr <- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit <- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p <- df_all_fdr01$p_ECT[i]
  
  pathwayname <- df$pathwayName[df$factor == factor & df$trait ==trait]
  
  eff <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = "black"),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = "black"),
          text= element_text(family="Times"),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                             "\nECT p-value =", ect_p),
               color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
    geom_hline(yintercept = 0, color = "grey", linetype = "dashed")
  
  
  print(paste0("TOP ",i))
  print(eff)
  
}



```