---
title: "The number of pairs at different cutoffs, ECT: bin AF"
author: "XSun"
date: "2026-01-07"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---


```{r warning=FALSE, message=FALSE}

library(ggplot2)
library(tidyr)

celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
type <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")
names(type) <- celltypes
names(celltypes) <- type

```


**ECT strategy: When generating “random” GWAS effects of a SNP, we sample from GWAS SNPs with comparable AF with that SNP.**


# The number of pairs at different cutoffs for each permutation and real data

We permuted the genotype and expression data in CEDAR by shuffling sample labels 10 times and performed the entire pipeline on each permuted dataset.

We then examined, across the 10 permutations, the number of pairs passing different cutoffs.

## Cutoff: ECT p < 0.05

```{r warning=FALSE, message=FALSE}

cutoff <- 0.05


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary_bin/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_binAF/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])


p <- ggplot(df_long, aes(x = Permutation, y = value, color = metric)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2) +
  
  # dashed reference lines
  geom_hline(yintercept = num_passcutoff_real,
             linetype = "dashed",
             color = "#1f77b4") +
  
  # labels on dashed lines
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0("Real data: pairs at ECT p < ", cutoff," = ", num_passcutoff_real),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  scale_color_manual(
    values = c(
      num_passcutoff= "#1f77b4"
    ),
    labels = c(
      num_passcutoff= paste0("ECT p < ", cutoff)
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs",
    color = NULL
  ) +
  
  theme_bw(base_size = 12) +
  theme(
    legend.position = "top"
  ) + 
  scale_x_continuous(breaks = 1:10)

p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```




## Cutoff: ECT p < 0.01

```{r warning=FALSE, message=FALSE}

cutoff <- 0.01


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary_bin/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_binAF/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])


p <- ggplot(df_long, aes(x = Permutation, y = value, color = metric)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2) +
  
  # dashed reference lines
  geom_hline(yintercept = num_passcutoff_real,
             linetype = "dashed",
             color = "#1f77b4") +
  
  # labels on dashed lines
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0("Real data: pairs at ECT p < ", cutoff," = ", num_passcutoff_real),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  scale_color_manual(
    values = c(
      num_passcutoff= "#1f77b4"
    ),
    labels = c(
      num_passcutoff= paste0("ECT p < ", cutoff)
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs",
    color = NULL
  ) +
  
  theme_bw(base_size = 12) +
  theme(
    legend.position = "top"
  ) + 
  scale_x_continuous(breaks = 1:10)

p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```


## Cutoff: ECT p < 0.001

```{r warning=FALSE, message=FALSE}

cutoff <- 0.001


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary_bin/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_binAF/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])


p <- ggplot(df_long, aes(x = Permutation, y = value, color = metric)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2) +
  
  # dashed reference lines
  geom_hline(yintercept = num_passcutoff_real,
             linetype = "dashed",
             color = "#1f77b4") +
  
  # labels on dashed lines
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0("Real data: pairs at ECT p < ", cutoff," = ", num_passcutoff_real),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  scale_color_manual(
    values = c(
      num_passcutoff= "#1f77b4"
    ),
    labels = c(
      num_passcutoff= paste0("ECT p < ", cutoff)
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs",
    color = NULL
  ) +
  
  theme_bw(base_size = 12) +
  theme(
    legend.position = "top"
  ) + 
  scale_x_continuous(breaks = 1:10)

p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```

## Cutoff: ECT p < 0.0001

```{r warning=FALSE, message=FALSE}

cutoff <- 0.0001


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary_bin/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_binAF/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])


p <- ggplot(df_long, aes(x = Permutation, y = value, color = metric)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2) +
  
  # dashed reference lines
  geom_hline(yintercept = num_passcutoff_real,
             linetype = "dashed",
             color = "#1f77b4") +
  
  # labels on dashed lines
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0("Real data: pairs at ECT p < ", cutoff," = ", num_passcutoff_real),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  scale_color_manual(
    values = c(
      num_passcutoff= "#1f77b4"
    ),
    labels = c(
      num_passcutoff= paste0("ECT p < ", cutoff)
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs",
    color = NULL
  ) +
  
  theme_bw(base_size = 12) +
  theme(
    legend.position = "top"
  ) + 
  scale_x_continuous(breaks = 1:10)

p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```


## Cutoff: ECT p < 0.00001

```{r warning=FALSE, message=FALSE}

cutoff <- 0.00001


## permutation data
num_passcutoff <- c()
for (permu in 1:10){
  
  folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary_bin/perm",permu,"/")
  
  df_all_passcutoff <- c()
  for(celltype in celltypes){
    
    df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
    df <- df[df$num_supp_SNP >=5,]
    
    df_passcutoff <- df[df$p_ECT < cutoff,]
    df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff)
    
  }
  
  rownames(df_all_passcutoff) <- NULL
  num_passcutoff <- c(num_passcutoff, nrow(df_all_passcutoff))
}


df <- data.frame(
  Permutation = seq_along(num_passcutoff),
  num_passcutoff = num_passcutoff
)

df_long <- pivot_longer(
  df,
  cols = c(num_passcutoff),
  names_to = "metric",
  values_to = "value"
)

## real data
folder_ect_summary_real <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_binAF/"

df_all_real <- c()
for(celltype in celltypes){
  
  df_real <- readRDS(paste0(folder_ect_summary_real, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df_real <- df_real[df_real$num_supp_SNP >=5,]
  
  df_all_real <- rbind(df_all_real,df_real)
  
}

num_passcutoff_real <- nrow(df_all_real[df_all_real$p_ECT < cutoff,])


p <- ggplot(df_long, aes(x = Permutation, y = value, color = metric)) +
  geom_line(linewidth = 1) +
  geom_point(size = 2) +
  
  # dashed reference lines
  geom_hline(yintercept = num_passcutoff_real,
             linetype = "dashed",
             color = "#1f77b4") +
  
  # labels on dashed lines
  annotate(
    "text",
    x = Inf, y = num_passcutoff_real,
    label = paste0("Real data: pairs at ECT p < ", cutoff," = ", num_passcutoff_real),
    color = "#1f77b4",
    hjust = 1.05, vjust = -0.5,
    size = 3.5
  ) +
  
  scale_color_manual(
    values = c(
      num_passcutoff= "#1f77b4"
    ),
    labels = c(
      num_passcutoff= paste0("ECT p < ", cutoff)
    )
  ) +
  
  labs(
    x = "Permutation",
    y = "Number of pairs",
    color = NULL
  ) +
  
  theme_bw(base_size = 12) +
  theme(
    legend.position = "top"
  ) + 
  scale_x_continuous(breaks = 1:10)

p

DT::datatable(df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0("Number of pairs passing ECT p < ", cutoff)),options = list(pageLength = 10) )

```


<!-- # Examing the permutation with large number of pairs -->

<!-- ## Permutation 2 -->

<!-- ```{r warning=FALSE, message=FALSE} -->

<!-- permu <- 2 -->

<!-- folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary_bin/perm",permu,"/") -->

<!-- df_all_passcutoff <- c() -->
<!-- for(celltype in celltypes){ -->

<!--   df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS")) -->
<!--   df <- df[complete.cases(df$ECT_FDR_5suppSNP),] -->

<!--   df_passcutoff <- df[df$ECT_FDR_5suppSNP < 0.2,] -->
<!--   df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff) -->

<!-- } -->

<!-- rownames(df_all_passcutoff) <- NULL -->

<!-- df_all_passcutoff <- df_all_passcutoff[df_all_passcutoff$ECT_FDR_5suppSNP < 0.2,] -->
<!-- df_all_passcutoff <- df_all_passcutoff[order(df_all_passcutoff$ECT_FDR_5suppSNP, decreasing = F),] -->

<!-- DT::datatable(df_all_passcutoff,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Pairs with ECT FDR < 0.2 for permutation ', permu)),options = list(pageLength = 10) ) -->
<!-- ``` -->

<!-- ### effect size plots for the top pairs (ECT FDR < 0.1) -->


<!-- ```{r warning=FALSE, message=FALSE, fig.height=3, fig.width=4} -->

<!-- df_all_fdr01 <- df_all_passcutoff[df_all_passcutoff$ECT_FDR_5suppSNP < 0.1,] -->
<!-- df_all_fdr01$celltype <- celltypes[df_all_fdr01$celltype] -->

<!-- fdr_cutoff <- 0.2 -->

<!-- for (i in 1:nrow(df_all_fdr01)){ -->

<!--   celltype <- df_all_fdr01$celltype[i] -->
<!--   trait <- df_all_fdr01$trait_id[i] -->
<!--   factor <- df_all_fdr01$factor[i] -->

<!--   file_harmo <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/assoc_harmonize/perm",permu,"/",celltype, "-", trait, "_harmo.RDS") -->
<!--   dt_harmo <- readRDS(file_harmo) -->

<!--   dat <- dt_harmo[[factor]] -->
<!--   snp_select <- dat[dat$FDR.exposure < fdr_cutoff,] -->
<!--   snp_select$chr <- as.numeric(as.character(snp_select$chr.exposure)) -->
<!--   snp_select$chr <- factor(snp_select$chr, levels = sort(unique(snp_select$chr))) -->

<!--   fit <- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure) -->

<!--   ect_p <- df_all_fdr01$p_ECT[i] -->

<!--   pathwayname <- df$pathwayName[df$factor == factor & df$trait ==trait] -->

<!--   eff <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) + -->
<!--     geom_point(size = 4) +  -->
<!--     geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) + -->
<!--     theme_bw(base_line_size = 0.3) + -->
<!--     theme(axis.title.x = element_text(size = 16), -->
<!--           axis.text.x = element_text(size = 14, color = "black"), -->
<!--           axis.title.y = element_text(size =16), -->
<!--           axis.text.y = element_text(size = 14, color = "black"), -->
<!--           text= element_text(family="Times"), -->
<!--           legend.text = element_text(size=12), -->
<!--           legend.title = element_text(size=12)) +  -->
<!--     geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) + -->
<!--     geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1, -->
<!--                label = paste("R^2 = ", signif(summary(fit)$r.squared, 3),  -->
<!--                              "\nECT p-value =", ect_p), -->
<!--                color = "black", family = "Times", size =3, label.size = NA,fill=NA) + -->
<!--     xlab(expression(hat(beta)[factor])) +  -->
<!--     ylab(expression(hat(beta)[gwas])) + -->
<!--     geom_vline(xintercept = 0, color = "grey", linetype = "dashed") + -->
<!--     geom_hline(yintercept = 0, color = "grey", linetype = "dashed") -->


<!--   print(paste0("TOP ",i)) -->
<!--   print(eff) -->

<!-- } -->



<!-- ``` -->


<!-- ## Permutation 6 -->

<!-- ```{r warning=FALSE, message=FALSE} -->

<!-- permu <- 6 -->

<!-- folder_ect_summary <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_CEDAR_permu/ECT_summary_bin/perm",permu,"/") -->

<!-- df_all_passcutoff <- c() -->
<!-- for(celltype in celltypes){ -->

<!--   df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS")) -->
<!--   df <- df[complete.cases(df$ECT_FDR_5suppSNP),] -->

<!--   df_passcutoff <- df[df$ECT_FDR_5suppSNP < 0.2,] -->
<!--   df_all_passcutoff <- rbind(df_all_passcutoff,df_passcutoff) -->

<!-- } -->

<!-- rownames(df_all_passcutoff) <- NULL -->

<!-- df_all_passcutoff <- df_all_passcutoff[df_all_passcutoff$ECT_FDR_5suppSNP < 0.2,] -->
<!-- df_all_passcutoff <- df_all_passcutoff[order(df_all_passcutoff$ECT_FDR_5suppSNP, decreasing = F),] -->

<!-- DT::datatable(df_all_passcutoff,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',paste0('Pairs with ECT FDR < 0.2 for permutation ', permu)),options = list(pageLength = 10) ) -->
<!-- ``` -->




