---
title: "Tables and Figures"
author: "XSun"
date: "2025-11-05"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

```{r, warning=F, message=F}

library(gridExtra)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(EnsDb.Hsapiens.v75)
library(ggrepel)

celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
types <- c("B cell (CD19+)","Monocyte (CD14+)","Granulocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")
names(celltypes) <- types
names(types) <- celltypes

source("/project/xinhe/xsun/pathway_factor/data_v2/traits_finalselection_SLEremoved.R")

folder_ect_summary <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/"

qqplot <- function(pvalues,title=NULL) {
  
  pval <- pvalues[complete.cases(pvalues)]
  title <- title
  plotdata <- data.frame(observed = -log10(sort(as.numeric(pval))),
                         expected = -log10(ppoints(length(pval))))
  
  qq <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5,color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(title) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 14),
          axis.text.x = element_text(size = 12, color = "black"),
          axis.title.y = element_text(size = 14),
          axis.text.y = element_text(size = 12, color = "black"),
          text= element_text(family="Times"))
  
  return(qq)
  
}


```


# Main text table

## Table 1

```{r fig.height=3, fig.width=9, warning=F, message=F}

df_all_fdr02 <- c()
for(celltype in celltypes){
  
  df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EUR_SLEremoved_noMHCadded.RDS"))
  df <- df[complete.cases(df$ECT_FDR_5suppSNP),]
  
  df_fdr02 <- df[df$ECT_FDR_5suppSNP < 0.2,]
  df_all_fdr02 <- rbind(df_all_fdr02,df_fdr02)
  
}


df_all_fdr02$celltype <- types[df_all_fdr02$celltype]
df_all_fdr02 <- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP),]

df_all_fdr01 <- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP < 0.1,]

DT::datatable(df_all_fdr01,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Pairs with ECT FDR <= 0.1 '),options = list(pageLength = 10) )

```


# Main text figures

## Figure 3 

```{r fig.height=6, fig.width=12, warning=F, message=F}

trait <- "RET-ebi-a-GCST90002405"

df_trait_fdr01 <- c()
df_trait_fdr02 <- c()
for(celltype in celltypes){
  
  df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_EUR_SLEremoved_noMHCadded.RDS"))
  
  df_fdr01 <- df[df$trait == trait & df$ECT_FDR_5suppSNP< 0.1, ]
  df_fdr01 <- df_fdr01[complete.cases(df_fdr01$ECT_FDR_5suppSNP),]
  df_trait_fdr01 <- rbind(df_trait_fdr01,df_fdr01)
  
  df_fdr02 <- df[df$trait == trait & df$ECT_FDR_5suppSNP< 0.2, ]
  df_fdr02 <- df_fdr02[complete.cases(df_fdr02$ECT_FDR_5suppSNP),]
  df_trait_fdr02 <- rbind(df_trait_fdr02,df_fdr02)
  
}

df_trait_fdr02$celltype <- types[df_trait_fdr02$celltype]

# Prepare data
selected <- df_trait_fdr02 %>%
  mutate(
    lgectpval = -log10(p_ECT),
    pathwayName = paste0(pathwayName," ",factor)
  )

# Plot
p <- ggplot(selected) +
  geom_col(
    aes(x = lgectpval, 
        y = reorder(pathwayName, lgectpval), 
        fill = celltype),
    width = 0.75
  ) +
  geom_text(
    aes(0, y = pathwayName, label = pathwayName),
    hjust = 0,
    nudge_x = 0.1,
    colour = "white",
    size = 4.5,
    family = "Times"
  ) +
  geom_text(
    aes(lgectpval, y = pathwayName, label = num_supp_SNP),
    hjust = 0,
    nudge_x = 0.05,
    colour = "black",
    size = 4.5,
    family = "Times"
  ) +
  labs(
    x = expression(paste("-log"[10], "(ECT p-value)")),
    y = NULL,
    title = paste0(unique(selected$trait)," ECT FDR < 0.2")
  ) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    text = element_text(family = "Times"),
    plot.title = element_text(hjust = 0.5)
  ) +
  xlim(0, 5)

p
```

- Platelets

Activated platelets, which are recruited to sites of vascular injury, are known to release a variety of chemokines (Y. Chen et al.,2020). The resulting chemokine-receptor interactions are crucial regulators of hematopoiesis (Bonavita et al., 2018; Broxmeyer, 2008; Duchene et al., 2017), which can impact the reticulocyte counts.


1. Hippo signaling pathway

Hippo signalling pathway plays a crucial role in human megakaryoblastic cell differentiation and thrombopoiesis.[PMID: 27786336,27903710,27515249]. megakaryocytes' main function is producing platelets [PMID 26787738]

2. Purine metabolism

purines have been shown to play an important role in the physiology of platelets[PMID 37108190]. Platelets contain very high levels of adenine nucleotides (like ADP and ATP), which are stored in granules and released during activation by stimuli such as thrombin or collagen. Released ADP promotes aggregation, while other purine metabolites—ATP, AMP, adenosine, and AP4—inhibit platelet aggregation. [(]Agarwal, K.C. (1996). Purines and Regulation of Platelet Activation. In: Abd-Elfattah, AS.A., Wechsler, A.S. (eds) Purines and Myocardial Protection. Developments in Cardiovascular Medicine, vol 181. Springer, Boston, MA.]

3. C-type lectin receptor signaling pathway

C-type lectin receptor signaling—especially via CLEC-2—drives strong platelet activation, aggregation, and thrombo-inflammatory responses through Syk-dependent pathways.[PMID: 25150298]

4. Glutamatergic synapse 

glutamate activates AMPA receptors on platelet surface. AMPAR signaling increases platelet activation and promotes thrombosis, while blocking AMPAR reduces platelet activation. [PMID: 18283118]

5. cAMP signaling pathway

The cAMP signaling pathway represents the most potent endogenous system for the control of platelet activation through its ability to modulate multiple platelet functions[PMID: 26617518, 24100445, 26617518]


6. mTOR signaling pathway

platelets use an mTOR/S6K1 pathway to activate Rac1 and regulate platelet spreading on fibrinogen [PMID: 21757621]. mTOR regulates GPVI-mediated platelet activation [PMID: 33971888]

7. PI3K-Akt signaling pathway

The PI3K/Akt signaling pathway has been implicated in playing an important role in platelet activation during hemostasis and thrombosis [PMID: 30972780]




## Figure 4

### example 1, NF-kappa B signaling pathway - Platelet - HB-ebi-a-GCST90002384

```{r fig.height=3, fig.width=7.7, warning=F, message=F}

celltype <- "platelet"
trait <- "HB-ebi-a-GCST90002384"
factor <- "04064_PC1"
fdr_cutoff <- 0.2
  
file_harmo <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/",celltype, "-", trait, "_harmo.RDS")
dt_harmo <- readRDS(file_harmo)

df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EUR_SLEremoved_noMHCadded.RDS"))

dat <- dt_harmo[[factor]]
snp_select <- dat[dat$FDR.exposure < fdr_cutoff,]
snp_select$chr <- as.numeric(as.character(snp_select$chr.exposure))
snp_select$chr <- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))

fit <- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)

ect_p <- df$p_ECT[df$factor == factor & df$trait ==trait]

pathwayname <- df$pathwayName[df$factor == factor & df$trait ==trait]

eff <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
  geom_point(size = 4) + 
  geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
  theme_bw(base_line_size = 0.3) +
  theme(axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = "black"),
        axis.title.y = element_text(size =16),
        axis.text.y = element_text(size = 14, color = "black"),
        text= element_text(family="Times"),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12)) + 
  geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
  geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
             label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                           "\nECT p-value =", ect_p),
             color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
  xlab(expression(hat(beta)[factor])) + 
  ylab(expression(hat(beta)[gwas])) +
  geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
  geom_hline(yintercept = 0, color = "grey", linetype = "dashed")


#snp_select <- dat[dat$FDR.exposure > fdr_cutoff & dat$steiger_dir ==T,]
snp_select <- dat[dat$FDR.exposure > fdr_cutoff,]

fit <- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
coef <- summary(fit)
p <- round(coef$coefficients[,4], digits = 4)
#ect_p <- df$p_ECT[df$factor == factor & df$trait ==trait]

eff_rev <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome)) +
  geom_point(size = 4, color ="steelblue") + 
  theme_bw(base_line_size = 0.3) +
  theme(axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = "black"),
        axis.title.y = element_text(size =16),
        axis.text.y = element_text(size = 14, color = "black"),
        text= element_text(family="Times"),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12)) + 
  geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
  geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
             label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                           "\n p-value =", p),
             color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
  xlab(expression(hat(beta)[factor])) + 
  ylab(expression(hat(beta)[gwas])) +
  geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
  geom_hline(yintercept = 0, color = "grey", linetype = "dashed")



file_gene_loading <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/gene_loading/",celltype, "_", factor, "-", trait, ".RDS")
gene_loadings_pc <- readRDS(file_gene_loading)

gene_loadings_pc_abs <- abs(gene_loadings_pc)
gene_loadings_pc_abs_sort <- sort(gene_loadings_pc_abs,decreasing = T)
topgenes <- names(gene_loadings_pc_abs_sort)[1:10]

gene_loadings_pc <- as.data.frame(cbind(names(gene_loadings_pc),gene_loadings_pc))
names(gene_loadings_pc) <- c("gene_name","gene_loadings_pc")

# Extract gene information based on the gene symbol
gene_info <- genes(EnsDb.Hsapiens.v75, filter = ~ gene_name ==  gene_loadings_pc$gene_name)
gene_info <- as.data.frame(gene_info)
gene_info <- gene_info[!duplicated(gene_info$gene_name),]

gene_info <- merge(gene_info,gene_loadings_pc, by="gene_name")
gene_info <- gene_info[gene_info$seqnames %in%c(1:22),]

loading_plot <- gene_info[,c("seqnames","start","gene_name","gene_loadings_pc")]
colnames(loading_plot) <- c("chromosome_name","start_position","gene_name","loadings")
loading_plot$loadings <- round(as.numeric(loading_plot$loadings),digits = 4)

don <- loading_plot %>%
  
  # Compute chromosome size
  group_by(chromosome_name) %>%
  summarise(chr_len=max(start_position)) %>%
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(loading_plot, ., by=c("chromosome_name"="chromosome_name")) %>%
  
  # Add a cumulative position of each SNP
  arrange(chromosome_name, start_position) %>%
  mutate( start_positioncum=start_position+tot) %>%
  
  # Add highlight and annotation information
  #mutate( is_annotate=ifelse(gene_name %in% loading_plot_top10$gene_name, "yes", "no"))
  mutate( is_annotate=ifelse(gene_name %in% topgenes, "yes", "no"))
# Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
axisdf = don %>% group_by(chromosome_name) %>% summarize(center=( max(start_positioncum) + min(start_positioncum) ) / 2 )

manhplot_gene <- ggplot(don, aes(x=start_positioncum, y=loadings)) +
  
  # Show all points
  #rasterise(geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3),dpi=80 )+
  geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3) +
  scale_color_manual(values = rep(c("red", "steelblue"), 22)) +
  
  # custom X axis:
  scale_x_continuous( label = as.integer(axisdf$chromosome_name), breaks= axisdf$center ) +
  #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  
  #ylab(expression(paste("-log"[10],"(p-value)"))) +
  #ylim(0,max(-log10(don$loadings)) + 2) +
  #ylim(min(as.numeric(don$loadings)),max(as.numeric(don$loadings))) +
  #ylab("") +
  ylab("Gene Loadings") +
  xlab("Chromosome") +
  
  # Custom the theme:
  theme_bw() +
  #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
  theme(
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_text(size = 18),
    axis.text.x = element_text(size = 16, color = "black"),
    axis.title.y = element_text(size = 18),
    axis.text.y = element_text(size = 16, color = "black"),
    text= element_text(family="Times"),
    #axis.text.y=element_blank(),
    #axis.ticks.y=element_blank()
  ) +
  geom_label_repel( data=subset(don, is_annotate=="yes"), aes(label=gene_name), size=4)


cat(paste0(celltype,"-", pathwayname, "-", trait ))


all <- grid.arrange(eff, eff_rev, nrow = 1)

print(manhplot_gene)
print(paste0("Top 10 genes with largest abs loading: ", paste0(topgenes, collapse = ",")))

```


- NF-kappa B signaling pathway -> Platelet -> hemoglobin

NF-κB activation in platelets promotes SNAP-23–dependent granule secretion, increases P-selectin surface expression, stimulates TXA₂ and ADP release, and facilitates inside-out activation of αIIbβ3 integrin to drive aggregation and thrombus formation. NF-κB activation in endothelial cells under inflammatory conditions upregulates adhesion molecules, enhancing platelet binding and promoting a pro-thrombotic vascular environment. [PMID: 31461836]

Thrombocytopenia increases mucosal and internal bleeding risk, leading to iron-deficiency anemia and reduced hemoglobin levels [PMID: 15248165]. Inflammatory cytokines such as IL-6 elevate platelet counts [PMID: 7951005] and induce hepcidin, which decreases iron availability and lowers hemoglobin [PMID: 15124018]. Platelet-derived S1P maintains endothelial barrier integrity; when this pathway is disrupted, vascular leak and red blood cell loss can occur [PMID: 36142188]. Additionally, conditions characterized by excessive platelet activation (e.g., TTP, HUS, DIC) produce platelet-rich microthrombi, causing schistocyte formation and hemolysis [PMID: 32310498], ultimately leading to hemoglobin reduction.

Also, Activated platelets, which are recruited to sites of vascular injury, are known to release a variety of chemokines (Y. Chen et al.,2020). The resulting chemokine-receptor interactions are crucial regulators of hematopoiesis (Bonavita et al., 2018; Broxmeyer, 2008; Duchene et al., 2017), leading to the change of hemoglobin.

- TOP loading genes -> Platelet


1. LY96

LY96 (MD-2) is the LPS-binding component of the TLR4–MD2 receptor complex. LPS binds directly to MD-2, enabling TLR4 activation and downstream inflammatory signaling. In platelets, LPS–TLR4–MD2 signaling promotes platelet activation and secretion [PMID: 19494325].

2. TNFRSF1A

TNFRSF1A encodes TNF receptor 1 (TNFR1). TNF-α signaling through TNFR1 enhances platelet activation and procoagulant activity, indicating a direct stimulatory effect of TNF-α on platelets [PMID: 26819099; 8679214].

3. PLCG2

PLCG2 encodes PLCγ2, the essential phospholipase required for platelet activation downstream of the collagen receptor GPVI, as well as integrin αIIbβ3 outside-in signaling. Loss of PLCγ2 severely impairs platelet activation [PMID: 25793864].

4. CD14

CD14 is a co-receptor that facilitates LPS–TLR4 signaling. Although platelets do not express membrane-bound CD14, soluble CD14 enables platelet TLR4 responsiveness. Together with LY96/MD-2, CD14 contributes to LPS-induced platelet activation and secretion [PMID: 24585776].

5. NFKBIA

NFKBIA encodes IκBα, the major inhibitory protein that restrains NF-κB activity. Because NF-κB regulates platelet activation, granule secretion, and integrin signaling, NFKBIA contributes to the control of NF-κB–dependent platelet responses, as discussed before.

6. NFKB1

NFKB1 encodes the NF-κB subunit p50, a core component of the platelet NF-κB signaling machinery. NF-κB activation in platelets influences secretion, aggregation, and thrombus formation, linking NFKB1 to platelet inflammatory and functional responses, as discussed before.

7. CFLAR

CFLAR encodes c-FLIP, a key anti-apoptotic regulator that modulates caspase-8 activity. c-FLIP affects TNF-α–driven signaling pathways [PMID: 23070002] and participates in mechanisms that regulate platelet lifespan and survival.

8. PARP1

PARP1 is a major substrate of activated caspases. Caspase signaling has both apoptotic and non-apoptotic roles in hematopoiesis, including megakaryocyte differentiation and platelet maturation. [PMID: 21176168].


<!-- 1. LY96 -->

<!-- LY96 (MD-2) is the LPS-binding component of the TLR4 receptor. LPS binds to MD-2, enabling TLR4–MD2 activation and downstream inflammatory signaling. LPS stimulates platelet secretion [PMID: 19494325].  -->

<!-- 2. TNFRSF1A -->

<!-- TNFRSF1A is TNF receptor 1. Increased levels of TNF-α are known to activate platelets[PMID: 26819099, 8679214].  -->

<!-- 3. PLCG2 -->

<!-- PLCG2 encodes PLCγ2, which is required for platelet activation. [PMID: 25793864] -->

<!-- 4. CD14 -->

<!-- CD14 is a coreceptor for LPS signaling through TLR4. Combine with the evidence of LY96, it is related to latelet secretion.[PMID: 24585776] -->

<!-- 5. NFKBIA  -->

<!-- NFKBIA is the gene that encodes IκBα, the major inhibitory protein that keeps NF-κB inactive under resting conditions. We have discussed the role of  NF-κB in platelets before.  -->

<!-- 6. NFKB1 -->

<!-- Critical part of platelet NF-κB machinery. We have discussed the role of  NF-κB in platelets -->

<!-- 7. CFLAR  -->

<!-- CFLAR encodes c-FLIP, is a master anti-apoptotic regulator and resistance factor that suppresses tumor necrosis factor-α (TNF-α) [PMID: 23070002], which has been discussed earlier.  -->

<!-- 8. PARP1  -->

<!-- PARP1 is a key substrate of activated caspases, which are central regulators of apoptosis and also have important non-apoptotic roles in hematopoiesis, including platelet maturation. [PMID: 21176168] -->

### example 2, Leukocyte transendothelial migration - T cell (CD 8+) - IBD-ebi-a-GCST003043

```{r fig.height=3, fig.width=7.7, warning=F, message=F}

celltype <- "thymocyte"
trait <- "IBD-ebi-a-GCST003043"
factor <- "04670_PC1"
fdr_cutoff <- 0.2
  
file_harmo <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/",celltype, "-", trait, "_harmo.RDS")
dt_harmo <- readRDS(file_harmo)

df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EUR_SLEremoved_noMHCadded.RDS"))

dat <- dt_harmo[[factor]]
snp_select <- dat[dat$FDR.exposure < fdr_cutoff,]
snp_select$chr <- as.numeric(as.character(snp_select$chr.exposure))
snp_select$chr <- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))

fit <- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)

ect_p <- df$p_ECT[df$factor == factor & df$trait ==trait]

pathwayname <- df$pathwayName[df$factor == factor & df$trait ==trait]

eff <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
  geom_point(size = 4) + 
  geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
  theme_bw(base_line_size = 0.3) +
  theme(axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = "black"),
        axis.title.y = element_text(size =16),
        axis.text.y = element_text(size = 14, color = "black"),
        text= element_text(family="Times"),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12)) + 
  geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
  geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
             label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                           "\nECT p-value =", ect_p),
             color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
  xlab(expression(hat(beta)[factor])) + 
  ylab(expression(hat(beta)[gwas])) +
  geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
  geom_hline(yintercept = 0, color = "grey", linetype = "dashed")


#snp_select <- dat[dat$FDR.exposure > fdr_cutoff & dat$steiger_dir ==T,]
snp_select <- dat[dat$FDR.exposure > fdr_cutoff,]

fit <- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
coef <- summary(fit)
p <- round(coef$coefficients[,4], digits = 4)
#ect_p <- df$p_ECT[df$factor == factor & df$trait ==trait]

eff_rev <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome)) +
  geom_point(size = 4, color ="steelblue") + 
  theme_bw(base_line_size = 0.3) +
  theme(axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 14, color = "black"),
        axis.title.y = element_text(size =16),
        axis.text.y = element_text(size = 14, color = "black"),
        text= element_text(family="Times"),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12)) + 
  geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
  geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
             label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                           "\n p-value =", p),
             color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
  xlab(expression(hat(beta)[factor])) + 
  ylab(expression(hat(beta)[gwas])) +
  geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
  geom_hline(yintercept = 0, color = "grey", linetype = "dashed")



file_gene_loading <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/gene_loading/",celltype, "_", factor, "-", trait, ".RDS")
gene_loadings_pc <- readRDS(file_gene_loading)

gene_loadings_pc_abs <- abs(gene_loadings_pc)
gene_loadings_pc_abs_sort <- sort(gene_loadings_pc_abs,decreasing = T)
topgenes <- names(gene_loadings_pc_abs_sort)[1:10]

gene_loadings_pc <- as.data.frame(cbind(names(gene_loadings_pc),gene_loadings_pc))
names(gene_loadings_pc) <- c("gene_name","gene_loadings_pc")

# Extract gene information based on the gene symbol
gene_info <- genes(EnsDb.Hsapiens.v75, filter = ~ gene_name ==  gene_loadings_pc$gene_name)
gene_info <- as.data.frame(gene_info)
gene_info <- gene_info[!duplicated(gene_info$gene_name),]

gene_info <- merge(gene_info,gene_loadings_pc, by="gene_name")
gene_info <- gene_info[gene_info$seqnames %in%c(1:22),]

loading_plot <- gene_info[,c("seqnames","start","gene_name","gene_loadings_pc")]
colnames(loading_plot) <- c("chromosome_name","start_position","gene_name","loadings")
loading_plot$loadings <- round(as.numeric(loading_plot$loadings),digits = 4)

don <- loading_plot %>%
  
  # Compute chromosome size
  group_by(chromosome_name) %>%
  summarise(chr_len=max(start_position)) %>%
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(loading_plot, ., by=c("chromosome_name"="chromosome_name")) %>%
  
  # Add a cumulative position of each SNP
  arrange(chromosome_name, start_position) %>%
  mutate( start_positioncum=start_position+tot) %>%
  
  # Add highlight and annotation information
  #mutate( is_annotate=ifelse(gene_name %in% loading_plot_top10$gene_name, "yes", "no"))
  mutate( is_annotate=ifelse(gene_name %in% topgenes, "yes", "no"))
# Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
axisdf = don %>% group_by(chromosome_name) %>% summarize(center=( max(start_positioncum) + min(start_positioncum) ) / 2 )

manhplot_gene <- ggplot(don, aes(x=start_positioncum, y=loadings)) +
  
  # Show all points
  #rasterise(geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3),dpi=80 )+
  geom_point( aes(color=as.factor(chromosome_name)), alpha=0.8, size=3) +
  scale_color_manual(values = rep(c("red", "steelblue"), 22)) +
  
  # custom X axis:
  scale_x_continuous( label = as.integer(axisdf$chromosome_name), breaks= axisdf$center ) +
  #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  
  #ylab(expression(paste("-log"[10],"(p-value)"))) +
  #ylim(0,max(-log10(don$loadings)) + 2) +
  #ylim(min(as.numeric(don$loadings)),max(as.numeric(don$loadings))) +
  #ylab("") +
  ylab("Gene Loadings") +
  xlab("Chromosome") +
  
  # Custom the theme:
  theme_bw() +
  #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
  theme(
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_text(size = 18),
    axis.text.x = element_text(size = 16, color = "black"),
    axis.title.y = element_text(size = 18),
    axis.text.y = element_text(size = 16, color = "black"),
    text= element_text(family="Times"),
    #axis.text.y=element_blank(),
    #axis.ticks.y=element_blank()
  ) +
  geom_label_repel( data=subset(don, is_annotate=="yes"), aes(label=gene_name), size=4)


cat(paste0(celltype,"-", pathwayname, "-", trait ))


all <- grid.arrange(eff, eff_rev, nrow = 1)

print(manhplot_gene)
print(paste0("Top 10 genes with largest abs loading: ", paste0(topgenes, collapse = ",")))

```


- Leukocyte transendothelial migration -> CD8⁺ T cells -> IBD

Leukocyte transendothelial migration (TEM) is the process by which white blood cells (leukocytes) exit the bloodstream and enter tissues [PMID: 21073340, 33855018]. Because CD8⁺ T cells are leukocytes, TEM directly enables their recruitment to inflammatory sites, allowing them to exert their cytotoxic and cytokine-mediated functions.

Emerging evidence shows that CD8⁺ T cells play important roles in IBD. Chronic intestinal infection can drive epithelial barrier leakage through type I IFN signaling and CD8⁺ T-cell activity, and blocking CD8⁺ T cells prevents this barrier breakdown [PMID: 32880630]. At the same time, CD8⁺ T cells in IBD are highly heterogeneous: cytotoxic (Tc1) and IL-17–producing (Tc17) subsets promote inflammation, whereas other CD8⁺ subsets may be neutral or even protective [PMID: 34707610].

- TOP loading genes -> CD8⁺ T cells

1 & 2 . ITGAL & ITGB2

<!-- ITGAL is essential for CD8⁺ T-cell activation, where its product, the CD11a α-subunit of LFA-1, supports magnesium-dependent immune synapse formation and antigen-specific cytotoxicity [PMID: 39605903].  -->

ITGAL and ITGB2 form LFA-1, which has been shown to facilitate T-cell activation by functioning as a scavenger receptor on CD8⁺ dendritic cells, enabling them to acquire antigen from antigen-bearing DCs [PMID: 17641014]. 

3. RAC2 

RAC2 is a member of the Rac subfamily of Rho GTPases. It plays essential roles in T-cell development, cytoskeletal remodeling, and survival[PMID: 19088377, 11581314, 18579797].  **In the absence of Rac2, the motility of T cells in response to CXCL12, CCL19, and CCL21 was partially reduced [PMID: 20870900]**

4. SIPA1

Sipa1 was shown to inhibit  thymocyte development [PMID: 37545505]

5. PIK3R2 

PIK3R2 is a key component of the PI3K–Akt signaling pathway, which plays a central role in T-cell exhaustion[PMID  17164340]

6. MAPK13

MAPK13 phosphorylates TCF1, leading to increase of stem-like T cells [PMID 38272565].

7. MSN 

MSN (Moesin) are implicated in promoting T cell activation and polarity [PMID: 19124745]

8. ACTN4

**ACTN4 has been shown to play important roles in cytoskeleton organization, cell adhesion, and cell migration [PMID: 25885339]**


# SI figures

The main change for figure S1-S3 is: more factors are included since we decreased the gene num cutoff from 40 to 20.

## S1 hsq and SE

```{r fig.height=5, fig.width=7.7, warning=F, message=F}

p1 <- list()
p2 <- list()
for (celltype in celltypes){
  
  file_hsq <- paste0("/project/xinhe/xsun/pathway_factor/analysis/7.analysis_for_factors/results/heritability/hsq_summary_", celltype,".txt")
  
  H <- read.table(file_hsq, header = T)
  sum_sorted <- H[order(as.numeric(H$h2)),]
  sum_sorted$order <- seq(1,nrow(H),by=1)
  
  p1[[celltype]] <- ggplot(sum_sorted, aes(x=order, y=h2)) + theme_bw(base_line_size =0.3) +
    geom_errorbar(aes(ymin=h2-SE, ymax=h2+SE), width=0.1, color = "grey",alpha=0.6) +
    geom_point(size = 0.8, color = "steelblue") + 
    ggtitle(types[celltype]) + theme(plot.title = element_text(hjust = 0.5)) + 
    xlab("PCs sorted by heritability") + ylab("Heritability ± SE") + 
    ylim(-2,3)  + 
    theme(axis.title.x = element_text(size = 14),
          axis.text.x = element_text(size = 12, color = "black"),
          axis.title.y = element_text(size = 14),
          axis.text.y = element_text(size = 12, color = "black"),
          text= element_text(family="Times")) 
  
  plotdata <- data.frame(observed = -log10(sort(H$Pval)),
                         expected = -log10(ppoints(nrow(H))))
  
  
  p2[[celltype]] <- ggplot(plotdata) + theme_bw(base_line_size =0.3) +
    geom_point(aes(expected, observed), shape = 1, size = 1.5, color = "steelblue") +
    geom_abline(intercept = 0, slope = 1, alpha = 0.6, color = "red") +
    ggtitle(types[celltype]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = expression(paste("Expected -log"[10],"(p-value)")),
         y = expression(paste("Observed -log"[10],"(p-value)"))) +
    
    theme(axis.title.x = element_text(size = 10),
          axis.text.x = element_text(size = 8, color = "black"),
          axis.title.y = element_text(size = 10),
          axis.text.y = element_text(size = 8, color = "black"),
          text= element_text(family="Times")) 
  
  
  
  
}


all_interval <- grid.arrange(p1[[1]],p1[[2]],p1[[3]],p1[[4]],p1[[5]],p1[[6]], nrow = 2)

```


## S2 hsq qqplot for p-values

```{r fig.height=5, fig.width=7.7, warning=F, message=F}


all_qq <- grid.arrange(p2[[1]],p2[[2]],p2[[3]],p2[[4]],p2[[5]],p2[[6]], nrow = 2)


```


## S3 Number of independent signals for factor gwas (factors filtered by hsq >0.9 & hsq pvalue < 0.05)


```{r fig.height=5, fig.width=7.7, warning=F, message=F}


p1 <- list()
for (celltype in celltypes){

  file <- paste0("/project/xinhe/xsun/pathway_factor/analysis/7.analysis_for_factors/results/num_gwas_signals_", celltype, ".RDS")
  num_signal_all <- readRDS(file)

  num_signal_all <- as.data.frame(num_signal_all)
  colnames(num_signal_all) <- "num"
  #p1[[i]] <-  ggplot(num_signal_all, aes(x=num)) + geom_histogram(breaks = seq(0, max(num_signal_all), by = 1),color="white",fill = "steelblue")+
  p1[[celltype]] <-  ggplot(num_signal_all, aes(x=num)) + geom_histogram(bins = 30,color="white",fill = "steelblue")+
    theme_bw(base_line_size =0.3) +
    ggtitle(types[celltype]) + theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "# of independent signals",
         y = "Count") +

    theme(axis.title.x = element_text(size = 14),
          axis.text.x = element_text(size = 12, color = "black"),
          axis.title.y = element_text(size = 14),
          axis.text.y = element_text(size = 12, color = "black"),
          text= element_text(family="Times")) +

    annotate(geom = "text", x = Inf, y = Inf, label = paste0("The total number of tested factor = ",nrow(num_signal_all)),
             hjust = 1, vjust = 1,  size = 2,  family = "Times",   color = "black")


}

all1 <- grid.arrange(p1[[1]],p1[[2]],p1[[3]],p1[[4]],p1[[5]],p1[[6]], nrow = 2)


```




## S4 2 examples for ACAT

same examples with initial submission. but the snps are a bit different

```{r fig.height=3, fig.width=5.5, warning=F, message=F}


folder_harmo <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/"


celltype <- "B_cell"
trait <- "lymphc-ebi-a-GCST90002388"
factor <- "04014_PC3"

file_harmo <- paste0(folder_harmo,celltype,"-",trait,"_harmo.RDS")
list_harmo <- readRDS(file_harmo)

df_harmo <- list_harmo[[factor]]

p1 <- qqplot(df_harmo$pval.exposure, title = "3rd PC of Ras signaling pathway \n- lymphocyte count")


celltype <- "B_cell"
trait <- "IBD-ebi-a-GCST003043"
factor <- "04612_PC5"

file_harmo <- paste0(folder_harmo,celltype,"-",trait,"_harmo.RDS")
list_harmo <- readRDS(file_harmo)

df_harmo <- list_harmo[[factor]]

p2 <- qqplot(df_harmo$pval.exposure, title = "5th PC of Antigen processing and \npresentation pathway - IBD")


all <- grid.arrange(p1,p2, nrow = 1)


```

## S5 Number of supporting SNPs per pair

difference: the current cutoff is >=5SNPs and >=20genes

### EUR


```{r fig.height=5, fig.width=7.7, warning=F, message=F}

p <- list()
for(celltype in celltypes){
  
  df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EUR.RDS"))
  df <- df[df$trait %in% traits_EUR,]
  
  summary <- df %>%
    count(num_supp_SNP, name = "Freq")
  
  # Plot
  p[[celltype]] <- ggplot(summary, aes(x = num_supp_SNP, y = Freq)) +
    geom_bar(stat = "identity", fill = "steelblue") +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "# of supporting SNPs in each pair", y = "Count") +
    geom_text(
      aes(label = ifelse(num_supp_SNP %% 5 == 0, as.character(Freq), "")),
      colour = "black", size = 2, vjust = -0.1, family = "Times"
    ) +
    annotate(
      geom = "text",
      x = Inf, y = Inf,
      label = paste0("# of candidate pairs = ", sum(summary$Freq[summary$num_supp_SNP >= 5])),
      hjust = 1.1, vjust = 1.5, size = 3, family = "Times", color = "black"
    ) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = "black"),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = "black"),
      text = element_text(family = "Times")
    ) +
    scale_x_continuous(breaks = seq(0, max(summary$num_supp_SNP), 5))
  
}


all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)



```



### EAS


```{r fig.height=5, fig.width=7.7, warning=F, message=F}

p <- list()
for(celltype in celltypes){
  
  df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EAS.RDS"))
  df <- df[df$trait %in% traits_EAS,]
  
  summary <- df %>%
    count(num_supp_SNP, name = "Freq")
  
  # Plot
  p[[celltype]] <- ggplot(summary, aes(x = num_supp_SNP, y = Freq)) +
    geom_bar(stat = "identity", fill = "steelblue") +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "# of supporting SNPs in each pair", y = "Count") +
    geom_text(
      aes(label = ifelse(num_supp_SNP %% 5 == 0, as.character(Freq), "")),
      colour = "black", size = 2, vjust = -0.1, family = "Times"
    ) +
    annotate(
      geom = "text",
      x = Inf, y = Inf,
      label = paste0("# of candidate pairs = ", sum(summary$Freq[summary$num_supp_SNP >= 5])),
      hjust = 1.1, vjust = 1.5, size = 3, family = "Times", color = "black"
    ) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = "black"),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = "black"),
      text = element_text(family = "Times")
    ) +
    scale_x_continuous(breaks = seq(0, max(summary$num_supp_SNP), 5))
  
}


all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)



```


## S6 ECT p-values

### EUR

```{r fig.height=5, fig.width=7.7, warning=F, message=F}

p <- list()
for(celltype in celltypes){
  
  df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EUR.RDS"))
  df <- df[df$trait %in% traits_EUR,]
  
  df <- df[complete.cases(df$p_ECT),]
  
  summary_p <- df %>%
    mutate(bin = cut(p_ECT, breaks = seq(0, 1, by = 0.05), include.lowest = TRUE)) %>%
    count(bin, name = "Freq")
  
  # Correctly handle both [ and ( at the start of interval labels
  summary_p <- summary_p %>%
    mutate(
      bin_low  = as.numeric(sub("[^0-9\\.]*([0-9\\.]+),.*", "\\1", bin)),
      bin_high = as.numeric(sub(".*,([0-9\\.]+)\\]", "\\1", bin)),
      bin_mid  = (bin_low + bin_high) / 2
    )
  
  p[[celltype]] <- ggplot(summary_p, aes(x = bin_mid, y = Freq)) +
    geom_bar(stat = "identity", fill = "steelblue") +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "ECT p-value", y = "Count") +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = "black"),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = "black"),
      text = element_text(family = "Times")
    ) +
    scale_x_continuous(breaks = seq(0, 1, 0.1), limits = c(0, 1))
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)

p <- list()
for(celltype in celltypes){
  
  df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EUR.RDS"))
  df <- df[df$trait %in% traits_EUR,]
  
  df <- df[complete.cases(df$p_ECT),]
  
  p[[celltype]] <- qqplot(pvalues = df$p_ECT, title = types[celltype])
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)


```





### EAS

```{r fig.height=5, fig.width=7.7, warning=F, message=F}

p <- list()
for(celltype in celltypes){
  
  df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EAS.RDS"))
  df <- df[df$trait %in% traits_EAS,]
  
  df <- df[complete.cases(df$p_ECT),]
  
  summary_p <- df %>%
    mutate(bin = cut(p_ECT, breaks = seq(0, 1, by = 0.05), include.lowest = TRUE)) %>%
    count(bin, name = "Freq")
  
  # Correctly handle both [ and ( at the start of interval labels
  summary_p <- summary_p %>%
    mutate(
      bin_low  = as.numeric(sub("[^0-9\\.]*([0-9\\.]+),.*", "\\1", bin)),
      bin_high = as.numeric(sub(".*,([0-9\\.]+)\\]", "\\1", bin)),
      bin_mid  = (bin_low + bin_high) / 2
    )
  
  p[[celltype]] <- ggplot(summary_p, aes(x = bin_mid, y = Freq)) +
    geom_bar(stat = "identity", fill = "steelblue") +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "ECT p-value", y = "Count") +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = "black"),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = "black"),
      text = element_text(family = "Times")
    ) +
    scale_x_continuous(breaks = seq(0, 1, 0.1), limits = c(0, 1))
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)

p <- list()
for(celltype in celltypes){
  
  df <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_final/", celltype,"_ECT_summary_EAS.RDS"))
  df <- df[df$trait %in% traits_EAS,]
  
  df <- df[complete.cases(df$p_ECT),]
  
  p[[celltype]] <- qqplot(pvalues = df$p_ECT, title = types[celltype])
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)


```



## S7 ECT storey pi1

### EUR


```{r fig.height=3, fig.width=5, warning=F, message=F}

load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/ECT_story_EUR.rdata")

values <- pi1_all
labels <- c("B cell \n(CD19+)","Monocyte \n(CD14+)","Leukocyte \n(CD15+)","Platelet","T cell \n(CD4+)","T cell \n(CD8+)")
data <- data.frame(labels, values)

# Create the bar plot
p <- ggplot(data, aes(x=labels, y=values, fill=labels)) +
  geom_bar(stat="identity", color="steelblue", fill="steelblue") +
  labs(x="Cell Types", y="Storey's Pi1") +
  theme_minimal()  +
  
  theme(axis.title.x = element_text(size = 14),
        axis.text.x = element_text(size = 12, color = "black", angle = 45, hjust = 1,vjust = 1.1),
        axis.title.y = element_text(size = 14),
        axis.text.y = element_text(size = 12, color = "black"),
        text= element_text(family="Times")) 

p

```

### EAS


```{r fig.height=3, fig.width=5, warning=F, message=F}

load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/ECT_story_EAS.rdata")

values <- pi1_all
labels <- c("B cell \n(CD19+)","Monocyte \n(CD14+)","Leukocyte \n(CD15+)","Platelet","T cell \n(CD4+)","T cell \n(CD8+)")
data <- data.frame(labels, values)

# Create the bar plot
p <- ggplot(data, aes(x=labels, y=values, fill=labels)) +
  geom_bar(stat="identity", color="steelblue", fill="steelblue") +
  labs(x="Cell Types", y="Storey's Pi1") +
  theme_minimal()  +
  
  theme(axis.title.x = element_text(size = 14),
        axis.text.x = element_text(size = 12, color = "black", angle = 45, hjust = 1,vjust = 1.1),
        axis.title.y = element_text(size = 14),
        axis.text.y = element_text(size = 12, color = "black"),
        text= element_text(family="Times")) 

p

```