---
title: "Comparing with eQTLGen"
author: "XSun"
date: "2025-10-24"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

We checked if the supporting SNPs are trans-eQTLs. The trans-eQTLs are downloaded from https://eqtlgen.org/trans-eqtls.html

# If supporting SNPs are trans-eqtls, comparing with totally trait-related SNPs

We have 989 supporting SNPs for all pairs with ECT FDR < 0.2, when filter by gene num >= 20, SNP num >= 5 . There are 329 unique SNPs. 

For each trait, we extracted the SNPs with p-value < 5E-8, then use the collected SNPs as the pool for random SNP, sampled 10*329 SNPs from it, use them as the random SNPs

We test whether supporting genes are more enriched in trans-eQTLs than the random genes.


```{r}

supp_snps <- readRDS("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/supp_snps_ECTFDR02.RDS")
snps <- supp_snps$SNP
snps <- snps[duplicated(snps)]

traits <- unique(supp_snps$trait)

snp_trait_related_all <- c()
for(trait in traits){
  
  snps_trait <- read.table(paste0("/project/xinhe/xsun/pathway_factor/data_v2/GWAS/",trait,"_pval5E8.txt"), header = T, fill = T)
  snp_trait_related_all <- c(snp_trait_related_all, snps_trait$SNP)
  
}
snp_trait_related_all <- unique(snp_trait_related_all)


trans_eqtlgen <- read.table(gzfile("/project/xinhe/xsun/pathway_factor/analysis//3.results_analysis/data/data_eqtlgen/2018-09-04-trans-eQTLsFDR0.05-CohortInfoRemoved-BonferroniAdded.txt.gz"),header = T)

num_supp_trans <- sum(snps %in% trans_eqtlgen$SNP)
num_supp_nottrans <- sum(!snps %in% trans_eqtlgen$SNP)

#write.table(snps, file = "results/supp_snps_ECTp0.0005_unique.txt", quote = F, col.names = F, row.names = F)

ramdon_snp <- snp_trait_related_all[sample(1:length(snp_trait_related_all),size = 10*length(snps))]

num_random_trans <- sum(ramdon_snp  %in% trans_eqtlgen$SNP)
num_random_notrans <- sum(!ramdon_snp  %in% trans_eqtlgen$SNP)


mat <- matrix(c(num_supp_trans, num_supp_nottrans, num_random_trans, num_random_notrans), nrow = 2, byrow = TRUE,
              dimnames = list(
                c("Supp_SNP", "Random"),
                c("Trans", "NotTrans")
              ))

mat <- mat+1

print(mat)

# Fisher's exact test
fisher.test(mat)
 
```


# If SNP-gene pairs were reported by eQTLGen

We use top 10 genes with largest gene loading for a factor. Creat gene-snp pairs for the top genes and supporting SNPs.


```{r}

load("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/results/trans_pairs.rdata")

print(paste0("Total gene-transqtl pairs in eQTLGen = ", nrow(pairs_all[pairs_all$trans == T,])))

DT::datatable(pairs_all[pairs_all$trans == T,],caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','gene-suppoting SNP pair reported by eQTLGen'),options = list(pageLength = 10) )


```




