---
title: "Summarizing ECT results -- ≥ 5 supporting SNPs & ≥ 20 genes, CEDAR, bin AF"
author: "XSun"
date: "2026-01-07"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Introduction

We applied cutoffs of 
+
- `supporting SNP ≥ 5` and 
- `gene number in pathway ≥ 20` 

to filter the pairs before performing FDR control. All results presented below are based on this setting.

**ECT strategy: When generating “random” GWAS effects of a SNP, we sample from GWAS SNPs with comparable AF with that SNP.**


```{r warning=F, message=F}
library(dplyr)
library(ggplot2)
library(gridExtra)
library(forcats)
library(tidyr)

# library(ensembldb)
# library(EnsDb.Hsapiens.v75)
# library(ggrepel)
# library(ggrastr)

celltypes <- c("B_cell","CD14_positive_monocyte","CD15_positive_leukocyte","platelet","T_cell","thymocyte")
types <- c("B cell (CD19+)","Monocyte (CD14+)","Leukocyte (CD15+)","Platelet","T cell (CD4+)","T cell (CD8+)")
names(types) <- celltypes
names(celltypes) <- types

source("/project/xinhe/xsun/pathway_factor/data_v2/traits_finalselection_SLEremoved.R")

folder_ect_summary <- "/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_binAF/"

DT::datatable(matrix())
```

# Number of supporting variants

```{r fig.height=6, fig.width=9, warning=F, message=F}

p <- list()
for(celltype in celltypes){

  df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df <- df[df$trait_id %in% traits_EUR,]

  summary <- df %>%
    count(num_supp_SNP, name = "Freq")

  # Plot
  p[[celltype]] <- ggplot(summary, aes(x = num_supp_SNP, y = Freq)) +
    geom_bar(stat = "identity", fill = "steelblue") +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "# of supporting SNPs in each pair", y = "# of factor–trait pairs") +
    geom_text(
      aes(label = ifelse(num_supp_SNP %% 5 == 0, as.character(Freq), "")),
      colour = "black", size = 2, vjust = -0.1, family = "Times"
    ) +
    annotate(
      geom = "text",
      x = Inf, y = Inf,
      label = paste0("# of candidate pairs = ", sum(summary$Freq[summary$num_supp_SNP >= 5])),
      hjust = 1.1, vjust = 1.5, size = 3, family = "Times", color = "black"
    ) +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = "black"),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = "black"),
      text = element_text(family = "Times")
    ) +
    scale_x_continuous(breaks = seq(0, max(summary$num_supp_SNP), 5))

}


all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)

```


# Histogram for ECT p-values

```{r fig.height=6, fig.width=9, warning=F, message=F}

p <- list()
for(celltype in celltypes){

  df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df <- df[df$trait_id %in% traits_EUR,]

  df <- df[complete.cases(df$p_ECT),]

  summary_p <- df %>%
    mutate(bin = cut(p_ECT, breaks = seq(0, 1, by = 0.05), include.lowest = TRUE)) %>%
    count(bin, name = "Freq")

  # Correctly handle both [ and ( at the start of interval labels
  summary_p <- summary_p %>%
    mutate(
      bin_low  = as.numeric(sub("[^0-9\\.]*([0-9\\.]+),.*", "\\1", bin)),
      bin_high = as.numeric(sub(".*,([0-9\\.]+)\\]", "\\1", bin)),
      bin_mid  = (bin_low + bin_high) / 2
    )

  p[[celltype]] <- ggplot(summary_p, aes(x = bin_mid, y = Freq)) +
    geom_bar(stat = "identity", fill = "steelblue") +
    theme_bw(base_line_size = 0.3) +
    ggtitle(types[celltype]) +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "ECT p-value", y = "Count") +
    theme(
      axis.title.x = element_text(size = 10),
      axis.text.x  = element_text(size = 8, color = "black"),
      axis.title.y = element_text(size = 10),
      axis.text.y  = element_text(size = 8, color = "black"),
      text = element_text(family = "Times")
    ) +
    scale_x_continuous(breaks = seq(0, 1, 0.1), limits = c(0, 1))

}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)


```

# Number of significant pairs at different cutoffs, for each cell type

```{r fig.height=3, fig.width=4, warning=F, message=F}

num_fdr01 <- c()
num_fdr02 <- c()
for(celltype in celltypes){

  df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))

  df_fdr01 <- df[df$ECT_FDR_5suppSNP < 0.1,]
  df_fdr01 <- df_fdr01[complete.cases(df_fdr01$p_ECT),]
  num_fdr01 <- c(num_fdr01, nrow(df_fdr01))

  df_fdr02 <- df[df$ECT_FDR_5suppSNP < 0.2 & df$ECT_FDR_5suppSNP >= 0.1,]
  df_fdr02 <- df_fdr02[complete.cases(df_fdr02$p_ECT),]
  num_fdr02 <- c(num_fdr02, nrow(df_fdr02))

}

sum <- data.frame(celltype = types,
                  num_fdr01 = num_fdr01,
                  num_fdr02 = num_fdr02)


df <- sum %>%
  pivot_longer(cols = starts_with("num_fdr"), names_to = "event", values_to = "total") %>%
  mutate(
    event = recode(event,
                   "num_fdr01" = "FDR < 0.1",
                   "num_fdr02" = "0.1 < FDR < 0.2")
  )

df$event <- factor(df$event, levels = c("FDR < 0.1", "0.1 < FDR < 0.2"))

# Plot
p1 <- ggplot(df, aes(x = fct_inorder(celltype), y = total, fill = event)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.9), alpha = 0.9) +
  scale_fill_manual(
    name = "group",
    values = c("#F8766D", "darkturquoise"),
    labels = c("FDR < 0.1", "0.1 < FDR < 0.2")
  ) +
  geom_text(
    aes(label = total),
    position = position_dodge(width = 0.9),
    color = "black",
    size = 3,
    vjust = -0.3
  ) +
  labs(x = "Cell types", y = "Count") +
  ggtitle("# of pairs") +
  ylim(0, 30) +
  theme_bw(base_line_size = 0.3) +
  theme(
    axis.title.x = element_text(size = 14),
    axis.text.x = element_text(size = 8, color = "black", angle = 45, hjust = 1, vjust = 1.1),
    axis.title.y = element_text(size = 14),
    axis.text.y = element_text(size = 12, color = "black"),
    text = element_text(family = "Times")
  )

p1
```



# Detailed results for all pairs at ECT FDR < 0.2

```{r fig.height=3, fig.width=9, warning=F, message=F}

df_all_fdr02 <- c()
df_largeECTp <- c()
for(celltype in celltypes){

  df <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df <- df[complete.cases(df$ECT_FDR_5suppSNP),]

  df_fdr02 <- df[df$ECT_FDR_5suppSNP < 0.2,]
  df_all_fdr02 <- rbind(df_all_fdr02,df_fdr02)
  
  df_sort <- df[order(df$p_ECT, decreasing = T),]
  df_largeECTp <- rbind(df_largeECTp , df_sort[1,])

}

rownames(df_all_fdr02) <- NULL

#df_all_fdr02$celltype <- types[df_all_fdr02$celltype]
df_all_fdr02 <- df_all_fdr02[order(df_all_fdr02$ECT_FDR_5suppSNP),]

df_all_fdr01 <- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP < 0.1,]

DT::datatable(df_all_fdr02,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Pairs with ECT FDR <= 0.2 '),options = list(pageLength = 10) )

```


## effect size plots for the top pairs (ECT FDR < 0.1)


```{r warning=FALSE, message=FALSE, fig.height=3, fig.width=8}

df_all_fdr01 <- df_all_fdr02[df_all_fdr02$ECT_FDR_5suppSNP < 0.1,]
df_all_fdr01$celltype <- celltypes[df_all_fdr01$celltype]

fdr_cutoff <- 0.2

for (i in 1:nrow(df_all_fdr01)){
  
  celltype <- df_all_fdr01$celltype[i]
  trait <- df_all_fdr01$trait_id[i]
  factor <- df_all_fdr01$factor[i]
  pwy <- df_all_fdr01$pathwayName[i]
  
  file_harmo <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/",celltype, "-", trait, "_harmo.RDS")
  dt_harmo <- readRDS(file_harmo)
  
  dat <- dt_harmo[[factor]]
  snp_select <- dat[dat$FDR.exposure < fdr_cutoff,]
  snp_select$chr <- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr <- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit <- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p <- df_all_fdr01$p_ECT[i]
  
  pathwayname <- df$pathwayName[df$factor == factor & df$trait ==trait]
  
  eff_chr <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = "black"),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = "black"),
          text= element_text(family="Times"),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                             "\nECT p-value =", ect_p),
               color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
    geom_hline(yintercept = 0, color = "grey", linetype = "dashed") +
    ggtitle(paste0(pwy, "-", trait))
  
  
  eaf_breaks <- c(0, 0.01, 0.05, 0.10, 0.20, 0.30, 0.50)
  eaf_labels <- c("<0.01","0.01-0.05", "0.05-0.1", "0.1-0.2",
                  "0.2-0.3", "0.3-0.5")
  
  snp_select$AF_bin <- cut(
    snp_select[["eaf.exposure"]],
    breaks = eaf_breaks, labels = eaf_labels,
    include.lowest = TRUE, right = TRUE
  )
  
  eff_AF <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=AF_bin)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=AF_bin), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = "black"),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = "black"),
          text= element_text(family="Times"),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                             "\nECT p-value =", ect_p),
               color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
    geom_hline(yintercept = 0, color = "grey", linetype = "dashed") +
    ggtitle(paste0(pwy, "-", trait))
  
  all <- grid.arrange(eff_chr, eff_AF, nrow = 1)
  print(all)
}



```


## effect size plots for the pairs with large ECT p-values


```{r warning=FALSE, message=FALSE, fig.height=3, fig.width=8}

df_largeECTp$celltype <- celltypes[df_largeECTp$celltype]

fdr_cutoff <- 0.2

for (i in 1:nrow(df_largeECTp)){
  
  celltype <- df_largeECTp$celltype[i]
  trait <- df_largeECTp$trait_id[i]
  factor <- df_largeECTp$factor[i]
  pwy <- df_largeECTp$pathwayName[i]
  
  file_harmo <- paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/assoc_harmonize/",celltype, "-", trait, "_harmo.RDS")
  dt_harmo <- readRDS(file_harmo)
  
  dat <- dt_harmo[[factor]]
  snp_select <- dat[dat$FDR.exposure < fdr_cutoff,]
  snp_select$chr <- as.numeric(as.character(snp_select$chr.exposure))
  snp_select$chr <- factor(snp_select$chr, levels = sort(unique(snp_select$chr)))
  
  fit <- lm(snp_select$beta.outcome ~ 0 + snp_select$beta.exposure)
  
  ect_p <- df_largeECTp$p_ECT[i]
  
  pathwayname <- df$pathwayName[df$factor == factor & df$trait ==trait]
  
  eff_chr <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=chr)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=chr), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = "black"),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = "black"),
          text= element_text(family="Times"),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                             "\nECT p-value =", ect_p),
               color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
    geom_hline(yintercept = 0, color = "grey", linetype = "dashed") +
    ggtitle(paste0(pwy, "-", trait))
  
  
  eaf_breaks <- c(0, 0.01, 0.05, 0.10, 0.20, 0.30, 0.50)
  eaf_labels <- c("<0.01","0.01-0.05", "0.05-0.1", "0.1-0.2",
                  "0.2-0.3", "0.3-0.5")
  
  snp_select$AF_bin <- cut(
    snp_select[["eaf.exposure"]],
    breaks = eaf_breaks, labels = eaf_labels,
    include.lowest = TRUE, right = TRUE
  )
  
  eff_AF <- ggplot(snp_select, aes(x=beta.exposure, y=beta.outcome, color=AF_bin)) +
    geom_point(size = 4) + 
    geom_errorbar(aes(ymin=beta.outcome-se.outcome, ymax=beta.outcome+se.outcome, color=AF_bin), width=0.01, alpha=0.6) +
    theme_bw(base_line_size = 0.3) +
    theme(axis.title.x = element_text(size = 16),
          axis.text.x = element_text(size = 14, color = "black"),
          axis.title.y = element_text(size =16),
          axis.text.y = element_text(size = 14, color = "black"),
          text= element_text(family="Times"),
          legend.text = element_text(size=12),
          legend.title = element_text(size=12)) + 
    geom_abline(slope = coef(fit)[1], intercept = 0, color = "grey", size = 0.5) +
    geom_label(aes(x = Inf, y = Inf), hjust = 1, vjust = 1,
               label = paste("R^2 = ", signif(summary(fit)$r.squared, 3), 
                             "\nECT p-value =", ect_p),
               color = "black", family = "Times", size =3, label.size = NA,fill=NA) +
    xlab(expression(hat(beta)[factor])) + 
    ylab(expression(hat(beta)[gwas])) +
    geom_vline(xintercept = 0, color = "grey", linetype = "dashed") +
    geom_hline(yintercept = 0, color = "grey", linetype = "dashed") +
    ggtitle(paste0(pwy, "-", trait))
  
  all <- grid.arrange(eff_chr, eff_AF, nrow = 1)
  print(all)
}



```





# Comparing the ECT p-values from old and new ECT strategy

```{r fig.height=6, fig.width=9, warning=F, message=F, results='asis'}

library(DT)
library(htmltools)
library(knitr)

p <- list()
tables <- list()
for(celltype in celltypes){
  
  df_new <- readRDS(paste0(folder_ect_summary, celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df_new$id <- paste0(df_new$factor,"-", df_new$trait_id)
  df_new <- df_new[df_new$num_supp_SNP >=5,]
  df_new <- df_new[c("id","p_ECT","ECT_FDR_5suppSNP")]
  
  df_old <- readRDS(paste0("/project/xinhe/xsun/pathway_factor/analysis/1.ECT_v2/ECT_summary_originalECT/", celltype,"_ECT_summary_poolalltraits_selected.RDS"))
  df_old$id <- paste0(df_old$factor,"-", df_old$trait_id)
  df_old <- df_old[df_old$num_supp_SNP >=5,]
  df_old <- df_old[c("id","p_ECT","ECT_FDR_5suppSNP","pathwayName","trait_id","trait_name")]
  
  df_merge <- merge(df_new,df_old, by = "id")
  colnames(df_merge)[1:5] <- c("id","p_ECT_new","FDR_ECT_new" ,"p_ECT_old","FDR_ECT_old")
  
  df_plot <- df_merge %>%
    filter(!is.na(p_ECT_new), !is.na(p_ECT_old)) %>%
    mutate(
      logp_old = -log10(p_ECT_old),
      logp_new = -log10(p_ECT_new)
    )
  
  p[[celltype]] <- ggplot(df_plot, aes(x = logp_old, y = logp_new)) +
    geom_point(alpha = 0.6, size = 2) +
    geom_abline(slope = 1, intercept = 0,
                linetype = "dashed", color = "red") +
    ggtitle(types[celltype]) +
    labs(
      x = expression(-log[10](p[ECT~old_strategy])),
      y = expression(-log[10](p[ECT~new_binAF]))
    ) +
    theme_classic()
  
  df_show <- df_merge[order(df_merge$p_ECT_new, decreasing = F),]
  df_show <- df_show[1:20,]
  
  tables[[celltype]] <- DT::datatable(
    df_show,
    caption = tags$caption(
      style = 'caption-side: left; text-align: left; color:black; font-size:150%;',
      paste0('Pairs with lowest new ECT p-values - ', types[celltype])
    ),
    options = list(pageLength = 10)
  )
  
  tables[[celltype]] <- datatable(
    df_show,
    caption = tags$caption(
      style = 'caption-side:left; text-align:left; color:black; font-size:140%;',
      paste0('Pairs with lowest new ECT p-values - ', types[celltype])
    ),
    options = list(pageLength = 10)
  )
  
  # print(
  #   DT::datatable(
  #     df_show,
  #     caption = tags$caption(
  #       style = 'caption-side: left; text-align: left; color:black; font-size:150%;',
  #       paste0('Pairs with lowest new ECT p-values - ', types[celltype])
  #     ),
  #     options = list(pageLength = 10)
  #   )
  # )
  # 
  # cat("\n\n")
  
  
}

all <- grid.arrange(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],p[[6]], nrow = 2)

for(celltype in celltypes) {
 
  cat(knit_print(tables[[celltype]]))
}

```