---
title: "Onek1k Data Overview"
author: "XSun"
date: "2025-12-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Introduction

Data citation: Yazar S., Alquicira-Hernández J., Wing K., Senabouth A., Gordon G., Andersen S., Lu Q., Rowson A., Taylor T., Clarke L., Maccora L., Chen C., Cook A., Ye J., Fairfax K., Hewitt A., Powell J. Single cell eQTL mapping identified cell type specific control of autoimmune disease. Science, 376, 6589 (2022)

## Basic infomation summary

- Dimensions (genes x cells):  32738 x 1272489 
- Number of Genes:  32738 
- Number of Cells:  1272489 

- Assays:  RNA 
  - Layers:  counts, data 
  - Variable Features:  0 

- Metadata Columns:
 [1] "orig.ident"   "nCount_RNA"   "nFeature_RNA" "pool"         "individual"   "percent.mt"   "latent"       "nCount_SCT"  
 [9] "nFeature_SCT" "cell_type"    "cell_label"   "sex"          "age"         

- Cell Type Counts:

        B IN        B Mem       CD4 ET       CD4 NC     CD4 SOX4       CD8 ET       CD8 NC    CD8 S100B           DC Erythrocytes 
       82068        48023        61786       463528         4065       205077       133482        34528         8690          680 
      Mono C      Mono NC           NK         NK R       Plasma    Platelets 
       38233        15166       159820         9677         3625         4041 

- Individual number: 981 

```{r}

```

## Processing 

### Gene expression data

The CEDAR dataset includes six immune cell types (CD4⁺ T cells, CD8⁺ T cells, CD19⁺ B cells, CD14⁺ monocytes, CD15⁺ granulocytes, and platelets). Because the OneK1K dataset does not include CD15⁺ granulocytes, we restricted our analysis to the remaining five shared cell types. All OneK1K cells were first subset by cell type and donor identity, and pseudobulk expression matrices were generated by summing raw UMI counts across all cells belonging to the same donor. This resulted in one gene-by-donor count matrix per cell type.

Specifically, cells annotated as B IN or B Mem were combined into a single B cell group; CD4 ET, CD4 NC, and CD4 SOX4 were grouped as CD4+ T cell; CD8 ET, CD8 NC, and CD8 S100B were grouped as CD8+ T cell; Mono C and Mono NC were grouped as monocyte; and Platelets were retained as platelet.

| **Grouped Cell Type**              | **Original OneK1K Cell Labels**            |
| ---------------------------------- | ------------------------------------------ |
| **B cell**                         | B IN, B Mem                                |
| **CD4 T cell**                     | CD4 ET, CD4 NC, CD4 SOX4                   |
| **CD8 T cell**                     | CD8 ET, CD8 NC, CD8 S100B                  |
| **Monocyte**                       | Mono C, Mono NC                            |
| **Platelet**                       | Platelets                                  |

Each pseudobulk matrix was then processed and normalized. Only genes annotated as protein_coding were retained. After filtering, to account for differences in sequencing depth across donors, we applied counts-per-million (CPM) normalization using `edgeR`’s `cpm()` function with `log = TRUE` and a `prior count of 1`, producing `log₂-CPM` values that stabilize variance—particularly for low-abundance genes—and reduce library-size–related heterogeneity. For each cell type, the resulting normalized log₂-CPM matrix was used in downstream analyses.

### Genotype data

The process of genotype data is the same with the CEDAR data set.





